---
title: "Investigating pleiotropy between depression and autoimmune diseases using the UK Biobank"
author: "Kylie P. Glanville"
output:
  html_document:
    toc: true
    toc_depth: 4
    toc_float: true
    number_sections: true
---

```{r load_libraries, include=FALSE, message=FALSE}

knitr::opts_chunk$set(echo = TRUE)
library(knitr)
library(kableExtra)
library(data.table)
library(VennDiagram)
library(eulerr)
library(scales)
library(gridExtra)
library(Hmisc)
library(ggplot2)
library(tibble)
library(tidyverse)
library(dplyr)
library(stringr)
library(formattable)
library(fmsb)
library(psych)

```

```{bash print_pheno_columns, echo=FALSE, eval=FALSE}

# Print required phenotype columns

#Demographics
#   f.21022.0.0    Age.at.recruitment.0.0
#   f.31.0.0       Sex.0.0
#   f.189.0.0      Townsend.deprivation.index.at.recruitment.0.0
#   f.21001.0.0    Body.mass.index.BMI.0.0
#   f.21001.1.0    Body.mass.index.BMI.1.0
#   f.21001.2.0    Body.mass.index.BMI.2.0
#   f.20116.0.0    Smoking.status.0.0
#   f.20116.1.0    Smoking.status.1.0
#   f.20116.2.0    Smoking.status.2.0

awk '{print $1,$8329,$1655,$1943,$8317,$8318,$8319,$7671,$7672,$7673}' /file_path/phenotypes.txt > demographics

#Self-report
#   f.20002.0.0    Non.cancer.illness.code.self.reported.0.0 - 
#   f.20002.2.28   Non.cancer.illness.code.self.reported.2.28

paste <(awk '{print $1}' /file_path/phenotypes.txt) <(cut -f5793-5879 /file_path/phenotypes.txt) > self_report

#Treatment/medications
#   f.20003.0.0    Treatment.medication.code.0.0 - 
#   f.20003.2.47   Treatment.medication.code.2.47

paste <(awk '{print $1}' /file_path/phenotypes.txt) <(cut -f5880-6023 /file_path/phenotypes.txt) > medications

```

```{r extract_hes_data_portal, echo=FALSE, eval=FALSE}

# downloaded ICD coding from UKB website
# description: ICD10 - WHO International Classification of Diseases = "coding10.tsv"

coding_icd10 <- as.data.frame(fread("/file_path/coding19.tsv", header = T))

# extracted ICD-10 codes relevant to each autoimmune disease to build sql scripts to extract primary and secondary diagnoses.
# submitted the following SQL scripts to extract ICD data from HES portal

# ankylosing spondylitis
sql_as_pri.txt:
  SELECT eid,admidate,disdate,epidur,diag_icd10 AS primary_diag
FROM hesin
WHERE diag_icd10 = 'M45'
OR diag_icd10 = 'M45X0'
OR diag_icd10 = 'M45X1'
OR diag_icd10 = 'M45X2'
OR diag_icd10 = 'M45X3'
OR diag_icd10 = 'M45X4'
OR diag_icd10 = 'M45X5'
OR diag_icd10 = 'M45X6'
OR diag_icd10 = 'M45X7'
OR diag_icd10 = 'M45X8'
OR diag_icd10 = 'M45X9'

sql_as_sec.txt:
  SELECT eid,admidate,disdate,epidur,secondary_diag FROM
(SELECT * FROM
(SELECT eid,record_id,diag_icd10 AS secondary_diag
FROM hesin_diag10
WHERE diag_icd10 = 'M45'
OR diag_icd10 = 'M45X0'
OR diag_icd10 = 'M45X1'
OR diag_icd10 = 'M45X2'
OR diag_icd10 = 'M45X3'
OR diag_icd10 = 'M45X4'
OR diag_icd10 = 'M45X5'
OR diag_icd10 = 'M45X6'
OR diag_icd10 = 'M45X7'
OR diag_icd10 = 'M45X8'
OR diag_icd10 = 'M45X9') AS b
LEFT JOIN hesin AS a ON a.record_id = b.record_id) AS c

# autoimmune thyroid disease
sql_at_pri.txt:
  SELECT eid,admidate,disdate,epidur,diag_icd10 AS primary_diag
FROM hesin
WHERE diag_icd10 ='E063'
OR diag_icd10 ='E050'

sql_at_sec.txt:
  SELECT eid,admidate,disdate,epidur,secondary_diag FROM
(SELECT * FROM
(SELECT eid,record_id,diag_icd10 AS secondary_diag
FROM hesin_diag10
WHERE diag_icd10 ='E063'
OR diag_icd10 ='E050') AS b
LEFT JOIN hesin AS a ON a.record_id = b.record_id) AS c

# coeliac disease
sql_coeliac_pri.txt:
  SELECT eid,admidate,disdate,epidur,diag_icd10 AS primary_diag
FROM hesin
WHERE diag_icd10 = 'K900'

sql_coeliac_sec.txt:
  SELECT eid,admidate,disdate,epidur,secondary_diag FROM
(SELECT * FROM
(SELECT eid,record_id,diag_icd10 AS secondary_diag
FROM hesin_diag10
WHERE diag_icd10 = 'K900') AS b
LEFT JOIN hesin AS a ON a.record_id = b.record_id) AS c

# hidradenitis suppurativa
sql_hidradentis_pri.txt:
  SELECT eid,admidate,disdate,epidur,diag_icd10 AS primary_diag
FROM hesin
WHERE diag_icd10 = 'L732'

sql_hidradentis_sec.txt:
  SELECT eid,admidate,disdate,epidur,secondary_diag FROM
(SELECT * FROM
(SELECT eid,record_id,diag_icd10 AS secondary_diag
FROM hesin_diag10
WHERE diag_icd10 = 'L732') AS b
LEFT JOIN hesin AS a ON a.record_id = b.record_id) AS c

# inflammatory bowel disease
sql_ibd_pri.txt:
  SELECT eid,admidate,disdate,epidur,diag_icd10 AS primary_diag
FROM hesin
WHERE diag_icd10 = 'K50'
OR diag_icd10 = 'K500'
OR diag_icd10 = 'K501'
OR diag_icd10 = 'K508'
OR diag_icd10 = 'K509'
OR diag_icd10 = 'K51'
OR diag_icd10 = 'K510'
OR diag_icd10 = 'K511'
OR diag_icd10 = 'K512'
OR diag_icd10 = 'K513'
OR diag_icd10 = 'K514'
OR diag_icd10 = 'K515'
OR diag_icd10 = 'K518'
OR diag_icd10 = 'K519'

sql_ibd_sec.txt:
  SELECT eid,admidate,disdate,epidur,secondary_diag FROM
(SELECT * FROM
(SELECT eid,record_id,diag_icd10 AS secondary_diag
FROM hesin_diag10
WHERE diag_icd10 = 'K50'
OR diag_icd10 = 'K500'
OR diag_icd10 = 'K501'
OR diag_icd10 = 'K508'
OR diag_icd10 = 'K509'
OR diag_icd10 = 'K51'
OR diag_icd10 = 'K510'
OR diag_icd10 = 'K511'
OR diag_icd10 = 'K512'
OR diag_icd10 = 'K513'
OR diag_icd10 = 'K514'
OR diag_icd10 = 'K515'
OR diag_icd10 = 'K518'
OR diag_icd10 = 'K519') AS b
LEFT JOIN hesin AS a ON a.record_id = b.record_id) AS c

# myasthenia gravis
sql_mg_pri.txt:
  SELECT eid,admidate,disdate,epidur,diag_icd10 AS primary_diag
FROM hesin
WHERE diag_icd10 = 'G700'

sql_mg_sec.txt:
  SELECT eid,admidate,disdate,epidur,secondary_diag FROM
(SELECT * FROM
(SELECT eid,record_id,diag_icd10 AS secondary_diag
FROM hesin_diag10
WHERE diag_icd10 = 'G700') AS b
LEFT JOIN hesin AS a ON a.record_id = b.record_id) AS c

# multiple sclerosis
sql_ms_pri.txt:
  SELECT eid,admidate,disdate,epidur,diag_icd10 AS primary_diag
FROM hesin
WHERE diag_icd10 = 'G35'

sql_ms_sec.txt:
  SELECT eid,admidate,disdate,epidur,secondary_diag FROM
(SELECT * FROM
(SELECT eid,record_id,diag_icd10 AS secondary_diag
FROM hesin_diag10
WHERE diag_icd10 = 'G35') AS b
LEFT JOIN hesin AS a ON a.record_id = b.record_id) AS c

# pernicious anaemia
sql_pa_pri.txt:
  SELECT eid,admidate,disdate,epidur,diag_icd10 AS primary_diag
FROM hesin
WHERE diag_icd10 ='D510'

sql_pa_sec.txt:
  SELECT eid,admidate,disdate,epidur,secondary_diag FROM
(SELECT * FROM
(SELECT eid,record_id,diag_icd10 AS secondary_diag
FROM hesin_diag10
WHERE diag_icd10 ='D510') AS b
LEFT JOIN hesin AS a ON a.record_id = b.record_id) AS c

# pemphigoid, pemphigus
sql_pemphigoid_gus_pri.txt:
  SELECT eid,admidate,disdate,epidur,diag_icd10 AS primary_diag
FROM hesin
WHERE diag_icd10 = 'L12'
OR diag_icd10 = 'L120'
OR diag_icd10 = 'L121'
OR diag_icd10 = 'L122'
OR diag_icd10 = 'L123'
OR diag_icd10 = 'L128'
OR diag_icd10 = 'L129'
OR diag_icd10 = 'L10'
OR diag_icd10 = 'L100'
OR diag_icd10 = 'L101'
OR diag_icd10 = 'L102'
OR diag_icd10 = 'L103'
OR diag_icd10 = 'L104'
OR diag_icd10 = 'L105'
OR diag_icd10 = 'L108'
OR diag_icd10 = 'L109'

sql_pemphigoid_gus_sec.txt:
  SELECT eid,admidate,disdate,epidur,secondary_diag FROM
(SELECT * FROM
(SELECT eid,record_id,diag_icd10 AS secondary_diag
FROM hesin_diag10
WHERE diag_icd10 = 'L12'
OR diag_icd10 = 'L120'
OR diag_icd10 = 'L121'
OR diag_icd10 = 'L122'
OR diag_icd10 = 'L123'
OR diag_icd10 = 'L128'
OR diag_icd10 = 'L129'
OR diag_icd10 = 'L10'
OR diag_icd10 = 'L100'
OR diag_icd10 = 'L101'
OR diag_icd10 = 'L102'
OR diag_icd10 = 'L103'
OR diag_icd10 = 'L104'
OR diag_icd10 = 'L105'
OR diag_icd10 = 'L108'
OR diag_icd10 = 'L109') AS b
LEFT JOIN hesin AS a ON a.record_id = b.record_id) AS c

# polymyalgia rheumatica
sql_pr_pri.txt:
  SELECT eid,admidate,disdate,epidur,diag_icd10 AS primary_diag
FROM hesin
WHERE diag_icd10 = 'M315'
OR diag_icd10 = 'M353'

sql_pr_sec.txt:
  SELECT eid,admidate,disdate,epidur,secondary_diag FROM
(SELECT * FROM
(SELECT eid,record_id,diag_icd10 AS secondary_diag
FROM hesin_diag10
WHERE diag_icd10 = 'M315'
OR diag_icd10 = 'M353') AS b
LEFT JOIN hesin AS a ON a.record_id = b.record_id) AS c

# psoriatic arthritis
sql_psa_pri.txt:
  SELECT eid,admidate,disdate,epidur,diag_icd10 AS primary_diag
FROM hesin
WHERE diag_icd10 = 'L405'
OR diag_icd10 = 'M073'
OR diag_icd10 = 'M0730'
OR diag_icd10 = 'M0731'
OR diag_icd10 = 'M0732'
OR diag_icd10 = 'M0733'
OR diag_icd10 = 'M0734'
OR diag_icd10 = 'M0735'
OR diag_icd10 = 'M0736'
OR diag_icd10 = 'M0737'
OR diag_icd10 = 'M0738'
OR diag_icd10 = 'M0739'

sql_psa_sec.txt:
  SELECT eid,admidate,disdate,epidur,secondary_diag FROM
(SELECT * FROM
(SELECT eid,record_id,diag_icd10 AS secondary_diag
FROM hesin_diag10
WHERE diag_icd10 = 'L405'
OR diag_icd10 = 'M073'
OR diag_icd10 = 'M0730'
OR diag_icd10 = 'M0731'
OR diag_icd10 = 'M0732'
OR diag_icd10 = 'M0733'
OR diag_icd10 = 'M0734'
OR diag_icd10 = 'M0735'
OR diag_icd10 = 'M0736'
OR diag_icd10 = 'M0737'
OR diag_icd10 = 'M0738'
OR diag_icd10 = 'M0739') AS b
LEFT JOIN hesin AS a ON a.record_id = b.record_id) AS c

# psoriasis
sql_psoriasis_pri.txt:
  SELECT eid,admidate,disdate,epidur,diag_icd10 AS primary_diag
FROM hesin
WHERE diag_icd10 = 'L40'
OR diag_icd10 = 'L400'
OR diag_icd10 = 'L401'
OR diag_icd10 = 'L402'
OR diag_icd10 = 'L403'
OR diag_icd10 = 'L404'
OR diag_icd10 = 'L408'
OR diag_icd10 = 'L409'

sql_psoriasis_sec.txt:
  SELECT eid,admidate,disdate,epidur,secondary_diag FROM
(SELECT * FROM
(SELECT eid,record_id,diag_icd10 AS secondary_diag
FROM hesin_diag10
WHERE diag_icd10 = 'L40'
OR diag_icd10 = 'L400'
OR diag_icd10 = 'L401'
OR diag_icd10 = 'L402'
OR diag_icd10 = 'L403'
OR diag_icd10 = 'L404'
OR diag_icd10 = 'L408'
OR diag_icd10 = 'L409') AS b
LEFT JOIN hesin AS a ON a.record_id = b.record_id) AS c

# rheumatoid arthritis
sql_ra_pri.txt:
  SELECT eid,admidate,disdate,epidur,diag_icd10 AS primary_diag
FROM hesin
WHERE diag_icd10 ='M05'
OR diag_icd10 ='M050'
OR diag_icd10 ='M0500'
OR diag_icd10 ='M0501'
OR diag_icd10 ='M0502'
OR diag_icd10 ='M0503'
OR diag_icd10 ='M0504'
OR diag_icd10 ='M0505'
OR diag_icd10 ='M0506'
OR diag_icd10 ='M0507'
OR diag_icd10 ='M0508'
OR diag_icd10 ='M0509'
OR diag_icd10 ='M051'
OR diag_icd10 ='M0510'
OR diag_icd10 ='M0511'
OR diag_icd10 ='M0512'
OR diag_icd10 ='M0513'
OR diag_icd10 ='M0514'
OR diag_icd10 ='M0515'
OR diag_icd10 ='M0516'
OR diag_icd10 ='M0517'
OR diag_icd10 ='M0518'
OR diag_icd10 ='M0519'
OR diag_icd10 ='M052'
OR diag_icd10 ='M0520'
OR diag_icd10 ='M0521'
OR diag_icd10 ='M0522'
OR diag_icd10 ='M0523'
OR diag_icd10 ='M0524'
OR diag_icd10 ='M0525'
OR diag_icd10 ='M0526'
OR diag_icd10 ='M0527'
OR diag_icd10 ='M0528'
OR diag_icd10 ='M0529'
OR diag_icd10 ='M053'
OR diag_icd10 ='M0530'
OR diag_icd10 ='M0531'
OR diag_icd10 ='M0532'
OR diag_icd10 ='M0533'
OR diag_icd10 ='M0534'
OR diag_icd10 ='M0535'
OR diag_icd10 ='M0536'
OR diag_icd10 ='M0537'
OR diag_icd10 ='M0538'
OR diag_icd10 ='M0539'
OR diag_icd10 ='M058'
OR diag_icd10 ='M0580'
OR diag_icd10 ='M0581'
OR diag_icd10 ='M0582'
OR diag_icd10 ='M0583'
OR diag_icd10 ='M0584'
OR diag_icd10 ='M0585'
OR diag_icd10 ='M0586'
OR diag_icd10 ='M0587'
OR diag_icd10 ='M0588'
OR diag_icd10 ='M0589'
OR diag_icd10 ='M059'
OR diag_icd10 ='M0590'
OR diag_icd10 ='M0591'
OR diag_icd10 ='M0592'
OR diag_icd10 ='M0593'
OR diag_icd10 ='M0594'
OR diag_icd10 ='M0595'
OR diag_icd10 ='M0596'
OR diag_icd10 ='M0597'
OR diag_icd10 ='M0598'
OR diag_icd10 ='M0599'
OR diag_icd10 ='M06'
OR diag_icd10 ='M060'
OR diag_icd10 ='M0600'
OR diag_icd10 ='M0601'
OR diag_icd10 ='M0602'
OR diag_icd10 ='M0603'
OR diag_icd10 ='M0604'
OR diag_icd10 ='M0605'
OR diag_icd10 ='M0606'
OR diag_icd10 ='M0607'
OR diag_icd10 ='M0608'
OR diag_icd10 ='M0609'
OR diag_icd10 ='M061'
OR diag_icd10 ='M0610'
OR diag_icd10 ='M0611'
OR diag_icd10 ='M0612'
OR diag_icd10 ='M0613'
OR diag_icd10 ='M0614'
OR diag_icd10 ='M0615'
OR diag_icd10 ='M0616'
OR diag_icd10 ='M0617'
OR diag_icd10 ='M0618'
OR diag_icd10 ='M0619'
OR diag_icd10 ='M062'
OR diag_icd10 ='M0620'
OR diag_icd10 ='M0621'
OR diag_icd10 ='M0622'
OR diag_icd10 ='M0623'
OR diag_icd10 ='M0624'
OR diag_icd10 ='M0625'
OR diag_icd10 ='M0626'
OR diag_icd10 ='M0627'
OR diag_icd10 ='M0628'
OR diag_icd10 ='M0629'
OR diag_icd10 ='M063'
OR diag_icd10 ='M0630'
OR diag_icd10 ='M0631'
OR diag_icd10 ='M0632'
OR diag_icd10 ='M0633'
OR diag_icd10 ='M0634'
OR diag_icd10 ='M0635'
OR diag_icd10 ='M0636'
OR diag_icd10 ='M0637'
OR diag_icd10 ='M0638'
OR diag_icd10 ='M0639'
OR diag_icd10 ='M064'
OR diag_icd10 ='M0640'
OR diag_icd10 ='M0641'
OR diag_icd10 ='M0642'
OR diag_icd10 ='M0643'
OR diag_icd10 ='M0644'
OR diag_icd10 ='M0645'
OR diag_icd10 ='M0646'
OR diag_icd10 ='M0647'
OR diag_icd10 ='M0648'
OR diag_icd10 ='M0649'
OR diag_icd10 ='M068'
OR diag_icd10 ='M0680'
OR diag_icd10 ='M0681'
OR diag_icd10 ='M0682'
OR diag_icd10 ='M0683'
OR diag_icd10 ='M0684'
OR diag_icd10 ='M0685'
OR diag_icd10 ='M0686'
OR diag_icd10 ='M0687'
OR diag_icd10 ='M0688'
OR diag_icd10 ='M0689'
OR diag_icd10 ='M069'
OR diag_icd10 ='M0690'
OR diag_icd10 ='M0691'
OR diag_icd10 ='M0692'
OR diag_icd10 ='M0693'
OR diag_icd10 ='M0694'
OR diag_icd10 ='M0695'
OR diag_icd10 ='M0696'
OR diag_icd10 ='M0697'
OR diag_icd10 ='M0698'
OR diag_icd10 ='M0699'

sql_ra_sec.txt:
  SELECT eid,admidate,disdate,epidur,secondary_diag FROM
(SELECT * FROM
(SELECT eid,record_id,diag_icd10 AS secondary_diag
FROM hesin_diag10
WHERE diag_icd10 ='M05'
OR diag_icd10 ='M050'
OR diag_icd10 ='M0500'
OR diag_icd10 ='M0501'
OR diag_icd10 ='M0502'
OR diag_icd10 ='M0503'
OR diag_icd10 ='M0504'
OR diag_icd10 ='M0505'
OR diag_icd10 ='M0506'
OR diag_icd10 ='M0507'
OR diag_icd10 ='M0508'
OR diag_icd10 ='M0509'
OR diag_icd10 ='M051'
OR diag_icd10 ='M0510'
OR diag_icd10 ='M0511'
OR diag_icd10 ='M0512'
OR diag_icd10 ='M0513'
OR diag_icd10 ='M0514'
OR diag_icd10 ='M0515'
OR diag_icd10 ='M0516'
OR diag_icd10 ='M0517'
OR diag_icd10 ='M0518'
OR diag_icd10 ='M0519'
OR diag_icd10 ='M052'
OR diag_icd10 ='M0520'
OR diag_icd10 ='M0521'
OR diag_icd10 ='M0522'
OR diag_icd10 ='M0523'
OR diag_icd10 ='M0524'
OR diag_icd10 ='M0525'
OR diag_icd10 ='M0526'
OR diag_icd10 ='M0527'
OR diag_icd10 ='M0528'
OR diag_icd10 ='M0529'
OR diag_icd10 ='M053'
OR diag_icd10 ='M0530'
OR diag_icd10 ='M0531'
OR diag_icd10 ='M0532'
OR diag_icd10 ='M0533'
OR diag_icd10 ='M0534'
OR diag_icd10 ='M0535'
OR diag_icd10 ='M0536'
OR diag_icd10 ='M0537'
OR diag_icd10 ='M0538'
OR diag_icd10 ='M0539'
OR diag_icd10 ='M058'
OR diag_icd10 ='M0580'
OR diag_icd10 ='M0581'
OR diag_icd10 ='M0582'
OR diag_icd10 ='M0583'
OR diag_icd10 ='M0584'
OR diag_icd10 ='M0585'
OR diag_icd10 ='M0586'
OR diag_icd10 ='M0587'
OR diag_icd10 ='M0588'
OR diag_icd10 ='M0589'
OR diag_icd10 ='M059'
OR diag_icd10 ='M0590'
OR diag_icd10 ='M0591'
OR diag_icd10 ='M0592'
OR diag_icd10 ='M0593'
OR diag_icd10 ='M0594'
OR diag_icd10 ='M0595'
OR diag_icd10 ='M0596'
OR diag_icd10 ='M0597'
OR diag_icd10 ='M0598'
OR diag_icd10 ='M0599'
OR diag_icd10 ='M06'
OR diag_icd10 ='M060'
OR diag_icd10 ='M0600'
OR diag_icd10 ='M0601'
OR diag_icd10 ='M0602'
OR diag_icd10 ='M0603'
OR diag_icd10 ='M0604'
OR diag_icd10 ='M0605'
OR diag_icd10 ='M0606'
OR diag_icd10 ='M0607'
OR diag_icd10 ='M0608'
OR diag_icd10 ='M0609'
OR diag_icd10 ='M061'
OR diag_icd10 ='M0610'
OR diag_icd10 ='M0611'
OR diag_icd10 ='M0612'
OR diag_icd10 ='M0613'
OR diag_icd10 ='M0614'
OR diag_icd10 ='M0615'
OR diag_icd10 ='M0616'
OR diag_icd10 ='M0617'
OR diag_icd10 ='M0618'
OR diag_icd10 ='M0619'
OR diag_icd10 ='M062'
OR diag_icd10 ='M0620'
OR diag_icd10 ='M0621'
OR diag_icd10 ='M0622'
OR diag_icd10 ='M0623'
OR diag_icd10 ='M0624'
OR diag_icd10 ='M0625'
OR diag_icd10 ='M0626'
OR diag_icd10 ='M0627'
OR diag_icd10 ='M0628'
OR diag_icd10 ='M0629'
OR diag_icd10 ='M063'
OR diag_icd10 ='M0630'
OR diag_icd10 ='M0631'
OR diag_icd10 ='M0632'
OR diag_icd10 ='M0633'
OR diag_icd10 ='M0634'
OR diag_icd10 ='M0635'
OR diag_icd10 ='M0636'
OR diag_icd10 ='M0637'
OR diag_icd10 ='M0638'
OR diag_icd10 ='M0639'
OR diag_icd10 ='M064'
OR diag_icd10 ='M0640'
OR diag_icd10 ='M0641'
OR diag_icd10 ='M0642'
OR diag_icd10 ='M0643'
OR diag_icd10 ='M0644'
OR diag_icd10 ='M0645'
OR diag_icd10 ='M0646'
OR diag_icd10 ='M0647'
OR diag_icd10 ='M0648'
OR diag_icd10 ='M0649'
OR diag_icd10 ='M068'
OR diag_icd10 ='M0680'
OR diag_icd10 ='M0681'
OR diag_icd10 ='M0682'
OR diag_icd10 ='M0683'
OR diag_icd10 ='M0684'
OR diag_icd10 ='M0685'
OR diag_icd10 ='M0686'
OR diag_icd10 ='M0687'
OR diag_icd10 ='M0688'
OR diag_icd10 ='M0689'
OR diag_icd10 ='M069'
OR diag_icd10 ='M0690'
OR diag_icd10 ='M0691'
OR diag_icd10 ='M0692'
OR diag_icd10 ='M0693'
OR diag_icd10 ='M0694'
OR diag_icd10 ='M0695'
OR diag_icd10 ='M0696'
OR diag_icd10 ='M0697'
OR diag_icd10 ='M0698'
OR diag_icd10 ='M0699') AS b
LEFT JOIN hesin AS a ON a.record_id = b.record_id) AS c

# sjogrens syndrome
sql_sjogrens_pri.txt:
  SELECT eid,admidate,disdate,epidur,diag_icd10 AS primary_diag
FROM hesin
WHERE OR diag_icd10 = 'M350'

sql_sjogrens_sec.txt:
  SELECT eid,admidate,disdate,epidur,secondary_diag FROM
(SELECT * FROM
(SELECT eid,record_id,diag_icd10 AS secondary_diag
FROM hesin_diag10
WHERE diag_icd10 = 'M350') AS b
LEFT JOIN hesin AS a ON a.record_id = b.record_id) AS c

# systemic lupus erythematosus
sql_sle_pri.txt:
  SELECT eid,admidate,disdate,epidur,diag_icd10 AS primary_diag
FROM hesin
WHEREOR diag_icd10 = 'L93'
OR diag_icd10 = 'L930'
OR diag_icd10 = 'L931'
OR diag_icd10 = 'L932'
OR diag_icd10 = 'M32'
OR diag_icd10 = 'M320'
OR diag_icd10 = 'M321'
OR diag_icd10 = 'M328'
OR diag_icd10 = 'M329'
OR diag_icd10 = 'M3290'

sql_sle_sec.txt:
  SELECT eid,admidate,disdate,epidur,secondary_diag FROM
(SELECT * FROM
(SELECT eid,record_id,diag_icd10 AS secondary_diag
FROM hesin_diag10
WHERE diag_icd10 = 'L93'
OR diag_icd10 = 'L930'
OR diag_icd10 = 'L931'
OR diag_icd10 = 'L932'
OR diag_icd10 = 'M32'
OR diag_icd10 = 'M320'
OR diag_icd10 = 'M321'
OR diag_icd10 = 'M328'
OR diag_icd10 = 'M329'
OR diag_icd10 = 'M3290') AS b
LEFT JOIN hesin AS a ON a.record_id = b.record_id) AS c

# type 1 diabetes
sql_t1d_pri.txt:
  SELECT eid,admidate,disdate,epidur,diag_icd10 AS primary_diag
FROM hesin
WHERE diag_icd10 = 'E10'
OR diag_icd10 = 'E100'
OR diag_icd10 = 'E101'
OR diag_icd10 = 'E102'
OR diag_icd10 = 'E103'
OR diag_icd10 = 'E104'
OR diag_icd10 = 'E105'
OR diag_icd10 = 'E106'
OR diag_icd10 = 'E107'
OR diag_icd10 = 'E108'
OR diag_icd10 = 'E109'

sql_t1d_sec.txt:
  SELECT eid,admidate,disdate,epidur,secondary_diag FROM
(SELECT * FROM
(SELECT eid,record_id,diag_icd10 AS secondary_diag
FROM hesin_diag10
WHERE diag_icd10 = 'E10'
OR diag_icd10 = 'E100'
OR diag_icd10 = 'E101'
OR diag_icd10 = 'E102'
OR diag_icd10 = 'E103'
OR diag_icd10 = 'E104'
OR diag_icd10 = 'E105'
OR diag_icd10 = 'E106'
OR diag_icd10 = 'E107'
OR diag_icd10 = 'E108'
OR diag_icd10 = 'E109') AS b
LEFT JOIN hesin AS a ON a.record_id = b.record_id) AS c

```

```{r recode_self_report_columns, echo=FALSE, eval=FALSE}

# searched self-report codes for autoimmune diseases on UKB website and made "self_report_codes.txt": 
# 1331	pernicious anaemia	
# 1428	thyroiditis	
# 1522	grave's disease	
# 1222	type 1 diabetes	
# 1261	multiple sclerosis	
# 1437	myasthenia gravis1	
# 1260	myasthenia gravis2	
# 1456	malabsorption/coeliac disease	
# 1462	crohns disease	
# 1463	ulcerative colitis
# 1345	pemphigoid/pemphigus	
# 1453	psoriasis	
# 1313	ankylosing spondylitis	
# 1377	polymyalgia rheumatica	
# 1477	psoriatic arthropathy
# 1464	rheumatoid arthritis	
# 1382	sjogren's syndrome/sicca syndrome	
# 1381	systemic lupus erythematosis/sle

# Recode self-report columns to 0/1 using CARS package in R. 
  
library(CARS)
library(data.table)

# read in data
self_report <- read.table("/file_path/self_report", header=TRUE, stringsAsFactors=FALSE, na.strings=c("NA",""))
coding_self <- as.data.frame(fread("/file_path/self_report_codes.txt", header = F))

# make vector of codes
coding_self <- coding_self[,1]

# make a copy of df of UKB self report 
self_report_copy <- self_report

# make df filled with NA (number of columns = self_report - 1 (no ID column))
temp_self_report_copy=as.data.frame(matrix(NA, ncol=87, nrow=nrow(self_report_copy)))

# populate empty df with 0/1 by recoding the self code in each cell, then sum along the rows to create new column for each diagnosis
for (i in coding_self) {
  temp_self_report_copy[,1:87] <- lapply(self_report_copy[,2:88], function(x)car::recode(x, paste("'",i,"'=1; else=0", sep="")))
  self_report_copy[[paste(i)]] <- rowSums(temp_self_report_copy[,1:87])
}

# print required columns
self_report_copy <- self_report_copy[,c(1,89:ncol(self_report_copy))]

# write tables
write.table(self_report_copy, "/file_path/self_report_autoimmune", row.names=F, col.name=T, quote=F)
  
```

```{r recode_medication_columns, echo=FALSE, eval=FALSE}

# Created lists of medication codes (using "coding4.tsv" from UKB website) corresponding to each autoimmune disease. 
# Recoded self-reported medications columns to 0/1 for each list of medications corresponding to each autoimmune disease. Example of recoding script below for ankylosing spondylitis
  
library(CARS)
library(data.table)

# read in data
medications <- read.table("/file_path/medications", header=TRUE, stringsAsFactors=FALSE, na.strings=c("NA",""))
ankylosing_codes <- as.data.frame(fread("/file_path/new_med_ankylosing.txt", header = F))

# make vector of codes
ankylosing_codes <- ankylosing_codes[,1]

# make a copy of df of UKB medications
medications_copy <- medications

# make df filled with NA (number of columns = self_report - 1 (no ID column))
temp_medications_copy=as.data.frame(matrix(NA, ncol=144, nrow=nrow(medications_copy)))

# populate empty df with 0/1 by recoding the medication code in each cell, then sum along the rows to create new column for each medication
for (i in ankylosing_codes) {
  temp_medications_copy[,1:144] <- lapply(medications_copy[,2:145], function(x)car::recode(x, paste("'",i,"'=1; else=0", sep="")))
  medications_copy[[paste(i)]] <- rowSums(temp_medications_copy[,1:144])
}

# print required columns
medications_copy <- medications_copy[,c(1,146:ncol(medications_copy))]

# write tables
write.table(medications_copy, "/file_path/meds_new_ankylosing", row.names=F, col.name=T, quote=F)

```


```{r download_icd_self_report_medications_coding, echo=FALSE}

# downloaded ICD coding from UKB website
# Description: ICD10 - WHO International Classification of Diseases = "coding10.tsv"

coding_icd10 <- as.data.frame(fread("/file_path/coding19.tsv", header = T))
coding_icd10 <- coding_icd10[,1:2]

# self-report UKB codes
my_codes <- as.data.frame(fread("/file_path/self_report_codes.txt", header = F))

# downloaded medication coding from UKB website "coding4.tsv"
coding_meds <- as.data.frame(fread("/file_path/coding4.tsv", header = T))

```

```{r download_qc_ids, echo=FALSE} 

# read in participants IDs passing genetic QC
qc_ids <- read.table("/file_path/post_qc_id_list.fam", header = F) 
qc_ids <- qc_ids %>% 
  mutate(eid = coalesce(as.integer(V1)))

```

```{r download_hes_data, echo=FALSE, message=FALSE} 

# read in output from HES portal - with primary and secondaary diagnoses for each disorder in participants

# ankylosing spondylitis
as_pri <- read_tsv("/file_path/as_pri_hes.tsv")
as_sec <- read_tsv("/file_path/as_sec_hes.tsv")

# autoimmune thyroid disease
at_pri <- read_tsv("/file_path/at_pri_hes.tsv")
at_sec <- read_tsv("/file_path/at_sec_hes.tsv")

# coeliac disease
coeliac_pri <- read_tsv("/file_path/coeliac_pri_hes.tsv")
coeliac_sec <- read_tsv("/file_path/coeliac_sec_hes.tsv")

# hidradenitis suppurativa
hidradentis_pri <- read_tsv("/file_path/hidradentis_pri_hes.tsv")
hidradentis_sec <- read_tsv("/file_path/hidradentis_sec_hes.tsv")

# inflammatory bowel disease
ibd_pri <- read_tsv("/file_path/ibd_pri_hes.tsv")
ibd_sec <- read_tsv("/file_path/ibd_sec_hes.tsv")

# myasthenia gravis
mg_pri <- read_tsv("/file_path/mg_pri_hes.tsv")
mg_sec <- read_tsv("/file_path/mg_sec_hes.tsv")

# multiple sclerosis
ms_pri <- read_tsv("/file_path/ms_pri_hes.tsv")
ms_sec <- read_tsv("/file_path/ms_sec_hes.tsv")

# pernicious anaemia
pa_pri <- read_tsv("/file_path/pa_pri_hes.tsv")
pa_sec <- read_tsv("/file_path/pa_sec_hes.tsv")

# pemphigoid, pemphigus
pemphigoid_gus_pri <- read_tsv("/file_path/pemphigoid_gus_pri_hes.tsv")
pemphigoid_gus_sec <- read_tsv("/file_path/pemphigoid_gus_sec_hes.tsv")

# polymyalgia rheumatica
pr_pri <- read_tsv("/file_path/pr_pri_hes.tsv")
pr_sec <- read_tsv("/file_path/pr_sec_hes.tsv")

# psoriatic arthritis
psa_pri <- read_tsv("/file_path/psa_pri_hes.tsv")
psa_sec <- read_tsv("/file_path/psa_sec_hes.tsv")

# psoriasis
psoriasis_pri <- read_tsv("/file_path/psoriasis_pri_hes.tsv")
psoriasis_sec <- read_tsv("/file_path/psoriasis_sec_hes.tsv")

# rheumatoid arthritis
ra_pri <- read_tsv("/file_path/ra_pri_hes.tsv")
ra_sec <- read_tsv("/file_path/ra_sec_hes.tsv")

# sjogrens syndrome
sjogrens_pri <- read_tsv("/file_path/sjogrens_pri_hes.tsv")
sjogrens_sec <- read_tsv("/file_path/sjogrens_sec_hes.tsv")

# systemic lupus erythematosus
sle_pri <- read_tsv("/file_path/sle_pri_hes.tsv")
sle_sec <- read_tsv("/file_path/sle_sec_hes.tsv")

# type 1 diabetes
t1d_pri <- read_tsv("/file_path/t1d_pri_hes.tsv")
t1d_sec <- read_tsv("/file_path/t1d_sec_hes.tsv")

```

```{r download_self_report_data, echo=FALSE}

# read in self-report participant data
self_report <- fread("/file_path/self_report_autoimmune.txt", header = T) 

```

```{r download_medication_data, include=FALSE}

# read in medication data for participants

meds_ankylosing <- as.data.frame(fread("/file_path/meds_new_ankylosing", header = T))
meds_arthritis <- as.data.frame(fread("/file_path/meds_new_arthritis", header = T))  
meds_crohn <- as.data.frame(fread("/file_path/meds_new_crohn", header = T)) # meds for IBD     
meds_diabetes <- as.data.frame(fread("/file_path/meds_new_diabetes", header = T))   
meds_hidradenitis <- as.data.frame(fread("/file_path/meds_new_hidradenitis", header = T))
meds_lupus <- as.data.frame(fread("/file_path/meds_new_lupus", header = T))      
meds_myasthenia <- as.data.frame(fread("/file_path/meds_new_myasthenia", header = T)) 
meds_pemphigoid <- as.data.frame(fread("/file_path/meds_new_pemphigoid", header = T)) 
meds_polymyalgia <- as.data.frame(fread("/file_path/meds_new_polymyalgia", header = T))
meds_psoriasis <- as.data.frame(fread("/file_path/meds_new_psoriasis", header = T))  
meds_psoriatic <- as.data.frame(fread("/file_path/meds_new_psoriatic", header = T))  
meds_sclerosis <- as.data.frame(fread("/file_path/meds_new_sclerosis", header = T))  
meds_sjogren <- as.data.frame(fread("/file_path/meds_new_sjogren", header = T))    
meds_thyroid <- as.data.frame(fread("/file_path/meds_new_thyroid", header = T))    
meds_ulcerative <- as.data.frame(fread("/file_path/meds_new_ulcerative", header = T)) 

```

```{r download_demographics, echo=FALSE}

# read in demographic data for participants
demographics <- as.data.frame(fread("/file_path/demographics", header = T))

```


```{r qc_demographics, echo=FALSE}

# print required columns
demographics <- demographics[,c(1:5,8)]
colnames(demographics) <- c("eid","Age","Sex","SES","BMI","Current Smoker")

# recode smoking: change to 1 = current, 0 = not current
# Smoking Coding
# -3	Prefer not to answer
# 0	Never
# 1	Previous
# 2	Current

demographics$`Current Smoker`[demographics$`Current Smoker`==-3 | demographics$`Current Smoker`==1] <- 0
demographics$`Current Smoker`[demographics$`Current Smoker`==2] <- 1

# filter on participant IDs after genetic QC
demographics <- left_join(qc_ids, demographics, by = "eid") %>% 
  dplyr::select(-(V1:V6)) 

```

```{r qc_hes_data, echo=FALSE}

# Step 1: remove duplicated lines. Code removes duplicate lines where all cells are identical 
# Step 2: group by participant ID and remove lines where the admission date is repeated for the same ICD code. This will retain lines where >1 primary dx has been assigned under the same admission. 
# create QC function for steps 1 and 2 and run over dataframes for each disease, first for primary then secondary

qc_hes_pri <- function(x) {
  dplyr::distinct(x, .keep_all = TRUE) %>% 
  group_by(eid) %>% 
  dplyr::distinct(admidate, primary_diag, .keep_all = TRUE)}
  
as_pri_qc1 <- qc_hes_pri(as_pri)
at_pri_qc1 <- qc_hes_pri(at_pri)
coeliac_pri_qc1 <- qc_hes_pri(coeliac_pri)
hidradentis_pri_qc1 <- qc_hes_pri(hidradentis_pri)
ibd_pri_qc1 <- qc_hes_pri(ibd_pri)
mg_pri_qc1 <- qc_hes_pri(mg_pri)
ms_pri_qc1 <- qc_hes_pri(ms_pri)
pa_pri_qc1 <- qc_hes_pri(pa_pri)
pemphigoid_gus_pri_qc1 <- qc_hes_pri(pemphigoid_gus_pri)
pr_pri_qc1 <- qc_hes_pri(pr_pri)
psa_pri_qc1 <- qc_hes_pri(psa_pri)
psoriasis_pri_qc1 <- qc_hes_pri(psoriasis_pri)
ra_pri_qc1 <- qc_hes_pri(ra_pri)
sjogrens_pri_qc1 <- qc_hes_pri(sjogrens_pri)
sle_pri_qc1 <- qc_hes_pri(sle_pri)
t1d_pri_qc1 <- qc_hes_pri(t1d_pri)

qc_hes_sec <- function(x) {
  dplyr::distinct(x, .keep_all = TRUE) %>% 
  group_by(eid) %>% 
  dplyr::distinct(admidate, secondary_diag, .keep_all = TRUE)}

as_sec_qc1 <- qc_hes_sec(as_sec)
at_sec_qc1 <- qc_hes_sec(at_sec)
coeliac_sec_qc1 <- qc_hes_sec(coeliac_sec)
hidradentis_sec_qc1 <- qc_hes_sec(hidradentis_sec)
ibd_sec_qc1 <- qc_hes_sec(ibd_sec)
mg_sec_qc1 <- qc_hes_sec(mg_sec)
ms_sec_qc1 <- qc_hes_sec(ms_sec)
pa_sec_qc1 <- qc_hes_sec(pa_sec)
pemphigoid_gus_sec_qc1 <- qc_hes_sec(pemphigoid_gus_sec)
pr_sec_qc1 <- qc_hes_sec(pr_sec)
psa_sec_qc1 <- qc_hes_sec(psa_sec)
psoriasis_sec_qc1 <- qc_hes_sec(psoriasis_sec)
ra_sec_qc1 <- qc_hes_sec(ra_sec)
sjogrens_sec_qc1 <- qc_hes_sec(sjogrens_sec)
sle_sec_qc1 <- qc_hes_sec(sle_sec)
t1d_sec_qc1 <- qc_hes_sec(t1d_sec)

```

```{r format_hes_data_and_filter_on_qc_ids, echo=FALSE}

# create function to put hes data into column format; count frequency of ICD code assigned to each participant. Make columns for each ICD code and rows for each participant ID with count of frequency per ICD code column.  

# for primary diagnoses
column_format_pri <- function(x) {x %>% 
  group_by(eid) %>%
  count(primary_diag) %>%
  spread(key=primary_diag,value=n) %>%
  replace(., is.na(.), 0)
}

as_pri_format<-column_format_pri(as_pri_qc1)
at_pri_format<-column_format_pri(at_pri_qc1)
coeliac_pri_format<-column_format_pri(coeliac_pri_qc1)
hidradentis_pri_format<-column_format_pri(hidradentis_pri_qc1)
ibd_pri_format<-column_format_pri(ibd_pri_qc1)
mg_pri_format<-column_format_pri(mg_pri_qc1)
ms_pri_format<-column_format_pri(ms_pri_qc1)
pa_pri_format<-column_format_pri(pa_pri_qc1)
pemphigoid_gus_pri_format<-column_format_pri(pemphigoid_gus_pri_qc1)
pr_pri_format<-column_format_pri(pr_pri_qc1)
psa_pri_format<-column_format_pri(psa_pri_qc1)
psoriasis_pri_format<-column_format_pri(psoriasis_pri_qc1)
ra_pri_format<-column_format_pri(ra_pri_qc1)
sjogrens_pri_format<-column_format_pri(sjogrens_pri_qc1)
sle_pri_format<-column_format_pri(sle_pri_qc1)
# remove M32.0 from sle
sle_pri_format <- sle_pri_format[,-5]
t1d_pri_format<-column_format_pri(t1d_pri_qc1)

# for secondary diagnoses
column_format_sec <- function(x) {x %>% 
  group_by(eid) %>%
  count(secondary_diag) %>%
  spread(key=secondary_diag,value=n) %>%
  replace(., is.na(.), 0)
}

as_sec_format<-column_format_sec(as_sec_qc1)
at_sec_format<-column_format_sec(at_sec_qc1)
coeliac_sec_format<-column_format_sec(coeliac_sec_qc1)
hidradentis_sec_format<-column_format_sec(hidradentis_sec_qc1)
ibd_sec_format<-column_format_sec(ibd_sec_qc1)
mg_sec_format<-column_format_sec(mg_sec_qc1)
ms_sec_format<-column_format_sec(ms_sec_qc1)
pa_sec_format<-column_format_sec(pa_sec_qc1)
pemphigoid_gus_sec_format<-column_format_sec(pemphigoid_gus_sec_qc1)
pr_sec_format<-column_format_sec(pr_sec_qc1)
psa_sec_format<-column_format_sec(psa_sec_qc1)
psoriasis_sec_format<-column_format_sec(psoriasis_sec_qc1)
ra_sec_format<-column_format_sec(ra_sec_qc1)
sjogrens_sec_format<-column_format_sec(sjogrens_sec_qc1)
sle_sec_format<-column_format_sec(sle_sec_qc1)
# remove M32.0 from sle
sle_sec_format <- sle_sec_format[,-5]
t1d_sec_format<-column_format_sec(t1d_sec_qc1)

# merge formatted dataframes with IDs passing genetic qc. 

merge_qc <- function(x) {
  left_join(qc_ids, x, by = "eid") %>% 
  dplyr::select(-(V1:V6)) %>% 
  replace(., is.na(.), 0)
}

as_pri_format_qc2 <- merge_qc(as_pri_format)
at_pri_format_qc2 <- merge_qc(at_pri_format)
coeliac_pri_format_qc2 <- merge_qc(coeliac_pri_format)
hidradentis_pri_format_qc2 <- merge_qc(hidradentis_pri_format)
ibd_pri_format_qc2 <- merge_qc(ibd_pri_format)
mg_pri_format_qc2 <- merge_qc(mg_pri_format)
ms_pri_format_qc2 <- merge_qc(ms_pri_format)
pa_pri_format_qc2 <- merge_qc(pa_pri_format)
pemphigoid_gus_pri_format_qc2 <- merge_qc(pemphigoid_gus_pri_format)
pr_pri_format_qc2 <- merge_qc(pr_pri_format)
psa_pri_format_qc2 <- merge_qc(psa_pri_format)
psoriasis_pri_format_qc2 <- merge_qc(psoriasis_pri_format)
ra_pri_format_qc2 <- merge_qc(ra_pri_format)
sjogrens_pri_format_qc2 <- merge_qc(sjogrens_pri_format)
sle_pri_format_qc2 <- merge_qc(sle_pri_format)
t1d_pri_format_qc2 <- merge_qc(t1d_pri_format)

as_sec_format_qc2 <- merge_qc(as_sec_format)
at_sec_format_qc2 <- merge_qc(at_sec_format)
coeliac_sec_format_qc2 <- merge_qc(coeliac_sec_format)
hidradentis_sec_format_qc2 <- merge_qc(hidradentis_sec_format)
ibd_sec_format_qc2 <- merge_qc(ibd_sec_format)
mg_sec_format_qc2 <- merge_qc(mg_sec_format)
ms_sec_format_qc2 <- merge_qc(ms_sec_format)
pa_sec_format_qc2 <- merge_qc(pa_sec_format)
pemphigoid_gus_sec_format_qc2 <- merge_qc(pemphigoid_gus_sec_format)
pr_sec_format_qc2 <- merge_qc(pr_sec_format)
psa_sec_format_qc2 <- merge_qc(psa_sec_format)
psoriasis_sec_format_qc2 <- merge_qc(psoriasis_sec_format)
ra_sec_format_qc2 <- merge_qc(ra_sec_format)
sjogrens_sec_format_qc2 <- merge_qc(sjogrens_sec_format)
sle_sec_format_qc2 <- merge_qc(sle_sec_format)
t1d_sec_format_qc2 <- merge_qc(t1d_sec_format)

```

```{r hes_dx_count_pp, include=FALSE}

# make function to count primary and secondary dx for each individual across all ICD codes pertaining to each autoimmune disease
count_dx_pp <- function(x,y) {
  data.frame("X1" = x[,1], "X2" = rowSums(x[,2:ncol(x)]), "X3"= rowSums(y[,2:ncol(y)])) %>% 
  rename("ID" = X1, "Primary count" = X2, "Secondary count" = X3)  
}

# run function across primary and secondary columns for disorders with multiple ICD codes
as_count_pp <- count_dx_pp(as_pri_format_qc2,as_sec_format_qc2)
at_count_pp <- count_dx_pp(at_pri_format_qc2,at_sec_format_qc2)
ibd_count_pp <- count_dx_pp(ibd_pri_format_qc2,ibd_sec_format_qc2)
pemphigoid_gus_count_pp <- count_dx_pp(pemphigoid_gus_pri_format_qc2,pemphigoid_gus_sec_format_qc2)
pr_count_pp <- count_dx_pp(pr_pri_format_qc2,pr_sec_format_qc2)
psa_count_pp <- count_dx_pp(psa_pri_format_qc2,psa_sec_format_qc2)
psoriasis_count_pp <- count_dx_pp(psoriasis_pri_format_qc2,psoriasis_sec_format_qc2)
ra_count_pp <- count_dx_pp(ra_pri_format_qc2,ra_sec_format_qc2)
sle_count_pp <- count_dx_pp(sle_pri_format_qc2,sle_sec_format_qc2)
t1d_count_pp <- count_dx_pp(t1d_pri_format_qc2,t1d_sec_format_qc2)

# make function to merge primary and secondary columns for single code disorders 
count_dx_pp_single <- function(x,y) {
  data.frame("X1" = x[,1], "X2" = x[,2], "X3"= y[,2]) %>% 
  rename("ID" = X1, "Primary count" = X2, "Secondary count" = X3)  
}

# merge primary and secondary columns for single code disorders 
coeliac_count_pp <- count_dx_pp_single(coeliac_pri_format_qc2,coeliac_sec_format_qc2)
hidradentis_count_pp <- count_dx_pp_single(hidradentis_pri_format_qc2,hidradentis_sec_format_qc2)
mg_count_pp <- count_dx_pp_single(mg_pri_format_qc2,mg_sec_format_qc2)
ms_count_pp <- count_dx_pp_single(ms_pri_format_qc2,ms_sec_format_qc2)
pa_count_pp <- count_dx_pp_single(pa_pri_format_qc2,pa_sec_format_qc2)
sjogrens_count_pp <- count_dx_pp_single(sjogrens_pri_format_qc2,sjogrens_sec_format_qc2)

```

```{r hes_dx_count_across_people_for_subcode_table, include=FALSE}

# make function to count dx across participants to create summary table of number of participants with at least one diagnosis per ICD code
count_dx <- function(x) {
  data.frame("x1" = names(x[,-1]), "x2" = colSums(x[,-1] > 0), stringsAsFactors = FALSE, row.names = NULL)
}

# make function to merge ICD codes with the disorder name of each code
merge_dx <- function(x) {
  merge(coding_icd10, x, by.x = "coding", by.y = "x1")
}

# create primary and secondary summary tables for multiple icd subcodes
as_dx_pri <- count_dx(as_pri_format_qc2) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning,"Primary dx" = x2)
at_dx_pri <- count_dx(at_pri_format_qc2) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning,"Primary dx" = x2)
ibd_dx_pri <- count_dx(ibd_pri_format_qc2) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning,"Primary dx" = x2)
pemphigoid_gus_dx_pri <- count_dx(pemphigoid_gus_pri_format_qc2) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning,"Primary dx" = x2)
pr_dx_pri <- count_dx(pr_pri_format_qc2) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning,"Primary dx" = x2)
psa_dx_pri <- count_dx(psa_pri_format_qc2) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning,"Primary dx" = x2)
psoriasis_dx_pri <- count_dx(psoriasis_pri_format_qc2) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning,"Primary dx" = x2)
ra_dx_pri <- count_dx(ra_pri_format_qc2) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning,"Primary dx" = x2)
sle_dx_pri <- count_dx(sle_pri_format_qc2) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning,"Primary dx" = x2)
t1d_dx_pri <- count_dx(t1d_pri_format_qc2) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning,"Primary dx" = x2)

as_dx_sec <- count_dx(as_sec_format_qc2) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning, "Secondary dx" = x2)
at_dx_sec <- count_dx(at_sec_format_qc2) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning, "Secondary dx" = x2)
ibd_dx_sec <- count_dx(ibd_sec_format_qc2) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning, "Secondary dx" = x2)
pemphigoid_gus_dx_sec <- count_dx(pemphigoid_gus_sec_format_qc2) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning, "Secondary dx" = x2)
pr_dx_sec <- count_dx(pr_sec_format_qc2) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning, "Secondary dx" = x2)
psa_dx_sec <- count_dx(psa_sec_format_qc2) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning, "Secondary dx" = x2)
psoriasis_dx_sec <- count_dx(psoriasis_sec_format_qc2) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning, "Secondary dx" = x2)
ra_dx_sec <- count_dx(ra_sec_format_qc2) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning, "Secondary dx" = x2)
sle_dx_sec <- count_dx(sle_sec_format_qc2) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning, "Secondary dx" = x2)
t1d_dx_sec <- count_dx(t1d_sec_format_qc2) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes" = meaning, "Secondary dx" = x2)

# create primary and secondary tables single icd codes
# first create copy of df with dummy column
dummy_col <- function(x) {
  x[,c(1:2,2)]
}

coeliac_pri_format_qc2_dummy <- dummy_col(coeliac_pri_format_qc2)
hidradentis_pri_format_qc2_dummy <- dummy_col(hidradentis_pri_format_qc2)
mg_pri_format_qc2_dummy <- dummy_col(mg_pri_format_qc2)
ms_pri_format_qc2_dummy <- dummy_col(ms_pri_format_qc2)
pa_pri_format_qc2_dummy <- dummy_col(pa_pri_format_qc2)
sjogrens_pri_format_qc2_dummy <- dummy_col(sjogrens_pri_format_qc2)

coeliac_sec_format_qc2_dummy <- dummy_col(coeliac_sec_format_qc2)
hidradentis_sec_format_qc2_dummy <- dummy_col(hidradentis_sec_format_qc2)
mg_sec_format_qc2_dummy <- dummy_col(mg_sec_format_qc2)
ms_sec_format_qc2_dummy <- dummy_col(ms_sec_format_qc2)
pa_sec_format_qc2_dummy <- dummy_col(pa_sec_format_qc2)
sjogrens_sec_format_qc2_dummy <- dummy_col(sjogrens_sec_format_qc2)

# run function to create primary and secondary tables for single icd codes 
coeliac_dx_pri <- count_dx(coeliac_pri_format_qc2_dummy) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes"=meaning,"Primary dx"=x2)
hidradentis_dx_pri <- count_dx(hidradentis_pri_format_qc2_dummy) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes"=meaning,"Primary dx"=x2)
mg_dx_pri <- count_dx(mg_pri_format_qc2_dummy) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes"=meaning,"Primary dx"=x2)
ms_dx_pri <- count_dx(ms_pri_format_qc2_dummy) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes"=meaning,"Primary dx"=x2)
pa_dx_pri <- count_dx(pa_pri_format_qc2_dummy) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes"=meaning,"Primary dx"=x2)
sjogrens_dx_pri <- count_dx(sjogrens_pri_format_qc2_dummy) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes"=meaning,"Primary dx"=x2)

coeliac_dx_sec <- count_dx(coeliac_sec_format_qc2_dummy) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes"=meaning,"Secondary dx"=x2)
hidradentis_dx_sec <- count_dx(hidradentis_sec_format_qc2_dummy) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes"=meaning,"Secondary dx"=x2)
mg_dx_sec <- count_dx(mg_sec_format_qc2_dummy) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes"=meaning,"Secondary dx"=x2)
ms_dx_sec <- count_dx(ms_sec_format_qc2_dummy) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes"=meaning,"Secondary dx"=x2)
pa_dx_sec <- count_dx(pa_sec_format_qc2_dummy) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes"=meaning,"Secondary dx"=x2)
sjogrens_dx_sec <- count_dx(sjogrens_sec_format_qc2_dummy) %>% merge_dx() %>% dplyr::select("ICD-10 sub-codes"=meaning,"Secondary dx"=x2)

# make function to merge primary and secondary columns across diseases
merge_pri_sec <- function(x,y) {
  merge(x, y, by = "ICD-10 sub-codes", all = TRUE) %>% 
  replace(., is.na(.), 0)
}

as_subcodes <- merge_pri_sec(as_dx_pri,as_dx_sec)
at_subcodes <- merge_pri_sec(at_dx_pri,at_dx_sec)
ibd_subcodes <- merge_pri_sec(ibd_dx_pri,ibd_dx_sec)
pemphigoid_gus_subcodes <- merge_pri_sec(pemphigoid_gus_dx_pri,pemphigoid_gus_dx_sec)
pr_subcodes <- merge_pri_sec(pr_dx_pri,pr_dx_sec)
psa_subcodes <- merge_pri_sec(psa_dx_pri,psa_dx_sec)
psoriasis_subcodes <- merge_pri_sec(psoriasis_dx_pri,psoriasis_dx_sec)
ra_subcodes <- merge_pri_sec(ra_dx_pri,ra_dx_sec)
sle_subcodes <- merge_pri_sec(sle_dx_pri,sle_dx_sec)
t1d_subcodes <- merge_pri_sec(t1d_dx_pri,t1d_dx_sec)
coeliac_subcodes <- merge_pri_sec(coeliac_dx_pri,coeliac_dx_sec)
hidradentis_subcodes <- merge_pri_sec(hidradentis_dx_pri,hidradentis_dx_sec)
mg_subcodes <- merge_pri_sec(mg_dx_pri,mg_dx_sec)
ms_subcodes <- merge_pri_sec(ms_dx_pri,ms_dx_sec)
pa_subcodes <- merge_pri_sec(pa_dx_pri,pa_dx_sec)
sjogrens_subcodes <- merge_pri_sec(sjogrens_dx_pri,sjogrens_dx_sec)

```

```{r self_report_count_pp_filter_on_qc_ids, echo=FALSE}

# filter on participant ids passing genetic qc
self_report <- left_join(qc_ids, self_report, by = c("eid" = "f.eid")) %>% 
  dplyr::select(-(V1:V6)) %>% 
  replace(., is.na(.), 0)

# add column names - replace ukb code with name of disorder
colnames(self_report) <- c("ID",my_codes$V2)

# combine myasthenia gravis into one column (this was assigned two codes by ukb for same disordr)
self_report$`myasthenia gravis` <- self_report$`myasthenia gravis1`+self_report$`myasthenia gravis2`
self_report <- self_report[,c(1:6,20,9:19)]

# make a separate dataframe for each disorder and collapse columns where necessary (e.g. combine crohn and ulcerative colitis into inflammatory bowel disease)

pa_self<-self_report[,c(1,2)]
colnames(pa_self) <- c("ID","Self-report")

coeliac_self<-self_report[,c(1,8)]
colnames(coeliac_self) <- c("ID","Self-report")

at_self<-self_report[,c(1,3:4)]
at_self$combined <- at_self$thyroiditis+at_self$`grave's disease`
at_self <- at_self[,c(1,4)]
colnames(at_self) <- c("ID","Self-report")

t1d_self<-self_report[,c(1,5)]
colnames(t1d_self) <- c("ID","Self-report")

ms_self<-self_report[,c(1,6)]
colnames(ms_self) <- c("ID","Self-report")

mg_self<-self_report[,c(1,7)]
colnames(mg_self) <- c("ID","Self-report")

ibd_self<-self_report[,c(1,9:10)]
ibd_self$combined <- ibd_self$`crohns disease`+ibd_self$`ulcerative colitis`
ibd_self <- ibd_self[,c(1,4)]
colnames(ibd_self) <- c("ID","Self-report")

pemphigoid_gus_self<-self_report[,c(1,11)]
colnames(pemphigoid_gus_self) <- c("ID","Self-report")

psoriasis_self<-self_report[,c(1,12)]
colnames(psoriasis_self) <- c("ID","Self-report")

as_self<-self_report[,c(1,13)]
colnames(as_self) <- c("ID","Self-report")

pr_self<-self_report[,c(1,14)]
colnames(pr_self) <- c("ID","Self-report")

psa_self<-self_report[,c(1,15)]
colnames(psa_self) <- c("ID","Self-report")

ra_self<-self_report[,c(1,16)]
colnames(ra_self) <- c("ID","Self-report")

sjogrens_self<-self_report[,c(1,17)]
colnames(sjogrens_self) <- c("ID","Self-report")

sle_self<-self_report[,c(1,18)]
colnames(sle_self) <- c("ID","Self-report")

```

```{r make_self_report_summary, echo=FALSE}

# make summary table showing count of participants who endorse each disorder
summary_self_report <- data.frame("x1" = names(self_report[,-1]), "x2" = as.numeric(format(colSums(self_report[,-1] > 0), scientific = F)), stringsAsFactors = F, row.names = NULL) %>% 
  rename("temp" = x1, "Number of people" = x2) 

summary_self_report$'Self-reported disorder' <- capitalize(summary_self_report$temp)
summary_self_report <- summary_self_report[,c(3,2)]

```

```{r medications_count_pp_filter_on_qc_ids, echo=FALSE, include=FALSE}

# change medication code to medication name
(names( meds_ankylosing ) <- coding_meds[ match( 
names( meds_ankylosing ) , coding_meds[ , 'coding' ] ) , 'meaning' ])
colnames(meds_ankylosing) [1]  <- "f.eid"

(names( meds_arthritis ) <- coding_meds[ match( 
names( meds_arthritis ) , coding_meds[ , 'coding' ] ) , 'meaning' ])
colnames(meds_arthritis) [1]  <- "f.eid"

(names( meds_crohn ) <- coding_meds[ match( 
names( meds_crohn ) , coding_meds[ , 'coding' ] ) , 'meaning' ])
colnames(meds_crohn) [1]  <- "f.eid"

(names( meds_diabetes ) <- coding_meds[ match( 
names( meds_diabetes ) , coding_meds[ , 'coding' ] ) , 'meaning' ])
colnames(meds_diabetes) [1]  <- "f.eid"

(names( meds_hidradenitis ) <- coding_meds[ match( 
names( meds_hidradenitis ) , coding_meds[ , 'coding' ] ) , 'meaning' ])
colnames(meds_hidradenitis) [1]  <- "f.eid"

(names( meds_lupus ) <- coding_meds[ match( 
names( meds_lupus ) , coding_meds[ , 'coding' ] ) , 'meaning' ])
colnames(meds_lupus) [1]  <- "f.eid"

(names( meds_myasthenia ) <- coding_meds[ match( 
names( meds_myasthenia ) , coding_meds[ , 'coding' ] ) , 'meaning' ])
colnames(meds_myasthenia) [1]  <- "f.eid"

(names( meds_pemphigoid ) <- coding_meds[ match( 
names( meds_pemphigoid ) , coding_meds[ , 'coding' ] ) , 'meaning' ])
colnames(meds_pemphigoid) [1]  <- "f.eid"

(names( meds_polymyalgia ) <- coding_meds[ match( 
names( meds_polymyalgia ) , coding_meds[ , 'coding' ] ) , 'meaning' ])
colnames(meds_polymyalgia) [1]  <- "f.eid"

(names( meds_psoriasis ) <- coding_meds[ match( 
names( meds_psoriasis ) , coding_meds[ , 'coding' ] ) , 'meaning' ])
colnames(meds_psoriasis) [1]  <- "f.eid"

(names( meds_psoriatic ) <- coding_meds[ match( 
names( meds_psoriatic ) , coding_meds[ , 'coding' ] ) , 'meaning' ])
colnames(meds_psoriatic) [1]  <- "f.eid"

(names( meds_sclerosis ) <- coding_meds[ match( 
names( meds_sclerosis ) , coding_meds[ , 'coding' ] ) , 'meaning' ])
colnames(meds_sclerosis) [1]  <- "f.eid"

(names( meds_sjogren ) <- coding_meds[ match( 
names( meds_sjogren ) , coding_meds[ , 'coding' ] ) , 'meaning' ])
colnames(meds_sjogren) [1]  <- "f.eid"

(names( meds_thyroid ) <- coding_meds[ match( 
names( meds_thyroid ) , coding_meds[ , 'coding' ] ) , 'meaning' ])
colnames(meds_thyroid) [1]  <- "f.eid"

(names( meds_ulcerative ) <- coding_meds[ match( 
names( meds_ulcerative ) , coding_meds[ , 'coding' ] ) , 'meaning' ])
colnames(meds_ulcerative) [1]  <- "f.eid"

# filter on participants passing genetic qc
meds_ankylosing <- left_join(qc_ids, meds_ankylosing, by = c("eid" = "f.eid")) %>% 
  dplyr::select(-(V1:V6)) %>% 
  replace(., is.na(.), 0)

meds_arthritis <- left_join(qc_ids, meds_arthritis, by = c("eid" = "f.eid")) %>% 
  dplyr::select(-(V1:V6)) %>% 
  replace(., is.na(.), 0)

meds_crohn <- left_join(qc_ids, meds_crohn, by = c("eid" = "f.eid")) %>% 
  dplyr::select(-(V1:V6)) %>% 
  replace(., is.na(.), 0)

meds_diabetes <- left_join(qc_ids, meds_diabetes, by = c("eid" = "f.eid")) %>% 
  dplyr::select(-(V1:V6)) %>% 
  replace(., is.na(.), 0)

meds_hidradenitis <- left_join(qc_ids, meds_hidradenitis, by = c("eid" = "f.eid")) %>% 
  dplyr::select(-(V1:V6)) %>% 
  replace(., is.na(.), 0)

meds_lupus <- left_join(qc_ids, meds_lupus, by = c("eid" = "f.eid")) %>% 
  dplyr::select(-(V1:V6)) %>% 
  replace(., is.na(.), 0)

meds_myasthenia <- left_join(qc_ids, meds_myasthenia, by = c("eid" = "f.eid")) %>% 
  dplyr::select(-(V1:V6)) %>% 
  replace(., is.na(.), 0)

meds_pemphigoid <- left_join(qc_ids, meds_pemphigoid, by = c("eid" = "f.eid")) %>% 
  dplyr::select(-(V1:V6)) %>% 
  replace(., is.na(.), 0)

meds_polymyalgia <- left_join(qc_ids, meds_polymyalgia, by = c("eid" = "f.eid")) %>% 
  dplyr::select(-(V1:V6)) %>% 
  replace(., is.na(.), 0)

meds_psoriasis <- left_join(qc_ids, meds_psoriasis, by = c("eid" = "f.eid")) %>% 
  dplyr::select(-(V1:V6)) %>% 
  replace(., is.na(.), 0)

meds_psoriatic <- left_join(qc_ids, meds_psoriatic, by = c("eid" = "f.eid")) %>% 
  dplyr::select(-(V1:V6)) %>% 
  replace(., is.na(.), 0)

meds_sclerosis <- left_join(qc_ids, meds_sclerosis, by = c("eid" = "f.eid")) %>% 
  dplyr::select(-(V1:V6)) %>% 
  replace(., is.na(.), 0)

meds_sjogren <- left_join(qc_ids, meds_sjogren, by = c("eid" = "f.eid")) %>% 
  dplyr::select(-(V1:V6)) %>% 
  replace(., is.na(.), 0)

meds_thyroid <- left_join(qc_ids, meds_thyroid, by = c("eid" = "f.eid")) %>% 
  dplyr::select(-(V1:V6)) %>% 
  replace(., is.na(.), 0)

meds_ulcerative <- left_join(qc_ids, meds_ulcerative, by = c("eid" = "f.eid")) %>% 
  dplyr::select(-(V1:V6)) %>% 
  replace(., is.na(.), 0)

# count number of medications endorsed by each participant (collapse medication classes into one column)
count_meds_pp <- function(x) {
  data.frame("X1" = x[,1], "X2" = rowSums(x[,2:ncol(x)]))
}

meds_ankylosing_count_pp <- count_meds_pp(meds_ankylosing)
names(meds_ankylosing_count_pp) <- c("ID","Medication")

meds_arthritis_count_pp <- count_meds_pp(meds_arthritis)
names(meds_arthritis_count_pp) <- c("ID","Medication")

meds_crohn_count_pp <- count_meds_pp(meds_crohn)
names(meds_crohn_count_pp) <- c("ID","Medication")

meds_diabetes_count_pp <- meds_diabetes
names(meds_diabetes_count_pp) <- c("ID","Medication")

meds_hidradenitis_count_pp <- count_meds_pp(meds_hidradenitis)
names(meds_hidradenitis_count_pp) <- c("ID","Medication")

meds_lupus_count_pp <- count_meds_pp(meds_lupus)
names(meds_lupus_count_pp) <- c("ID","Medication")

meds_myasthenia_count_pp <- count_meds_pp(meds_myasthenia)
names(meds_myasthenia_count_pp) <- c("ID","Medication")

meds_pemphigoid_count_pp <- count_meds_pp(meds_pemphigoid)
names(meds_pemphigoid_count_pp) <- c("ID","Medication")

meds_polymyalgia_count_pp <- count_meds_pp(meds_polymyalgia)
names(meds_polymyalgia_count_pp) <- c("ID","Medication")

meds_psoriasis_count_pp <- count_meds_pp(meds_psoriasis)
names(meds_psoriasis_count_pp) <- c("ID","Medication")

meds_psoriatic_count_pp <- count_meds_pp(meds_psoriatic)
names(meds_psoriatic_count_pp) <- c("ID","Medication")

meds_sclerosis_count_pp <- count_meds_pp(meds_sclerosis)
names(meds_sclerosis_count_pp) <- c("ID","Medication")

meds_sjogren_count_pp <- count_meds_pp(meds_sjogren)
names(meds_sjogren_count_pp) <- c("ID","Medication")

meds_thyroid_count_pp <- count_meds_pp(meds_thyroid)
names(meds_thyroid_count_pp) <- c("ID","Medication")

meds_ulcerative_count_pp <- count_meds_pp(meds_ulcerative)
names(meds_ulcerative_count_pp) <- c("ID","Medication")

# Note - meds_crohn and meds_ulcerative include the same medications, therefore, make one df called ibd

meds_ibd_count_pp <- meds_ulcerative_count_pp

```

```{r make_df_for_each_disorder_across_measures, echo=FALSE}

# make df for each disorder by combining columns for 1) count of primary and secondary ICD dx, 2) self-report endorsement for disorder, 3) count of medications corresponding to each disorder. 

# no drugs
pa_all_measures <- plyr::join_all(list(pa_count_pp, pa_self), by='ID')
coeliac_all_measures <- plyr::join_all(list(coeliac_count_pp, coeliac_self), by='ID')

# drugs
at_all_measures <- plyr::join_all(list(at_count_pp, at_self, meds_thyroid_count_pp), by='ID')
t1d_all_measures <- plyr::join_all(list(t1d_count_pp, t1d_self, meds_diabetes_count_pp), by='ID')
ms_all_measures <- plyr::join_all(list(ms_count_pp, ms_self, meds_sclerosis_count_pp), by='ID')
mg_all_measures <- plyr::join_all(list(mg_count_pp, mg_self, meds_myasthenia_count_pp), by='ID')
ibd_all_measures <- plyr::join_all(list(ibd_count_pp, ibd_self, meds_ibd_count_pp), by='ID')
hidradentis_all_measures <- plyr::join_all(list(hidradentis_count_pp, meds_hidradenitis_count_pp), by='ID')
pemphigoid_gus_all_measures <- plyr::join_all(list(pemphigoid_gus_count_pp, pemphigoid_gus_self, meds_pemphigoid_count_pp), by='ID')
psoriasis_all_measures <- plyr::join_all(list(psoriasis_count_pp, psoriasis_self, meds_psoriasis_count_pp), by='ID')
as_all_measures  <-  plyr::join_all(list(as_count_pp, as_self, meds_ankylosing_count_pp), by='ID')
pr_all_measures <- plyr::join_all(list(pr_count_pp, pr_self, meds_polymyalgia_count_pp), by='ID')
psa_all_measures <- plyr::join_all(list(psa_count_pp, psa_self, meds_psoriatic_count_pp), by='ID')
ra_all_measures <- plyr::join_all(list(ra_count_pp, ra_self, meds_arthritis_count_pp), by='ID')
sjogrens_all_measures <- plyr::join_all(list(sjogrens_count_pp, sjogrens_self, meds_sjogren_count_pp), by='ID')
sle_all_measures <- plyr::join_all(list(sle_count_pp, sle_self, meds_lupus_count_pp), by='ID')

```

```{r make_super_controls, echo=FALSE}

# make control group screened for ICD codes, self-report, and medications endorsed for any of the autoimmune diseases

# join dataframes for each disorder
all_df <- plyr::join_all(list(pa_all_measures, coeliac_all_measures, at_all_measures, t1d_all_measures, ms_all_measures, mg_all_measures, ibd_all_measures, hidradentis_all_measures, pemphigoid_gus_all_measures, psoriasis_all_measures, as_all_measures, pr_all_measures, psa_all_measures, ra_all_measures, sjogrens_all_measures, sle_all_measures), by='ID')

# make control column
all_df$sum <- rowSums(all_df[,2:ncol(all_df)])
sum_measures <- all_df[,c(1,63)]
super_controls <- sum_measures %>% 
  filter(sum == 0)

```


```{r make_one_possible_one_probable_for_venn, echo=FALSE}

# create dataframes for each disorder with one column for 'possible' cases and one column for 'probable' cases

# create function
poss_prob_fun <- function(x) {
  as_tibble(x) %>% 
    mutate(
      Possible = ifelse(
    `Primary count` > 0 | 
    `Secondary count` > 0 | 
    `Self-report` > 0, 1, 0)
    ) %>% 
    mutate(
      Probable = ifelse(
    `Primary count` > 1 | 
    `Primary count` == 1 & `Secondary count` == 1 | 
    `Primary count` > 1 & `Secondary count` == 1 | 
    `Secondary count` > 1 | 
    `Primary count` == 1 & `Secondary count` > 1 | 
    `Primary count` > 1 & `Secondary count` > 1 | 
    `Primary count` > 0 & `Self-report` > 0 |
    `Secondary count` > 0 & `Self-report` > 0 |
    `Primary count` > 0 & `Medication` > 0 |
    `Secondary count` > 0 & `Medication` > 0 |
    `Self-report` > 0 & `Medication` > 0, 1, 0)
    ) %>% 
    dplyr::select(ID, Possible, Probable)
}

# run function across disorders with medication and self report
as_poss_prob <- poss_prob_fun(as_all_measures)
at_poss_prob <- poss_prob_fun(at_all_measures)
ibd_poss_prob <- poss_prob_fun(ibd_all_measures)
mg_poss_prob <- poss_prob_fun(mg_all_measures)
ms_poss_prob <- poss_prob_fun(ms_all_measures)
pemphigoid_gus_poss_prob <- poss_prob_fun(pemphigoid_gus_all_measures)
pr_poss_prob <- poss_prob_fun(pr_all_measures)
psa_poss_prob <- poss_prob_fun(psa_all_measures)
psoriasis_poss_prob <- poss_prob_fun(psoriasis_all_measures)
ra_poss_prob <- poss_prob_fun(ra_all_measures)
sjogrens_poss_prob <- poss_prob_fun(sjogrens_all_measures)
sle_poss_prob <- poss_prob_fun(sle_all_measures)
t1d_poss_prob <- poss_prob_fun(t1d_all_measures)

# create function for disorders with no medication treatment
poss_prob_fun_no_med <- function(x) {
  as_tibble(x) %>% 
    mutate(
      Possible = ifelse(
    `Primary count` > 0 | 
    `Secondary count` > 0 | 
    `Self-report` > 0, 1, 0)
    ) %>% 
    mutate(
      Probable = ifelse(
    `Primary count` > 1 | 
    `Primary count` == 1 & `Secondary count` == 1 | 
    `Primary count` > 1 & `Secondary count` == 1 | 
    `Secondary count` > 1 | 
    `Primary count` == 1 & `Secondary count` > 1 | 
    `Primary count` > 1 & `Secondary count` > 1 | 
    `Primary count` > 0 & `Self-report` > 0 |
    `Secondary count` > 0 & `Self-report` > 0 , 1, 0)
    ) %>% 
    dplyr::select(ID, Possible, Probable)
}

coeliac_poss_prob <- poss_prob_fun_no_med(coeliac_all_measures)
pa_poss_prob <- poss_prob_fun_no_med(pa_all_measures)

# for disorders with no self-report measure
hidradentis_poss_prob <- as_tibble(hidradentis_all_measures) %>% 
    mutate(
      Possible = ifelse(
    `Primary count` > 0 | 
    `Secondary count` > 0, 1, 0)
    ) %>% 
    mutate(
      Probable = ifelse(
    `Primary count` > 1 | 
    `Primary count` == 1 & `Secondary count` == 1 | 
    `Primary count` > 1 & `Secondary count` == 1 | 
    `Secondary count` > 1 | 
    `Primary count` == 1 & `Secondary count` > 1 | 
    `Primary count` > 1 & `Secondary count` > 1 | 
    `Primary count` > 0 & `Medication` > 0 |
    `Secondary count` > 0 & `Medication` > 0, 1, 0)
    ) %>% 
    dplyr::select(ID, Possible, Probable)

```

```{r make_one_possible_one_probable_for_summary_table_and_pheno_file, echo=FALSE}

# make df for each disorder, joining possible cases with supercontrols, then probable cases with supercontrols

# create temporary supercontrol files with matching columns names 
poss_super_controls <- super_controls
colnames(poss_super_controls) <- c("ID", "Possible")

prob_super_controls <- super_controls
colnames(prob_super_controls) <- c("ID", "Probable")

# create function to select possible cases and bind with supercontrols
select_poss <- function(x) {
  x %>% 
  filter(Possible == 1) %>% 
  dplyr::select(ID, Possible) %>% 
  rbind(., poss_super_controls) %>% 
    `rownames<-`(NULL)
}

# create function to select probable cases and bind with supercontrols
select_prob <- function(x) {
  x %>% 
  filter(Probable == 1) %>% 
  dplyr::select(ID, Probable) %>% 
  rbind(., prob_super_controls) %>% 
    `rownames<-`(NULL)
}

# create possible files 
as_poss <- select_poss(as_poss_prob)
at_poss <- select_poss(at_poss_prob)
ibd_poss <- select_poss(ibd_poss_prob)
mg_poss <- select_poss(mg_poss_prob)
ms_poss <- select_poss(ms_poss_prob)
pemphigoid_gus_poss <- select_poss(pemphigoid_gus_poss_prob)
pr_poss <- select_poss(pr_poss_prob)
psa_poss <- select_poss(psa_poss_prob)
psoriasis_poss <- select_poss(psoriasis_poss_prob)
ra_poss <- select_poss(ra_poss_prob)
sjogrens_poss <- select_poss(sjogrens_poss_prob)
sle_poss <- select_poss(sle_poss_prob)
t1d_poss <- select_poss(t1d_poss_prob)
coeliac_poss <- select_poss(coeliac_poss_prob)
pa_poss <- select_poss(pa_poss_prob)
hidradentis_poss <- select_poss(hidradentis_poss_prob)

# create probable files 
as_prob <- select_prob(as_poss_prob)
at_prob <- select_prob(at_poss_prob)
ibd_prob <- select_prob(ibd_poss_prob)
mg_prob <- select_prob(mg_poss_prob)
ms_prob <- select_prob(ms_poss_prob)
pemphigoid_gus_prob <- select_prob(pemphigoid_gus_poss_prob)
pr_prob <- select_prob(pr_poss_prob)
psa_prob <- select_prob(psa_poss_prob)
psoriasis_prob <- select_prob(psoriasis_poss_prob)
ra_prob <- select_prob(ra_poss_prob)
sjogrens_prob <- select_prob(sjogrens_poss_prob)
sle_prob <- select_prob(sle_poss_prob)
t1d_prob <- select_prob(t1d_poss_prob)
coeliac_prob <- select_prob(coeliac_poss_prob)
pa_prob <- select_prob(pa_poss_prob)
hidradentis_prob <- select_prob(hidradentis_poss_prob)

# create function to bind possible and probable columns for each disease, add NA where necessary and make columns headings suitable for PRSice
merge_poss_prob <- function(x,y) {
  full_join(x, y, by = "ID") %>% 
    mutate(FID = ID, IID = ID) %>% 
    dplyr::select(FID, IID, Possible, Probable)
}

as_poss_prob_prs <- merge_poss_prob(as_poss,as_prob)
at_poss_prob_prs <- merge_poss_prob(at_poss,at_prob)
ibd_poss_prob_prs <- merge_poss_prob(ibd_poss,ibd_prob)
mg_poss_prob_prs <- merge_poss_prob(mg_poss,mg_prob)
ms_poss_prob_prs <- merge_poss_prob(ms_poss,ms_prob)
pemphigoid_gus_poss_prob_prs <- merge_poss_prob(pemphigoid_gus_poss,pemphigoid_gus_prob)
pr_poss_prob_prs <- merge_poss_prob(pr_poss,pr_prob)
psa_poss_prob_prs <- merge_poss_prob(psa_poss,psa_prob)
psoriasis_poss_prob_prs <- merge_poss_prob(psoriasis_poss,psoriasis_prob)
ra_poss_prob_prs <- merge_poss_prob(ra_poss,ra_prob)
sjogrens_poss_prob_prs <- merge_poss_prob(sjogrens_poss,sjogrens_prob)
sle_poss_prob_prs <- merge_poss_prob(sle_poss,sle_prob)
t1d_poss_prob_prs <- merge_poss_prob(t1d_poss,t1d_prob)
coeliac_poss_prob_prs <- merge_poss_prob(coeliac_poss,coeliac_prob)
pa_poss_prob_prs <- merge_poss_prob(pa_poss,pa_prob)
hidradentis_poss_prob_prs <- merge_poss_prob(hidradentis_poss,hidradentis_prob)

```

```{r print_one_possible_one_probable_to_file, echo=FALSE}

# print phenotype dfs to file

write.table(as_poss_prob_prs, file="/file_path/new_as_poss_prob_prs", row.names=F, col.name=T, quote=F, sep="\t")
write.table(at_poss_prob_prs, file="/file_path/new_at_poss_prob_prs", row.names=F, col.name=T, quote=F, sep="\t")
write.table(ibd_poss_prob_prs, file="/file_path/new_ibd_poss_prob_prs", row.names=F, col.name=T, quote=F, sep="\t")
write.table(mg_poss_prob_prs, file="/file_path/new_mg_poss_prob_prs", row.names=F, col.name=T, quote=F, sep="\t")
write.table(ms_poss_prob_prs, file="/file_path/new_ms_poss_prob_prs", row.names=F, col.name=T, quote=F, sep="\t")
write.table(pemphigoid_gus_poss_prob_prs, file="/file_path/new_pemphigoid_gus_poss_prob_prs", row.names=F, col.name=T, quote=F, sep="\t")
write.table(pr_poss_prob_prs, file="/file_path/new_pr_poss_prob_prs", row.names=F, col.name=T, quote=F, sep="\t")
write.table(psa_poss_prob_prs, file="/file_path/new_psa_poss_prob_prs", row.names=F, col.name=T, quote=F, sep="\t")
write.table(psoriasis_poss_prob_prs, file="/file_path/new_psoriasis_poss_prob_prs", row.names=F, col.name=T, quote=F, sep="\t")
write.table(ra_poss_prob_prs, file="/file_path/new_ra_poss_prob_prs", row.names=F, col.name=T, quote=F, sep="\t")
write.table(sjogrens_poss_prob_prs, file="/file_path/new_sjogrens_poss_prob_prs", row.names=F, col.name=T, quote=F, sep="\t")
write.table(sle_poss_prob_prs, file="/file_path/new_sle_poss_prob_prs", row.names=F, col.name=T, quote=F, sep="\t")
write.table(t1d_poss_prob_prs, file="/file_path/new_t1d_poss_prob_prs", row.names=F, col.name=T, quote=F, sep="\t")
write.table(coeliac_poss_prob_prs, file="/file_path/new_coeliac_poss_prob_prs", row.names=F, col.name=T, quote=F, sep="\t")
write.table(pa_poss_prob_prs, file="/file_path/new_pa_poss_prob_prs", row.names=F, col.name=T, quote=F, sep="\t")
write.table(hidradentis_poss_prob_prs, file="/file_path/new_hidradentis_poss_prob_prs", row.names=F, col.name=T, quote=F, sep="\t")

```

```{r make_one_df_all_autoimmune, echo=FALSE}

# make one df containing all diseases

as_poss_prob_prs <- fread("/file_path/new_as_poss_prob_prs", header = TRUE)
colnames(as_poss_prob_prs) <- c("FID","IID","ankylosing_poss","ankylosing_prob")

at_poss_prob_prs <- fread("/file_path/new_at_poss_prob_prs", header = TRUE)
colnames(at_poss_prob_prs) <- c("FID","IID","thyroid_poss","thyroid_prob")

coeliac_poss_prob_prs <- fread("/file_path/new_coeliac_poss_prob_prs", header = TRUE)
colnames(coeliac_poss_prob_prs) <- c("FID","IID","coeliac_poss","coeliac_prob")

hidradentis_poss_prob_prs <- fread("/file_path/new_hidradentis_poss_prob_prs", header = TRUE)
colnames(hidradentis_poss_prob_prs) <- c("FID","IID","hidradentis_poss","hidradentis_prob")

ibd_poss_prob_prs <- fread("/file_path/new_ibd_poss_prob_prs", header = TRUE)
colnames(ibd_poss_prob_prs) <- c("FID","IID","ibd_poss","ibd_prob")

mg_poss_prob_prs <- fread("/file_path/new_mg_poss_prob_prs", header = TRUE)
colnames(mg_poss_prob_prs) <- c("FID","IID","myasthenia_poss","myasthenia_prob")

ms_poss_prob_prs <- fread("/file_path/new_ms_poss_prob_prs", header = TRUE)
colnames(ms_poss_prob_prs) <- c("FID","IID","ms_poss","ms_prob")

pa_poss_prob_prs <- fread("/file_path/new_pa_poss_prob_prs", header = TRUE)
colnames(pa_poss_prob_prs) <- c("FID","IID","pernecious_poss","pernecious_prob")

pemphigoid_gus_poss_prob_prs <- fread("/file_path/new_pemphigoid_gus_poss_prob_prs", header = TRUE)
colnames(pemphigoid_gus_poss_prob_prs) <- c("FID","IID","pemphigoid_poss","pemphigoid_prob")

pr_poss_prob_prs <- fread("/file_path/new_pr_poss_prob_prs", header = TRUE)
colnames(pr_poss_prob_prs) <- c("FID","IID","polymyalgia_poss","polymyalgia_prob")

psa_poss_prob_prs <- fread("/file_path/new_psa_poss_prob_prs", header = TRUE)
colnames(psa_poss_prob_prs) <- c("FID","IID","psa_poss","psa_prob")

psoriasis_poss_prob_prs <- fread("/file_path/new_psoriasis_poss_prob_prs", header = TRUE)
colnames(psoriasis_poss_prob_prs) <- c("FID","IID","psoriasis_poss","psoriasis_prob")

ra_poss_prob_prs <- fread("/file_path/new_ra_poss_prob_prs", header = TRUE)
colnames(ra_poss_prob_prs) <- c("FID","IID","arthritis_poss","arthritis_prob")

sjogrens_poss_prob_prs <- fread("/file_path/new_sjogrens_poss_prob_prs", header = TRUE)
colnames(sjogrens_poss_prob_prs) <- c("FID","IID","sjogrens_poss","sjogrens_prob")

sle_poss_prob_prs <- fread("/file_path/new_sle_poss_prob_prs", header = TRUE)
colnames(sle_poss_prob_prs) <- c("FID","IID","sle_poss","sle_prob")

t1d_poss_prob_prs <- fread("/file_path/new_t1d_poss_prob_prs", header = TRUE)
colnames(t1d_poss_prob_prs) <- c("FID","IID","t1d_poss","t1d_prob")

# make one df
autoimmune_all <- plyr::join_all(list(as_poss_prob_prs, at_poss_prob_prs, coeliac_poss_prob_prs, hidradentis_poss_prob_prs, ibd_poss_prob_prs, mg_poss_prob_prs, ms_poss_prob_prs, pa_poss_prob_prs, pemphigoid_gus_poss_prob_prs, pr_poss_prob_prs, psa_poss_prob_prs, psoriasis_poss_prob_prs, ra_poss_prob_prs, sjogrens_poss_prob_prs, sle_poss_prob_prs, t1d_poss_prob_prs), by = c("FID","IID"), type = 'full')

write.table(autoimmune_all, file="/file_path/new_autoimmune_all", row.names=F, col.name=T, quote=F, sep="\t")

```

```{r make_icd_self_meds_possible_for_venn, echo=FALSE}

# make dfs with ICD primary and secondary collapsed into one column, and self-report and medications for use with venn diagrams

# create function to collapse ICD into one column and select required columns
icd_self_med <- function(x) {
  x %>% 
  mutate(ICD = `Primary count` + `Secondary count`) %>% 
  dplyr::select(ID, ICD, `Self-report`, Medication)
}

# all measures
at_icd_self_med <- icd_self_med(at_all_measures)
t1d_icd_self_med <- icd_self_med(t1d_all_measures)
ms_icd_self_med <- icd_self_med(ms_all_measures)
mg_icd_self_med <- icd_self_med(mg_all_measures)
ibd_icd_self_med <- icd_self_med(ibd_all_measures)
pemphigoid_gus_icd_self_med <- icd_self_med(pemphigoid_gus_all_measures)
psoriasis_icd_self_med <- icd_self_med(psoriasis_all_measures)
as_icd_self_med <- icd_self_med(as_all_measures)
pr_icd_self_med <- icd_self_med(pr_all_measures)
psa_icd_self_med <- icd_self_med(psa_all_measures)
ra_icd_self_med <- icd_self_med(ra_all_measures)
sjogrens_icd_self_med <- icd_self_med(sjogrens_all_measures)
sle_icd_self_med <- icd_self_med(sle_all_measures)

# no drugs
icd_self <- function(x) {
  x %>% 
  mutate(ICD = `Primary count` + `Secondary count`) %>% 
  dplyr::select(ID, ICD, `Self-report`)
}

pa_icd_self_med <- icd_self(pa_all_measures)
coeliac_icd_self_med <- icd_self(coeliac_all_measures)

# no self-report
icd_med <- function(x) {
  x %>% 
  mutate(ICD = `Primary count` + `Secondary count`) %>% 
  dplyr::select(ID, ICD, Medication)
}

hidradentis_icd_self_med <- icd_med(hidradentis_all_measures)

```

```{r make_eight_icd_groups_for_matrix, echo=FALSE}

# count combinations of ICD codes to display in matrix format. 

# create function to make eight icd phenotypes
icd_cases <- function(x) {
data.frame(
"one_zero" = ifelse(x$`Primary count` == 1 & x$`Secondary count` == 0, 1, 0),
"multiple_zero" = ifelse(x$`Primary count` > 1 & x$`Secondary count` == 0, 1, 0),
"zero_one" = ifelse(x$`Primary count` == 0 & x$`Secondary count` == 1, 1, 0),
"one_one" = ifelse(x$`Primary count` == 1 & x$`Secondary count` == 1, 1, 0),
"multiple_one" = ifelse(x$`Primary count` > 1 & x$`Secondary count` == 1, 1, 0),
"zero_multiple" = ifelse(x$`Primary count` == 0 & x$`Secondary count` > 1, 1, 0),
"one_multiple" = ifelse(x$`Primary count` == 1 & x$`Secondary count` > 1, 1, 0),
"multiple_multiple" = ifelse(x$`Primary count` > 1 & x$`Secondary count` > 1, 1, 0)
)
}

as_temp_df <- icd_cases(as_count_pp)
at_temp_df <- icd_cases(at_count_pp)
ibd_temp_df <- icd_cases(ibd_count_pp)
pemphigoid_gus_temp_df <- icd_cases(pemphigoid_gus_count_pp)
pr_temp_df <- icd_cases(pr_count_pp)
psa_temp_df <- icd_cases(psa_count_pp)
psoriasis_temp_df <- icd_cases(psoriasis_count_pp)
ra_temp_df <- icd_cases(ra_count_pp)
sle_temp_df <- icd_cases(sle_count_pp)
t1d_temp_df <- icd_cases(t1d_count_pp)
coeliac_temp_df <- icd_cases(coeliac_count_pp)
hidradentis_temp_df <- icd_cases(hidradentis_count_pp)
mg_temp_df <- icd_cases(mg_count_pp)
ms_temp_df <- icd_cases(ms_count_pp)
pa_temp_df <- icd_cases(pa_count_pp)
sjogrens_temp_df <- icd_cases(sjogrens_count_pp)

# create function to make matrix
icd_matrix <- function(x) {
  data.frame(rbind(
  c(format(nrow(super_controls), big.mark = ","), format(sum(x$one_zero == 1), big.mark = ","), format(sum(x$multiple_zero == 1), big.mark = ",")),
  c(format(sum(x$zero_one == 1), big.mark = ","), format(sum(x$one_one == 1), big.mark = ","), format(sum(x$multiple_one == 1), big.mark = ",")),
  c(format(sum(x$zero_multiple == 1), big.mark = ","), format(sum(x$one_multiple == 1), big.mark = ","), format(sum(x$multiple_multiple == 1), big.mark = ","))))
}

as_temp_matrix <- icd_matrix(as_temp_df)
at_temp_matrix <- icd_matrix(at_temp_df)
ibd_temp_matrix <- icd_matrix(ibd_temp_df)
pemphigoid_gus_temp_matrix <- icd_matrix(pemphigoid_gus_temp_df)
pr_temp_matrix <- icd_matrix(pr_temp_df)
psa_temp_matrix <- icd_matrix(psa_temp_df)
psoriasis_temp_matrix <- icd_matrix(psoriasis_temp_df)
ra_temp_matrix <- icd_matrix(ra_temp_df)
sle_temp_matrix <- icd_matrix(sle_temp_df)
t1d_temp_matrix <- icd_matrix(t1d_temp_df)
coeliac_temp_matrix <- icd_matrix(coeliac_temp_df)
hidradentis_temp_matrix <- icd_matrix(hidradentis_temp_df)
mg_temp_matrix <- icd_matrix(mg_temp_df)
ms_temp_matrix <- icd_matrix(ms_temp_df)
pa_temp_matrix <- icd_matrix(pa_temp_df)
sjogrens_temp_matrix <- icd_matrix(sjogrens_temp_df)

```

```{r make_possible_summary, include=FALSE}

# Make table of demographic information for possible cases. Remove pemphigoid and hidradenitis due to small n. 

# create df for all possible cases - read in files with all autoimmune phenotypes
all_autoimmune <- fread("/file_path/new_autoimmune_all", header = TRUE)

# select possible columns
autoimmune_possible <- all_autoimmune %>%
  select(IID, contains("poss"), -pemphigoid_poss, -hidradentis_poss)

# make column with count of diagnoses per person
autoimmune_possible$count_pp_possible <- rowSums(autoimmune_possible[,2:ncol(autoimmune_possible)], na.rm = T)

# make column with 1 for cases of any disorder, 0 without disorder 
autoimmune_possible$any_possible <- ifelse(autoimmune_possible$count_pp_possible>0, 1, 0)

# make df with required columns
any_poss <- autoimmune_possible %>%
  select(IID, any_possible) %>%
  rename(ID=IID, Possible=any_possible)

# select possible cases and bind with supercontrols, thus removing Hidradenitis Suppurativa and Pemphigoid/Pemphigus cases from controls
any_poss <- select_poss(any_poss)

# create temporary demographics df with ID column renamed from eid
demographics1 <- demographics
colnames(demographics1) <- c("ID", "Age", "Sex", "SES", "BMI", "Current Smoker")

# join possible phenotypes with demographics1
join_demo <- function(x) {
  left_join(x, demographics1, by = "ID") 
}

as_poss_demo <- join_demo(as_poss)
at_poss_demo <- join_demo(at_poss)
ibd_poss_demo <- join_demo(ibd_poss)
mg_poss_demo <- join_demo(mg_poss)
ms_poss_demo <- join_demo(ms_poss)
pr_poss_demo <- join_demo(pr_poss)
psa_poss_demo <- join_demo(psa_poss)
psoriasis_poss_demo <- join_demo(psoriasis_poss)
ra_poss_demo <- join_demo(ra_poss)
sjogrens_poss_demo <- join_demo(sjogrens_poss)
sle_poss_demo <- join_demo(sle_poss)
t1d_poss_demo <- join_demo(t1d_poss)
coeliac_poss_demo <- join_demo(coeliac_poss)
pa_poss_demo <- join_demo(pa_poss)
any_poss_demo <- join_demo(any_poss)

# select only possible cases
poss_select <- function(x) {
  x %>%
  filter(Possible == 1)
}

as_poss_only_demo <- poss_select(as_poss_demo)
at_poss_only_demo <- poss_select(at_poss_demo)
ibd_poss_only_demo <- poss_select(ibd_poss_demo)
mg_poss_only_demo <- poss_select(mg_poss_demo)
ms_poss_only_demo <- poss_select(ms_poss_demo)
pr_poss_only_demo <- poss_select(pr_poss_demo)
psa_poss_only_demo <- poss_select(psa_poss_demo)
psoriasis_poss_only_demo <- poss_select(psoriasis_poss_demo)
ra_poss_only_demo <- poss_select(ra_poss_demo)
sjogrens_poss_only_demo <- poss_select(sjogrens_poss_demo)
sle_poss_only_demo <- poss_select(sle_poss_demo)
t1d_poss_only_demo <- poss_select(t1d_poss_demo)
coeliac_poss_only_demo <- poss_select(coeliac_poss_demo)
pa_poss_only_demo <- poss_select(pa_poss_demo)
any_poss_only_demo <- poss_select(any_poss_demo)

# summarise data
summarise_possible <- function(x) {
  data.frame(
  Count = format(nrow(x), big.mark = ","), 
  Prevalence = paste(formattable::percent(round(nrow(x)/(nrow(x)+nrow(super_controls)), digits = 4))), 
  "Age (SD)" = paste(format(mean(x$Age, na.rm = T), digits = 3, nsmall = 1), paste("(", format(sd(x$Age, na.rm = T), digits = 3, nsmall = 2), ")", sep = ""), sep = " "),  
  "Female(%)" = percent(round(sum(x$Sex==0, na.rm = T)/nrow(x), digits = 2), digits = 0),
  "SES (SD)" = paste(format(mean(x$SES, na.rm = T), digits = 2, nsmall = 2), paste("(", format(sd(x$SES, na.rm = T), digits = 2, nsmall = 2), ")", sep = ""), sep = " "),
  "Current smoker (%)" = percent(round(sum(x$`Current Smoker`==1, na.rm = T)/nrow(x), digits = 2), digits = 0),
  "BMI (SD)" = paste(format(mean(x$BMI, na.rm = T), digits = 3, nsmall = 1), paste("(", format(sd(x$BMI, na.rm = T), digits = 3, nsmall = 2), ")", sep = ""), sep = " ")
)
}

as_poss_desc <- summarise_possible(as_poss_only_demo)
at_poss_desc <- summarise_possible(at_poss_only_demo)
ibd_poss_desc <- summarise_possible(ibd_poss_only_demo)
mg_poss_desc <- summarise_possible(mg_poss_only_demo)
ms_poss_desc <- summarise_possible(ms_poss_only_demo)
pr_poss_desc <- summarise_possible(pr_poss_only_demo)
psa_poss_desc <- summarise_possible(psa_poss_only_demo)
psoriasis_poss_desc <- summarise_possible(psoriasis_poss_only_demo)
ra_poss_desc <- summarise_possible(ra_poss_only_demo)
sjogrens_poss_desc <- summarise_possible(sjogrens_poss_only_demo)
sle_poss_desc <- summarise_possible(sle_poss_only_demo)
t1d_poss_desc <- summarise_possible(t1d_poss_only_demo)
coeliac_poss_desc <- summarise_possible(coeliac_poss_only_demo)
pa_poss_desc <- summarise_possible(pa_poss_only_demo)
any_poss_desc <- summarise_possible(any_poss_only_demo)

# descriptives row for controls

demo_controls <- left_join(super_controls, demographics1, by = "ID")

desc_controls <- data.frame(
  Count = format(nrow(demo_controls), big.mark = ","), 
    Prevalence = "NA", 
    "Age (SD)" = paste(format(mean(demo_controls$Age, na.rm = T), digits = 3, nsmall = 1), paste("(", format(sd(demo_controls$Age, na.rm = T), digits = 3, nsmall = 2), ")", sep = ""), sep = " "),  
    "Female(%)" = percent(round(sum(demo_controls$Sex==0, na.rm = T)/nrow(demo_controls), digits = 2), digits = 0),
    "SES (SD)" = paste(format(mean(demo_controls$SES, na.rm = T), digits = 2, nsmall = 2), paste("(", format(sd(demo_controls$SES, na.rm = T), digits = 2, nsmall = 2), ")", sep = ""), sep = " "),
    "Current smoker (%)" = percent(round(sum(demo_controls$`Current Smoker`==1, na.rm = T)/nrow(demo_controls), digits = 2), digits = 0),
    "BMI (SD)" = paste(format(mean(demo_controls$BMI, na.rm = T), digits = 3, nsmall = 1), paste("(", format(sd(demo_controls$BMI, na.rm = T), digits = 3, nsmall = 2), ")", sep = ""), sep = " ")
)

desc_controls$Prevalence <- as.character(desc_controls$Prevalence)
  
# combine data
possible_summary <- rbind(desc_controls,
                          pa_poss_desc,
                          at_poss_desc,
                          t1d_poss_desc,
                          ms_poss_desc,
                          mg_poss_desc,
                          coeliac_poss_desc,
                          ibd_poss_desc,
                          psoriasis_poss_desc,
                          as_poss_desc,
                          pr_poss_desc,
                          psa_poss_desc,
                          ra_poss_desc,
                          sjogrens_poss_desc,
                          sle_poss_desc,
                          any_poss_desc)

colnames(possible_summary) <- c("Count","UKB.Prev.","Mean age (SD)","Female","SES* (SD)","Smoker","Mean BMI (SD)")
rownames(possible_summary) <- c("Controls","Pernicious Anemia","Autoimmune Thyroid Disease","Type 1 diabetes","Multiple Sclerosis","Myasthenia Gravis","Coeliac","Inflammatory Bowel Disease","Psoriasis","Ankylosing Spondylitis","Polymyalgia Rheumatica/GCA","Psoriatic Arthritis","Rheumatoid Arthritis","Sjögren Syndrome","Systemic Lupus Erythematosus","Any Possible Autoimmune Disease")

### add population prevalences

possible_summary$Pop.Prev. <- c("NA","0.10%","2.00%","0.30%","0.10%","0.02%","1.00%","0.50%","2.00%","0.55%","0.85%","0.50%","1.00%","0.70%","0.10%","NA")

possible_summary <- possible_summary[, c(1:2,8,3:7)]

```

```{r make_probable_summary, include=FALSE}

# Make table of demographic information for probable cases. Remove pemphigoid and hidradenitis due to small n. 

# create df for all probable cases - read in files with all autoimmune phenotypes
autoimmune_all <- fread("/file_path/new_autoimmune_all", header = TRUE)

# select probable columns
autoimmune_probable <- all_autoimmune %>%
  select(IID, contains("prob"), -pemphigoid_prob, -hidradentis_prob)

# make column with count of diagnoses per person
autoimmune_probable$count_pp_probable <- rowSums(autoimmune_probable[,2:ncol(autoimmune_probable)], na.rm = T)

# make column with 1 for cases of any disorder, 0 without disorder 
autoimmune_probable$any_probable <- ifelse(autoimmune_probable$count_pp_probable>0, 1, 0)

# make df with required columns
any_prob <- autoimmune_probable %>%
  select(IID, any_probable) %>%
  rename(ID=IID, Probable=any_probable)

# select probable cases and bind with supercontrols, thus removing Hidradenitis Suppurativa and Pemphigoid/Pemphigus cases from controls
any_prob <- select_prob(any_prob)

# join probable phenotypes with demographics1
as_prob_demo <- join_demo(as_prob)
at_prob_demo <- join_demo(at_prob)
ibd_prob_demo <- join_demo(ibd_prob)
mg_prob_demo <- join_demo(mg_prob)
ms_prob_demo <- join_demo(ms_prob)
pr_prob_demo <- join_demo(pr_prob)
psa_prob_demo <- join_demo(psa_prob)
psoriasis_prob_demo <- join_demo(psoriasis_prob)
ra_prob_demo <- join_demo(ra_prob)
sjogrens_prob_demo <- join_demo(sjogrens_prob)
sle_prob_demo <- join_demo(sle_prob)
t1d_prob_demo <- join_demo(t1d_prob)
coeliac_prob_demo <- join_demo(coeliac_prob)
pa_prob_demo <- join_demo(pa_prob)
any_prob_demo <- join_demo(any_prob)

# select only probable cases
prob_select <- function(x) {
  x %>%
  filter(Probable == 1)
}

as_prob_only_demo <- prob_select(as_prob_demo)
at_prob_only_demo <- prob_select(at_prob_demo)
ibd_prob_only_demo <- prob_select(ibd_prob_demo)
mg_prob_only_demo <- prob_select(mg_prob_demo)
ms_prob_only_demo <- prob_select(ms_prob_demo)
pr_prob_only_demo <- prob_select(pr_prob_demo)
psa_prob_only_demo <- prob_select(psa_prob_demo)
psoriasis_prob_only_demo <- prob_select(psoriasis_prob_demo)
ra_prob_only_demo <- prob_select(ra_prob_demo)
sjogrens_prob_only_demo <- prob_select(sjogrens_prob_demo)
sle_prob_only_demo <- prob_select(sle_prob_demo)
t1d_prob_only_demo <- prob_select(t1d_prob_demo)
coeliac_prob_only_demo <- prob_select(coeliac_prob_demo)
pa_prob_only_demo <- prob_select(pa_prob_demo)
any_prob_only_demo <- prob_select(any_prob_demo)

# summarise data
summarise_probable <- function(x) {
  data.frame(
  Count = format(nrow(x), big.mark = ","), 
  Prevalence = paste(formattable::percent(round(nrow(x)/(nrow(x)+nrow(super_controls)), digits = 4))), 
  "Age (SD)" = paste(format(mean(x$Age, na.rm = T), digits = 3, nsmall = 1), paste("(", format(sd(x$Age, na.rm = T), digits = 3, nsmall = 2), ")", sep = ""), sep = " "),  
  "Female(%)" = percent(round(sum(x$Sex==0, na.rm = T)/nrow(x), digits = 2), digits = 0),
  "SES (SD)" = paste(format(mean(x$SES, na.rm = T), digits = 2, nsmall = 2), paste("(", format(sd(x$SES, na.rm = T), digits = 2, nsmall = 2), ")", sep = ""), sep = " "),
  "Current smoker (%)" = percent(round(sum(x$`Current Smoker`==1, na.rm = T)/nrow(x), digits = 2), digits = 0),
  "BMI (SD)" = paste(format(mean(x$BMI, na.rm = T), digits = 3, nsmall = 1), paste("(", format(sd(x$BMI, na.rm = T), digits = 3, nsmall = 2), ")", sep = ""), sep = " ")
)
}

as_prob_desc <- summarise_probable(as_prob_only_demo)
at_prob_desc <- summarise_probable(at_prob_only_demo)
ibd_prob_desc <- summarise_probable(ibd_prob_only_demo)
mg_prob_desc <- summarise_probable(mg_prob_only_demo)
ms_prob_desc <- summarise_probable(ms_prob_only_demo)
pr_prob_desc <- summarise_probable(pr_prob_only_demo)
psa_prob_desc <- summarise_probable(psa_prob_only_demo)
psoriasis_prob_desc <- summarise_probable(psoriasis_prob_only_demo)
ra_prob_desc <- summarise_probable(ra_prob_only_demo)
sjogrens_prob_desc <- summarise_probable(sjogrens_prob_only_demo)
sle_prob_desc <- summarise_probable(sle_prob_only_demo)
t1d_prob_desc <- summarise_probable(t1d_prob_only_demo)
coeliac_prob_desc <- summarise_probable(coeliac_prob_only_demo)
pa_prob_desc <- summarise_probable(pa_prob_only_demo)
any_prob_desc <- summarise_probable(any_prob_only_demo)

# combine data
probable_summary <- rbind.data.frame(desc_controls,
                          pa_prob_desc,
                          at_prob_desc,
                          t1d_prob_desc,
                          ms_prob_desc,
                          mg_prob_desc,
                          coeliac_prob_desc,
                          ibd_prob_desc,
                          psoriasis_prob_desc,
                          as_prob_desc,
                          pr_prob_desc,
                          psa_prob_desc,
                          ra_prob_desc,
                          sjogrens_prob_desc,
                          sle_prob_desc, 
                          any_prob_desc, stringsAsFactors = FALSE)

colnames(probable_summary) <- c("Count","UKB.Prev.","Mean age (SD)","Female","SES* (SD)","Smoker","Mean BMI (SD)")
rownames(probable_summary) <- c("Controls","Pernicious Anemia","Autoimmune Thyroid Disease","Type 1 diabetes","Multiple Sclerosis","Myasthenia Gravis","Coeliac","Inflammatory Bowel Disease","Psoriasis","Ankylosing Spondylitis","Polymyalgia Rheumatica/GCA","Psoriatic Arthritis","Rheumatoid Arthritis","Sjögren Syndrome","Systemic Lupus Erythematosus","Any Probable Autoimmune Disease")

### add population prevalences

probable_summary$Pop.Prev. <- c("NA","0.10%","2.00%","0.30%","0.10%","0.02%","1.00%","0.50%","2.00%","0.55%","0.85%","0.50%","1.00%","0.70%","0.10%","NA")

probable_summary <- probable_summary[, c(1:2,8,3:7)]

```

```{r make_medication_summary_disorder_drug_pairing, echo=FALSE}

# make summary tables for count of medications endorsed by participants corresponding to each disease. 

summary_meds_thyroid <- data.frame("x" = "Autoimmune Thyroid Disease (Grave's and Thyroiditis)", "x1" = names(meds_thyroid[,-1]), "x2" = as.integer(format(colSums(meds_thyroid[,-1] > 0)), scientific = F), stringsAsFactors = F, row.names = NULL) %>%
  rename(Disease = x, Medication = x1, Count = x2)

summary_meds_diabetes <- data.frame("x" = "Type 1 diabetes", "x1" = "insulin product", "x2" = as.integer(format(sum(meds_diabetes[,-1] > 0)), scientific = F), stringsAsFactors = F, row.names = NULL) %>%
  rename(Disease = x, Medication = x1, Count = x2)

summary_meds_sclerosis <- data.frame("x" = "Multiple Sclerosis", "x1" = names(meds_sclerosis[,-1]), "x2" = as.integer(format(colSums(meds_sclerosis[,-1] > 0)), scientific = F), stringsAsFactors = F, row.names = NULL) %>%
  rename(Disease = x, Medication = x1, Count = x2)

summary_meds_myasthenia <- data.frame("x" = "Myasthenia Gravis", "x1" = names(meds_myasthenia[,-1]), "x2" = as.integer(format(colSums(meds_myasthenia[,-1] > 0)), scientific = F), stringsAsFactors = F, row.names = NULL) %>%
  rename(Disease = x, Medication = x1, Count = x2)

summary_meds_crohn <- data.frame("x" = "Inflammatory Bowel Disease", "x1" = names(meds_crohn[,-1]), "x2" = as.integer(format(colSums(meds_crohn[,-1] > 0)), scientific = F), stringsAsFactors = F, row.names = NULL) %>%
  rename(Disease = x, Medication = x1, Count = x2)

summary_meds_hidradenitis <- data.frame("x" = "Hidradenitis Suppurativa", "x1" = names(meds_hidradenitis[,-1]), "x2" = as.integer(format(colSums(meds_hidradenitis[,-1] > 0)), scientific = F), stringsAsFactors = F, row.names = NULL) %>%
  rename(Disease = x, Medication = x1, Count = x2)

summary_meds_pemphigoid <- data.frame("x" = "Pemphigoid/Pemphigus", "x1" = names(meds_pemphigoid[,-1]), "x2" = as.integer(format(colSums(meds_pemphigoid[,-1] > 0)), scientific = F), stringsAsFactors = F, row.names = NULL) %>%
  rename(Disease = x, Medication = x1, Count = x2)

summary_meds_psoriasis <- data.frame("x" = "Psoriasis", "x1" = names(meds_psoriasis[,-1]), "x2" = as.integer(format(colSums(meds_psoriasis[,-1] > 0)), scientific = F), stringsAsFactors = F, row.names = NULL) %>%
  rename(Disease = x, Medication = x1, Count = x2)

summary_meds_ankylosing <- data.frame("x" = "Ankylosing Spondylitis", "x1" = names(meds_ankylosing[,-1]), "x2" = as.integer(format(colSums(meds_ankylosing[,-1] > 0)), scientific = F), stringsAsFactors = F, row.names = NULL) %>%
  rename(Disease = x, Medication = x1, Count = x2)

summary_meds_polymyalgia <- data.frame("x" = "Polymyalgia Rheumatica/Giant Cell Arteritis", "x1" = names(meds_polymyalgia[,-1]), "x2" = as.integer(format(colSums(meds_polymyalgia[,-1] > 0)), scientific = F), stringsAsFactors = F, row.names = NULL) %>%
  rename(Disease = x, Medication = x1, Count = x2)

summary_meds_psoriatic <- data.frame("x" = "Psoriatic Arthritis", "x1" = names(meds_psoriatic[,-1]), "x2" = as.integer(format(colSums(meds_psoriatic[,-1] > 0)), scientific = F), stringsAsFactors = F, row.names = NULL) %>%
  rename(Disease = x, Medication = x1, Count = x2)

summary_meds_arthritis <- data.frame("x" = "Rheumatoid Arthritis", "x1" = names(meds_arthritis[,-1]), "x2" = as.integer(format(colSums(meds_arthritis[,-1] > 0)), scientific = F), stringsAsFactors = F, row.names = NULL) %>%
  rename(Disease = x, Medication = x1, Count = x2)

summary_meds_sjogren <- data.frame("x" = "Sjögren Syndrome", "x1" = names(meds_sjogren[,-1]), "x2" = as.integer(format(colSums(meds_sjogren[,-1] > 0)), scientific = F), stringsAsFactors = F, row.names = NULL) %>%
  rename(Disease = x, Medication = x1, Count = x2)

summary_meds_lupus <- data.frame("x" = "Systemic Lupus Erythematosus", "x1" = names(meds_lupus[,-1]), "x2" = as.integer(format(colSums(meds_lupus[,-1] > 0)), scientific = F), stringsAsFactors = F, row.names = NULL) %>%
  rename(Disease = x, Medication = x1, Count = x2)


# order by frequency

summary_meds_thyroid <- summary_meds_thyroid[order(-summary_meds_thyroid$Count),] 
rownames(summary_meds_thyroid) <- NULL
summary_meds_diabetes <- summary_meds_diabetes[order(-summary_meds_diabetes$Count),] 
rownames(summary_meds_diabetes) <- NULL
summary_meds_sclerosis <- summary_meds_sclerosis[order(-summary_meds_sclerosis$Count),] 
rownames(summary_meds_sclerosis) <- NULL
summary_meds_myasthenia <- summary_meds_myasthenia[order(-summary_meds_myasthenia$Count),] 
rownames(summary_meds_myasthenia) <- NULL
summary_meds_crohn <- summary_meds_crohn[order(-summary_meds_crohn$Count),] 
rownames(summary_meds_crohn) <- NULL
summary_meds_hidradenitis <- summary_meds_hidradenitis[order(-summary_meds_hidradenitis$Count),] 
rownames(summary_meds_hidradenitis) <- NULL
summary_meds_pemphigoid <- summary_meds_pemphigoid[order(-summary_meds_pemphigoid$Count),] 
rownames(summary_meds_pemphigoid) <- NULL
summary_meds_psoriasis <- summary_meds_psoriasis[order(-summary_meds_psoriasis$Count),] 
rownames(summary_meds_psoriasis) <- NULL
summary_meds_ankylosing <- summary_meds_ankylosing[order(-summary_meds_ankylosing$Count),] 
rownames(summary_meds_ankylosing) <- NULL
summary_meds_polymyalgia <- summary_meds_polymyalgia[order(-summary_meds_polymyalgia$Count),] 
rownames(summary_meds_polymyalgia) <- NULL
summary_meds_psoriatic <- summary_meds_psoriatic[order(-summary_meds_psoriatic$Count),]
rownames(summary_meds_psoriatic) <- NULL
summary_meds_arthritis <- summary_meds_arthritis[order(-summary_meds_arthritis$Count),] 
rownames(summary_meds_arthritis) <- NULL
summary_meds_sjogren <- summary_meds_sjogren[order(-summary_meds_sjogren$Count),] 
rownames(summary_meds_sjogren) <- NULL
summary_meds_lupus <- summary_meds_lupus[order(-summary_meds_lupus$Count),] 
rownames(summary_meds_lupus) <- NULL

```

</br>

# Demographic information: possible and probable

</br>

**Table:** Demographic information for **POSSIBLE** autoimmune cases and controls in the UK Biobank

```{r print_possible_summary, echo = FALSE, results = 'asis'}

kable(possible_summary, escape = F, align = 'r') %>%
  kable_styling(c("striped", "condensed"), full_width = T, font_size = 13) %>%
  pack_rows("Blood", 2, 2, label_row_css = "background-color: #666; color: #fff;") %>%
  pack_rows("Endocrine system", 3, 4, label_row_css = "background-color: #666; color: #fff;") %>%
  pack_rows("Nervous system", 5, 6, label_row_css = "background-color: #666; color: #fff;") %>%
  pack_rows("Digestive system", 7, 8, label_row_css = "background-color: #666; color: #fff;") %>%
  pack_rows("Skin", 9, 9, label_row_css = "background-color: #666; color: #fff;") %>%
  pack_rows("Musculoskeletal system and connective tissue", 10, 15, label_row_css = "background-color: #666; color: #fff;") %>% 
  pack_rows("Total", 16, 16, label_row_css = "background-color: #666; color: #fff;") %>% 
  add_indent(1)

```

</br>

**Table:** Demographic information for **PROBABLE** autoimmune cases and controls in the UK Biobank

```{r print_probable_summary, echo = FALSE, results = 'asis'}

kable(probable_summary, escape = F, align = 'r') %>%
  kable_styling(c("striped", "condensed"), full_width = T, font_size = 13) %>%
  pack_rows("Blood", 2, 2, label_row_css = "background-color: #666; color: #fff;") %>%
  pack_rows("Endocrine system", 3, 4, label_row_css = "background-color: #666; color: #fff;") %>%
  pack_rows("Nervous system", 5, 6, label_row_css = "background-color: #666; color: #fff;") %>%
  pack_rows("Digestive system", 7, 8, label_row_css = "background-color: #666; color: #fff;") %>%
  pack_rows("Skin", 9, 9, label_row_css = "background-color: #666; color: #fff;") %>%
  pack_rows("Musculoskeletal system and connective tissue", 10, 15, label_row_css = "background-color: #666; color: #fff;") %>% 
  pack_rows("Total", 16, 16, label_row_css = "background-color: #666; color: #fff;") %>% 
  add_indent(1) 

```

</br>

## Case-control differences in sociodemographic characteristics

</br>

**Table:** Possible autoimmune cases (N = 28,479) vs controls (N = 324,074)

</br>

```{r ttest_demographics_controls_v_possible, echo=FALSE, results = 'asis'}

# Test for significant differences in demographic variables between possible autoimmune cases and controls

age_controls_v_poss <- t.test(demo_controls$Age, any_poss_only_demo$Age) 
table_age_controls_v_poss <- data.frame(
  "Demographic" = "Mean age",
  "Estimate (controls)" = as.data.frame(age_controls_v_poss$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(age_controls_v_poss$estimate)[2,1],
  "t-statistic" = as.data.frame(age_controls_v_poss$statistic)[1,1],
  "df" = as.data.frame(age_controls_v_poss$parameter)[1,1],
  "p-value" = formatC(age_controls_v_poss$p.value, format = "e", digits = 0)
)

sex_controls_v_poss <- t.test(demo_controls$Sex, any_poss_only_demo$Sex) 
table_sex_controls_v_poss <- data.frame(
  "Demographic" = "Female",
  "Estimate (controls)" = 1-(as.data.frame(sex_controls_v_poss$estimate)[1,1]),
  "Estimate (cases)" = 1-(as.data.frame(sex_controls_v_poss$estimate)[2,1]),
  "t-statistic" = as.data.frame(sex_controls_v_poss$statistic)[1,1],
  "df" = as.data.frame(sex_controls_v_poss$parameter)[1,1],
  "p-value" = formatC(sex_controls_v_poss$p.value, format = "e", digits = 0)
)

ses_controls_v_poss <- t.test(demo_controls$SES, any_poss_only_demo$SES) 
table_ses_controls_v_poss <- data.frame(
  "Demographic" = "SES",
  "Estimate (controls)" = as.data.frame(ses_controls_v_poss$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(ses_controls_v_poss$estimate)[2,1],
  "t-statistic" = as.data.frame(ses_controls_v_poss$statistic)[1,1],
  "df" = as.data.frame(ses_controls_v_poss$parameter)[1,1],
  "p-value" = formatC(ses_controls_v_poss$p.value, format = "e", digits = 0)
)

smoke_controls_v_poss <- t.test(demo_controls$`Current Smoker`, any_poss_only_demo$`Current Smoker`) 
table_smoke_controls_v_poss <- data.frame(
  "Demographic" = "Current smoker",
  "Estimate (controls)" = as.data.frame(smoke_controls_v_poss$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(smoke_controls_v_poss$estimate)[2,1],
  "t-statistic" = as.data.frame(smoke_controls_v_poss$statistic)[1,1],
  "df" = as.data.frame(smoke_controls_v_poss$parameter)[1,1],
  "p-value" = formatC(smoke_controls_v_poss$p.value, format = "e", digits = 0)
)

bmi_controls_v_poss <- t.test(demo_controls$BMI, any_poss_only_demo$BMI) 
table_bmi_controls_v_poss <- data.frame(
  "Demographic" = "Mean BMI",
  "Estimate (controls)" = as.data.frame(bmi_controls_v_poss$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(bmi_controls_v_poss$estimate)[2,1],
  "t-statistic" = as.data.frame(bmi_controls_v_poss$statistic)[1,1],
  "df" = as.data.frame(bmi_controls_v_poss$parameter)[1,1],
  "p-value" = formatC(bmi_controls_v_poss$p.value, format = "e", digits = 0)
)

controls_vs_poss <- rbind(table_age_controls_v_poss,
                  table_sex_controls_v_poss,
                  table_ses_controls_v_poss,
                  table_smoke_controls_v_poss,
                  table_bmi_controls_v_poss)

colnames(controls_vs_poss) <- c("Demographic","Estimate (controls)","Estimate (cases)","t-statistic","df","p-value")

controls_vs_poss$`Estimate (controls)` <- round(controls_vs_poss$`Estimate (controls)`, digits = 2)
controls_vs_poss$`Estimate (cases)` <- round(controls_vs_poss$`Estimate (cases)`, digits = 2)
controls_vs_poss$`t-statistic` <- round(controls_vs_poss$`t-statistic`, digits = 0)

kable(controls_vs_poss, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Table:** Probable autoimmune cases (N = 16,824) vs controls (N = 324,074)

</br>

```{r ttest_demographics_controls_v_probable, echo=FALSE, results = 'asis'}

# Test for significant differences in demographic variables between probable autoimmune cases and controls

age_controls_v_prob <- t.test(demo_controls$Age, any_prob_only_demo$Age) 
table_age_controls_v_prob <- data.frame(
  "Demographic" = "Mean age",
  "Estimate (controls)" = as.data.frame(age_controls_v_prob$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(age_controls_v_prob$estimate)[2,1],
  "t-statistic" = as.data.frame(age_controls_v_prob$statistic)[1,1],
  "df" = as.data.frame(age_controls_v_prob$parameter)[1,1],
  "p-value" = formatC(age_controls_v_prob$p.value, format = "e", digits = 0)
)

sex_controls_v_prob <- t.test(demo_controls$Sex, any_prob_only_demo$Sex) 
table_sex_controls_v_prob <- data.frame(
  "Demographic" = "Female",
  "Estimate (controls)" = 1-(as.data.frame(sex_controls_v_prob$estimate)[1,1]),
  "Estimate (cases)" = 1-(as.data.frame(sex_controls_v_prob$estimate)[2,1]),
  "t-statistic" = as.data.frame(sex_controls_v_prob$statistic)[1,1],
  "df" = as.data.frame(sex_controls_v_prob$parameter)[1,1],
  "p-value" = formatC(sex_controls_v_prob$p.value, format = "e", digits = 0)
)

ses_controls_v_prob <- t.test(demo_controls$SES, any_prob_only_demo$SES) 
table_ses_controls_v_prob <- data.frame(
  "Demographic" = "SES",
  "Estimate (controls)" = as.data.frame(ses_controls_v_prob$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(ses_controls_v_prob$estimate)[2,1],
  "t-statistic" = as.data.frame(ses_controls_v_prob$statistic)[1,1],
  "df" = as.data.frame(ses_controls_v_prob$parameter)[1,1],
  "p-value" = formatC(ses_controls_v_prob$p.value, format = "e", digits = 0)
)

smoke_controls_v_prob <- t.test(demo_controls$`Current Smoker`, any_prob_only_demo$`Current Smoker`) 
table_smoke_controls_v_prob <- data.frame(
  "Demographic" = "Current smoker",
  "Estimate (controls)" = as.data.frame(smoke_controls_v_prob$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(smoke_controls_v_prob$estimate)[2,1],
  "t-statistic" = as.data.frame(smoke_controls_v_prob$statistic)[1,1],
  "df" = as.data.frame(smoke_controls_v_prob$parameter)[1,1],
  "p-value" = formatC(smoke_controls_v_prob$p.value, format = "e", digits = 0)
)

bmi_controls_v_prob <- t.test(demo_controls$BMI, any_prob_only_demo$BMI) 
table_bmi_controls_v_prob <- data.frame(
  "Demographic" = "Mean BMI",
  "Estimate (controls)" = as.data.frame(bmi_controls_v_prob$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(bmi_controls_v_prob$estimate)[2,1],
  "t-statistic" = as.data.frame(bmi_controls_v_prob$statistic)[1,1],
  "df" = as.data.frame(bmi_controls_v_prob$parameter)[1,1],
  "p-value" = formatC(bmi_controls_v_prob$p.value, format = "e", digits = 0)
)

controls_vs_prob <- rbind(table_age_controls_v_prob,
                  table_sex_controls_v_prob,
                  table_ses_controls_v_prob,
                  table_smoke_controls_v_prob,
                  table_bmi_controls_v_prob)

colnames(controls_vs_prob) <- c("Demographic","Estimate (controls)","Estimate (cases)","t-statistic","df","p-value")

controls_vs_prob$`Estimate (controls)` <- round(controls_vs_prob$`Estimate (controls)`, digits = 2)
controls_vs_prob$`Estimate (cases)` <- round(controls_vs_prob$`Estimate (cases)`, digits = 2)
controls_vs_prob$`t-statistic` <- round(controls_vs_prob$`t-statistic`, digits = 0)

kable(controls_vs_prob, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

# Summary information for each disorder

</br>

## Pernicious Anemia

---

```{r pa_create_venn_diagrams, echo=FALSE}

# possible and probable
pa_pp <- euler(pa_poss_prob[,2:3] > 0)

# possible: ICD, self and medication
pa_overlap <- euler(pa_icd_self_med[,2:3] > 0)

```

```{r pa_print_venn_diagrams, echo=FALSE, results = 'asis', fig.align='center', fig.height=5, fig.width=10}

pap1 <- plot(pa_pp, quantities = TRUE, fill = c("dodgerblue2", "firebrick2"))
pap2 <- plot(pa_overlap, quantities = TRUE, fill = c("lightgreen", "plum2"))
grid.arrange(pap1, pap2, ncol = 2)

```

**Venn diagrams:** *Left:* probable cases as a subset of possible; *Right:* overlapping measures in the UK Biobank

</br>

**Matrix of ICD-10 primary and secondary dx - cells show number of individuals with each combination of dx**

```{r pa_print_icd_matrix, echo = FALSE, results = 'asis'}

pa_matrix <- matrix(ncol = 3, nrow = 3)
pa_matrix[1,1] <- paste0(pa_temp_matrix[1,1], footnote_marker_number(1))
pa_matrix[1,2] <- paste0(pa_temp_matrix[1,2], footnote_marker_number(2))
pa_matrix[1,3] <- paste0(pa_temp_matrix[1,3], footnote_marker_number(3))
pa_matrix[2,1] <- paste0(pa_temp_matrix[2,1], footnote_marker_number(2))
pa_matrix[2,2] <- paste0(pa_temp_matrix[2,2], footnote_marker_number(3))
pa_matrix[2,3] <- paste0(pa_temp_matrix[2,3], footnote_marker_number(3))
pa_matrix[3,1] <- paste0(pa_temp_matrix[3,1], footnote_marker_number(3))
pa_matrix[3,2] <- paste0(pa_temp_matrix[3,2], footnote_marker_number(3))
pa_matrix[3,3] <- paste0(pa_temp_matrix[3,3], footnote_marker_number(3))

colnames(pa_matrix) <- c("0 primary dx","1 primary dx",">1 primary dx")
rownames(pa_matrix) <- c("0 secondary dx","1 secondary dx",">1 secondary dx")

kable(pa_matrix, escape = F, align = 'r') %>%
  kable_styling(bootstrap_options = "striped") %>%
  column_spec(1, bold = TRUE, color = "white", background = "#666") %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>%
  footnote(general = " ",
           number = c("Controls", "Possible cases", "Probable cases"))

```

```{r pa_remove_zero_sums_from_icd_subcodes_table, echo = FALSE, include = FALSE}

pa_subcodes_final <- pa_subcodes[apply(pa_subcodes[,-1], 1, function(x) !all(x==0)),]
rownames(pa_subcodes_final) <- NULL

```

</br>

**ICD-10 codes used to construct matrix above. Columns show count of individuals who have received at least one dx** 

```{r pa_print_icd_subcodes_table, echo = FALSE, results = 'asis'}

kable(pa_subcodes_final, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%   
  pack_rows("D51 Vitamin B 12 deficiency anaemia", 1, 1, label_row_css = "background-color: #888; color: #fff;") %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")
 
```

</br>

**Count of self-reported autoimmune condition**

```{r pa_print_self_report, echo=FALSE, results = 'asis'}

pa_table_self <- summary_self_report[1,] %>% 
  `rownames<-`(NULL)

kable(pa_table_self, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

## Autoimmune Thyroid Disease

---

```{r at_create_venn_diagrams, echo=FALSE}

# possible and probable
at_pp <- euler(at_poss_prob[,2:3] > 0)

# possible: ICD, self and medication
at_overlap <- euler(at_icd_self_med[,2:4] > 0)

```

```{r at_print_venn_diagrams, echo=FALSE, results = 'asis', fig.align='center', fig.height=5, fig.width=10}

atp1 <- plot(at_pp, quantities = TRUE, fill = c("royalblue2", "firebrick2"))
atp2 <- plot(at_overlap, quantities = TRUE, fill = c("lightgreen", "plum2", "lightyellow"))
grid.arrange(atp1, atp2, ncol = 2)

```

**Venn diagrams:** *Left:* probable cases as a subset of possible; *Right:* overlapping measures in the UK Biobank

</br>

**Matrix of ICD-10 primary and secondary dx - cells show number of individuals with each combination of dx**

```{r at_print_icd_matrix, echo = FALSE, results = 'asis'}

at_matrix <- matrix(ncol = 3, nrow = 3)
at_matrix[1,1] <- paste0(at_temp_matrix[1,1], footnote_marker_number(1))
at_matrix[1,2] <- paste0(at_temp_matrix[1,2], footnote_marker_number(2))
at_matrix[1,3] <- paste0(at_temp_matrix[1,3], footnote_marker_number(3))
at_matrix[2,1] <- paste0(at_temp_matrix[2,1], footnote_marker_number(2))
at_matrix[2,2] <- paste0(at_temp_matrix[2,2], footnote_marker_number(3))
at_matrix[2,3] <- paste0(at_temp_matrix[2,3], footnote_marker_number(3))
at_matrix[3,1] <- paste0(at_temp_matrix[3,1], footnote_marker_number(3))
at_matrix[3,2] <- paste0(at_temp_matrix[3,2], footnote_marker_number(3))
at_matrix[3,3] <- paste0(at_temp_matrix[3,3], footnote_marker_number(3))

colnames(at_matrix) <- c("0 primary dx","1 primary dx",">1 primary dx")
rownames(at_matrix) <- c("0 secondary dx","1 secondary dx",">1 secondary dx")

kable(at_matrix, escape = F, align = 'r') %>%
  kable_styling(bootstrap_options = "striped") %>%
  column_spec(1, bold = TRUE, color = "white", background = "#666") %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>%
  footnote(general = " ",
           number = c("Controls", "Possible cases", "Probable cases"))

```

```{r at_remove_zero_sums_from_icd_subcodes_table, echo = FALSE, include = FALSE}

at_subcodes_final <- at_subcodes[apply(at_subcodes[,-1], 1, function(x) !all(x==0)),]
rownames(at_subcodes_final) <- NULL
```

</br>

**ICD-10 codes used to construct matrix above. Columns show count of individuals who have received at least one dx** 

```{r at_print_icd_subcodes_table, echo = FALSE, results = 'asis'}

kable(at_subcodes_final, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  pack_rows("E05 Thyrotoxicosis [hyperthyroidism]", 1, 1, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("E06 Thyroiditis", 2, 2, label_row_css = "background-color: #888; color: #fff;") %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Count of self-reported autoimmune condition**

```{r at_print_self_report, echo=FALSE, results = 'asis'}

at_table_self <- summary_self_report[2:3,] %>% 
  `rownames<-`(NULL)

kable(at_table_self, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Count of self-reported prescription medications** 

```{r at_print_meds, echo=FALSE, results = 'asis'}

kable(summary_meds_thyroid, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed"), ) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>% 
  collapse_rows(columns = 1, valign = "top") %>% 
  column_spec(1, width = "30em") 

```

</br>

## Type 1 diabetes

---

```{r t1d_create_venn_diagrams, echo=FALSE}

# possible and probable
t1d_pp <- euler(t1d_poss_prob[,2:3] > 0)

# possible: ICD, self and medication
t1d_overlap <- euler(t1d_icd_self_med[,2:4] > 0)

```

```{r t1d_print_venn_diagrams, echo=FALSE, results = 'asis', fig.align='center', fig.height=5, fig.width=10}

t1dp1 <- plot(t1d_pp, quantities = TRUE, fill = c("dodgerblue2", "firebrick2"))
t1dp2 <- plot(t1d_overlap, quantities = TRUE, fill = c("lightgreen", "plum2", "lightyellow"))
grid.arrange(t1dp1, t1dp2, ncol = 2)

```

**Venn diagrams:** *Left:* probable cases as a subset of possible; *Right:* overlapping measures in the UK Biobank

</br>

**Matrix of ICD-10 primary and secondary dx - cells show number of individuals with each combination of dx**

```{r t1d_print_icd_matrix, echo = FALSE, results = 'asis'}

t1d_matrix <- matrix(ncol = 3, nrow = 3)
t1d_matrix[1,1] <- paste0(t1d_temp_matrix[1,1], footnote_marker_number(1))
t1d_matrix[1,2] <- paste0(t1d_temp_matrix[1,2], footnote_marker_number(2))
t1d_matrix[1,3] <- paste0(t1d_temp_matrix[1,3], footnote_marker_number(3))
t1d_matrix[2,1] <- paste0(t1d_temp_matrix[2,1], footnote_marker_number(2))
t1d_matrix[2,2] <- paste0(t1d_temp_matrix[2,2], footnote_marker_number(3))
t1d_matrix[2,3] <- paste0(t1d_temp_matrix[2,3], footnote_marker_number(3))
t1d_matrix[3,1] <- paste0(t1d_temp_matrix[3,1], footnote_marker_number(3))
t1d_matrix[3,2] <- paste0(t1d_temp_matrix[3,2], footnote_marker_number(3))
t1d_matrix[3,3] <- paste0(t1d_temp_matrix[3,3], footnote_marker_number(3))

colnames(t1d_matrix) <- c("0 primary dx","1 primary dx",">1 primary dx")
rownames(t1d_matrix) <- c("0 secondary dx","1 secondary dx",">1 secondary dx")

kable(t1d_matrix, escape = F, align = 'r') %>%
  kable_styling(bootstrap_options = "striped") %>%
  column_spec(1, bold = TRUE, color = "white", background = "#666") %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>%
  footnote(general = " ",
           number = c("Controls", "Possible cases", "Probable cases"))

```

```{r t1d_remove_zero_sums_from_icd_subcodes_table, echo = FALSE, include = FALSE}

t1d_subcodes_final <- t1d_subcodes[apply(t1d_subcodes[,-1], 1, function(x) !all(x==0)),]
rownames(t1d_subcodes_final) <- NULL

```

</br>

**ICD-10 codes used to construct matrix above. Columns show count of individuals who have received at least one dx** 

```{r t1d_print_icd_subcodes_table, echo = FALSE, results = 'asis'}


kable(t1d_subcodes_final, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  pack_rows("E10 Type 1 diabetes mellitus", 1, 10, label_row_css = "background-color: #888; color: #fff;") %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Count of self-reported autoimmune condition**

```{r t1d_print_self_report, echo=FALSE, results = 'asis'}

t1d_table_self <- summary_self_report[4,] %>% 
  `rownames<-`(NULL)

kable(t1d_table_self, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Count of self-reported prescription medications** 

```{r t1d_print_meds, echo=FALSE, results = 'asis'}

kable(summary_meds_diabetes, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed"), ) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>% 
  collapse_rows(columns = 1, valign = "top") %>% 
  column_spec(1, width = "30em") 

```

</br>

## Multiple Sclerosis

---

```{r ms_create_venn_diagrams, echo=FALSE}

# possible and probable
ms_pp <- euler(ms_poss_prob[,2:3] > 0)

# possible: ICD, self and medication
ms_overlap <- euler(ms_icd_self_med[,2:4] > 0)

```

```{r ms_print_venn_diagrams, echo=FALSE, results = 'asis', fig.align='center', fig.height=5, fig.width=10}

msp1 <- plot(ms_pp, quantities = TRUE, fill = c("dodgerblue2", "firebrick2"))
msp2 <- plot(ms_overlap, quantities = TRUE, fill = c("lightgreen", "plum2", "lightyellow"))
grid.arrange(msp1, msp2, ncol = 2)

```

**Venn diagrams:** *Left:* probable cases as a subset of possible; *Right:* overlapping measures in the UK Biobank

</br>

**Matrix of ICD-10 primary and secondary dx - cells show number of individuals with each combination of dx**

```{r ms_print_icd_matrix, echo = FALSE, results = 'asis'}

ms_matrix <- matrix(ncol = 3, nrow = 3)
ms_matrix[1,1] <- paste0(ms_temp_matrix[1,1], footnote_marker_number(1))
ms_matrix[1,2] <- paste0(ms_temp_matrix[1,2], footnote_marker_number(2))
ms_matrix[1,3] <- paste0(ms_temp_matrix[1,3], footnote_marker_number(3))
ms_matrix[2,1] <- paste0(ms_temp_matrix[2,1], footnote_marker_number(2))
ms_matrix[2,2] <- paste0(ms_temp_matrix[2,2], footnote_marker_number(3))
ms_matrix[2,3] <- paste0(ms_temp_matrix[2,3], footnote_marker_number(3))
ms_matrix[3,1] <- paste0(ms_temp_matrix[3,1], footnote_marker_number(3))
ms_matrix[3,2] <- paste0(ms_temp_matrix[3,2], footnote_marker_number(3))
ms_matrix[3,3] <- paste0(ms_temp_matrix[3,3], footnote_marker_number(3))

colnames(ms_matrix) <- c("0 primary dx","1 primary dx",">1 primary dx")
rownames(ms_matrix) <- c("0 secondary dx","1 secondary dx",">1 secondary dx")

kable(ms_matrix, escape = F, align = 'r') %>%
  kable_styling(bootstrap_options = "striped") %>%
  column_spec(1, bold = TRUE, color = "white", background = "#666") %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>%
  footnote(general = " ",
           number = c("Controls", "Possible cases", "Probable cases"))

```

```{r ms_remove_zero_sums_from_icd_subcodes_table, echo = FALSE, include = FALSE}

ms_subcodes_final <- ms_subcodes[apply(ms_subcodes[,-1], 1, function(x) !all(x==0)),]
rownames(ms_subcodes_final) <- NULL

```

</br>

**ICD-10 codes used to construct matrix above. Columns show count of individuals who have received at least one dx** 

```{r ms_print_icd_subcodes_table, echo = FALSE, results = 'asis'}

kable(ms_subcodes_final, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  pack_rows("G35 Multiple sclerosis", 1, 1, label_row_css = "background-color: #888; color: #fff;") %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Count of self-reported autoimmune condition**

```{r ms_print_self_report, echo=FALSE, results = 'asis'}

ms_table_self <- summary_self_report[5,] %>% 
  `rownames<-`(NULL)

kable(ms_table_self, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Count of self-reported prescription medications** 

```{r ms_print_meds, echo=FALSE, results = 'asis'}

kable(summary_meds_sclerosis, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed"), ) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>% 
  collapse_rows(columns = 1, valign = "top") %>% 
  column_spec(1, width = "30em") 

```

</br>

## Myasthenia Gravis

---

```{r mg_create_venn_diagrams, echo=FALSE}

# possible and probable
mg_pp <- euler(mg_poss_prob[,2:3] > 0)

# possible: ICD, self and medication
mg_overlap <- euler(mg_icd_self_med[,2:4] > 0)

```

```{r mg_print_venn_diagrams, echo=FALSE, results = 'asis', fig.align='center', fig.height=5, fig.width=10}

mgp1 <- plot(mg_pp, quantities = TRUE, fill = c("dodgerblue2", "firebrick2"))
mgp2 <- plot(mg_overlap, quantities = TRUE, fill = c("lightgreen", "plum2", "lightyellow"))
grid.arrange(mgp1, mgp2, ncol = 2)

```

**Venn diagrams:** *Left:* probable cases as a subset of possible; *Right:* overlapping measures in the UK Biobank

</br>

**Matrix of ICD-10 primary and secondary dx - cells show number of individuals with each combination of dx**

```{r mg_print_icd_matrix, echo = FALSE, results = 'asis'}

mg_matrix <- matrix(ncol = 3, nrow = 3)
mg_matrix[1,1] <- paste0(mg_temp_matrix[1,1], footnote_marker_number(1))
mg_matrix[1,2] <- paste0(mg_temp_matrix[1,2], footnote_marker_number(2))
mg_matrix[1,3] <- paste0(mg_temp_matrix[1,3], footnote_marker_number(3))
mg_matrix[2,1] <- paste0(mg_temp_matrix[2,1], footnote_marker_number(2))
mg_matrix[2,2] <- paste0(mg_temp_matrix[2,2], footnote_marker_number(3))
mg_matrix[2,3] <- paste0(mg_temp_matrix[2,3], footnote_marker_number(3))
mg_matrix[3,1] <- paste0(mg_temp_matrix[3,1], footnote_marker_number(3))
mg_matrix[3,2] <- paste0(mg_temp_matrix[3,2], footnote_marker_number(3))
mg_matrix[3,3] <- paste0(mg_temp_matrix[3,3], footnote_marker_number(3))

colnames(mg_matrix) <- c("0 primary dx","1 primary dx",">1 primary dx")
rownames(mg_matrix) <- c("0 secondary dx","1 secondary dx",">1 secondary dx")

kable(mg_matrix, escape = F, align = 'r') %>%
  kable_styling(bootstrap_options = "striped") %>%
  column_spec(1, bold = TRUE, color = "white", background = "#666") %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>%
  footnote(general = " ",
           number = c("Controls", "Possible cases", "Probable cases"))

```

```{r mg_remove_zero_sums_from_icd_subcodes_table, echo = FALSE, include = FALSE}

mg_subcodes_final <- mg_subcodes[apply(mg_subcodes[,-1], 1, function(x) !all(x==0)),]
rownames(mg_subcodes_final) <- NULL

```

</br>

**ICD-10 codes used to construct matrix above. Columns show count of individuals who have received at least one dx** 

```{r mg_print_icd_subcodes_table, echo = FALSE, results = 'asis'}

kable(mg_subcodes_final, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  pack_rows("G70 Myasthenia gravis and other myoneural disorders", 1, 1, label_row_css = "background-color: #888; color: #fff;") %>%
   row_spec(0, bold = TRUE, color = "white", background = "#666")
	 
```

</br>

**Count of self-reported autoimmune condition**

```{r mg_print_self_report, echo=FALSE, results = 'asis'}

mg_table_self <- summary_self_report[6,] %>% 
  `rownames<-`(NULL)

kable(mg_table_self, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Count of self-reported prescription medications** 

```{r mg_print_meds, echo=FALSE, results = 'asis'}

kable(summary_meds_myasthenia, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed"), ) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>% 
  collapse_rows(columns = 1, valign = "top") %>% 
  column_spec(1, width = "30em") 

```

</br>

## Coeliac

---

```{r coeliac_create_venn_diagrams, echo=FALSE}

# possible and probable
coeliac_pp <- euler(coeliac_poss_prob[,2:3] > 0)

# possible: ICD, self and medication
coeliac_overlap <- euler(coeliac_icd_self_med[,2:3] > 0)

```

```{r coeliac_print_venn_diagrams, echo=FALSE, results = 'asis', fig.align='center', fig.height=5, fig.width=10}

coeliacp1 <- plot(coeliac_pp, quantities = TRUE, fill = c("dodgerblue2", "firebrick2"))
coeliacp2 <- plot(coeliac_overlap, quantities = TRUE, fill = c("lightgreen", "plum2"))
grid.arrange(coeliacp1, coeliacp2, ncol = 2)

```

**Venn diagrams:** *Left:* probable cases as a subset of possible; *Right:* overlapping measures in the UK Biobank

</br>

**Matrix of ICD-10 primary and secondary dx - cells show number of individuals with each combination of dx**

```{r coeliac_print_icd_matrix, echo = FALSE, results = 'asis'}

coeliac_matrix <- matrix(ncol = 3, nrow = 3)
coeliac_matrix[1,1] <- paste0(coeliac_temp_matrix[1,1], footnote_marker_number(1))
coeliac_matrix[1,2] <- paste0(coeliac_temp_matrix[1,2], footnote_marker_number(2))
coeliac_matrix[1,3] <- paste0(coeliac_temp_matrix[1,3], footnote_marker_number(3))
coeliac_matrix[2,1] <- paste0(coeliac_temp_matrix[2,1], footnote_marker_number(2))
coeliac_matrix[2,2] <- paste0(coeliac_temp_matrix[2,2], footnote_marker_number(3))
coeliac_matrix[2,3] <- paste0(coeliac_temp_matrix[2,3], footnote_marker_number(3))
coeliac_matrix[3,1] <- paste0(coeliac_temp_matrix[3,1], footnote_marker_number(3))
coeliac_matrix[3,2] <- paste0(coeliac_temp_matrix[3,2], footnote_marker_number(3))
coeliac_matrix[3,3] <- paste0(coeliac_temp_matrix[3,3], footnote_marker_number(3))

colnames(coeliac_matrix) <- c("0 primary dx","1 primary dx",">1 primary dx")
rownames(coeliac_matrix) <- c("0 secondary dx","1 secondary dx",">1 secondary dx")

kable(coeliac_matrix, escape = F, align = 'r') %>%
  kable_styling(bootstrap_options = "striped") %>%
  column_spec(1, bold = TRUE, color = "white", background = "#666") %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>%
  footnote(general = " ",
           number = c("Controls", "Possible cases", "Probable cases"))

```

```{r coeliac_remove_zero_sums_from_icd_subcodes_table, echo = FALSE, include = FALSE}

coeliac_subcodes_final <- coeliac_subcodes[apply(coeliac_subcodes[,-1], 1, function(x) !all(x==0)),]
rownames(coeliac_subcodes_final) <- NULL

```

</br>

**ICD-10 codes used to construct matrix above. Columns show count of individuals who have received at least one dx** 

```{r coeliac_print_icd_subcodes_table, echo = FALSE, results = 'asis'}

kable(coeliac_subcodes_final, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  pack_rows("K90 Intestinal malabsorption", 1, 1, label_row_css = "background-color: #888; color: #fff;") %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Count of self-reported autoimmune condition**

```{r coeliac_print_self_report, echo=FALSE, results = 'asis'}

coeliac_table_self <- summary_self_report[7,] %>% 
  `rownames<-`(NULL)

kable(coeliac_table_self, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

## Inflammatory Bowel Disease

---

```{r ibd_create_venn_diagrams, echo=FALSE}

# possible and probable
ibd_pp <- euler(ibd_poss_prob[,2:3] > 0)

# possible: ICD, self and medication
ibd_overlap <- euler(ibd_icd_self_med[,2:4] > 0)

```

```{r ibd_print_venn_diagrams, echo=FALSE, results = 'asis', fig.align='center', fig.height=5, fig.width=10}

ibdp1 <- plot(ibd_pp, quantities = TRUE, fill = c("dodgerblue2", "firebrick2"))
ibdp2 <- plot(ibd_overlap, quantities = TRUE, fill = c("lightgreen", "plum2", "lightyellow"))
grid.arrange(ibdp1, ibdp2, ncol = 2)

```

**Venn diagrams:** *Left:* probable cases as a subset of possible; *Right:* overlapping measures in the UK Biobank

</br>

**Matrix of ICD-10 primary and secondary dx - cells show number of individuals with each combination of dx**

```{r ibd_print_icd_matrix, echo = FALSE, results = 'asis'}

ibd_matrix <- matrix(ncol = 3, nrow = 3)
ibd_matrix[1,1] <- paste0(ibd_temp_matrix[1,1], footnote_marker_number(1))
ibd_matrix[1,2] <- paste0(ibd_temp_matrix[1,2], footnote_marker_number(2))
ibd_matrix[1,3] <- paste0(ibd_temp_matrix[1,3], footnote_marker_number(3))
ibd_matrix[2,1] <- paste0(ibd_temp_matrix[2,1], footnote_marker_number(2))
ibd_matrix[2,2] <- paste0(ibd_temp_matrix[2,2], footnote_marker_number(3))
ibd_matrix[2,3] <- paste0(ibd_temp_matrix[2,3], footnote_marker_number(3))
ibd_matrix[3,1] <- paste0(ibd_temp_matrix[3,1], footnote_marker_number(3))
ibd_matrix[3,2] <- paste0(ibd_temp_matrix[3,2], footnote_marker_number(3))
ibd_matrix[3,3] <- paste0(ibd_temp_matrix[3,3], footnote_marker_number(3))

colnames(ibd_matrix) <- c("0 primary dx","1 primary dx",">1 primary dx")
rownames(ibd_matrix) <- c("0 secondary dx","1 secondary dx",">1 secondary dx")

kable(ibd_matrix, escape = F, align = 'r') %>%
  kable_styling(bootstrap_options = "striped") %>%
  column_spec(1, bold = TRUE, color = "white", background = "#666") %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>%
  footnote(general = " ",
           number = c("Controls", "Possible cases", "Probable cases"))

```

```{r ibd_remove_zero_sums_from_icd_subcodes_table, echo = FALSE, include = FALSE}

ibd_subcodes_final <- ibd_subcodes[apply(ibd_subcodes[,-1], 1, function(x) !all(x==0)),]
rownames(ibd_subcodes_final) <- NULL

```

</br>

**ICD-10 codes used to construct matrix above. Columns show count of individuals who have received at least one dx** 

```{r ibd_print_icd_subcodes_table, echo = FALSE, results = 'asis'}

kable(ibd_subcodes_final, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  pack_rows("K50 Crohn disease [regional enteritis]", 1, 4, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("K51 Ulcerative colitis", 5, nrow(ibd_subcodes_final), label_row_css = "background-color: #888; color: #fff;") %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Count of self-reported autoimmune condition**

```{r ibd_print_self_report, echo=FALSE, results = 'asis'}

ibd_table_self <- summary_self_report[8:9,] %>% 
  `rownames<-`(NULL)

kable(ibd_table_self, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Count of self-reported prescription medications** 

```{r ibd_print_meds, echo=FALSE, results = 'asis'}

kable(summary_meds_crohn, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed"), ) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>% 
  collapse_rows(columns = 1, valign = "top") %>% 
  column_spec(1, width = "30em") 

```

</br>

## Psoriasis

---

```{r psoriasis_create_venn_diagrams, echo=FALSE}

# possible and probable
psoriasis_pp <- euler(psoriasis_poss_prob[,2:3] > 0)

# possible: ICD, self and medication
psoriasis_overlap <- euler(psoriasis_icd_self_med[,2:4] > 0)

```

```{r psoriasis_print_venn_diagrams, echo=FALSE, results = 'asis', fig.align='center', fig.height=5, fig.width=10}

psoriasisp1 <- plot(psoriasis_pp, quantities = TRUE, fill = c("dodgerblue2", "firebrick2"))
psoriasisp2 <- plot(psoriasis_overlap, quantities = TRUE, fill = c("lightgreen", "plum2", "lightyellow"))
grid.arrange(psoriasisp1, psoriasisp2, ncol = 2)

```

**Venn diagrams:** *Left:* probable cases as a subset of possible; *Right:* overlapping measures in the UK Biobank

</br>

**Matrix of ICD-10 primary and secondary dx - cells show number of individuals with each combination of dx**

```{r psoriasis_print_icd_matrix, echo = FALSE, results = 'asis'}

psoriasis_matrix <- matrix(ncol = 3, nrow = 3)
psoriasis_matrix[1,1] <- paste0(psoriasis_temp_matrix[1,1], footnote_marker_number(1))
psoriasis_matrix[1,2] <- paste0(psoriasis_temp_matrix[1,2], footnote_marker_number(2))
psoriasis_matrix[1,3] <- paste0(psoriasis_temp_matrix[1,3], footnote_marker_number(3))
psoriasis_matrix[2,1] <- paste0(psoriasis_temp_matrix[2,1], footnote_marker_number(2))
psoriasis_matrix[2,2] <- paste0(psoriasis_temp_matrix[2,2], footnote_marker_number(3))
psoriasis_matrix[2,3] <- paste0(psoriasis_temp_matrix[2,3], footnote_marker_number(3))
psoriasis_matrix[3,1] <- paste0(psoriasis_temp_matrix[3,1], footnote_marker_number(3))
psoriasis_matrix[3,2] <- paste0(psoriasis_temp_matrix[3,2], footnote_marker_number(3))
psoriasis_matrix[3,3] <- paste0(psoriasis_temp_matrix[3,3], footnote_marker_number(3))

colnames(psoriasis_matrix) <- c("0 primary dx","1 primary dx",">1 primary dx")
rownames(psoriasis_matrix) <- c("0 secondary dx","1 secondary dx",">1 secondary dx")

kable(psoriasis_matrix, escape = F, align = 'r') %>%
  kable_styling(bootstrap_options = "striped") %>%
  column_spec(1, bold = TRUE, color = "white", background = "#666") %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>%
  footnote(general = " ",
           number = c("Controls", "Possible cases", "Probable cases"))

```

```{r psoriasis_remove_zero_sums_from_icd_subcodes_table, echo = FALSE, include = FALSE}

psoriasis_subcodes_final <- psoriasis_subcodes[apply(psoriasis_subcodes[,-1], 1, function(x) !all(x==0)),]
rownames(psoriasis_subcodes_final) <- NULL

```

</br>

**ICD-10 codes used to construct matrix above. Columns show count of individuals who have received at least one dx** 

```{r psoriasis_print_icd_subcodes_table, echo = FALSE, results = 'asis'}

kable(psoriasis_subcodes_final, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  pack_rows("L40 Psoriasis", 1, 6, label_row_css = "background-color: #888; color: #fff;") %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Count of self-reported autoimmune condition**

```{r psoriasis_print_self_report, echo=FALSE, results = 'asis'}

psoriasis_table_self <- summary_self_report[11,] %>% 
  `rownames<-`(NULL)

kable(psoriasis_table_self, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Count of self-reported prescription medications** 

```{r psoriasis_print_meds, echo=FALSE, results = 'asis'}

kable(summary_meds_psoriasis, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed"), ) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>% 
  collapse_rows(columns = 1, valign = "top") %>% 
  column_spec(1, width = "30em") 

```

</br>

## Ankylosing Spondylitis

---

```{r as_create_venn_diagrams, echo=FALSE}

# possible and probable
as_pp <- euler(as_poss_prob[,2:3] > 0)

# possible: ICD, self and medication
as_overlap <- euler(as_icd_self_med[,2:4] > 0)

```

```{r as_print_venn_diagrams, echo=FALSE, results = 'asis', fig.align='center', fig.height=5, fig.width=10}

asp1 <- plot(as_pp, quantities = TRUE, fill = c("dodgerblue2", "firebrick2"))
asp2 <- plot(as_overlap, quantities = TRUE, fill = c("lightgreen", "plum2", "lightyellow"))
grid.arrange(asp1, asp2, ncol = 2)

```

**Venn diagrams:** *Left:* probable cases as a subset of possible; *Right:* overlapping measures in the UK Biobank

</br>

**Matrix of ICD-10 primary and secondary dx - cells show number of individuals with each combination of dx**

```{r as_print_icd_matrix, echo = FALSE, results = 'asis'}

as_matrix <- matrix(ncol = 3, nrow = 3)
as_matrix[1,1] <- paste0(as_temp_matrix[1,1], footnote_marker_number(1))
as_matrix[1,2] <- paste0(as_temp_matrix[1,2], footnote_marker_number(2))
as_matrix[1,3] <- paste0(as_temp_matrix[1,3], footnote_marker_number(3))
as_matrix[2,1] <- paste0(as_temp_matrix[2,1], footnote_marker_number(2))
as_matrix[2,2] <- paste0(as_temp_matrix[2,2], footnote_marker_number(3))
as_matrix[2,3] <- paste0(as_temp_matrix[2,3], footnote_marker_number(3))
as_matrix[3,1] <- paste0(as_temp_matrix[3,1], footnote_marker_number(3))
as_matrix[3,2] <- paste0(as_temp_matrix[3,2], footnote_marker_number(3))
as_matrix[3,3] <- paste0(as_temp_matrix[3,3], footnote_marker_number(3))

colnames(as_matrix) <- c("0 primary dx","1 primary dx",">1 primary dx")
rownames(as_matrix) <- c("0 secondary dx","1 secondary dx",">1 secondary dx")

kable(as_matrix, escape = F, align = 'r') %>%
  kable_styling(bootstrap_options = "striped") %>%
  column_spec(1, bold = TRUE, color = "white", background = "#666") %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>%
  footnote(general = " ",
           number = c("Controls", "Possible cases", "Probable cases"))

```

```{r as_remove_zero_sums_from_icd_subcodes_table, echo = FALSE, include = FALSE}

as_subcodes_final <- as_subcodes[apply(as_subcodes[,-1], 1, function(x) !all(x==0)),]
rownames(as_subcodes_final) <- NULL

```

</br>

**ICD-10 codes used to construct matrix above. Columns show count of individuals who have received at least one dx** 

```{r as_print_icd_subcodes_table, echo = FALSE, results = 'asis'}

kable(as_subcodes_final, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  pack_rows("M45 Ankylosing spondylitis ", 1, 9, label_row_css = "background-color: #888; color: #fff;") %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Count of self-reported autoimmune condition**

```{r as_print_self_report, echo=FALSE, results = 'asis'}

as_table_self <- summary_self_report[12,] %>% 
  `rownames<-`(NULL)

kable(as_table_self, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Count of self-reported prescription medications** 

```{r as_print_meds, echo=FALSE, results = 'asis'}

kable(summary_meds_ankylosing, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed"), ) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>% 
  collapse_rows(columns = 1, valign = "top") %>% 
  column_spec(1, width = "30em") 

```

</br>

## Polymyalgia Rheumatica/Giant Cell Arteritis

---

```{r pr_create_venn_diagrams, echo=FALSE}

# possible and probable
pr_pp <- euler(pr_poss_prob[,2:3] > 0)

# possible: ICD, self and medication
pr_overlap <- euler(pr_icd_self_med[,2:4] > 0)

```

```{r pr_print_venn_diagrams, echo=FALSE, results = 'asis', fig.align='center', fig.height=5, fig.width=10}

prp1 <- plot(pr_pp, quantities = TRUE, fill = c("dodgerblue2", "firebrick2"))
prp2 <- plot(pr_overlap, quantities = TRUE, fill = c("lightgreen", "plum2", "lightyellow"))
grid.arrange(prp1, prp2, ncol = 2)

```

**Venn diagrams:** *Left:* probable cases as a subset of possible; *Right:* overlapping measures in the UK Biobank

</br>

**Matrix of ICD-10 primary and secondary dx - cells show number of individuals with each combination of dx**

```{r pr_print_icd_matrix, echo = FALSE, results = 'asis'}

pr_matrix <- matrix(ncol = 3, nrow = 3)
pr_matrix[1,1] <- paste0(pr_temp_matrix[1,1], footnote_marker_number(1))
pr_matrix[1,2] <- paste0(pr_temp_matrix[1,2], footnote_marker_number(2))
pr_matrix[1,3] <- paste0(pr_temp_matrix[1,3], footnote_marker_number(3))
pr_matrix[2,1] <- paste0(pr_temp_matrix[2,1], footnote_marker_number(2))
pr_matrix[2,2] <- paste0(pr_temp_matrix[2,2], footnote_marker_number(3))
pr_matrix[2,3] <- paste0(pr_temp_matrix[2,3], footnote_marker_number(3))
pr_matrix[3,1] <- paste0(pr_temp_matrix[3,1], footnote_marker_number(3))
pr_matrix[3,2] <- paste0(pr_temp_matrix[3,2], footnote_marker_number(3))
pr_matrix[3,3] <- paste0(pr_temp_matrix[3,3], footnote_marker_number(3))

colnames(pr_matrix) <- c("0 primary dx","1 primary dx",">1 primary dx")
rownames(pr_matrix) <- c("0 secondary dx","1 secondary dx",">1 secondary dx")

kable(pr_matrix, escape = F, align = 'r') %>%
  kable_styling(bootstrap_options = "striped") %>%
  column_spec(1, bold = TRUE, color = "white", background = "#666") %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>%
  footnote(general = " ",
           number = c("Controls", "Possible cases", "Probable cases"))

```

```{r pr_remove_zero_sums_from_icd_subcodes_table, echo = FALSE, include = FALSE}

pr_subcodes_final <- pr_subcodes[apply(pr_subcodes[,-1], 1, function(x) !all(x==0)),]
rownames(pr_subcodes_final) <- NULL

```

</br>

**ICD-10 codes used to construct matrix above. Columns show count of individuals who have received at least one dx** 

```{r pr_print_icd_subcodes_table, echo = FALSE, results = 'asis'}

kable(pr_subcodes_final, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  pack_rows("M31 Other necrotizing vasculopathies", 1, 1, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("M35 Other systemic involvement of connective tissue", 2, nrow(pr_subcodes_final), label_row_css = "background-color: #888; color: #fff;") %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Count of self-reported autoimmune condition**

```{r pr_print_self_report, echo=FALSE, results = 'asis'}

pr_table_self <- summary_self_report[13,] %>% 
  `rownames<-`(NULL)

kable(pr_table_self, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Count of self-reported prescription medications** 

```{r pr_print_meds, echo=FALSE, results = 'asis'}

kable(summary_meds_polymyalgia, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed"), ) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>% 
  collapse_rows(columns = 1, valign = "top") %>% 
  column_spec(1, width = "30em") 

```

</br>

## Psoriatic Arthritis

---

```{r psa_create_venn_diagrams, echo=FALSE}

# possible and probable
psa_pp <- euler(psa_poss_prob[,2:3] > 0)

# possible: ICD, self and medication
psa_overlap <- euler(psa_icd_self_med[,2:4] > 0)

```

```{r psa_print_venn_diagrams, echo=FALSE, results = 'asis', fig.align='center', fig.height=5, fig.width=10}

psap1 <- plot(psa_pp, quantities = TRUE, fill = c("dodgerblue2", "firebrick2"))
psap2 <- plot(psa_overlap, quantities = TRUE, fill = c("lightgreen", "plum2", "lightyellow"))
grid.arrange(psap1, psap2, ncol = 2)

```

**Venn diagrams:** *Left:* probable cases as a subset of possible; *Right:* overlapping measures in the UK Biobank

</br>

**Matrix of ICD-10 primary and secondary dx - cells show number of individuals with each combination of dx**

```{r psa_print_icd_matrix, echo = FALSE, results = 'asis'}

psa_matrix <- matrix(ncol = 3, nrow = 3)
psa_matrix[1,1] <- paste0(psa_temp_matrix[1,1], footnote_marker_number(1))
psa_matrix[1,2] <- paste0(psa_temp_matrix[1,2], footnote_marker_number(2))
psa_matrix[1,3] <- paste0(psa_temp_matrix[1,3], footnote_marker_number(3))
psa_matrix[2,1] <- paste0(psa_temp_matrix[2,1], footnote_marker_number(2))
psa_matrix[2,2] <- paste0(psa_temp_matrix[2,2], footnote_marker_number(3))
psa_matrix[2,3] <- paste0(psa_temp_matrix[2,3], footnote_marker_number(3))
psa_matrix[3,1] <- paste0(psa_temp_matrix[3,1], footnote_marker_number(3))
psa_matrix[3,2] <- paste0(psa_temp_matrix[3,2], footnote_marker_number(3))
psa_matrix[3,3] <- paste0(psa_temp_matrix[3,3], footnote_marker_number(3))

colnames(psa_matrix) <- c("0 primary dx","1 primary dx",">1 primary dx")
rownames(psa_matrix) <- c("0 secondary dx","1 secondary dx",">1 secondary dx")

kable(psa_matrix, escape = F, align = 'r') %>%
  kable_styling(bootstrap_options = "striped") %>%
  column_spec(1, bold = TRUE, color = "white", background = "#666") %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>%
  footnote(general = " ",
           number = c("Controls", "Possible cases", "Probable cases"))

```

```{r psa_remove_zero_sums_from_icd_subcodes_table, echo = FALSE, include = FALSE}

psa_subcodes_final <- psa_subcodes[apply(psa_subcodes[,-1], 1, function(x) !all(x==0)),]
rownames(psa_subcodes_final) <- NULL

```

</br>

**ICD-10 codes used to construct matrix above. Columns show count of individuals who have received at least one dx** 

```{r psa_print_icd_subcodes_table, echo = FALSE, results = 'asis'}

kable(psa_subcodes_final, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  pack_rows("L40 Psoriasis", 1, 1, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("M07 Psoriatic and enteropathic arthropathies", 2, nrow(psa_subcodes_final), label_row_css = "background-color: #888; color: #fff;") %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Count of self-reported autoimmune condition**

```{r psa_print_self_report, echo=FALSE, results = 'asis'}

psa_table_self <- summary_self_report[14,] %>% 
  `rownames<-`(NULL)

kable(psa_table_self, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Count of self-reported prescription medications** 

```{r psa_print_meds, echo=FALSE, results = 'asis'}

kable(summary_meds_psoriatic, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed"), ) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>% 
  collapse_rows(columns = 1, valign = "top") %>% 
  column_spec(1, width = "30em") 

```

</br>

## Rheumatoid Arthritis

---

```{r ra_create_venn_diagrams, echo=FALSE}

# possible and probable
ra_pp <- euler(ra_poss_prob[,2:3] > 0)

# possible: ICD, self and medication
ra_overlap <- euler(ra_icd_self_med[,2:4] > 0)

```

```{r ra_print_venn_diagrams, echo=FALSE, results = 'asis', fig.align='center', fig.height=5, fig.width=10}

rap1 <- plot(ra_pp, quantities = TRUE, fill = c("dodgerblue2", "firebrick2"))
rap2 <- plot(ra_overlap, quantities = TRUE, fill = c("lightgreen", "plum2", "lightyellow"))
grid.arrange(rap1, rap2, ncol = 2)

```

**Venn diagrams:** *Left:* probable cases as a subset of possible; *Right:* overlapping measures in the UK Biobank

</br>

**Matrix of ICD-10 primary and secondary dx - cells show number of individuals with each combination of dx**

```{r ra_print_icd_matrix, echo = FALSE, results = 'asis'}

ra_matrix <- matrix(ncol = 3, nrow = 3)
ra_matrix[1,1] <- paste0(ra_temp_matrix[1,1], footnote_marker_number(1))
ra_matrix[1,2] <- paste0(ra_temp_matrix[1,2], footnote_marker_number(2))
ra_matrix[1,3] <- paste0(ra_temp_matrix[1,3], footnote_marker_number(3))
ra_matrix[2,1] <- paste0(ra_temp_matrix[2,1], footnote_marker_number(2))
ra_matrix[2,2] <- paste0(ra_temp_matrix[2,2], footnote_marker_number(3))
ra_matrix[2,3] <- paste0(ra_temp_matrix[2,3], footnote_marker_number(3))
ra_matrix[3,1] <- paste0(ra_temp_matrix[3,1], footnote_marker_number(3))
ra_matrix[3,2] <- paste0(ra_temp_matrix[3,2], footnote_marker_number(3))
ra_matrix[3,3] <- paste0(ra_temp_matrix[3,3], footnote_marker_number(3))

colnames(ra_matrix) <- c("0 primary dx","1 primary dx",">1 primary dx")
rownames(ra_matrix) <- c("0 secondary dx","1 secondary dx",">1 secondary dx")

kable(ra_matrix, escape = F, align = 'r') %>%
  kable_styling(bootstrap_options = "striped") %>%
  column_spec(1, bold = TRUE, color = "white", background = "#666") %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>%
  footnote(general = " ",
           number = c("Controls", "Possible cases", "Probable cases"))

```

```{r ra_remove_zero_sums_from_icd_subcodes_table, echo = FALSE, include = FALSE}

ra_subcodes_final <- ra_subcodes[apply(ra_subcodes[,-1], 1, function(x) !all(x==0)),]
rownames(ra_subcodes_final) <- NULL

```

</br>

**ICD-10 codes used to construct matrix above. Columns show count of individuals who have received at least one dx** 

```{r ra_print_icd_subcodes_table, echo = FALSE, results = 'asis'}

kable(ra_subcodes_final, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  pack_rows("M05 Seropositive rheumatoid arthritis", 1, 34, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("M06 Other rheumatoid arthritis", 35, nrow(ra_subcodes_final), label_row_css = "background-color: #888; color: #fff;") %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Count of self-reported autoimmune condition**

```{r ra_print_self_report, echo=FALSE, results = 'asis'}

ra_table_self <- summary_self_report[15,] %>% 
  `rownames<-`(NULL)

kable(ra_table_self, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Count of self-reported prescription medications** 

```{r ra_print_meds, echo=FALSE, results = 'asis'}

kable(summary_meds_arthritis, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed"), ) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>% 
  collapse_rows(columns = 1, valign = "top") %>% 
  column_spec(1, width = "30em") 

```

</br>

## Sjögren Syndrome

---

```{r sjogrens_create_venn_diagrams, echo=FALSE}

# possible and probable
sjogrens_pp <- euler(sjogrens_poss_prob[,2:3] > 0)

# possible: ICD, self and medication
sjogrens_overlap <- euler(sjogrens_icd_self_med[,2:4] > 0)

```

```{r sjogrens_print_venn_diagrams, echo=FALSE, results = 'asis', fig.align='center', fig.height=5, fig.width=10}

sjogrensp1 <- plot(sjogrens_pp, quantities = TRUE, fill = c("dodgerblue2", "firebrick2"))
sjogrensp2 <- plot(sjogrens_overlap, quantities = TRUE, fill = c("lightgreen", "plum2", "lightyellow"))
grid.arrange(sjogrensp1, sjogrensp2, ncol = 2)

```

**Venn diagrams:** *Left:* probable cases as a subset of possible; *Right:* overlapping measures in the UK Biobank

</br>

**Matrix of ICD-10 primary and secondary dx - cells show number of individuals with each combination of dx**

```{r sjogrens_print_icd_matrix, echo = FALSE, results = 'asis'}

sjogrens_matrix <- matrix(ncol = 3, nrow = 3)
sjogrens_matrix[1,1] <- paste0(sjogrens_temp_matrix[1,1], footnote_marker_number(1))
sjogrens_matrix[1,2] <- paste0(sjogrens_temp_matrix[1,2], footnote_marker_number(2))
sjogrens_matrix[1,3] <- paste0(sjogrens_temp_matrix[1,3], footnote_marker_number(3))
sjogrens_matrix[2,1] <- paste0(sjogrens_temp_matrix[2,1], footnote_marker_number(2))
sjogrens_matrix[2,2] <- paste0(sjogrens_temp_matrix[2,2], footnote_marker_number(3))
sjogrens_matrix[2,3] <- paste0(sjogrens_temp_matrix[2,3], footnote_marker_number(3))
sjogrens_matrix[3,1] <- paste0(sjogrens_temp_matrix[3,1], footnote_marker_number(3))
sjogrens_matrix[3,2] <- paste0(sjogrens_temp_matrix[3,2], footnote_marker_number(3))
sjogrens_matrix[3,3] <- paste0(sjogrens_temp_matrix[3,3], footnote_marker_number(3))

colnames(sjogrens_matrix) <- c("0 primary dx","1 primary dx",">1 primary dx")
rownames(sjogrens_matrix) <- c("0 secondary dx","1 secondary dx",">1 secondary dx")

kable(sjogrens_matrix, escape = F, align = 'r') %>%
  kable_styling(bootstrap_options = "striped") %>%
  column_spec(1, bold = TRUE, color = "white", background = "#666") %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>%
  footnote(general = " ",
           number = c("Controls", "Possible cases", "Probable cases"))

```

```{r sjogrens_remove_zero_sums_from_icd_subcodes_table, echo = FALSE, include = FALSE}

sjogrens_subcodes_final <- sjogrens_subcodes[apply(sjogrens_subcodes[,-1], 1, function(x) !all(x==0)),]
rownames(sjogrens_subcodes_final) <- NULL

```

</br>

**ICD-10 codes used to construct matrix above. Columns show count of individuals who have received at least one dx** 

```{r sjogrens_print_icd_subcodes_table, echo = FALSE, results = 'asis'}

kable(sjogrens_subcodes_final, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  pack_rows("M35 Other systemic involvement of connective tissue", 1, 1, label_row_css = "background-color: #888; color: #fff;") %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Count of self-reported autoimmune condition**

```{r sjogrens_print_self_report, echo=FALSE, results = 'asis'}

sjogrens_table_self <- summary_self_report[16,] %>% 
  `rownames<-`(NULL)

kable(sjogrens_table_self, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Count of self-reported prescription medications** 

```{r sjogrens_print_meds, echo=FALSE, results = 'asis'}

kable(summary_meds_sjogren, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed"), ) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>% 
  collapse_rows(columns = 1, valign = "top") %>% 
  column_spec(1, width = "30em") 
```

</br>

## Systemic Lupus Erythematosus

---

```{r sle_create_venn_diagrams, echo=FALSE}

# possible and probable
sle_pp <- euler(sle_poss_prob[,2:3] > 0)

# possible: ICD, self and medication
sle_overlap <- euler(sle_icd_self_med[,2:4] > 0)

```

```{r sle_print_venn_diagrams, echo=FALSE, results = 'asis', fig.align='center', fig.height=5, fig.width=10}

slep1 <- plot(sle_pp, quantities = TRUE, fill = c("dodgerblue2", "firebrick2"))
slep2 <- plot(sle_overlap, quantities = TRUE, fill = c("lightgreen", "plum2", "lightyellow"))
grid.arrange(slep1, slep2, ncol = 2)

```

**Venn diagrams:** *Left:* probable cases as a subset of possible; *Right:* overlapping measures in the UK Biobank

</br>

**Matrix of ICD-10 primary and secondary dx - cells show number of individuals with each combination of dx**

```{r sle_print_icd_matrix, echo = FALSE, results = 'asis'}

sle_matrix <- matrix(ncol = 3, nrow = 3)
sle_matrix[1,1] <- paste0(sle_temp_matrix[1,1], footnote_marker_number(1))
sle_matrix[1,2] <- paste0(sle_temp_matrix[1,2], footnote_marker_number(2))
sle_matrix[1,3] <- paste0(sle_temp_matrix[1,3], footnote_marker_number(3))
sle_matrix[2,1] <- paste0(sle_temp_matrix[2,1], footnote_marker_number(2))
sle_matrix[2,2] <- paste0(sle_temp_matrix[2,2], footnote_marker_number(3))
sle_matrix[2,3] <- paste0(sle_temp_matrix[2,3], footnote_marker_number(3))
sle_matrix[3,1] <- paste0(sle_temp_matrix[3,1], footnote_marker_number(3))
sle_matrix[3,2] <- paste0(sle_temp_matrix[3,2], footnote_marker_number(3))
sle_matrix[3,3] <- paste0(sle_temp_matrix[3,3], footnote_marker_number(3))

colnames(sle_matrix) <- c("0 primary dx","1 primary dx",">1 primary dx")
rownames(sle_matrix) <- c("0 secondary dx","1 secondary dx",">1 secondary dx")

kable(sle_matrix, escape = F, align = 'r') %>%
  kable_styling(bootstrap_options = "striped") %>%
  column_spec(1, bold = TRUE, color = "white", background = "#666") %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>%
  footnote(general = " ",
           number = c("Controls", "Possible cases", "Probable cases"))

```

```{r sle_remove_zero_sums_from_icd_subcodes_table, echo = FALSE, include = FALSE}

sle_subcodes_final <- sle_subcodes[apply(sle_subcodes[,-1], 1, function(x) !all(x==0)),]
rownames(sle_subcodes_final) <- NULL

```

</br>

**ICD-10 codes used to construct matrix above. Columns show count of individuals who have received at least one dx** 

```{r sle_print_icd_subcodes_table, echo = FALSE, results = 'asis'}

kable(sle_subcodes_final, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  pack_rows("L93 Lupus erythematosus", 1, 3, label_row_css = "background-color: #888; color: #fff;") %>% 
  pack_rows("M32 Systemic lupus erythematosus", 4, nrow(sle_subcodes_final), label_row_css = "background-color: #888; color: #fff;") %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Count of self-reported autoimmune condition**

```{r sle_print_self_report, echo=FALSE, results = 'asis'}

sle_table_self <- summary_self_report[17,] %>% 
  `rownames<-`(NULL)

kable(sle_table_self, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Count of self-reported prescription medications** 

```{r sle_print_meds, echo=FALSE, results = 'asis'}

kable(summary_meds_lupus, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed"), ) %>% 
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>% 
  collapse_rows(columns = 1, valign = "top") %>% 
  column_spec(1, width = "30em") 

```

</br>

# Autoimmune polygenic score analysis

---

</br>

```{bash prs_scripts, echo=FALSE, eval=FALSE}

## PRS Coeliac 
Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/CELI01.gz \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/new_autoimmune_all \
--pheno-col coeliac_poss,coeliac_prob \
--binary-target T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.01,0.01 \
--quantile 10 \
--perm 10000 \
--out /file_path/coeliac_prs

## PRS IBD 
Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/INFB01.gz \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/new_autoimmune_all \
--pheno-col ibd_poss,ibd_prob \
--binary-target T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.005,0.005 \
--quantile 10 \
--perm 10000 \
--out /file_path/ibd_prs


## PRS Myasthenia Gravis
module load apps/R/3.6.0

Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/myasthenia_gravis_sum_stats_New_AllResults.2_NoSex.txt \
--A1 ALLELE1 \
--A2 ALLELE2 \
--pvalue PVALUE \
--snp MARKER \
--stat OR \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/new_autoimmune_all \
--pheno-col myasthenia_poss,myasthenia_prob \
--binary-target T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.0002,0.0002 \
--quantile 10 \
--perm 10000 \
--out /file_path/myasthenia_prs


## PRS Multiple Sclerosis
Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/SCLE02.gz \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/new_autoimmune_all \
--pheno-col ms_poss,ms_prob \
--binary-target T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.001,0.001 \
--quantile 10 \
--perm 10000 \
--out /file_path/ms_prs


## PRS Rheumatoid Arthritis 
Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/RHEU02.gz \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/new_autoimmune_all \
--pheno-col arthritis_poss,arthritis_prob \
--binary-target T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.01,0.01 \
--quantile 10 \
--perm 10000 \
--out /file_path/ra_prs


## PRS Lupus 
Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/LUPU01.gz \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/new_autoimmune_all \
--pheno-col sle_poss,sle_prob \
--binary-target T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.001,0.001 \
--quantile 10 \
--perm 10000 \
--out /file_path/lupus_prs

## PRS Psoriasis 
Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/metaanalysis_inversevariance_7datasets1_RS_ID.tbl \
--A1 Allele1 \
--A2 Allele2 \
--pvalue P.value \
--snp SNP \
--stat Effect \
--beta \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/new_autoimmune_all \
--pheno-col psoriasis_poss,psoriasis_prob \
--binary-target T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.02,0.02 \
--quantile 10 \
--perm 10000 \
--out /file_path/psoriasis_prs


## PRS PsA 
Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/psa_sumstats.txt \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/new_autoimmune_all \
--pheno-col psa_poss,psa_prob \
--binary-target T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.005,0.005 \
--quantile 10 \
--perm 10000 \
--out /file_path/psa_prs

```

```{r prepare_prs_poss_prob_df_for_plot, include=FALSE}

# Read in data from PRS analyses for plotting

ms_best <- as.data.frame(fread("/file_path/ms_prs.summary", header = T, sep = "\t"))
ms_best$Disorder <- c("Multiple\nSclerosis","Multiple\nSclerosis")
ms_best$Base <- c("Multiple Sclerosis","Multiple Sclerosis")
ms_best$Phenotype <- c("Multiple Sclerosis","Multiple Sclerosis")
ms_best$Type <- c("Possible","Probable")

coeliac_best <- as.data.frame(fread("/file_path/coeliac_prs.summary", header = T, sep = "\t"))
coeliac_best$Disorder <- c("Coeliac","Coeliac")
coeliac_best$Base <- c("Coeliac","Coeliac")
coeliac_best$Phenotype <- c("Coeliac","Coeliac")
coeliac_best$Type <- c("Possible","Probable")

ibd_best <- as.data.frame(fread("/file_path/ibd_prs.summary", header = T, sep = "\t"))
ibd_best$Disorder <- c("Inflammatory\nBowel Disease","Inflammatory\nBowel Disease")
ibd_best$Base <- c("Inflammatory Bowel Disease","Inflammatory Bowel Disease")
ibd_best$Phenotype <- c("Inflammatory Bowel Disease","Inflammatory Bowel Disease")
ibd_best$Type <- c("Possible","Probable")

psa_best <- as.data.frame(fread("/file_path/psa_prs.summary", header = T, sep = "\t"))
psa_best$Disorder <- c("Psoriatic\nArthritis","Psoriatic\nArthritis")
psa_best$Base <- c("Psoriatic Arthritis","Psoriatic Arthritis")
psa_best$Phenotype <- c("Psoriatic Arthritis","Psoriatic Arthritis")
psa_best$Type <- c("Possible","Probable")

ra_best <- as.data.frame(fread("/file_path/ra_prs.summary", header = T, sep = "\t"))
ra_best$Disorder <- c("Rheumatoid\nArthritis","Rheumatoid\nArthritis")
ra_best$Base <- c("Rheumatoid Arthritis","Rheumatoid Arthritis")
ra_best$Phenotype <- c("Rheumatoid Arthritis","Rheumatoid Arthritis")
ra_best$Type <- c("Possible","Probable")

sle_best <- as.data.frame(fread("/file_path/lupus_prs.summary", header = T, sep = "\t"))
sle_best$Disorder <- c("Systemic Lupus\nErythematosus","Systemic Lupus\nErythematosus")
sle_best$Base <- c("Systemic Lupus Erythematosus","Systemic Lupus Erythematosus")
sle_best$Phenotype <- c("Systemic Lupus Erythematosus","Systemic Lupus Erythematosus")
sle_best$Type <- c("Possible","Probable")

mg_best <- as.data.frame(fread("/file_path/myasthenia_prs.summary", header = T, sep = "\t"))
mg_best$Disorder <- c("Myasthenia\nGravis","Myasthenia\nGravis")
mg_best$Base <- c("Myasthenia Gravis","Myasthenia Gravis")
mg_best$Phenotype <- c("Myasthenia Gravis","Myasthenia Gravis")
mg_best$Type <- c("Possible","Probable")

psoriasis_best <- as.data.frame(fread("/file_path/psoriasis_prs.summary", header = T, sep = "\t"))
psoriasis_best$Disorder <- c("Psoriasis","Psoriasis")
psoriasis_best$Base <- c("Psoriasis","Psoriasis")
psoriasis_best$Phenotype <- c("Psoriasis","Psoriasis")
psoriasis_best$Type <- c("Possible","Probable")

all <- rbind(ms_best, coeliac_best, ibd_best, psa_best, ra_best, sle_best, mg_best, psoriasis_best)
colnames(all)[13] <- "Empirical"
all$Empirical <- formatC(all$Empirical,format="e",digits = 0)
all$P <- formatC(all$P,format="e",digits = 0)
all$print.p <- sub("e", "*x*10^", all$P)
all$print.empirical <- sub("e", "*x*10^", all$Empirical)

```

```{r prs_poss_prob_plot, include=FALSE}

autoimmune_all <- fread("/file_path/new_autoimmune_all", header = TRUE)

## Plot

Figure <- ggplot(data = all, aes(x = as.factor(Disorder), y = PRS.R2.adj, fill = as.factor(Type))) +
  geom_bar(stat = "identity", position = position_dodge(), width = 0.9) +
  scale_fill_manual(values = c("dodgerblue2", "firebrick2")) +
  geom_text(aes(label=paste(print.p)), position=position_dodge(width=0.9), 
                             vjust=-1, hjust=0.5, size = 3, angle = 0, parse = T) +
  ylab(expression(paste("R"^"2"," ","liability"))) +
  xlab("\nAutoimmune Disease") +
  theme(axis.text=element_text(size=12),
        axis.title=element_text(size=14)) +
  theme(panel.grid.minor.x=element_blank(), 
                         panel.grid.major.x=element_blank(), 
                         panel.grid.minor.y=element_blank(), 
                         panel.grid.major.y=element_blank(),
                         panel.background = element_rect(fill = "white"),
                         axis.line.y = element_line(color="grey"),
                         axis.line.x = element_line(color="grey"),
                         legend.title = element_text(size = 12),
                         legend.text = element_text(size = 12),
                        plot.title = element_text(hjust = 0.5, margin = margin(b=25), size = 14)) +
  scale_y_continuous(limits=c(0,0.06), expand=c(0,0)) +
  annotate("text", x = 1:8, y = Inf,  vjust=2, fontface=4, colour = "dodgerblue2", size=4,
                            label = c(paste(formatC(sum(autoimmune_all$coeliac_poss==1, na.rm = T), big.mark = ",")),
                                      paste(formatC(sum(autoimmune_all$ibd_poss==1, na.rm = T), big.mark = ",")),
                                      paste(formatC(sum(autoimmune_all$ms_poss==1, na.rm = T), big.mark = ",")),
                                      paste(formatC(sum(autoimmune_all$myasthenia_poss==1, na.rm = T), big.mark = ",")),
                                      paste(formatC(sum(autoimmune_all$psoriasis_poss==1, na.rm = T), big.mark = ",")),
                                      paste(formatC(sum(autoimmune_all$psa_poss==1, na.rm = T), big.mark = ",")),
                                      paste(formatC(sum(autoimmune_all$arthritis_poss==1, na.rm = T), big.mark = ",")),
                                      paste(formatC(sum(autoimmune_all$sle_poss==1, na.rm = T), big.mark = ",")))) +
  annotate("text", x = 1:8, y = Inf,  vjust=4, fontface=4, colour = "firebrick2", size=4,
                            label = c(paste(formatC(sum(autoimmune_all$coeliac_prob==1, na.rm = T), big.mark = ",")),
                                      paste(formatC(sum(autoimmune_all$ibd_prob==1, na.rm = T), big.mark = ",")),
                                      paste(formatC(sum(autoimmune_all$ms_prob==1, na.rm = T), big.mark = ",")),
                                      paste(formatC(sum(autoimmune_all$myasthenia_prob==1, na.rm = T), big.mark = ",")),
                                      paste(formatC(sum(autoimmune_all$psoriasis_prob==1, na.rm = T), big.mark = ",")),
                                      paste(formatC(sum(autoimmune_all$psa_prob==1, na.rm = T), big.mark = ",")),
                                      paste(formatC(sum(autoimmune_all$arthritis_prob==1, na.rm = T), big.mark = ",")),
                                      paste(formatC(sum(autoimmune_all$sle_prob==1, na.rm = T), big.mark = ",")))) + 
  labs(fill = "Autoimmune\nPhenotype\n(target trait)")

```

```{r print_plot_prs, echo=FALSE, results = 'asis', fig.align='center', fig.height=5, fig.width=12}

Figure

```

**Figure:**  Polygenic scores predicting case/control status in the UK Biobank for eight autoimmune diseases. P-values are shown atop each column and the number of cases in the UKB are shown at the top of the plot (possible = blue, probable = red).

</br>

**Table:** PRS for autoimmune diseases predicting autoimmune case/control status

```{r print_prs_table, echo=FALSE, results = 'asis'}

# Print full table of results

all$Target <- paste0(all$Phenotype," (",all$Type,")")
all$Standard.Error <- round(all$Standard.Error, digits = 3)
table <- all[,c(15,19,3,12,8,4,5,9,10,11)]
table$Prevalence <- percent(table$Prevalence)
table$PRS.R2.adj <- percent(table$PRS.R2.adj)
table$PRS.R2 <- percent(table$PRS.R2)
table$Coefficient <- round(table$Coefficient, digits = 2)
table <- table[order(table$Base),]
rownames(table) <- NULL

kable(table, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666", align = "centre")

```

</br>

## PRS at all thresholds

</br>

```{r prepare_plot_all_thresholds, echo=FALSE, results = 'asis', fig.align='center', fig.height=6, fig.width=18}

# Prepare plots for PRS results at all p-value thresholds

# Read and format data
celiac <- read.table("/file_path/coeliac_prs.prsice", header = T) 
celiac$Set <- ifelse(grepl("poss", celiac$Pheno), "Coeliac Disease (Possible)", "Coeliac Disease (Probable)")

ibd <- read.table("/file_path/ibd_prs.prsice", header = T) 
ibd$Set <- ifelse(grepl("poss", ibd$Pheno), "Inflammatory Bowel Disease (Possible)", "Inflammatory Bowel Disease (Probable)")

ms <- read.table("/file_path/ms_prs.prsice", header = T) 
ms$Set <- ifelse(grepl("poss", ms$Pheno), "Multiple Sclerosis (Possible)", "Multiple Sclerosis (Probable)")

mg <- read.table("/file_path/myasthenia_prs.prsice", header = T) 
mg$Set <- ifelse(grepl("poss", mg$Pheno), "Myasthenia Gravis (Possible)", "Myasthenia Gravis (Probable)")

ps <- read.table("/file_path/psoriasis_prs.prsice", header = T) 
ps$Set <- ifelse(grepl("poss", ps$Pheno), "Psoriasis (Possible)", "Psoriasis (Probable)")

psa <- read.table("/file_path/psa_prs.prsice", header = T) 
psa$Set <- ifelse(grepl("poss", psa$Pheno), "Psoriatic Arthritis (Possible)", "Psoriatic Arthritis (Probable)")

ra <- read.table("/file_path/ra_prs.prsice", header = T) 
ra$Set <- ifelse(grepl("poss", ra$Pheno), "Rheumatoid Arthritis (Possible)", "Rheumatoid Arthritis (Probable)")

sle <- read.table("/file_path/lupus_prs.prsice", header = T) 
sle$Set <- ifelse(grepl("poss", sle$Pheno), "Systemic Lupus Erythematosus (Possible)", "Systemic Lupus Erythematosus (Probable)")

# Make list of dfs
thresholds <- list(celiac, ibd, ms, mg, ps, psa, ra, sle)

# format p-value column for plot
for (i in 1:(length(thresholds))) {
  thresholds[[i]]$print.p <- formatC(thresholds[[i]]$P, digits = 2)
}
for (i in 1:(length(thresholds))) {
  thresholds[[i]]$print.p <- sub("e", "*x*10^", thresholds[[i]]$print.p)
}

# Create plot function
threshold_plot <- function(x) {
  ggplot(x, aes(x = as.factor(Threshold), y = R2, fill = -log10(P))) +
  geom_bar(stat = "identity", position = position_dodge()) +
  scale_fill_gradient2(low = "dodgerblue", high = "firebrick", mid = "dodgerblue", midpoint=1e-4,
                       name = bquote(atop(-log[10] ~ model, italic(P) - value),)) +
  ylab(expression(paste("PRS model fit:  ", R ^ 2, sep = " "))) +
  xlab(expression(italic(P) - value ~ threshold ~ (italic(P)[T]))) +
  theme(axis.text.x = element_text(angle=45, hjust=1, size = 12),
        axis.text.y = element_text(size = 12),
        axis.title = element_text(size = 14),
        plot.title = element_text(hjust = 0.5, size = 18),
        strip.text.x = element_text(size = 12, face = "bold"),
        panel.grid = element_blank()) +
  scale_y_continuous(expand = expand_scale(mult = c(0, .3))) +
  geom_text(aes(label = paste(print.p)), vjust = -1.5, hjust = 0, angle = 45, cex = 3.5, parse = T) +
  facet_grid(~ Set)
}  

celiac <- as.data.frame(thresholds[1])
ibd <- as.data.frame(thresholds[2])
ms <- as.data.frame(thresholds[3])
mg <- as.data.frame(thresholds[4])
ps <- as.data.frame(thresholds[5])
psa <- as.data.frame(thresholds[6])
ra <- as.data.frame(thresholds[7])
sle <- as.data.frame(thresholds[8])

celiac_plot <- threshold_plot(celiac)
ibd_plot <- threshold_plot(ibd)
ms_plot <- threshold_plot(ms)
mg_plot <- threshold_plot(mg)
ps_plot <- threshold_plot(ps)
psa_plot <- threshold_plot(psa)
ra_plot <- threshold_plot(ra)
sle_plot <- threshold_plot(sle)

```

```{r print_thresholds1, echo=FALSE, results = 'asis', fig.align='center', fig.height=20, fig.width=15}

grid.arrange(celiac_plot, ibd_plot, ms_plot, mg_plot, ncol = 1)

```

```{r print_thresholds2, echo=FALSE, results = 'asis', fig.align='center', fig.height=20, fig.width=15}

grid.arrange(ps_plot, psa_plot, ra_plot, sle_plot, ncol = 1)

```

</br>

# Depression phenotyping

</br>

## Demographic information: any and stringent

</br>

```{r depression_demographics, echo=FALSE, message = FALSE, results = 'asis'}

# Read in depression phenotype data (depression phenotypes defined in "Multiple measures of depression to enhance validity of major depressive disorder in the UK Biobank", DOI: https://doi.org/10.1192/bjo.2020.145)

depression <- read.table("/file_path/dep_broad_narrow", header = T) 

# Make df for any, stringent and controls; merge with demographic data
any_dep <- depression %>%
  select(ID, Broad) %>%
  filter(Broad==1) %>%
  na.omit() %>%
  left_join(., demographics1)

stringent_dep <- depression %>%
  select(ID, Narrow) %>%
  filter(Narrow==1) %>%
  na.omit() %>%
  left_join(., demographics1)

controls_dep <- depression %>%
  select(ID, Narrow) %>%
  filter(Narrow==0) %>%
  rename(Controls = Narrow) %>%
  na.omit() %>%
  left_join(., demographics1)

# Summarise demographics
summarise_dep <- function(x) {
  data.frame(
  Count = format(nrow(x), big.mark = ","), 
  Prevalence = paste(formattable::percent(round(nrow(x)/(nrow(x)+nrow(controls_dep)), digits = 4))), 
  "Age (SD)" = paste(format(mean(x$Age, na.rm = T), digits = 3, nsmall = 1), paste("(", format(sd(x$Age, na.rm = T), digits = 3, nsmall = 2), ")", sep = ""), sep = " "),  
  "Female(%)" = percent(round(sum(x$Sex==0, na.rm = T)/nrow(x), digits = 2), digits = 0),
  "SES (SD)" = paste(format(mean(x$SES, na.rm = T), digits = 2, nsmall = 2), paste("(", format(sd(x$SES, na.rm = T), digits = 2, nsmall = 2), ")", sep = ""), sep = " "),
  "Current smoker (%)" = percent(round(sum(x$`Current Smoker`==1, na.rm = T)/nrow(x), digits = 2), digits = 0),
  "BMI (SD)" = paste(format(mean(x$BMI, na.rm = T), digits = 3, nsmall = 1), paste("(", format(sd(x$BMI, na.rm = T), digits = 3, nsmall = 2), ")", sep = ""), sep = " ")
)
}

any_dep_desc <- summarise_dep(any_dep)
stringent_dep_desc <- summarise_dep(stringent_dep)
controls_dep_desc <- summarise_dep(controls_dep)

# combine data
depression_summary <- rbind.data.frame(controls_dep_desc, any_dep_desc, stringent_dep_desc, stringsAsFactors = FALSE)

depression_summary[1,2] <- NA
colnames(depression_summary) <- c("Count","UKB.Prev.","Mean age (SD)","Female","SES* (SD)","Smoker","Mean BMI (SD)")
rownames(depression_summary) <- c("Controls","Any Depression","Stringent Depression")

### Join with population prevalences

depression_summary$Pop.Prev. <- c("NA","15%","15%")

depression_summary <- depression_summary[, c(1:2,8,3:7)]

# Print table with summary of demographic data for any, stringent and controls
kable(depression_summary, escape = F, align = 'r') %>%
  kable_styling(c("striped", "condensed"), full_width = T, font_size = 13) %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

## Case-control differences in sociodemographic characteristics

</br>

**Table:** Any depression cases (N = 65,075) vs controls (N = 232,552)

```{r ttest_demographics_ss_vs_anydep, echo=FALSE, results = 'asis'}

# Compare demographic variables between controls and any depression

# Semi-screened vs Any Depression Phenotype
age_ss_vs_anydep <- t.test(controls_dep$Age, any_dep$Age) 
table_age_ss_vs_anydep <- data.frame(
  "Demographic" = "Mean age",
  "Estimate (controls)" = as.data.frame(age_ss_vs_anydep$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(age_ss_vs_anydep$estimate)[2,1],
  "t-statistic" = as.data.frame(age_ss_vs_anydep$statistic)[1,1],
  "df" = as.data.frame(age_ss_vs_anydep$parameter)[1,1],
  "p-value" = formatC(age_ss_vs_anydep$p.value, format = "e", digits = 0)
)

sex_ss_vs_anydep <- t.test(controls_dep$Sex, any_dep$Sex) 
table_sex_ss_vs_anydep <- data.frame(
  "Demographic" = "Female",
  "Estimate (controls)" = 1-(as.data.frame(sex_ss_vs_anydep$estimate)[1,1]),
  "Estimate (cases)" = 1-(as.data.frame(sex_ss_vs_anydep$estimate)[2,1]),
  "t-statistic" = as.data.frame(sex_ss_vs_anydep$statistic)[1,1],
  "df" = as.data.frame(sex_ss_vs_anydep$parameter)[1,1],
  "p-value" = formatC(sex_ss_vs_anydep$p.value, format = "e", digits = 0)
)

ses_ss_vs_anydep <- t.test(controls_dep$SES, any_dep$SES) 
table_ses_ss_vs_anydep <- data.frame(
  "Demographic" = "SES",
  "Estimate (controls)" = as.data.frame(ses_ss_vs_anydep$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(ses_ss_vs_anydep$estimate)[2,1],
  "t-statistic" = as.data.frame(ses_ss_vs_anydep$statistic)[1,1],
  "df" = as.data.frame(ses_ss_vs_anydep$parameter)[1,1],
  "p-value" = formatC(ses_ss_vs_anydep$p.value, format = "e", digits = 0)
)

smoke_ss_vs_anydep <- t.test(controls_dep$`Current Smoker`, any_dep$`Current Smoker`) 
table_smoke_ss_vs_anydep <- data.frame(
  "Demographic" = "Current smoker",
  "Estimate (controls)" = as.data.frame(smoke_ss_vs_anydep$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(smoke_ss_vs_anydep$estimate)[2,1],
  "t-statistic" = as.data.frame(smoke_ss_vs_anydep$statistic)[1,1],
  "df" = as.data.frame(smoke_ss_vs_anydep$parameter)[1,1],
  "p-value" = formatC(smoke_ss_vs_anydep$p.value, format = "e", digits = 0)
)

bmi_ss_vs_anydep <- t.test(controls_dep$BMI, any_dep$BMI) 
table_bmi_ss_vs_anydep <- data.frame(
  "Demographic" = "Mean BMI",
  "Estimate (controls)" = as.data.frame(bmi_ss_vs_anydep$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(bmi_ss_vs_anydep$estimate)[2,1],
  "t-statistic" = as.data.frame(bmi_ss_vs_anydep$statistic)[1,1],
  "df" = as.data.frame(bmi_ss_vs_anydep$parameter)[1,1],
  "p-value" = formatC(bmi_ss_vs_anydep$p.value, format = "e", digits = 0)
)

ss_vs_anydep <- rbind(table_age_ss_vs_anydep,
                  table_sex_ss_vs_anydep,
                  table_ses_ss_vs_anydep,
                  table_smoke_ss_vs_anydep,
                  table_bmi_ss_vs_anydep)

colnames(ss_vs_anydep) <- c("Demographic","Estimate (controls)","Estimate (cases)","t-statistic","df","p-value")

ss_vs_anydep$`Estimate (controls)` <- round(ss_vs_anydep$`Estimate (controls)`, digits = 2)
ss_vs_anydep$`Estimate (cases)` <- round(ss_vs_anydep$`Estimate (cases)`, digits = 2)
ss_vs_anydep$`t-statistic` <- round(ss_vs_anydep$`t-statistic`, digits = 0)

kable(ss_vs_anydep, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

**Table:** Stringent depression cases (N = 14,625) vs controls (N = 232,552)

```{r ttest_demographics_ss_vs_stringentdep, echo=FALSE, results = 'asis'}

# Compare demographic variables between controls and stringent depression

# Semi-screened vs Stringent Depression Phenotype
age_ss_vs_stringentdep <- t.test(controls_dep$Age, stringent_dep$Age) 
table_age_ss_vs_stringentdep <- data.frame(
  "Demographic" = "Mean age",
  "Estimate (controls)" = as.data.frame(age_ss_vs_stringentdep$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(age_ss_vs_stringentdep$estimate)[2,1],
  "t-statistic" = as.data.frame(age_ss_vs_stringentdep$statistic)[1,1],
  "df" = as.data.frame(age_ss_vs_stringentdep$parameter)[1,1],
  "p-value" = formatC(age_ss_vs_stringentdep$p.value, format = "e", digits = 0)
)

sex_ss_vs_stringentdep <- t.test(controls_dep$Sex, stringent_dep$Sex) 
table_sex_ss_vs_stringentdep <- data.frame(
  "Demographic" = "Female",
  "Estimate (controls)" = 1-(as.data.frame(sex_ss_vs_stringentdep$estimate)[1,1]),
  "Estimate (cases)" = 1-(as.data.frame(sex_ss_vs_stringentdep$estimate)[2,1]),
  "t-statistic" = as.data.frame(sex_ss_vs_stringentdep$statistic)[1,1],
  "df" = as.data.frame(sex_ss_vs_stringentdep$parameter)[1,1],
  "p-value" = formatC(sex_ss_vs_stringentdep$p.value, format = "e", digits = 0)
)

ses_ss_vs_stringentdep <- t.test(controls_dep$SES, stringent_dep$SES) 
table_ses_ss_vs_stringentdep <- data.frame(
  "Demographic" = "SES",
  "Estimate (controls)" = as.data.frame(ses_ss_vs_stringentdep$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(ses_ss_vs_stringentdep$estimate)[2,1],
  "t-statistic" = as.data.frame(ses_ss_vs_stringentdep$statistic)[1,1],
  "df" = as.data.frame(ses_ss_vs_stringentdep$parameter)[1,1],
  "p-value" = formatC(ses_ss_vs_stringentdep$p.value, format = "e", digits = 0)
)

smoke_ss_vs_stringentdep <- t.test(controls_dep$`Current Smoker`, stringent_dep$`Current Smoker`) 
table_smoke_ss_vs_stringentdep <- data.frame(
  "Demographic" = "Current smoker",
  "Estimate (controls)" = as.data.frame(smoke_ss_vs_stringentdep$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(smoke_ss_vs_stringentdep$estimate)[2,1],
  "t-statistic" = as.data.frame(smoke_ss_vs_stringentdep$statistic)[1,1],
  "df" = as.data.frame(smoke_ss_vs_stringentdep$parameter)[1,1],
  "p-value" = formatC(smoke_ss_vs_stringentdep$p.value, format = "e", digits = 0)
)

bmi_ss_vs_stringentdep <- t.test(controls_dep$BMI, stringent_dep$BMI) 
table_bmi_ss_vs_stringentdep <- data.frame(
  "Demographic" = "Mean BMI",
  "Estimate (controls)" = as.data.frame(bmi_ss_vs_stringentdep$estimate)[1,1],
  "Estimate (cases)" = as.data.frame(bmi_ss_vs_stringentdep$estimate)[2,1],
  "t-statistic" = as.data.frame(bmi_ss_vs_stringentdep$statistic)[1,1],
  "df" = as.data.frame(bmi_ss_vs_stringentdep$parameter)[1,1],
  "p-value" = formatC(bmi_ss_vs_stringentdep$p.value, format = "e", digits = 0)
)

ss_vs_stringentdep <- rbind(table_age_ss_vs_stringentdep,
                  table_sex_ss_vs_stringentdep,
                  table_ses_ss_vs_stringentdep,
                  table_smoke_ss_vs_stringentdep,
                  table_bmi_ss_vs_stringentdep)

colnames(ss_vs_stringentdep) <- c("Demographic","Estimate (controls)","Estimate (cases)","t-statistic","df","p-value")

ss_vs_stringentdep$`Estimate (controls)` <- round(ss_vs_stringentdep$`Estimate (controls)`, digits = 2)
ss_vs_stringentdep$`Estimate (cases)` <- round(ss_vs_stringentdep$`Estimate (cases)`, digits = 2)
ss_vs_stringentdep$`t-statistic` <- round(ss_vs_stringentdep$`t-statistic`, digits = 0)

kable(ss_vs_stringentdep, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```

</br>

# Depression polygenic score analysis

</br>

```{bash prs_scripts_mdd_to_mdd, echo=FALSE, eval=FALSE}

## PRS for MDD predicing any and stringent depression
Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/DEPR06.gz \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/depression_gender \
--pheno-col Broad_all,Narrow_all \
--binary-target T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.15,0.15 \
--quantile 10 \
--perm 10000 \
--out /file_path/mdd_to_mdd_any_stringent

```

```{r prepare_prs_mdd_to_mdd_table, include=FALSE}

# Make table of results for MDD PRS predicting any and stringent depression

# Read in data
dep <- as.data.frame(fread("/file_path/mdd_to_mdd_any_stringent.summary", header = T, sep = "\t"))
dep$Base <- c("Major Depressive Disorder","Major Depressive Disorder")
dep$Target <- c("Depression (Any)","Depression (Stringent)")

# Note - p.value shows zero because R doesn't show p-values <5e-324. Therefore enter the minimum value in R. 
dep[1,11] <- 5e-324
dep$P <- as.numeric(dep$P)
dep$P <- formatC(dep$P,format="e",digits = 0)
dep$Standard.Error <- round(dep$Standard.Error, digits = 3)

# Format
dep <- dep[,c(14,15,3,12,8,4,5,9:11)]
dep$Prevalence <- percent(dep$Prevalence, digits = 0)
dep$PRS.R2.adj <- percent(dep$PRS.R2.adj)
dep$PRS.R2 <- percent(dep$PRS.R2)
dep$Coefficient <- round(dep$Coefficient, digits = 2)
rownames(dep) <- NULL

```

**Table:** PRS for MDD predicting depression case/control status

```{r print_prs_mdd_to_mdd_table, echo=FALSE}

kable(dep, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666", align = "centre")

```

</br>

```{r prepare_plot_all_thresholds_mdd_to_mdd, echo=FALSE, results = 'asis', fig.align='center', fig.height=6, fig.width=18}

# Prepare plot showing PRS at all p-value thresholds

# Read data
dep_prsice <- read.table("/file_path/mdd_to_mdd_any_stringent.prsice", header = T) 
dep_prsice$Pheno <- gsub("Broad_all","Depression (Any)", dep_prsice$Pheno)
dep_prsice$Pheno <- gsub("Narrow_all","Depression (Stringent)", dep_prsice$Pheno)

# Populate missing p-values
dep_prsice[2:8,6] <- 5e-324
dep_prsice$P <- as.numeric(dep_prsice$P)
dep_prsice$P.format <- formatC(dep_prsice$P,format="e",digits = 0)

# Format p-values for plotting
dep_prsice$print.p <- formatC(dep_prsice$P.format, digits = 2)
dep_prsice$print.p <- sub("e", "*x*10^", dep_prsice$print.p)

# Create plot 
threshold_plot <- ggplot(dep_prsice, aes(x = as.factor(Threshold), y = R2, fill = -log10(P))) +
  geom_bar(stat = "identity", position = position_dodge()) +
  scale_fill_gradient2(low = "dodgerblue", high = "firebrick", mid = "dodgerblue", midpoint=1e-4,
                       name = bquote(atop(-log[10] ~ model, italic(P) - value),)) +
  ylab(expression(paste("PRS model fit:  ", R ^ 2, sep = " "))) +
  xlab(expression(italic(P) - value ~ threshold ~ (italic(P)[T]))) +
  theme(axis.text.x = element_text(angle=45, hjust=1, size = 12),
        axis.text.y = element_text(size = 12),
        axis.title = element_text(size = 14),
        plot.title = element_text(hjust = 0.5, size = 18),
        strip.text.x = element_text(size = 12, face = "bold"),
        panel.grid = element_blank()) +
  scale_y_continuous(expand = expand_scale(mult = c(0, .3))) +
  geom_text(aes(label = paste(print.p)), vjust = -1.5, hjust = 0, angle = 45, cex = 3.5, parse = T) +
  facet_grid(~ Pheno)

```

```{r print_thresholds_mdd_to_mdd, echo=FALSE, results = 'asis', fig.align='center', fig.height=5, fig.width=15}

threshold_plot

```

</br>

# Phenotypic Overlap

</br>

## Prevalence of depression in autoimmune cases and controls

```{r prepare_prevalence_any_dep_in_poss_autoimmune, include=FALSE}

# Test for differences in any depression prevalence between possible autoimmune cases and controls

# Read autoimmune and depression phenotype files
autoimmune <-fread("/file_path/new_autoimmune_all", head = T, data.table = F)
depression <- fread("/file_path/dep_broad_narrow", head = T, data.table = F)

# Create dfs for broad (any depression)/possible first

# DEPRESSION - remove NA fields
dep_broad <- depression[,c(1,24)] %>%
  na.omit() %>%
  rename(IID = ID)

# AUTOIMMUNE DISEASE - create df for each possible autoimmune disease, removing NA and merging with broad depression

# Make df for each column of autoimmune df
for(i in seq(3, 33, 2)) {
  assign(paste("possible",i,sep="_"),autoimmune[,c(2,i)])
}

# Make list of dfs
MyList <- list(possible_3,possible_5,possible_7,possible_9,
               possible_11,possible_13,possible_15,possible_17,possible_19,
               possible_21,possible_23,possible_25,possible_27,possible_29,
               possible_31,possible_33)

# Remove NA
MyList <- lapply(MyList, na.omit)

# Merge with any dep
poss1 <- merge(MyList[[1]], dep_broad, by = "IID")
poss2 <- merge(MyList[[2]], dep_broad, by = "IID")
poss3 <- merge(MyList[[3]], dep_broad, by = "IID")
poss4 <- merge(MyList[[4]], dep_broad, by = "IID")
poss5 <- merge(MyList[[5]], dep_broad, by = "IID")
poss6 <- merge(MyList[[6]], dep_broad, by = "IID")
poss7 <- merge(MyList[[7]], dep_broad, by = "IID")
poss8 <- merge(MyList[[8]], dep_broad, by = "IID")
poss9 <- merge(MyList[[9]], dep_broad, by = "IID")
poss10 <- merge(MyList[[10]], dep_broad, by = "IID")
poss11 <- merge(MyList[[11]], dep_broad, by = "IID")
poss12 <- merge(MyList[[12]], dep_broad, by = "IID")
poss13 <- merge(MyList[[13]], dep_broad, by = "IID")
poss14 <- merge(MyList[[14]], dep_broad, by = "IID")
poss15 <- merge(MyList[[15]], dep_broad, by = "IID")
poss16 <- merge(MyList[[16]], dep_broad, by = "IID")

# Run chi-square
x <- c(nrow(poss1[ which(poss1[,3]==1 & poss1[,2]==1), ]),
       nrow(poss1[ which(poss1[,3]==1 & poss1[,2]==0), ]))
n <- c(sum(poss1[,2]==1), sum(poss1[,2]==0))
poss_a <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss2[ which(poss2[,3]==1 & poss2[,2]==1), ]),
       nrow(poss2[ which(poss2[,3]==1 & poss2[,2]==0), ]))
n <- c(sum(poss2[,2]==1), sum(poss2[,2]==0))
poss_b <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss3[ which(poss3[,3]==1 & poss3[,2]==1), ]),
       nrow(poss3[ which(poss3[,3]==1 & poss3[,2]==0), ]))
n <- c(sum(poss3[,2]==1), sum(poss3[,2]==0))
poss_c <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss4[ which(poss4[,3]==1 & poss4[,2]==1), ]),
       nrow(poss4[ which(poss4[,3]==1 & poss4[,2]==0), ]))
n <- c(sum(poss4[,2]==1), sum(poss4[,2]==0))
poss_d <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss5[ which(poss5[,3]==1 & poss5[,2]==1), ]),
       nrow(poss5[ which(poss5[,3]==1 & poss5[,2]==0), ]))
n <- c(sum(poss5[,2]==1), sum(poss5[,2]==0))
poss_e <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss6[ which(poss6[,3]==1 & poss6[,2]==1), ]),
       nrow(poss6[ which(poss6[,3]==1 & poss6[,2]==0), ]))
n <- c(sum(poss6[,2]==1), sum(poss6[,2]==0))
poss_f <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss7[ which(poss7[,3]==1 & poss7[,2]==1), ]),
       nrow(poss7[ which(poss7[,3]==1 & poss7[,2]==0), ]))
n <- c(sum(poss7[,2]==1), sum(poss7[,2]==0))
poss_g <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss8[ which(poss8[,3]==1 & poss8[,2]==1), ]),
       nrow(poss8[ which(poss8[,3]==1 & poss8[,2]==0), ]))
n <- c(sum(poss8[,2]==1), sum(poss8[,2]==0))
poss_h <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss9[ which(poss9[,3]==1 & poss9[,2]==1), ]),
       nrow(poss9[ which(poss9[,3]==1 & poss9[,2]==0), ]))
n <- c(sum(poss9[,2]==1), sum(poss9[,2]==0))
poss_i <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss10[ which(poss10[,3]==1 & poss10[,2]==1), ]),
       nrow(poss10[ which(poss10[,3]==1 & poss10[,2]==0), ]))
n <- c(sum(poss10[,2]==1), sum(poss10[,2]==0))
poss_j <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss11[ which(poss11[,3]==1 & poss11[,2]==1), ]),
       nrow(poss11[ which(poss11[,3]==1 & poss11[,2]==0), ]))
n <- c(sum(poss11[,2]==1), sum(poss11[,2]==0))
poss_k <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss12[ which(poss12[,3]==1 & poss12[,2]==1), ]),
       nrow(poss12[ which(poss12[,3]==1 & poss12[,2]==0), ]))
n <- c(sum(poss12[,2]==1), sum(poss12[,2]==0))
poss_l <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss13[ which(poss13[,3]==1 & poss13[,2]==1), ]),
       nrow(poss13[ which(poss13[,3]==1 & poss13[,2]==0), ]))
n <- c(sum(poss13[,2]==1), sum(poss13[,2]==0))
poss_m <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss14[ which(poss14[,3]==1 & poss14[,2]==1), ]),
       nrow(poss14[ which(poss14[,3]==1 & poss14[,2]==0), ]))
n <- c(sum(poss14[,2]==1), sum(poss14[,2]==0))
poss_n <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss15[ which(poss15[,3]==1 & poss15[,2]==1), ]),
       nrow(poss15[ which(poss15[,3]==1 & poss15[,2]==0), ]))
n <- c(sum(poss15[,2]==1), sum(poss15[,2]==0))
poss_o <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss16[ which(poss16[,3]==1 & poss16[,2]==1), ]),
       nrow(poss16[ which(poss16[,3]==1 & poss16[,2]==0), ]))
n <- c(sum(poss16[,2]==1), sum(poss16[,2]==0))
poss_p <- prop.test(x, n, correct = FALSE)


# BUILD TABLE OF RESULTS
p_poss <- data.frame(
  "Autoimmune Disease" = c("Ankylosing Spondylitis", 
                           "Autoimmune Thyroid Disease",
                           "Coeliac",
                           "Hidradenitis Suppurativa",
                           "Inflammatory Bowel Disease",
                           "Myasthenia Gravis",
                           "Multiple Sclerosis",
                           "Pernicious Anemia",
                           "Pemphigoid/Pemphigus",
                           "Polymyalgia Rheumatica/GCA",
                           "Psoriatic Arthritis",
                           "Psoriasis",
                           "Rheumatoid Arthritis",
                           "Sjögren Syndrome",
                           "Systemic Lupus Erythematosus",
                           "Type 1 diabetes"),
    "P-value case-control difference (any/poss)" = c(poss_a$p.value,
                poss_b$p.value,
                poss_c$p.value,
                poss_d$p.value,
                poss_e$p.value,
                poss_f$p.value,
                poss_g$p.value,
                poss_h$p.value,
                poss_i$p.value,
                poss_j$p.value,
                poss_k$p.value,
                poss_l$p.value,
                poss_m$p.value,
                poss_n$p.value,
                poss_o$p.value,
                poss_p$p.value)
  )

proportions_poss <- as.data.frame(rbind(poss_a$estimate,
                poss_b$estimate,
                poss_c$estimate,
                poss_d$estimate,
                poss_e$estimate,
                poss_f$estimate,
                poss_g$estimate,
                poss_h$estimate,
                poss_i$estimate,
                poss_j$estimate,
                poss_k$estimate,
                poss_l$estimate,
                poss_m$estimate,
                poss_n$estimate,
                poss_o$estimate,
                poss_p$estimate)) 
colnames(proportions_poss) <- c("% Any Dep in Poss Cases","% Any Dep in Auto Controls")

summary_any_dep_in_poss_auto <- cbind(p_poss, proportions_poss)
summary_any_dep_in_poss_auto <- summary_any_dep_in_poss_auto[,c(1,3:4,2)]

```

```{r prepare_prevalence_narrow_dep_in_probable_autoimmune, include=FALSE}

# Test for differences in stringent depression prevalence between probable autoimmune cases and controls

# Create dfs for narrow(stringnt)/probable first

# DEPRESSION - remove NA fields
dep_narrow <- depression[,c(1,25)] %>%
  na.omit() %>%
  rename(IID = ID)

# AUTOIMMUNE DISEASE - create df for each probable autoimmune disease, removing NA and merging with stringent depression

# Make df for each column of autoimmune df
for(i in seq(4, 34, 2)) {
  assign(paste("probable",i,sep="_"),autoimmune[,c(2,i)])
}

# Make list of dfs
MyList <- list(probable_4,probable_6,probable_8,probable_10,
               probable_12,probable_14,probable_16,probable_18,probable_20,
               probable_22,probable_24,probable_26,probable_28,probable_30,
               probable_32,probable_34)

# Remove NA
MyList <- lapply(MyList, na.omit)

# Merge with broad dep
prob1 <- merge(MyList[[1]], dep_narrow, by = "IID")
prob2 <- merge(MyList[[2]], dep_narrow, by = "IID")
prob3 <- merge(MyList[[3]], dep_narrow, by = "IID")
prob4 <- merge(MyList[[4]], dep_narrow, by = "IID")
prob5 <- merge(MyList[[5]], dep_narrow, by = "IID")
prob6 <- merge(MyList[[6]], dep_narrow, by = "IID")
prob7 <- merge(MyList[[7]], dep_narrow, by = "IID")
prob8 <- merge(MyList[[8]], dep_narrow, by = "IID")
prob9 <- merge(MyList[[9]], dep_narrow, by = "IID")
prob10 <- merge(MyList[[10]], dep_narrow, by = "IID")
prob11 <- merge(MyList[[11]], dep_narrow, by = "IID")
prob12 <- merge(MyList[[12]], dep_narrow, by = "IID")
prob13 <- merge(MyList[[13]], dep_narrow, by = "IID")
prob14 <- merge(MyList[[14]], dep_narrow, by = "IID")
prob15 <- merge(MyList[[15]], dep_narrow, by = "IID")
prob16 <- merge(MyList[[16]], dep_narrow, by = "IID")

# Run chi-square
x <- c(nrow(prob1[ which(prob1[,3]==1 & prob1[,2]==1), ]),
       nrow(prob1[ which(prob1[,3]==1 & prob1[,2]==0), ]))
n <- c(sum(prob1[,2]==1), sum(prob1[,2]==0))
prob_a <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob2[ which(prob2[,3]==1 & prob2[,2]==1), ]),
       nrow(prob2[ which(prob2[,3]==1 & prob2[,2]==0), ]))
n <- c(sum(prob2[,2]==1), sum(prob2[,2]==0))
prob_b <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob3[ which(prob3[,3]==1 & prob3[,2]==1), ]),
       nrow(prob3[ which(prob3[,3]==1 & prob3[,2]==0), ]))
n <- c(sum(prob3[,2]==1), sum(prob3[,2]==0))
prob_c <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob4[ which(prob4[,3]==1 & prob4[,2]==1), ]),
       nrow(prob4[ which(prob4[,3]==1 & prob4[,2]==0), ]))
n <- c(sum(prob4[,2]==1), sum(prob4[,2]==0))
prob_d <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob5[ which(prob5[,3]==1 & prob5[,2]==1), ]),
       nrow(prob5[ which(prob5[,3]==1 & prob5[,2]==0), ]))
n <- c(sum(prob5[,2]==1), sum(prob5[,2]==0))
prob_e <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob6[ which(prob6[,3]==1 & prob6[,2]==1), ]),
       nrow(prob6[ which(prob6[,3]==1 & prob6[,2]==0), ]))
n <- c(sum(prob6[,2]==1), sum(prob6[,2]==0))
prob_f <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob7[ which(prob7[,3]==1 & prob7[,2]==1), ]),
       nrow(prob7[ which(prob7[,3]==1 & prob7[,2]==0), ]))
n <- c(sum(prob7[,2]==1), sum(prob7[,2]==0))
prob_g <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob8[ which(prob8[,3]==1 & prob8[,2]==1), ]),
       nrow(prob8[ which(prob8[,3]==1 & prob8[,2]==0), ]))
n <- c(sum(prob8[,2]==1), sum(prob8[,2]==0))
prob_h <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob9[ which(prob9[,3]==1 & prob9[,2]==1), ]),
       nrow(prob9[ which(prob9[,3]==1 & prob9[,2]==0), ]))
n <- c(sum(prob9[,2]==1), sum(prob9[,2]==0))
prob_i <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob10[ which(prob10[,3]==1 & prob10[,2]==1), ]),
       nrow(prob10[ which(prob10[,3]==1 & prob10[,2]==0), ]))
n <- c(sum(prob10[,2]==1), sum(prob10[,2]==0))
prob_j <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob11[ which(prob11[,3]==1 & prob11[,2]==1), ]),
       nrow(prob11[ which(prob11[,3]==1 & prob11[,2]==0), ]))
n <- c(sum(prob11[,2]==1), sum(prob11[,2]==0))
prob_k <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob12[ which(prob12[,3]==1 & prob12[,2]==1), ]),
       nrow(prob12[ which(prob12[,3]==1 & prob12[,2]==0), ]))
n <- c(sum(prob12[,2]==1), sum(prob12[,2]==0))
prob_l <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob13[ which(prob13[,3]==1 & prob13[,2]==1), ]),
       nrow(prob13[ which(prob13[,3]==1 & prob13[,2]==0), ]))
n <- c(sum(prob13[,2]==1), sum(prob13[,2]==0))
prob_m <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob14[ which(prob14[,3]==1 & prob14[,2]==1), ]),
       nrow(prob14[ which(prob14[,3]==1 & prob14[,2]==0), ]))
n <- c(sum(prob14[,2]==1), sum(prob14[,2]==0))
prob_n <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob15[ which(prob15[,3]==1 & prob15[,2]==1), ]),
       nrow(prob15[ which(prob15[,3]==1 & prob15[,2]==0), ]))
n <- c(sum(prob15[,2]==1), sum(prob15[,2]==0))
prob_o <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob16[ which(prob16[,3]==1 & prob16[,2]==1), ]),
       nrow(prob16[ which(prob16[,3]==1 & prob16[,2]==0), ]))
n <- c(sum(prob16[,2]==1), sum(prob16[,2]==0))
prob_p <- prop.test(x, n, correct = FALSE)

# BUILD TABLE OF RESULTS
p_prob <- data.frame(
  "Autoimmune Disease" = c("Ankylosing Spondylitis", 
                           "Autoimmune Thyroid Disease",
                           "Coeliac",
                           "Hidradenitis Suppurativa",
                           "Inflammatory Bowel Disease",
                           "Myasthenia Gravis",
                           "Multiple Sclerosis",
                           "Pernicious Anemia",
                           "Pemphigoid/Pemphigus",
                           "Polymyalgia Rheumatica/GCA",
                           "Psoriatic Arthritis",
                           "Psoriasis",
                           "Rheumatoid Arthritis",
                           "Sjögren Syndrome",
                           "Systemic Lupus Erythematosus",
                           "Type 1 diabetes"),
   "P-value case-control difference (stringent/prob)" = c(prob_a$p.value,
                prob_b$p.value,
                prob_c$p.value,
                prob_d$p.value,
                prob_e$p.value,
                prob_f$p.value,
                prob_g$p.value,
                prob_h$p.value,
                prob_i$p.value,
                prob_j$p.value,
                prob_k$p.value,
                prob_l$p.value,
                prob_m$p.value,
                prob_n$p.value,
                prob_o$p.value,
                prob_p$p.value)
  )

proportions_prob <- as.data.frame(rbind(prob_a$estimate,
                prob_b$estimate,
                prob_c$estimate,
                prob_d$estimate,
                prob_e$estimate,
                prob_f$estimate,
                prob_g$estimate,
                prob_h$estimate,
                prob_i$estimate,
                prob_j$estimate,
                prob_k$estimate,
                prob_l$estimate,
                prob_m$estimate,
                prob_n$estimate,
                prob_o$estimate,
                prob_p$estimate)) 
colnames(proportions_prob) <- c("% Stringent Dep in Prob Cases","% Stringent Dep in Auto Controls")

summary_stringent_dep_in_prob_auto <- cbind(p_prob, proportions_prob)
summary_stringent_dep_in_prob_auto <- summary_stringent_dep_in_prob_auto[,c(1,3:4,2)]

```

```{r prepare_prevalence_broad_dep_in_probable_autoimmune, include=FALSE}

# Test for differences in any depression prevalence between probable autoimmune cases and controls

# Read autoimmune and depression phenotype files
autoimmune <-fread("/file_path/new_autoimmune_all", head = T, data.table = F)
depression <- fread("/file_path/dep_broad_narrow", head = T, data.table = F)

# Create dfs for broad/possible first

# DEPRESSION - remove NA fields
dep_broad <- depression[,c(1,24)] %>%
  na.omit() %>%
  rename(IID = ID)

# AUTOIMMUNE DISEASE - create df for each probable autoimmune disease, removing NA and merging with broad depression

# Make df for each column of autoimmune df
for(i in seq(4, 34, 2)) {
  assign(paste("probable",i,sep="_"),autoimmune[,c(2,i)])
}

# Make list of dfs
MyList <- list(probable_4,probable_6,probable_8,probable_10,
               probable_12,probable_14,probable_16,probable_18,probable_20,
               probable_22,probable_24,probable_26,probable_28,probable_30,
               probable_32,probable_34)

# Remove NA
MyList <- lapply(MyList, na.omit)

# Merge with broad dep
prob1 <- merge(MyList[[1]], dep_broad, by = "IID")
prob2 <- merge(MyList[[2]], dep_broad, by = "IID")
prob3 <- merge(MyList[[3]], dep_broad, by = "IID")
prob4 <- merge(MyList[[4]], dep_broad, by = "IID")
prob5 <- merge(MyList[[5]], dep_broad, by = "IID")
prob6 <- merge(MyList[[6]], dep_broad, by = "IID")
prob7 <- merge(MyList[[7]], dep_broad, by = "IID")
prob8 <- merge(MyList[[8]], dep_broad, by = "IID")
prob9 <- merge(MyList[[9]], dep_broad, by = "IID")
prob10 <- merge(MyList[[10]], dep_broad, by = "IID")
prob11 <- merge(MyList[[11]], dep_broad, by = "IID")
prob12 <- merge(MyList[[12]], dep_broad, by = "IID")
prob13 <- merge(MyList[[13]], dep_broad, by = "IID")
prob14 <- merge(MyList[[14]], dep_broad, by = "IID")
prob15 <- merge(MyList[[15]], dep_broad, by = "IID")
prob16 <- merge(MyList[[16]], dep_broad, by = "IID")

# Run chi-square
x <- c(nrow(prob1[ which(prob1[,3]==1 & prob1[,2]==1), ]),
       nrow(prob1[ which(prob1[,3]==1 & prob1[,2]==0), ]))
n <- c(sum(prob1[,2]==1), sum(prob1[,2]==0))
prob_a <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob2[ which(prob2[,3]==1 & prob2[,2]==1), ]),
       nrow(prob2[ which(prob2[,3]==1 & prob2[,2]==0), ]))
n <- c(sum(prob2[,2]==1), sum(prob2[,2]==0))
prob_b <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob3[ which(prob3[,3]==1 & prob3[,2]==1), ]),
       nrow(prob3[ which(prob3[,3]==1 & prob3[,2]==0), ]))
n <- c(sum(prob3[,2]==1), sum(prob3[,2]==0))
prob_c <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob4[ which(prob4[,3]==1 & prob4[,2]==1), ]),
       nrow(prob4[ which(prob4[,3]==1 & prob4[,2]==0), ]))
n <- c(sum(prob4[,2]==1), sum(prob4[,2]==0))
prob_d <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob5[ which(prob5[,3]==1 & prob5[,2]==1), ]),
       nrow(prob5[ which(prob5[,3]==1 & prob5[,2]==0), ]))
n <- c(sum(prob5[,2]==1), sum(prob5[,2]==0))
prob_e <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob6[ which(prob6[,3]==1 & prob6[,2]==1), ]),
       nrow(prob6[ which(prob6[,3]==1 & prob6[,2]==0), ]))
n <- c(sum(prob6[,2]==1), sum(prob6[,2]==0))
prob_f <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob7[ which(prob7[,3]==1 & prob7[,2]==1), ]),
       nrow(prob7[ which(prob7[,3]==1 & prob7[,2]==0), ]))
n <- c(sum(prob7[,2]==1), sum(prob7[,2]==0))
prob_g <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob8[ which(prob8[,3]==1 & prob8[,2]==1), ]),
       nrow(prob8[ which(prob8[,3]==1 & prob8[,2]==0), ]))
n <- c(sum(prob8[,2]==1), sum(prob8[,2]==0))
prob_h <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob9[ which(prob9[,3]==1 & prob9[,2]==1), ]),
       nrow(prob9[ which(prob9[,3]==1 & prob9[,2]==0), ]))
n <- c(sum(prob9[,2]==1), sum(prob9[,2]==0))
prob_i <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob10[ which(prob10[,3]==1 & prob10[,2]==1), ]),
       nrow(prob10[ which(prob10[,3]==1 & prob10[,2]==0), ]))
n <- c(sum(prob10[,2]==1), sum(prob10[,2]==0))
prob_j <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob11[ which(prob11[,3]==1 & prob11[,2]==1), ]),
       nrow(prob11[ which(prob11[,3]==1 & prob11[,2]==0), ]))
n <- c(sum(prob11[,2]==1), sum(prob11[,2]==0))
prob_k <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob12[ which(prob12[,3]==1 & prob12[,2]==1), ]),
       nrow(prob12[ which(prob12[,3]==1 & prob12[,2]==0), ]))
n <- c(sum(prob12[,2]==1), sum(prob12[,2]==0))
prob_l <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob13[ which(prob13[,3]==1 & prob13[,2]==1), ]),
       nrow(prob13[ which(prob13[,3]==1 & prob13[,2]==0), ]))
n <- c(sum(prob13[,2]==1), sum(prob13[,2]==0))
prob_m <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob14[ which(prob14[,3]==1 & prob14[,2]==1), ]),
       nrow(prob14[ which(prob14[,3]==1 & prob14[,2]==0), ]))
n <- c(sum(prob14[,2]==1), sum(prob14[,2]==0))
prob_n <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob15[ which(prob15[,3]==1 & prob15[,2]==1), ]),
       nrow(prob15[ which(prob15[,3]==1 & prob15[,2]==0), ]))
n <- c(sum(prob15[,2]==1), sum(prob15[,2]==0))
prob_o <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob16[ which(prob16[,3]==1 & prob16[,2]==1), ]),
       nrow(prob16[ which(prob16[,3]==1 & prob16[,2]==0), ]))
n <- c(sum(prob16[,2]==1), sum(prob16[,2]==0))
prob_p <- prop.test(x, n, correct = FALSE)

# BUILD TABLE OF RESULTS
p_prob <- data.frame(
  "Autoimmune Disease" = c("Ankylosing Spondylitis", 
                           "Autoimmune Thyroid Disease",
                           "Coeliac",
                           "Hidradenitis Suppurativa",
                           "Inflammatory Bowel Disease",
                           "Myasthenia Gravis",
                           "Multiple Sclerosis",
                           "Pernicious Anemia",
                           "Pemphigoid/Pemphigus",
                           "Polymyalgia Rheumatica/GCA",
                           "Psoriatic Arthritis",
                           "Psoriasis",
                           "Rheumatoid Arthritis",
                           "Sjögren Syndrome",
                           "Systemic Lupus Erythematosus",
                           "Type 1 diabetes"),
   "P-value case-control difference (any/prob)" = c(prob_a$p.value,
                prob_b$p.value,
                prob_c$p.value,
                prob_d$p.value,
                prob_e$p.value,
                prob_f$p.value,
                prob_g$p.value,
                prob_h$p.value,
                prob_i$p.value,
                prob_j$p.value,
                prob_k$p.value,
                prob_l$p.value,
                prob_m$p.value,
                prob_n$p.value,
                prob_o$p.value,
                prob_p$p.value)
  )

proportions_prob <- as.data.frame(rbind(prob_a$estimate,
                prob_b$estimate,
                prob_c$estimate,
                prob_d$estimate,
                prob_e$estimate,
                prob_f$estimate,
                prob_g$estimate,
                prob_h$estimate,
                prob_i$estimate,
                prob_j$estimate,
                prob_k$estimate,
                prob_l$estimate,
                prob_m$estimate,
                prob_n$estimate,
                prob_o$estimate,
                prob_p$estimate)) 
colnames(proportions_prob) <- c("% Any Dep in Prob Cases","% Any Dep in Auto Controls")

summary_any_dep_in_prob_auto <- cbind(p_prob, proportions_prob)
summary_any_dep_in_prob_auto <- summary_any_dep_in_prob_auto[,c(1,3:4,2)]

```

```{r prepare_prevalence_narrow_dep_in_poss_autoimmune, include=FALSE}

# Test for differences in stringent depression prevalence between possible autoimmune cases and controls

# Read autoimmune and depression phenotype files
autoimmune <-fread("/file_path/new_autoimmune_all", head = T, data.table = F)
depression <- fread("/file_path/dep_broad_narrow", head = T, data.table = F)

# Create dfs for stringent/possible first

# DEPRESSION - remove NA fields
dep_narrow <- depression[,c(1,25)] %>%
  na.omit() %>%
  rename(IID = ID)

# AUTOIMMUNE DISEASE - create df for each possible autoimmune disease, removing NA and merging with stringent depression

# Make df for each column of autoimmune df
for(i in seq(3, 33, 2)) {
  assign(paste("possible",i,sep="_"),autoimmune[,c(2,i)])
}

# Make list of dfs
MyList <- list(possible_3,possible_5,possible_7,possible_9,
               possible_11,possible_13,possible_15,possible_17,possible_19,
               possible_21,possible_23,possible_25,possible_27,possible_29,
               possible_31,possible_33)

# Remove NA
MyList <- lapply(MyList, na.omit)

# Merge with broad dep
poss1 <- merge(MyList[[1]], dep_narrow, by = "IID")
poss2 <- merge(MyList[[2]], dep_narrow, by = "IID")
poss3 <- merge(MyList[[3]], dep_narrow, by = "IID")
poss4 <- merge(MyList[[4]], dep_narrow, by = "IID")
poss5 <- merge(MyList[[5]], dep_narrow, by = "IID")
poss6 <- merge(MyList[[6]], dep_narrow, by = "IID")
poss7 <- merge(MyList[[7]], dep_narrow, by = "IID")
poss8 <- merge(MyList[[8]], dep_narrow, by = "IID")
poss9 <- merge(MyList[[9]], dep_narrow, by = "IID")
poss10 <- merge(MyList[[10]], dep_narrow, by = "IID")
poss11 <- merge(MyList[[11]], dep_narrow, by = "IID")
poss12 <- merge(MyList[[12]], dep_narrow, by = "IID")
poss13 <- merge(MyList[[13]], dep_narrow, by = "IID")
poss14 <- merge(MyList[[14]], dep_narrow, by = "IID")
poss15 <- merge(MyList[[15]], dep_narrow, by = "IID")
poss16 <- merge(MyList[[16]], dep_narrow, by = "IID")

# Run chi-square
x <- c(nrow(poss1[ which(poss1[,3]==1 & poss1[,2]==1), ]),
       nrow(poss1[ which(poss1[,3]==1 & poss1[,2]==0), ]))
n <- c(sum(poss1[,2]==1), sum(poss1[,2]==0))
poss_a <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss2[ which(poss2[,3]==1 & poss2[,2]==1), ]),
       nrow(poss2[ which(poss2[,3]==1 & poss2[,2]==0), ]))
n <- c(sum(poss2[,2]==1), sum(poss2[,2]==0))
poss_b <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss3[ which(poss3[,3]==1 & poss3[,2]==1), ]),
       nrow(poss3[ which(poss3[,3]==1 & poss3[,2]==0), ]))
n <- c(sum(poss3[,2]==1), sum(poss3[,2]==0))
poss_c <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss4[ which(poss4[,3]==1 & poss4[,2]==1), ]),
       nrow(poss4[ which(poss4[,3]==1 & poss4[,2]==0), ]))
n <- c(sum(poss4[,2]==1), sum(poss4[,2]==0))
poss_d <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss5[ which(poss5[,3]==1 & poss5[,2]==1), ]),
       nrow(poss5[ which(poss5[,3]==1 & poss5[,2]==0), ]))
n <- c(sum(poss5[,2]==1), sum(poss5[,2]==0))
poss_e <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss6[ which(poss6[,3]==1 & poss6[,2]==1), ]),
       nrow(poss6[ which(poss6[,3]==1 & poss6[,2]==0), ]))
n <- c(sum(poss6[,2]==1), sum(poss6[,2]==0))
poss_f <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss7[ which(poss7[,3]==1 & poss7[,2]==1), ]),
       nrow(poss7[ which(poss7[,3]==1 & poss7[,2]==0), ]))
n <- c(sum(poss7[,2]==1), sum(poss7[,2]==0))
poss_g <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss8[ which(poss8[,3]==1 & poss8[,2]==1), ]),
       nrow(poss8[ which(poss8[,3]==1 & poss8[,2]==0), ]))
n <- c(sum(poss8[,2]==1), sum(poss8[,2]==0))
poss_h <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss9[ which(poss9[,3]==1 & poss9[,2]==1), ]),
       nrow(poss9[ which(poss9[,3]==1 & poss9[,2]==0), ]))
n <- c(sum(poss9[,2]==1), sum(poss9[,2]==0))
poss_i <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss10[ which(poss10[,3]==1 & poss10[,2]==1), ]),
       nrow(poss10[ which(poss10[,3]==1 & poss10[,2]==0), ]))
n <- c(sum(poss10[,2]==1), sum(poss10[,2]==0))
poss_j <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss11[ which(poss11[,3]==1 & poss11[,2]==1), ]),
       nrow(poss11[ which(poss11[,3]==1 & poss11[,2]==0), ]))
n <- c(sum(poss11[,2]==1), sum(poss11[,2]==0))
poss_k <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss12[ which(poss12[,3]==1 & poss12[,2]==1), ]),
       nrow(poss12[ which(poss12[,3]==1 & poss12[,2]==0), ]))
n <- c(sum(poss12[,2]==1), sum(poss12[,2]==0))
poss_l <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss13[ which(poss13[,3]==1 & poss13[,2]==1), ]),
       nrow(poss13[ which(poss13[,3]==1 & poss13[,2]==0), ]))
n <- c(sum(poss13[,2]==1), sum(poss13[,2]==0))
poss_m <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss14[ which(poss14[,3]==1 & poss14[,2]==1), ]),
       nrow(poss14[ which(poss14[,3]==1 & poss14[,2]==0), ]))
n <- c(sum(poss14[,2]==1), sum(poss14[,2]==0))
poss_n <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss15[ which(poss15[,3]==1 & poss15[,2]==1), ]),
       nrow(poss15[ which(poss15[,3]==1 & poss15[,2]==0), ]))
n <- c(sum(poss15[,2]==1), sum(poss15[,2]==0))
poss_o <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss16[ which(poss16[,3]==1 & poss16[,2]==1), ]),
       nrow(poss16[ which(poss16[,3]==1 & poss16[,2]==0), ]))
n <- c(sum(poss16[,2]==1), sum(poss16[,2]==0))
poss_p <- prop.test(x, n, correct = FALSE)


# BUILD TABLE OF RESULTS
p_poss <- data.frame(
  "Autoimmune Disease" = c("Ankylosing Spondylitis", 
                           "Autoimmune Thyroid Disease",
                           "Coeliac",
                           "Hidradenitis Suppurativa",
                           "Inflammatory Bowel Disease",
                           "Myasthenia Gravis",
                           "Multiple Sclerosis",
                           "Pernicious Anemia",
                           "Pemphigoid/Pemphigus",
                           "Polymyalgia Rheumatica/GCA",
                           "Psoriatic Arthritis",
                           "Psoriasis",
                           "Rheumatoid Arthritis",
                           "Sjögren Syndrome",
                           "Systemic Lupus Erythematosus",
                           "Type 1 diabetes"),
    "P-value case-control difference (stringent/poss)" = c(poss_a$p.value,
                poss_b$p.value,
                poss_c$p.value,
                poss_d$p.value,
                poss_e$p.value,
                poss_f$p.value,
                poss_g$p.value,
                poss_h$p.value,
                poss_i$p.value,
                poss_j$p.value,
                poss_k$p.value,
                poss_l$p.value,
                poss_m$p.value,
                poss_n$p.value,
                poss_o$p.value,
                poss_p$p.value)
  )

proportions_poss <- as.data.frame(rbind(poss_a$estimate,
                poss_b$estimate,
                poss_c$estimate,
                poss_d$estimate,
                poss_e$estimate,
                poss_f$estimate,
                poss_g$estimate,
                poss_h$estimate,
                poss_i$estimate,
                poss_j$estimate,
                poss_k$estimate,
                poss_l$estimate,
                poss_m$estimate,
                poss_n$estimate,
                poss_o$estimate,
                poss_p$estimate)) 
colnames(proportions_poss) <- c("% Stringent Dep in Poss Cases","% Stringent Dep in Auto Controls")

summary_stringent_dep_in_poss_auto <- cbind(p_poss, proportions_poss)
summary_stringent_dep_in_poss_auto <- summary_stringent_dep_in_poss_auto[,c(1,3:4,2)]

```

**Table:** Prevalence of depression in autoimmune cases and controls

```{r print_prevalence_table_dep_in_auto, echo=FALSE}

# Print summary of depression prevalence in autoimmune cases and controls

# Join tables
summary_dep_in_auto <- plyr::join_all(list(summary_any_dep_in_poss_auto, summary_any_dep_in_prob_auto, summary_stringent_dep_in_poss_auto, summary_stringent_dep_in_prob_auto), by = "Autoimmune.Disease")

# Format percentage
summary_dep_in_auto[,2] <- percent(round(summary_dep_in_auto[,2], digits = 3), digits = 1)
summary_dep_in_auto[,3] <- percent(round(summary_dep_in_auto[,3], digits = 3), digits = 1)
summary_dep_in_auto[,5] <- percent(round(summary_dep_in_auto[,5], digits = 3), digits = 1)
summary_dep_in_auto[,6] <- percent(round(summary_dep_in_auto[,6], digits = 3), digits = 1)
summary_dep_in_auto[,8] <- percent(round(summary_dep_in_auto[,8], digits = 3), digits = 1)
summary_dep_in_auto[,9] <- percent(round(summary_dep_in_auto[,9], digits = 3), digits = 1)
summary_dep_in_auto[,11] <- percent(round(summary_dep_in_auto[,11], digits = 3), digits = 1)
summary_dep_in_auto[,12] <- percent(round(summary_dep_in_auto[,12], digits = 3), digits = 1)

# Format p-values
summary_dep_in_auto[,4] <- formatC(summary_dep_in_auto[,4],format="e",digits = 0)
summary_dep_in_auto[,7] <- formatC(summary_dep_in_auto[,7],format="e",digits = 0)
summary_dep_in_auto[,10] <- formatC(summary_dep_in_auto[,10],format="e",digits = 0)
summary_dep_in_auto[,13] <- formatC(summary_dep_in_auto[,13],format="e",digits = 0)

# Remove Hidradenitis and Pemphigoid (excluded due to low N)
summary_dep_in_auto <- summary_dep_in_auto[- grep("Hidradenitis", summary_dep_in_auto$Autoimmune.Disease),]
summary_dep_in_auto <- summary_dep_in_auto[- grep("Pemphigoid", summary_dep_in_auto$Autoimmune.Disease),]

# Order columns
summary_dep_in_auto$order <- c(9,2,6,7,5,4,1,10,11,8,12,13,14,3)
summary_dep_in_auto <- summary_dep_in_auto[order(summary_dep_in_auto$order),]
rownames(summary_dep_in_auto) <- NULL
summary_dep_in_auto <- summary_dep_in_auto[,1:13]

# Format column headings
summary_dep_in_auto <- summary_dep_in_auto %>%
  rename("P-value case-control difference (any/poss)" = P.value.case.control.difference..any.poss.,
         "P-value case-control difference (any/prob)" = P.value.case.control.difference..any.prob.,
        "P-value case-control difference (stringent/poss)" = P.value.case.control.difference..stringent.poss.,
         "P-value case-control difference (stringent/prob)" = P.value.case.control.difference..stringent.prob.)


kable(summary_dep_in_auto, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%   
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>% 
  pack_rows("Blood", 1, 1, label_row_css = "background-color: #888; color: #fff;") %>%
  pack_rows("Endocrine system", 2, 3, label_row_css = "background-color: #888; color: #fff;") %>%
  pack_rows("Nervous System", 4, 5, label_row_css = "background-color: #888; color: #fff;") %>%
  pack_rows("Digestive system", 6, 7, label_row_css = "background-color: #888; color: #fff;") %>%
  pack_rows("Skin", 8, 8, label_row_css = "background-color: #888; color: #fff;") %>%
  pack_rows("Musculoskeletal system and connective tissue", 9, 14, label_row_css = "background-color: #888; color: #fff;")

```

</br>

## Prevalence of autoimmune diseases in depression cases and controls

</br>

```{r prepare_prevalence_poss_autoimmune_in_broad_dep, include=FALSE}

# Test for differences in possible autoimmune prevalence between any depression cases and controls

# Read autoimmune and depression phenotype files
autoimmune <-fread("/file_path/new_autoimmune_all", head = T, data.table = F)
depression <- fread("/file_path/dep_broad_narrow", head = T, data.table = F)

# AUTOIMMUNE DISEASE - create df for each possible autoimmune disease, removing NA and merging with broad depression

# Make df for each column of autoimmune df
for(i in seq(3, 33, 2)) {
  assign(paste("possible",i,sep="_"),autoimmune[,c(2,i)])
}

# Make list of dfs
MyList <- list(possible_3,possible_5,possible_7,possible_9,
               possible_11,possible_13,possible_15,possible_17,possible_19,
               possible_21,possible_23,possible_25,possible_27,possible_29,
               possible_31,possible_33)

# Remove NA
MyList <- lapply(MyList, na.omit)


# DEPRESSION - remove NA fields
dep_broad <- depression[,c(1,24)] %>%
  na.omit() %>%
  rename(IID = ID)


# Merge with broad dep
poss1 <- merge(dep_broad, MyList[[1]], by = "IID")
poss2 <- merge(dep_broad, MyList[[2]], by = "IID")
poss3 <- merge(dep_broad, MyList[[3]], by = "IID")
poss4 <- merge(dep_broad, MyList[[4]],  by = "IID")
poss5 <- merge(dep_broad, MyList[[5]], by = "IID")
poss6 <- merge(dep_broad, MyList[[6]], by = "IID")
poss7 <- merge(dep_broad, MyList[[7]], by = "IID")
poss8 <- merge(dep_broad, MyList[[8]], by = "IID")
poss9 <- merge(dep_broad, MyList[[9]], by = "IID")
poss10 <- merge(dep_broad, MyList[[10]], by = "IID")
poss11 <- merge(dep_broad, MyList[[11]], by = "IID")
poss12 <- merge(dep_broad, MyList[[12]], by = "IID")
poss13 <- merge(dep_broad, MyList[[13]], by = "IID")
poss14 <- merge(dep_broad, MyList[[14]], by = "IID")
poss15 <- merge(dep_broad, MyList[[15]], by = "IID")
poss16 <- merge(dep_broad, MyList[[16]], by = "IID")

# Run chi-square
x <- c(nrow(poss1[ which(poss1[,3]==1 & poss1[,2]==1), ]),
       nrow(poss1[ which(poss1[,3]==1 & poss1[,2]==0), ]))
n <- c(sum(poss1[,2]==1), sum(poss1[,2]==0))
poss_a <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss2[ which(poss2[,3]==1 & poss2[,2]==1), ]),
       nrow(poss2[ which(poss2[,3]==1 & poss2[,2]==0), ]))
n <- c(sum(poss2[,2]==1), sum(poss2[,2]==0))
poss_b <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss3[ which(poss3[,3]==1 & poss3[,2]==1), ]),
       nrow(poss3[ which(poss3[,3]==1 & poss3[,2]==0), ]))
n <- c(sum(poss3[,2]==1), sum(poss3[,2]==0))
poss_c <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss4[ which(poss4[,3]==1 & poss4[,2]==1), ]),
       nrow(poss4[ which(poss4[,3]==1 & poss4[,2]==0), ]))
n <- c(sum(poss4[,2]==1), sum(poss4[,2]==0))
poss_d <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss5[ which(poss5[,3]==1 & poss5[,2]==1), ]),
       nrow(poss5[ which(poss5[,3]==1 & poss5[,2]==0), ]))
n <- c(sum(poss5[,2]==1), sum(poss5[,2]==0))
poss_e <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss6[ which(poss6[,3]==1 & poss6[,2]==1), ]),
       nrow(poss6[ which(poss6[,3]==1 & poss6[,2]==0), ]))
n <- c(sum(poss6[,2]==1), sum(poss6[,2]==0))
poss_f <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss7[ which(poss7[,3]==1 & poss7[,2]==1), ]),
       nrow(poss7[ which(poss7[,3]==1 & poss7[,2]==0), ]))
n <- c(sum(poss7[,2]==1), sum(poss7[,2]==0))
poss_g <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss8[ which(poss8[,3]==1 & poss8[,2]==1), ]),
       nrow(poss8[ which(poss8[,3]==1 & poss8[,2]==0), ]))
n <- c(sum(poss8[,2]==1), sum(poss8[,2]==0))
poss_h <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss9[ which(poss9[,3]==1 & poss9[,2]==1), ]),
       nrow(poss9[ which(poss9[,3]==1 & poss9[,2]==0), ]))
n <- c(sum(poss9[,2]==1), sum(poss9[,2]==0))
poss_i <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss10[ which(poss10[,3]==1 & poss10[,2]==1), ]),
       nrow(poss10[ which(poss10[,3]==1 & poss10[,2]==0), ]))
n <- c(sum(poss10[,2]==1), sum(poss10[,2]==0))
poss_j <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss11[ which(poss11[,3]==1 & poss11[,2]==1), ]),
       nrow(poss11[ which(poss11[,3]==1 & poss11[,2]==0), ]))
n <- c(sum(poss11[,2]==1), sum(poss11[,2]==0))
poss_k <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss12[ which(poss12[,3]==1 & poss12[,2]==1), ]),
       nrow(poss12[ which(poss12[,3]==1 & poss12[,2]==0), ]))
n <- c(sum(poss12[,2]==1), sum(poss12[,2]==0))
poss_l <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss13[ which(poss13[,3]==1 & poss13[,2]==1), ]),
       nrow(poss13[ which(poss13[,3]==1 & poss13[,2]==0), ]))
n <- c(sum(poss13[,2]==1), sum(poss13[,2]==0))
poss_m <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss14[ which(poss14[,3]==1 & poss14[,2]==1), ]),
       nrow(poss14[ which(poss14[,3]==1 & poss14[,2]==0), ]))
n <- c(sum(poss14[,2]==1), sum(poss14[,2]==0))
poss_n <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss15[ which(poss15[,3]==1 & poss15[,2]==1), ]),
       nrow(poss15[ which(poss15[,3]==1 & poss15[,2]==0), ]))
n <- c(sum(poss15[,2]==1), sum(poss15[,2]==0))
poss_o <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss16[ which(poss16[,3]==1 & poss16[,2]==1), ]),
       nrow(poss16[ which(poss16[,3]==1 & poss16[,2]==0), ]))
n <- c(sum(poss16[,2]==1), sum(poss16[,2]==0))
poss_p <- prop.test(x, n, correct = FALSE)


# BUILD TABLE OF RESULTS
p_poss <- data.frame(
  "Autoimmune Disease" = c("Ankylosing Spondylitis", 
                           "Autoimmune Thyroid Disease",
                           "Coeliac",
                           "Hidradenitis Suppurativa",
                           "Inflammatory Bowel Disease",
                           "Myasthenia Gravis",
                           "Multiple Sclerosis",
                           "Pernicious Anemia",
                           "Pemphigoid/Pemphigus",
                           "Polymyalgia Rheumatica/GCA",
                           "Psoriatic Arthritis",
                           "Psoriasis",
                           "Rheumatoid Arthritis",
                           "Sjögren Syndrome",
                           "Systemic Lupus Erythematosus",
                           "Type 1 diabetes"),
    "P-value case-control difference (poss/any)" = c(poss_a$p.value,
                poss_b$p.value,
                poss_c$p.value,
                poss_d$p.value,
                poss_e$p.value,
                poss_f$p.value,
                poss_g$p.value,
                poss_h$p.value,
                poss_i$p.value,
                poss_j$p.value,
                poss_k$p.value,
                poss_l$p.value,
                poss_m$p.value,
                poss_n$p.value,
                poss_o$p.value,
                poss_p$p.value)
  )

proportions_poss <- as.data.frame(rbind(poss_a$estimate,
                poss_b$estimate,
                poss_c$estimate,
                poss_d$estimate,
                poss_e$estimate,
                poss_f$estimate,
                poss_g$estimate,
                poss_h$estimate,
                poss_i$estimate,
                poss_j$estimate,
                poss_k$estimate,
                poss_l$estimate,
                poss_m$estimate,
                poss_n$estimate,
                poss_o$estimate,
                poss_p$estimate)) 
colnames(proportions_poss) <- c("% Poss Cases in Any Dep","% Poss Cases Dep Controls")

summary_poss_auto_in_any_dep <- cbind(p_poss, proportions_poss)
summary_poss_auto_in_any_dep <- summary_poss_auto_in_any_dep[,c(1,3:4,2)]

```

```{r prepare_prevalence_poss_autoimmune_in_narrow_dep, include=FALSE}

# Test for differences in possible autoimmune prevalence between stringent depression cases and controls

# Read autoimmune and depression phenotype files
autoimmune <-fread("/file_path/new_autoimmune_all", head = T, data.table = F)
depression <- fread("/file_path/dep_broad_narrow", head = T, data.table = F)

# AUTOIMMUNE DISEASE - create df for each possible autoimmune disease, removing NA and merging with stringent depression

# Make df for each column of autoimmune df
for(i in seq(3, 33, 2)) {
  assign(paste("possible",i,sep="_"),autoimmune[,c(2,i)])
}

# Make list of dfs
MyList <- list(possible_3,possible_5,possible_7,possible_9,
               possible_11,possible_13,possible_15,possible_17,possible_19,
               possible_21,possible_23,possible_25,possible_27,possible_29,
               possible_31,possible_33)

# Remove NA
MyList <- lapply(MyList, na.omit)


# DEPRESSION - remove NA fields
dep_narrow <- depression[,c(1,25)] %>%
  na.omit() %>%
  rename(IID = ID)


# Merge with narrow dep
poss1 <- merge(dep_narrow, MyList[[1]], by = "IID")
poss2 <- merge(dep_narrow, MyList[[2]], by = "IID")
poss3 <- merge(dep_narrow, MyList[[3]], by = "IID")
poss4 <- merge(dep_narrow, MyList[[4]],  by = "IID")
poss5 <- merge(dep_narrow, MyList[[5]], by = "IID")
poss6 <- merge(dep_narrow, MyList[[6]], by = "IID")
poss7 <- merge(dep_narrow, MyList[[7]], by = "IID")
poss8 <- merge(dep_narrow, MyList[[8]], by = "IID")
poss9 <- merge(dep_narrow, MyList[[9]], by = "IID")
poss10 <- merge(dep_narrow, MyList[[10]], by = "IID")
poss11 <- merge(dep_narrow, MyList[[11]], by = "IID")
poss12 <- merge(dep_narrow, MyList[[12]], by = "IID")
poss13 <- merge(dep_narrow, MyList[[13]], by = "IID")
poss14 <- merge(dep_narrow, MyList[[14]], by = "IID")
poss15 <- merge(dep_narrow, MyList[[15]], by = "IID")
poss16 <- merge(dep_narrow, MyList[[16]], by = "IID")

# Run chi-square
x <- c(nrow(poss1[ which(poss1[,3]==1 & poss1[,2]==1), ]),
       nrow(poss1[ which(poss1[,3]==1 & poss1[,2]==0), ]))
n <- c(sum(poss1[,2]==1), sum(poss1[,2]==0))
poss_a <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss2[ which(poss2[,3]==1 & poss2[,2]==1), ]),
       nrow(poss2[ which(poss2[,3]==1 & poss2[,2]==0), ]))
n <- c(sum(poss2[,2]==1), sum(poss2[,2]==0))
poss_b <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss3[ which(poss3[,3]==1 & poss3[,2]==1), ]),
       nrow(poss3[ which(poss3[,3]==1 & poss3[,2]==0), ]))
n <- c(sum(poss3[,2]==1), sum(poss3[,2]==0))
poss_c <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss4[ which(poss4[,3]==1 & poss4[,2]==1), ]),
       nrow(poss4[ which(poss4[,3]==1 & poss4[,2]==0), ]))
n <- c(sum(poss4[,2]==1), sum(poss4[,2]==0))
poss_d <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss5[ which(poss5[,3]==1 & poss5[,2]==1), ]),
       nrow(poss5[ which(poss5[,3]==1 & poss5[,2]==0), ]))
n <- c(sum(poss5[,2]==1), sum(poss5[,2]==0))
poss_e <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss6[ which(poss6[,3]==1 & poss6[,2]==1), ]),
       nrow(poss6[ which(poss6[,3]==1 & poss6[,2]==0), ]))
n <- c(sum(poss6[,2]==1), sum(poss6[,2]==0))
poss_f <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss7[ which(poss7[,3]==1 & poss7[,2]==1), ]),
       nrow(poss7[ which(poss7[,3]==1 & poss7[,2]==0), ]))
n <- c(sum(poss7[,2]==1), sum(poss7[,2]==0))
poss_g <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss8[ which(poss8[,3]==1 & poss8[,2]==1), ]),
       nrow(poss8[ which(poss8[,3]==1 & poss8[,2]==0), ]))
n <- c(sum(poss8[,2]==1), sum(poss8[,2]==0))
poss_h <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss9[ which(poss9[,3]==1 & poss9[,2]==1), ]),
       nrow(poss9[ which(poss9[,3]==1 & poss9[,2]==0), ]))
n <- c(sum(poss9[,2]==1), sum(poss9[,2]==0))
poss_i <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss10[ which(poss10[,3]==1 & poss10[,2]==1), ]),
       nrow(poss10[ which(poss10[,3]==1 & poss10[,2]==0), ]))
n <- c(sum(poss10[,2]==1), sum(poss10[,2]==0))
poss_j <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss11[ which(poss11[,3]==1 & poss11[,2]==1), ]),
       nrow(poss11[ which(poss11[,3]==1 & poss11[,2]==0), ]))
n <- c(sum(poss11[,2]==1), sum(poss11[,2]==0))
poss_k <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss12[ which(poss12[,3]==1 & poss12[,2]==1), ]),
       nrow(poss12[ which(poss12[,3]==1 & poss12[,2]==0), ]))
n <- c(sum(poss12[,2]==1), sum(poss12[,2]==0))
poss_l <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss13[ which(poss13[,3]==1 & poss13[,2]==1), ]),
       nrow(poss13[ which(poss13[,3]==1 & poss13[,2]==0), ]))
n <- c(sum(poss13[,2]==1), sum(poss13[,2]==0))
poss_m <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss14[ which(poss14[,3]==1 & poss14[,2]==1), ]),
       nrow(poss14[ which(poss14[,3]==1 & poss14[,2]==0), ]))
n <- c(sum(poss14[,2]==1), sum(poss14[,2]==0))
poss_n <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss15[ which(poss15[,3]==1 & poss15[,2]==1), ]),
       nrow(poss15[ which(poss15[,3]==1 & poss15[,2]==0), ]))
n <- c(sum(poss15[,2]==1), sum(poss15[,2]==0))
poss_o <- prop.test(x, n, correct = FALSE)

x <- c(nrow(poss16[ which(poss16[,3]==1 & poss16[,2]==1), ]),
       nrow(poss16[ which(poss16[,3]==1 & poss16[,2]==0), ]))
n <- c(sum(poss16[,2]==1), sum(poss16[,2]==0))
poss_p <- prop.test(x, n, correct = FALSE)


# BUILD TABLE OF RESULTS
p_poss <- data.frame(
  "Autoimmune Disease" = c("Ankylosing Spondylitis", 
                           "Autoimmune Thyroid Disease",
                           "Coeliac",
                           "Hidradenitis Suppurativa",
                           "Inflammatory Bowel Disease",
                           "Myasthenia Gravis",
                           "Multiple Sclerosis",
                           "Pernicious Anemia",
                           "Pemphigoid/Pemphigus",
                           "Polymyalgia Rheumatica/GCA",
                           "Psoriatic Arthritis",
                           "Psoriasis",
                           "Rheumatoid Arthritis",
                           "Sjögren Syndrome",
                           "Systemic Lupus Erythematosus",
                           "Type 1 diabetes"),
    "P-value case-control difference (poss/stringent)" = c(poss_a$p.value,
                poss_b$p.value,
                poss_c$p.value,
                poss_d$p.value,
                poss_e$p.value,
                poss_f$p.value,
                poss_g$p.value,
                poss_h$p.value,
                poss_i$p.value,
                poss_j$p.value,
                poss_k$p.value,
                poss_l$p.value,
                poss_m$p.value,
                poss_n$p.value,
                poss_o$p.value,
                poss_p$p.value)
  )

proportions_poss <- as.data.frame(rbind(poss_a$estimate,
                poss_b$estimate,
                poss_c$estimate,
                poss_d$estimate,
                poss_e$estimate,
                poss_f$estimate,
                poss_g$estimate,
                poss_h$estimate,
                poss_i$estimate,
                poss_j$estimate,
                poss_k$estimate,
                poss_l$estimate,
                poss_m$estimate,
                poss_n$estimate,
                poss_o$estimate,
                poss_p$estimate)) 
colnames(proportions_poss) <- c("% Poss Cases in Stringent Dep","% Poss Cases Dep Controls")

summary_poss_auto_in_stringent_dep <- cbind(p_poss, proportions_poss)
summary_poss_auto_in_stringent_dep <- summary_poss_auto_in_stringent_dep[,c(1,3:4,2)]

```

```{r prepare_prevalence_prob_autoimmune_in_broad_dep, include=FALSE}

# Test for differences in probable autoimmune prevalence between any depression cases and controls

# Read autoimmune and depression phenotype files
autoimmune <-fread("/file_path/new_autoimmune_all", head = T, data.table = F)
depression <- fread("/file_path/dep_broad_narrow", head = T, data.table = F)

# AUTOIMMUNE DISEASE - create df for each probable autoimmune disease, removing NA and merging with broad depression

# Make df for each column of autoimmune df
for(i in seq(4, 34, 2)) {
  assign(paste("probable",i,sep="_"),autoimmune[,c(2,i)])
}

# Make list of dfs
MyList <- list(probable_4,probable_6,probable_8,probable_10,
               probable_12,probable_14,probable_16,probable_18,probable_20,
               probable_22,probable_24,probable_26,probable_28,probable_30,
               probable_32,probable_34)

# Remove NA
MyList <- lapply(MyList, na.omit)

# DEPRESSION - remove NA fields
dep_broad <- depression[,c(1,24)] %>%
  na.omit() %>%
  rename(IID = ID)


# Merge with broad dep
prob1 <- merge(dep_broad, MyList[[1]], by = "IID")
prob2 <- merge(dep_broad, MyList[[2]], by = "IID")
prob3 <- merge(dep_broad, MyList[[3]], by = "IID")
prob4 <- merge(dep_broad, MyList[[4]],  by = "IID")
prob5 <- merge(dep_broad, MyList[[5]], by = "IID")
prob6 <- merge(dep_broad, MyList[[6]], by = "IID")
prob7 <- merge(dep_broad, MyList[[7]], by = "IID")
prob8 <- merge(dep_broad, MyList[[8]], by = "IID")
prob9 <- merge(dep_broad, MyList[[9]], by = "IID")
prob10 <- merge(dep_broad, MyList[[10]], by = "IID")
prob11 <- merge(dep_broad, MyList[[11]], by = "IID")
prob12 <- merge(dep_broad, MyList[[12]], by = "IID")
prob13 <- merge(dep_broad, MyList[[13]], by = "IID")
prob14 <- merge(dep_broad, MyList[[14]], by = "IID")
prob15 <- merge(dep_broad, MyList[[15]], by = "IID")
prob16 <- merge(dep_broad, MyList[[16]], by = "IID")

# Run chi-square
x <- c(nrow(prob1[ which(prob1[,3]==1 & prob1[,2]==1), ]),
       nrow(prob1[ which(prob1[,3]==1 & prob1[,2]==0), ]))
n <- c(sum(prob1[,2]==1), sum(prob1[,2]==0))
prob_a <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob2[ which(prob2[,3]==1 & prob2[,2]==1), ]),
       nrow(prob2[ which(prob2[,3]==1 & prob2[,2]==0), ]))
n <- c(sum(prob2[,2]==1), sum(prob2[,2]==0))
prob_b <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob3[ which(prob3[,3]==1 & prob3[,2]==1), ]),
       nrow(prob3[ which(prob3[,3]==1 & prob3[,2]==0), ]))
n <- c(sum(prob3[,2]==1), sum(prob3[,2]==0))
prob_c <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob4[ which(prob4[,3]==1 & prob4[,2]==1), ]),
       nrow(prob4[ which(prob4[,3]==1 & prob4[,2]==0), ]))
n <- c(sum(prob4[,2]==1), sum(prob4[,2]==0))
prob_d <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob5[ which(prob5[,3]==1 & prob5[,2]==1), ]),
       nrow(prob5[ which(prob5[,3]==1 & prob5[,2]==0), ]))
n <- c(sum(prob5[,2]==1), sum(prob5[,2]==0))
prob_e <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob6[ which(prob6[,3]==1 & prob6[,2]==1), ]),
       nrow(prob6[ which(prob6[,3]==1 & prob6[,2]==0), ]))
n <- c(sum(prob6[,2]==1), sum(prob6[,2]==0))
prob_f <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob7[ which(prob7[,3]==1 & prob7[,2]==1), ]),
       nrow(prob7[ which(prob7[,3]==1 & prob7[,2]==0), ]))
n <- c(sum(prob7[,2]==1), sum(prob7[,2]==0))
prob_g <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob8[ which(prob8[,3]==1 & prob8[,2]==1), ]),
       nrow(prob8[ which(prob8[,3]==1 & prob8[,2]==0), ]))
n <- c(sum(prob8[,2]==1), sum(prob8[,2]==0))
prob_h <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob9[ which(prob9[,3]==1 & prob9[,2]==1), ]),
       nrow(prob9[ which(prob9[,3]==1 & prob9[,2]==0), ]))
n <- c(sum(prob9[,2]==1), sum(prob9[,2]==0))
prob_i <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob10[ which(prob10[,3]==1 & prob10[,2]==1), ]),
       nrow(prob10[ which(prob10[,3]==1 & prob10[,2]==0), ]))
n <- c(sum(prob10[,2]==1), sum(prob10[,2]==0))
prob_j <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob11[ which(prob11[,3]==1 & prob11[,2]==1), ]),
       nrow(prob11[ which(prob11[,3]==1 & prob11[,2]==0), ]))
n <- c(sum(prob11[,2]==1), sum(prob11[,2]==0))
prob_k <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob12[ which(prob12[,3]==1 & prob12[,2]==1), ]),
       nrow(prob12[ which(prob12[,3]==1 & prob12[,2]==0), ]))
n <- c(sum(prob12[,2]==1), sum(prob12[,2]==0))
prob_l <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob13[ which(prob13[,3]==1 & prob13[,2]==1), ]),
       nrow(prob13[ which(prob13[,3]==1 & prob13[,2]==0), ]))
n <- c(sum(prob13[,2]==1), sum(prob13[,2]==0))
prob_m <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob14[ which(prob14[,3]==1 & prob14[,2]==1), ]),
       nrow(prob14[ which(prob14[,3]==1 & prob14[,2]==0), ]))
n <- c(sum(prob14[,2]==1), sum(prob14[,2]==0))
prob_n <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob15[ which(prob15[,3]==1 & prob15[,2]==1), ]),
       nrow(prob15[ which(prob15[,3]==1 & prob15[,2]==0), ]))
n <- c(sum(prob15[,2]==1), sum(prob15[,2]==0))
prob_o <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob16[ which(prob16[,3]==1 & prob16[,2]==1), ]),
       nrow(prob16[ which(prob16[,3]==1 & prob16[,2]==0), ]))
n <- c(sum(prob16[,2]==1), sum(prob16[,2]==0))
prob_p <- prop.test(x, n, correct = FALSE)

# BUILD TABLE OF RESULTS
p_prob <- data.frame(
  "Autoimmune Disease" = c("Ankylosing Spondylitis", 
                           "Autoimmune Thyroid Disease",
                           "Coeliac",
                           "Hidradenitis Suppurativa",
                           "Inflammatory Bowel Disease",
                           "Myasthenia Gravis",
                           "Multiple Sclerosis",
                           "Pernicious Anemia",
                           "Pemphigoid/Pemphigus",
                           "Polymyalgia Rheumatica/GCA",
                           "Psoriatic Arthritis",
                           "Psoriasis",
                           "Rheumatoid Arthritis",
                           "Sjögren Syndrome",
                           "Systemic Lupus Erythematosus",
                           "Type 1 diabetes"),
   "P-value case-control difference (prob/broad)" = c(prob_a$p.value,
                prob_b$p.value,
                prob_c$p.value,
                prob_d$p.value,
                prob_e$p.value,
                prob_f$p.value,
                prob_g$p.value,
                prob_h$p.value,
                prob_i$p.value,
                prob_j$p.value,
                prob_k$p.value,
                prob_l$p.value,
                prob_m$p.value,
                prob_n$p.value,
                prob_o$p.value,
                prob_p$p.value)
  )

proportions_prob <- as.data.frame(rbind(prob_a$estimate,
                prob_b$estimate,
                prob_c$estimate,
                prob_d$estimate,
                prob_e$estimate,
                prob_f$estimate,
                prob_g$estimate,
                prob_h$estimate,
                prob_i$estimate,
                prob_j$estimate,
                prob_k$estimate,
                prob_l$estimate,
                prob_m$estimate,
                prob_n$estimate,
                prob_o$estimate,
                prob_p$estimate)) 
colnames(proportions_prob) <- c("% Prob Cases in Any Dep","% Prob Cases in Dep Controls")

summary_prob_auto_in_any_dep <- cbind(p_prob, proportions_prob)
summary_prob_auto_in_any_dep <- summary_prob_auto_in_any_dep[,c(1,3:4,2)]

```

```{r prepare_prevalence_prob_autoimmune_in_narrow_dep, include=FALSE}

# Test for differences in probable autoimmune prevalence between stringent depression cases and controls

# Read autoimmune and depression phenotype files
autoimmune <-fread("/file_path/new_autoimmune_all", head = T, data.table = F)
depression <- fread("/file_path/dep_broad_narrow", head = T, data.table = F)

# AUTOIMMUNE DISEASE - create df for each probable autoimmune disease, removing NA and merging with broad depression

# Make df for each column of autoimmune df
for(i in seq(4, 34, 2)) {
  assign(paste("probable",i,sep="_"),autoimmune[,c(2,i)])
}

# Make list of dfs
MyList <- list(probable_4,probable_6,probable_8,probable_10,
               probable_12,probable_14,probable_16,probable_18,probable_20,
               probable_22,probable_24,probable_26,probable_28,probable_30,
               probable_32,probable_34)

# Remove NA
MyList <- lapply(MyList, na.omit)

# DEPRESSION - remove NA fields
dep_narrow <- depression[,c(1,25)] %>%
  na.omit() %>%
  rename(IID = ID)


# Merge with narrow dep
prob1 <- merge(dep_narrow, MyList[[1]], by = "IID")
prob2 <- merge(dep_narrow, MyList[[2]], by = "IID")
prob3 <- merge(dep_narrow, MyList[[3]], by = "IID")
prob4 <- merge(dep_narrow, MyList[[4]],  by = "IID")
prob5 <- merge(dep_narrow, MyList[[5]], by = "IID")
prob6 <- merge(dep_narrow, MyList[[6]], by = "IID")
prob7 <- merge(dep_narrow, MyList[[7]], by = "IID")
prob8 <- merge(dep_narrow, MyList[[8]], by = "IID")
prob9 <- merge(dep_narrow, MyList[[9]], by = "IID")
prob10 <- merge(dep_narrow, MyList[[10]], by = "IID")
prob11 <- merge(dep_narrow, MyList[[11]], by = "IID")
prob12 <- merge(dep_narrow, MyList[[12]], by = "IID")
prob13 <- merge(dep_narrow, MyList[[13]], by = "IID")
prob14 <- merge(dep_narrow, MyList[[14]], by = "IID")
prob15 <- merge(dep_narrow, MyList[[15]], by = "IID")
prob16 <- merge(dep_narrow, MyList[[16]], by = "IID")

# Run chi-square
x <- c(nrow(prob1[ which(prob1[,3]==1 & prob1[,2]==1), ]),
       nrow(prob1[ which(prob1[,3]==1 & prob1[,2]==0), ]))
n <- c(sum(prob1[,2]==1), sum(prob1[,2]==0))
prob_a <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob2[ which(prob2[,3]==1 & prob2[,2]==1), ]),
       nrow(prob2[ which(prob2[,3]==1 & prob2[,2]==0), ]))
n <- c(sum(prob2[,2]==1), sum(prob2[,2]==0))
prob_b <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob3[ which(prob3[,3]==1 & prob3[,2]==1), ]),
       nrow(prob3[ which(prob3[,3]==1 & prob3[,2]==0), ]))
n <- c(sum(prob3[,2]==1), sum(prob3[,2]==0))
prob_c <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob4[ which(prob4[,3]==1 & prob4[,2]==1), ]),
       nrow(prob4[ which(prob4[,3]==1 & prob4[,2]==0), ]))
n <- c(sum(prob4[,2]==1), sum(prob4[,2]==0))
prob_d <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob5[ which(prob5[,3]==1 & prob5[,2]==1), ]),
       nrow(prob5[ which(prob5[,3]==1 & prob5[,2]==0), ]))
n <- c(sum(prob5[,2]==1), sum(prob5[,2]==0))
prob_e <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob6[ which(prob6[,3]==1 & prob6[,2]==1), ]),
       nrow(prob6[ which(prob6[,3]==1 & prob6[,2]==0), ]))
n <- c(sum(prob6[,2]==1), sum(prob6[,2]==0))
prob_f <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob7[ which(prob7[,3]==1 & prob7[,2]==1), ]),
       nrow(prob7[ which(prob7[,3]==1 & prob7[,2]==0), ]))
n <- c(sum(prob7[,2]==1), sum(prob7[,2]==0))
prob_g <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob8[ which(prob8[,3]==1 & prob8[,2]==1), ]),
       nrow(prob8[ which(prob8[,3]==1 & prob8[,2]==0), ]))
n <- c(sum(prob8[,2]==1), sum(prob8[,2]==0))
prob_h <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob9[ which(prob9[,3]==1 & prob9[,2]==1), ]),
       nrow(prob9[ which(prob9[,3]==1 & prob9[,2]==0), ]))
n <- c(sum(prob9[,2]==1), sum(prob9[,2]==0))
prob_i <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob10[ which(prob10[,3]==1 & prob10[,2]==1), ]),
       nrow(prob10[ which(prob10[,3]==1 & prob10[,2]==0), ]))
n <- c(sum(prob10[,2]==1), sum(prob10[,2]==0))
prob_j <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob11[ which(prob11[,3]==1 & prob11[,2]==1), ]),
       nrow(prob11[ which(prob11[,3]==1 & prob11[,2]==0), ]))
n <- c(sum(prob11[,2]==1), sum(prob11[,2]==0))
prob_k <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob12[ which(prob12[,3]==1 & prob12[,2]==1), ]),
       nrow(prob12[ which(prob12[,3]==1 & prob12[,2]==0), ]))
n <- c(sum(prob12[,2]==1), sum(prob12[,2]==0))
prob_l <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob13[ which(prob13[,3]==1 & prob13[,2]==1), ]),
       nrow(prob13[ which(prob13[,3]==1 & prob13[,2]==0), ]))
n <- c(sum(prob13[,2]==1), sum(prob13[,2]==0))
prob_m <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob14[ which(prob14[,3]==1 & prob14[,2]==1), ]),
       nrow(prob14[ which(prob14[,3]==1 & prob14[,2]==0), ]))
n <- c(sum(prob14[,2]==1), sum(prob14[,2]==0))
prob_n <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob15[ which(prob15[,3]==1 & prob15[,2]==1), ]),
       nrow(prob15[ which(prob15[,3]==1 & prob15[,2]==0), ]))
n <- c(sum(prob15[,2]==1), sum(prob15[,2]==0))
prob_o <- prop.test(x, n, correct = FALSE)

x <- c(nrow(prob16[ which(prob16[,3]==1 & prob16[,2]==1), ]),
       nrow(prob16[ which(prob16[,3]==1 & prob16[,2]==0), ]))
n <- c(sum(prob16[,2]==1), sum(prob16[,2]==0))
prob_p <- prop.test(x, n, correct = FALSE)

# BUILD TABLE OF RESULTS
p_prob <- data.frame(
  "Autoimmune Disease" = c("Ankylosing Spondylitis", 
                           "Autoimmune Thyroid Disease",
                           "Coeliac",
                           "Hidradenitis Suppurativa",
                           "Inflammatory Bowel Disease",
                           "Myasthenia Gravis",
                           "Multiple Sclerosis",
                           "Pernicious Anemia",
                           "Pemphigoid/Pemphigus",
                           "Polymyalgia Rheumatica/GCA",
                           "Psoriatic Arthritis",
                           "Psoriasis",
                           "Rheumatoid Arthritis",
                           "Sjögren Syndrome",
                           "Systemic Lupus Erythematosus",
                           "Type 1 diabetes"),
   "P-value case-control difference (prob/stringent)" = c(prob_a$p.value,
                prob_b$p.value,
                prob_c$p.value,
                prob_d$p.value,
                prob_e$p.value,
                prob_f$p.value,
                prob_g$p.value,
                prob_h$p.value,
                prob_i$p.value,
                prob_j$p.value,
                prob_k$p.value,
                prob_l$p.value,
                prob_m$p.value,
                prob_n$p.value,
                prob_o$p.value,
                prob_p$p.value)
  )

proportions_prob <- as.data.frame(rbind(prob_a$estimate,
                prob_b$estimate,
                prob_c$estimate,
                prob_d$estimate,
                prob_e$estimate,
                prob_f$estimate,
                prob_g$estimate,
                prob_h$estimate,
                prob_i$estimate,
                prob_j$estimate,
                prob_k$estimate,
                prob_l$estimate,
                prob_m$estimate,
                prob_n$estimate,
                prob_o$estimate,
                prob_p$estimate)) 
colnames(proportions_prob) <- c("% Prob Cases in Stringent Dep","% Prob Cases in Dep Controls")

summary_prob_auto_in_stringent_dep <- cbind(p_prob, proportions_prob)
summary_prob_auto_in_stringent_dep <- summary_prob_auto_in_stringent_dep[,c(1,3:4,2)]

```

**Table:** Prevalence of autoimmune diseases in depression cases and controls

```{r print_prevalence_table_auto_in_dep, echo=FALSE}

# Make summary tables of autoimmune prevalence in depression cases and controls

# Join tables
summary_auto_in_dep <- plyr::join_all(list(summary_poss_auto_in_any_dep, summary_poss_auto_in_stringent_dep, summary_prob_auto_in_any_dep, summary_prob_auto_in_stringent_dep), by = "Autoimmune.Disease")

# Format percentage
summary_auto_in_dep[,2] <- percent(round(summary_auto_in_dep[,2], digits = 4), digits = 2)
summary_auto_in_dep[,3] <- percent(round(summary_auto_in_dep[,3], digits = 4), digits = 2)
summary_auto_in_dep[,5] <- percent(round(summary_auto_in_dep[,5], digits = 4), digits = 2)
summary_auto_in_dep[,6] <- percent(round(summary_auto_in_dep[,6], digits = 4), digits = 2)
summary_auto_in_dep[,8] <- percent(round(summary_auto_in_dep[,8], digits = 4), digits = 2)
summary_auto_in_dep[,9] <- percent(round(summary_auto_in_dep[,9], digits = 4), digits = 2)
summary_auto_in_dep[,11] <- percent(round(summary_auto_in_dep[,11], digits = 4), digits = 2)
summary_auto_in_dep[,12] <- percent(round(summary_auto_in_dep[,12], digits = 4), digits = 2)

# Format p-values
summary_auto_in_dep[,4] <- formatC(summary_auto_in_dep[,4],format="e",digits = 0)
summary_auto_in_dep[,7] <- formatC(summary_auto_in_dep[,7],format="e",digits = 0)
summary_auto_in_dep[,10] <- formatC(summary_auto_in_dep[,10],format="e",digits = 0)
summary_auto_in_dep[,13] <- formatC(summary_auto_in_dep[,13],format="e",digits = 0)

# Remove Hidradenitis and Pemphigoid (removed due to low N)
summary_auto_in_dep <- summary_auto_in_dep[- grep("Hidradenitis", summary_auto_in_dep$Autoimmune.Disease),]
summary_auto_in_dep <- summary_auto_in_dep[- grep("Pemphigoid", summary_auto_in_dep$Autoimmune.Disease),]

# Order columns
summary_auto_in_dep$order <- c(9,2,6,7,5,4,1,10,11,8,12,13,14,3)
summary_auto_in_dep <- summary_auto_in_dep[order(summary_auto_in_dep$order),]
rownames(summary_auto_in_dep) <- NULL
summary_auto_in_dep <- summary_auto_in_dep[,1:13]

# Format column headings
summary_auto_in_dep <- summary_auto_in_dep %>%
  rename("P-value case-control difference (poss/any)" = P.value.case.control.difference..poss.any.,
         "P-value case-control difference (poss/stringent)" = P.value.case.control.difference..poss.stringent.,
        "P-value case-control difference (prob/any)" = P.value.case.control.difference..prob.broad.,
         "P-value case-control difference (prob/stringent)" = P.value.case.control.difference..prob.stringent.)


kable(summary_auto_in_dep, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%   
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>% 
  pack_rows("Blood", 1, 1, label_row_css = "background-color: #888; color: #fff;") %>%
  pack_rows("Endocrine system", 2, 3, label_row_css = "background-color: #888; color: #fff;") %>%
  pack_rows("Nervous System", 4, 5, label_row_css = "background-color: #888; color: #fff;") %>%
  pack_rows("Digestive system", 6, 7, label_row_css = "background-color: #888; color: #fff;") %>%
  pack_rows("Skin", 8, 8, label_row_css = "background-color: #888; color: #fff;") %>%
  pack_rows("Musculoskeletal system and connective tissue", 9, 14, label_row_css = "background-color: #888; color: #fff;")

```

</br>

## Prevalence summaries

</br>

```{r prepare_prev_summary_dep_prev_in_auto, echo=FALSE, include=FALSE}

# Prepare summaries of depression prevalence in autoimmune cases and controls 

# Create probable and possible autoimmune dfs
autoimmune <-fread("/file_path/new_autoimmune_all", head = T, data.table = F)

# Make columns for controls only, possible only and probable only, with NAs for blank cells
autoimmune$controls <- ifelse(rowSums(autoimmune[, 3:ncol(autoimmune)], na.rm = T)>0, 0, 1)
table(autoimmune$controls)

# remove columns with hidradenitis and pemphigoid (removed due to low N)
autoimmune <- autoimmune[,-c(9,10,19,20)]

autoimmune$possible_auto <- ifelse(rowSums(autoimmune[, grep("poss", names(autoimmune))], na.rm = T)>0, 1, NA)
table(autoimmune$possible_auto)

autoimmune$possible_auto <- ifelse(autoimmune$controls==1, 0, autoimmune$possible_auto)
table(autoimmune$possible_auto)
sum(is.na(autoimmune$possible_auto))

autoimmune$probable_auto <- ifelse(rowSums(autoimmune[, grep("prob", names(autoimmune))], na.rm = T)>0, 1, NA)
table(autoimmune$probable_auto)

autoimmune$probable_auto <- ifelse(autoimmune$controls==1, 0, autoimmune$probable_auto)
table(autoimmune$probable_auto)
sum(is.na(autoimmune$probable_auto))

# Make autoimmune possible and probable dfs

auto_prob <- autoimmune[,c(2,33)] %>%
  na.omit()
row.names(auto_prob) <- NULL

auto_poss <- autoimmune[,c(2,32)] %>%
  na.omit()
row.names(auto_poss) <- NULL

# Use any and stringent depression df from above, with NAs removed

### Depression prevalence in autoimmune cases and controls

# descriptives: stringent depression prevalence in probable auto cases vs auto controls 

narrow_probable <- merge(dep_narrow, auto_prob)
nrow(narrow_probable)

prob_auto_narrow_dep <- nrow(
  narrow_probable[ which(narrow_probable$probable_auto==1 & narrow_probable$Narrow==1), ])
prop_prob_auto_narrow_dep <- prob_auto_narrow_dep / sum(narrow_probable$probable_auto==1, na.rm = T)

no_auto_narrow_dep <- nrow(
  narrow_probable[ which(narrow_probable$probable_auto==0 & narrow_probable$Narrow==1), ])
prop_no_auto_narrow_dep <- no_auto_narrow_dep / sum(narrow_probable$probable_auto==0, na.rm = T)

# chi-square test: stringent depression prevalence in probable auto cases vs auto controls

x <- c(nrow(narrow_probable[ which(narrow_probable$Narrow==1 & narrow_probable$probable_auto==1), ]), # 1st col = number ppl with stringent dep WITH probable probable autoimmune 
       nrow(narrow_probable[ which(narrow_probable$Narrow==1 & narrow_probable$probable_auto==0), ])) # 2nd col = number ppl with stringent dep WITHOUT probable autoimmune
x

n <- c(sum(narrow_probable$probable_auto==1), # 1st col = number ppl WITH probable autoimmune
       sum(narrow_probable$probable_auto==0)) # 2nd col = number ppl WITHOUT probable autoimmune
n

chi_narrow_prob <- prop.test(x, n, correct = FALSE)
chi_narrow_prob

# descriptives: stringent depression prevalence in possible auto cases vs auto controls 

narrow_possible <- merge(dep_narrow, auto_poss)
nrow(narrow_possible)

poss_auto_narrow_dep <- nrow(
  narrow_possible[ which(narrow_possible$possible_auto==1 & narrow_possible$Narrow==1), ])
prop_poss_auto_narrow_dep <- poss_auto_narrow_dep / sum(narrow_possible$possible_auto==1, na.rm = T)

no_auto_narrow_dep <- nrow(
  narrow_possible[ which(narrow_possible$possible_auto==0 & narrow_possible$Narrow==1), ])
prop_no_auto_narrow_dep <- no_auto_narrow_dep / sum(narrow_possible$possible_auto==0, na.rm = T)

# chi-square test: stringent depression prevalence in possible auto cases vs auto controls

x <- c(nrow(narrow_possible[ which(narrow_possible$Narrow==1 & narrow_possible$possible_auto==1), ]), # 1st col = number ppl with stringent dep WITH possible autoimmune 
       nrow(narrow_possible[ which(narrow_possible$Narrow==1 & narrow_possible$possible_auto==0), ])) # 2nd col = number ppl with stringent dep WITHOUT possible autoimmune
x

n <- c(sum(narrow_possible$possible_auto==1), # 1st col = number ppl WITH possible autoimmune
       sum(narrow_possible$possible_auto==0)) # 2nd col = number ppl WITHOUT possible autoimmune
n

chi_narrow_poss <- prop.test(x, n, correct = FALSE)
chi_narrow_poss


# descriptives: any depression prevalence in probable auto cases vs auto controls 

broad_probable <- merge(dep_broad, auto_prob)
nrow(broad_probable)

prob_auto_broad_dep <- nrow(
  broad_probable[ which(broad_probable$probable_auto==1 & broad_probable$Broad==1), ])
prop_prob_auto_broad_dep <- prob_auto_broad_dep / sum(broad_probable$probable_auto==1, na.rm = T)


no_auto_broad_dep <- nrow(
  broad_probable[ which(broad_probable$probable_auto==0 & broad_probable$Broad==1), ])
prop_no_auto_broad_dep <- no_auto_broad_dep / sum(broad_probable$probable_auto==0, na.rm = T)

# chi-square test: any depression prevalence in probable auto cases vs auto controls

x <- c(nrow(broad_probable[ which(broad_probable$Broad==1 & broad_probable$probable_auto==1), ]), # 1st col = number ppl with any dep WITH probable autoimmune 
       nrow(broad_probable[ which(broad_probable$Broad==1 & broad_probable$probable_auto==0), ])) # 2nd col = number ppl with any dep WITHOUT probable autoimmune
x

n <- c(sum(broad_probable$probable_auto==1), # 1st col = number ppl WITH probable autoimmune
       sum(broad_probable$probable_auto==0)) # 2nd col = number ppl WITHOUT probable autoimmune
n

chi_broad_prob <- prop.test(x, n, correct = FALSE)
chi_broad_prob


# descriptives: any depression prevalence in possible auto cases vs auto controls 

broad_possible <- merge(dep_broad, auto_poss)
nrow(broad_possible)

poss_auto_broad_dep <- nrow(
  broad_possible[ which(broad_possible$possible_auto==1 & broad_possible$Broad==1), ])
prop_poss_auto_broad_dep <- poss_auto_broad_dep / sum(broad_possible$possible_auto==1, na.rm = T)


no_auto_broad_dep <- nrow(
  broad_possible[ which(broad_possible$possible_auto==0 & broad_possible$Broad==1), ])
prop_no_auto_broad_dep <- no_auto_broad_dep / sum(broad_possible$possible_auto==0, na.rm = T)

# chi-square test: any depression prevalence in possible auto cases vs auto controls

x <- c(nrow(broad_possible[ which(broad_possible$Broad==1 & broad_possible$possible_auto==1), ]), # 1st col = number ppl with any dep WITH possible autoimmune 
       nrow(broad_possible[ which(broad_possible$Broad==1 & broad_possible$possible_auto==0), ])) # 2nd col = number ppl with any dep WITHOUT possible autoimmune
x

n <- c(sum(broad_possible$possible_auto==1), # 1st col = number ppl WITH possible autoimmune
       sum(broad_possible$possible_auto==0)) # 2nd col = number ppl WITHOUT possible autoimmune
n

chi_broad_poss <- prop.test(x, n, correct = FALSE)
chi_broad_poss

```

**Table:** Summary: prevalence of depression in autoimmune cases and controls 

```{r print_prev_summary_dep_prev_in_auto_with_p_values, echo=FALSE}

# Summary: prevalence of depression in autoimmune cases and controls

### Make table
dep_in_auto_table1 <- data.frame(matrix(ncol=5, nrow=2))
colnames(dep_in_auto_table1) <- c("Prevalence of Depression in Probable Autoimmune Disease",
                                  "P-value Prevalence of Depression in Probable Autoimmune Disease vs Controls",
                                  "Prevalence of Depression in Possible Autoimmune Disease",
                                  "P-value Prevalence of Depression in Possible Autoimmune Disease vs Controls",
                                  "Prevalence of Depression in Autoimmune Controls")
rownames(dep_in_auto_table1) <- c("Stringent Depression","Any Depression")

dep_in_auto_table1[1,1] <- chi_narrow_prob$estimate[1]
dep_in_auto_table1[1,2] <- chi_narrow_prob$p.value
dep_in_auto_table1[1,3] <- chi_narrow_poss$estimate[1]
dep_in_auto_table1[1,4] <- chi_narrow_poss$p.value
dep_in_auto_table1[1,5] <- chi_narrow_poss$estimate[2]

dep_in_auto_table1[2,1] <- chi_broad_prob$estimate[1]
dep_in_auto_table1[2,2] <- chi_broad_prob$p.value
dep_in_auto_table1[2,3] <- chi_broad_poss$estimate[1]
dep_in_auto_table1[2,4] <- chi_broad_poss$p.value
dep_in_auto_table1[2,5] <- chi_broad_poss$estimate[2]

dep_in_auto_table1$`Prevalence of Depression in Probable Autoimmune Disease` <- percent(dep_in_auto_table1$`Prevalence of Depression in Probable Autoimmune Disease`, digits = 2)
dep_in_auto_table1$`P-value Prevalence of Depression in Probable Autoimmune Disease vs Controls` <- formatC(dep_in_auto_table1$`P-value Prevalence of Depression in Probable Autoimmune Disease vs Controls`, format = "e", digits = 2)
dep_in_auto_table1$`Prevalence of Depression in Possible Autoimmune Disease` <- percent(dep_in_auto_table1$`Prevalence of Depression in Possible Autoimmune Disease`, digits = 2)
dep_in_auto_table1$`P-value Prevalence of Depression in Possible Autoimmune Disease vs Controls` <- formatC(dep_in_auto_table1$`P-value Prevalence of Depression in Possible Autoimmune Disease vs Controls`, format = "e", digits = 2)
dep_in_auto_table1$`Prevalence of Depression in Autoimmune Controls` <- percent(dep_in_auto_table1$`Prevalence of Depression in Autoimmune Controls`, digits = 2)

kable(dep_in_auto_table1, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%   
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>%   
  column_spec(1, bold = TRUE, color = "white", background = "#666") 

```

</br>

```{r prepare_prev_summary_auto_prev_in_dep, echo=FALSE, include=FALSE}

### Autoimmune prevalence in depression cases and controls

# descriptives: probable autoimmune prevalence in stringent depression cases vs depression controls 

probable_narrow <- merge(auto_prob, dep_narrow)
nrow(probable_narrow)

narrow_dep_prob_auto <- nrow(
  probable_narrow[ which(probable_narrow$Narrow==1 & probable_narrow$probable_auto==1), ]) 
prop_narrow_dep_prob_auto <- narrow_dep_prob_auto / sum(probable_narrow$Narrow==1, na.rm = T) 

no_dep_prob_auto <- nrow(
  probable_narrow[ which(probable_narrow$Narrow==0 & probable_narrow$probable_auto==1), ]) 
prop_no_dep_prob_auto <- no_dep_prob_auto / sum(probable_narrow$Narrow==0) 

# chi-square test: probable auto prevalence in narrow depression and controls

x <- c(nrow(probable_narrow[ which(probable_narrow$probable_auto==1 & probable_narrow$Narrow==1), ]), # 1st col = number ppl with probable autoimmune WITH stringent dep 
       nrow(probable_narrow[ which(probable_narrow$probable_auto==1 & probable_narrow$Narrow==0), ])) # 2nd col = number ppl with probable autoimmune WITHOUT dep
x

n <- c(sum(probable_narrow$Narrow==1), # 1st col = number ppl WITH stringent dep
       sum(probable_narrow$Narrow==0)) # 2nd col = number ppl WITHOUT dep
n

chi_prob_narrow <- prop.test(x, n, correct = FALSE)
chi_prob_narrow


# descriptives: probable autoimmune prevalence in any depression cases vs depression controls 

probable_broad <- merge(auto_prob, dep_broad)
nrow(probable_broad)

broad_dep_prob_auto <- nrow(
  probable_broad[ which(probable_broad$Broad==1 & probable_broad$probable_auto==1), ]) 
prop_broad_dep_prob_auto <- broad_dep_prob_auto / sum(probable_broad$Broad==1, na.rm = T) 
no_dep_prob_auto <- nrow(
  probable_broad[ which(probable_broad$Broad==0 & probable_broad$probable_auto==1), ]) 
prop_no_dep_prob_auto <- no_dep_prob_auto / sum(probable_broad$Broad==0) 

# chi-square test: probable auto prevalence in broad depression and controls

x <- c(nrow(probable_broad[ which(probable_broad$probable_auto==1 & probable_broad$Broad==1), ]), # 1st col = number ppl with probable autoimmune WITH any dep 
       nrow(probable_broad[ which(probable_broad$probable_auto==1 & probable_broad$Broad==0), ])) # 2nd col = number ppl with probable autoimmune WITHOUT dep
x

n <- c(sum(probable_broad$Broad==1), # 1st col = number ppl WITH any dep
       sum(probable_broad$Broad==0)) # 2nd col = number ppl WITHOUT dep
n

chi_prob_broad <- prop.test(x, n, correct = FALSE)
chi_prob_broad


# descriptives: possible autoimmune prevalence in stringent depression cases vs depression controls 

possible_narrow <- merge(auto_poss, dep_narrow)
nrow(possible_narrow)

narrow_dep_poss_auto <- nrow(
  possible_narrow[ which(possible_narrow$Narrow==1 & possible_narrow$possible_auto==1), ]) 
prop_narrow_dep_poss_auto <- narrow_dep_poss_auto / sum(possible_narrow$Narrow==1, na.rm = T) 
no_dep_poss_auto <- nrow(
  possible_narrow[ which(possible_narrow$Narrow==0 & possible_narrow$possible_auto==1), ]) 
prop_no_dep_poss_auto <- no_dep_poss_auto / sum(possible_narrow$Narrow==0) 

# chi-square test: possible auto prevalence in narrow depression and controls

x <- c(nrow(possible_narrow[ which(possible_narrow$possible_auto==1 & possible_narrow$Narrow==1), ]), # 1st col = number ppl with possible autoimmune WITH stringent dep 
       nrow(possible_narrow[ which(possible_narrow$possible_auto==1 & possible_narrow$Narrow==0), ])) # 2nd col = number ppl with possible autoimmune WITHOUT dep
x

n <- c(sum(possible_narrow$Narrow==1), # 1st col = number ppl WITH narrow dep
       sum(possible_narrow$Narrow==0)) # 2nd col = number ppl WITHOUT dep
n

chi_poss_narrow <- prop.test(x, n, correct = FALSE)
chi_poss_narrow


# descriptives: possible autoimmune prevalence in any depression cases vs depression controls 

possible_broad <- merge(auto_poss, dep_broad)
nrow(possible_broad)

broad_dep_poss_auto <- nrow(
  possible_broad[ which(possible_broad$Broad==1 & possible_broad$possible_auto==1), ]) 
prop_broad_dep_poss_auto <- broad_dep_poss_auto / sum(possible_broad$Broad==1, na.rm = T) 
no_dep_poss_auto <- nrow(
  possible_broad[ which(possible_broad$Broad==0 & possible_broad$possible_auto==1), ]) 
prop_no_dep_poss_auto <- no_dep_poss_auto / sum(possible_broad$Broad==0) 

# chi-square test: possible auto prevalence in broad depression and controls

x <- c(nrow(possible_broad[ which(possible_broad$possible_auto==1 & possible_broad$Broad==1), ]), # 1st col = number ppl with possible autoimmune WITH any dep 
       nrow(possible_broad[ which(possible_broad$possible_auto==1 & possible_broad$Broad==0), ])) # 2nd col = number ppl with possible autoimmune WITHOUT dep
x

n <- c(sum(possible_broad$Broad==1), # 1st col = number ppl WITH any dep
       sum(possible_broad$Broad==0)) # 2nd col = number ppl WITHOUT dep
n

chi_poss_broad <- prop.test(x, n, correct = FALSE)
chi_poss_broad

```

**Table:** Summary: prevalence of autoimmune disease in depression cases and controls 

```{r print_prev_summary_auto_prev_in_dep_with_p_values, echo=FALSE}

### Make table of summary - autoimmune prevalence in depression

auto_in_dep_table1 <- data.frame(matrix(ncol=5, nrow=2))
colnames(auto_in_dep_table1) <- c("Prevalence of Autoimmune Disease in Narrow Depression",
                                 "P-value Prevalence of Autoimmune Disease in Narrow Depression vs Controls",
                                 "Prevalence of Autoimmune Disease in Broad Depression",
                                 "P-value Prevalence of Autoimmune Disease in Broad Depression vs Controls",
                                 "Prevalence of Autoimmune Disease in Depression Controls")
rownames(auto_in_dep_table1) <- c("Probable Autoimmune","Possible Autoimmune")

auto_in_dep_table1[1,1] <- chi_prob_narrow$estimate[1]
auto_in_dep_table1[1,2] <- chi_prob_narrow$p.value
auto_in_dep_table1[1,3] <- chi_prob_broad$estimate[1]
auto_in_dep_table1[1,4] <- chi_prob_broad$p.value
auto_in_dep_table1[1,5] <- chi_prob_broad$estimate[2]
auto_in_dep_table1[2,1] <- chi_poss_narrow$estimate[1]
auto_in_dep_table1[2,2] <- chi_poss_narrow$p.value
auto_in_dep_table1[2,3] <- chi_poss_broad$estimate[1]
auto_in_dep_table1[2,4] <- chi_poss_broad$p.value
auto_in_dep_table1[2,5] <- chi_poss_broad$estimate[2]

auto_in_dep_table1$`Prevalence of Autoimmune Disease in Narrow Depression` <- percent(auto_in_dep_table1$`Prevalence of Autoimmune Disease in Narrow Depression`, digits = 2)
auto_in_dep_table1$`P-value Prevalence of Autoimmune Disease in Narrow Depression vs Controls` <- formatC(auto_in_dep_table1$`P-value Prevalence of Autoimmune Disease in Narrow Depression vs Controls`, format = "e",  digits = 2)
auto_in_dep_table1$`Prevalence of Autoimmune Disease in Broad Depression` <- percent(auto_in_dep_table1$`Prevalence of Autoimmune Disease in Broad Depression`, digits = 2)
auto_in_dep_table1$`P-value Prevalence of Autoimmune Disease in Broad Depression vs Controls` <- formatC(auto_in_dep_table1$`P-value Prevalence of Autoimmune Disease in Broad Depression vs Controls`, format = "e",  digits = 2)
auto_in_dep_table1$`Prevalence of Autoimmune Disease in Depression Controls` <- percent(auto_in_dep_table1$`Prevalence of Autoimmune Disease in Depression Controls`, digits = 2)

kable(auto_in_dep_table1, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%   
  row_spec(0, bold = TRUE, color = "white", background = "#666") %>%   
  column_spec(1, bold = TRUE, color = "white", background = "#666") 

```

</br>

# Cross-trait polygenic score analyses

</br>

```{r make_pheno_files_by_sex, echo=FALSE, eval=FALSE}

# Make separate phenotype files by sex for PRS analyses

# Autoimmune 

# Read data
autoimmune <-fread("/file_path/new_autoimmune_all", head = T, data.table = F)
covariates <-fread("/file_path/covariates.txt", head = T, data.table = F)

# Merge with covariates
autoimmune_covar <- left_join(autoimmune, covariates)

# Split by sex
autoimmune_female <- autoimmune_covar %>% 
  filter(sex == 0) %>%
  select(1:34)

autoimmune_male <- autoimmune_covar %>% 
  filter(sex == 1) %>%
  select(1:34)

# Write files
write.table(autoimmune_female, "/file_path/autoimmune_female", row.names=F, col.name=T, quote=F)
write.table(autoimmune_male, "/file_path/autoimmune_male", row.names=F, col.name=T, quote=F)

# Depression. note - broad = any depression, narrow = stringent depression

# Read data
depression <- fread("/file_path/dep_broad_narrow", head = T, data.table = F) 
covariates <-fread("/file_path/covariates.txt", head = T, data.table = F)

# Merge
depression_covar <- left_join(depression, covariates, by = c("ID" = "IID"))

# Split by sex
depression_female <- depression_covar %>% 
  rename(IID = ID, Broad_female = Broad, Narrow_female = Narrow) %>%
  filter(sex == 0) %>%
  select(c(26,1,24,25))

depression_male <- depression_covar %>% 
  rename(IID = ID, Broad_male = Broad, Narrow_male = Narrow) %>%
  filter(sex == 1) %>%
  select(c(26,1,24,25))

# Cut required columns for all sex
depression_all <- depression_covar %>% 
  rename(IID = ID, Broad_all = Broad, Narrow_all = Narrow) %>%
  select(c(26,1,24,25)) 

# Combine
depression_gender <- plyr::join_all(list(depression_all,depression_female,depression_male), type = 'full')

# Write files
write.table(depression_gender, "/file_path/depression_gender", row.names=F, col.name=T, quote=F)

```

```{r prs_scripts_mdd_to_autoimmune, echo=FALSE, eval=FALSE}

# PRSice scripts for MDD PRS predicting autoimmune phenotypes

## PRS MDD – Autoimmune case/control status in combined sample of males and females. Split autoimmune phenotypes across two jobs. 

# Job 1
Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/DEPR06.gz \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/new_autoimmune_all \
--pheno-col ankylosing_poss,ankylosing_prob,thyroid_poss,thyroid_prob,coeliac_poss,coeliac_prob,hidradentis_poss,hidradentis_prob,ibd_poss,ibd_prob,myasthenia_poss,myasthenia_prob,ms_poss,ms_prob,pernecious_poss,pernecious_prob \
--binary-target T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.0055,0.0055,0.02,0.02,0.01,0.01,0.001,0.001,0.005,0.005,0.0002,0.0002,0.001,0.001,0.001,0.001 \
--quantile 10 \
--perm 10000 \
--out /file_path/mdd_prs_first

# Job 2
Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/DEPR06.gz \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/new_autoimmune_all \
--pheno-col pemphigoid_poss,pemphigoid_prob,polymyalgia_poss,polymyalgia_prob,psa_poss,psa_prob,psoriasis_poss,psoriasis_prob,arthritis_poss,arthritis_prob,sjogrens_poss,sjogrens_prob,sle_poss,sle_prob,t1d_poss,t1d_prob \
--binary-target T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.00005,0.00005,0.0085,0.0085,0.005,0.005,0.02,0.02,0.01,0.01,0.001,0.001,0.001,0.001,0.003,0.003 \
--quantile 10 \
--perm 10000 \
--out /file_path/mdd_prs_second


## PRS MDD – Autoimmune case/control status in females
Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/DEPR06.gz \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/autoimmune_female \
--pheno-col ankylosing_poss,ankylosing_prob,thyroid_poss,thyroid_prob,coeliac_poss,coeliac_prob,hidradentis_poss,hidradentis_prob,ibd_poss,ibd_prob,myasthenia_poss,myasthenia_prob,ms_poss,ms_prob,pernecious_poss,pernecious_prob,pemphigoid_poss,pemphigoid_prob,polymyalgia_poss,polymyalgia_prob,psa_poss,psa_prob,psoriasis_poss,psoriasis_prob,arthritis_poss,arthritis_prob,sjogrens_poss,sjogrens_prob,sle_poss,sle_prob,t1d_poss,t1d_prob \
--binary-target T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.0055,0.0055,0.02,0.02,0.01,0.01,0.001,0.001,0.005,0.005,0.0002,0.0002,0.001,0.001,0.001,0.001,0.00005,0.00005,0.0085,0.0085,0.005,0.005,0.02,0.02,0.01,0.01,0.001,0.001,0.001,0.001,0.003,0.003 \
--quantile 10 \
--out /file_path/female_mdd_prs

## PRS MDD – Autoimmune case/control status in males
Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/DEPR06.gz \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/autoimmune_male \
--pheno-col ankylosing_poss,ankylosing_prob,thyroid_poss,thyroid_prob,coeliac_poss,coeliac_prob,hidradentis_poss,hidradentis_prob,ibd_poss,ibd_prob,myasthenia_poss,myasthenia_prob,ms_poss,ms_prob,pernecious_poss,pernecious_prob,pemphigoid_poss,pemphigoid_prob,polymyalgia_poss,polymyalgia_prob,psa_poss,psa_prob,psoriasis_poss,psoriasis_prob,arthritis_poss,arthritis_prob,sjogrens_poss,sjogrens_prob,sle_poss,sle_prob,t1d_poss,t1d_prob \
--binary-target T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.0055,0.0055,0.02,0.02,0.01,0.01,0.001,0.001,0.005,0.005,0.0002,0.0002,0.001,0.001,0.001,0.001,0.00005,0.00005,0.0085,0.0085,0.005,0.005,0.02,0.02,0.01,0.01,0.001,0.001,0.001,0.001,0.003,0.003 \
--quantile 10 \
--out /file_path/male_mdd_prs

```

```{r prs_scripts_autoimmune_to_mdd, echo=FALSE, eval=FALSE}

# PRSice scripts for Autoimmune PRS predicting depression phenotypes (combined, males, females)

## PRS Celiac 
Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/CELI01.gz \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/depression_gender \
--pheno-col Broad_all,Narrow_all,Broad_female,Narrow_female,Broad_male,Narrow_male \
--binary-target T,T,T,T,T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.15,0.15,0.15,0.15,0.15,0.15 \
--quantile 10 \
--perm 10000 \
--out /file_path/coeliac_prs_to_mdd


## PRS IBD 
Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/INFB01.gz \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/depression_gender \
--pheno-col Broad_all,Narrow_all,Broad_female,Narrow_female,Broad_male,Narrow_male \
--binary-target T,T,T,T,T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.15,0.15,0.15,0.15,0.15,0.15 \
--quantile 10 \
--perm 10000 \
--out /file_path/ibd_prs_to_mdd


## PRS Myasthenia Gravis 
Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/myasthenia_gravis_sum_stats_New_AllResults.2_NoSex.txt \
--A1 ALLELE1 \
--A2 ALLELE2 \
--pvalue PVALUE \
--snp MARKER \
--stat OR \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/depression_gender \
--pheno-col Broad_all,Narrow_all,Broad_female,Narrow_female,Broad_male,Narrow_male \
--binary-target T,T,T,T,T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.15,0.15,0.15,0.15,0.15,0.15 \
--quantile 10 \
--perm 10000 \
--out /file_path/myasthenia_prs_to_mdd

## PRS Multiple Sclerosis 
Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/SCLE02.gz \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/depression_gender \
--pheno-col Broad_all,Narrow_all,Broad_female,Narrow_female,Broad_male,Narrow_male \
--binary-target T,T,T,T,T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.15,0.15,0.15,0.15,0.15,0.15 \
--quantile 10 \
--perm 10000 \
--out /file_path/ms_prs_to_mdd

## PRS Rheumatoid Arthritis 
Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/RHEU02.gz \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/depression_gender \
--pheno-col Broad_all,Narrow_all,Broad_female,Narrow_female,Broad_male,Narrow_male \
--binary-target T,T,T,T,T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.15,0.15,0.15,0.15,0.15,0.15 \
--quantile 10 \
--perm 10000 \
--out /file_path/ra_prs_to_mdd


## PRS Lupus 
Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/LUPU01.gz \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/depression_gender \
--pheno-col Broad_all,Narrow_all,Broad_female,Narrow_female,Broad_male,Narrow_male \
--binary-target T,T,T,T,T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.15,0.15,0.15,0.15,0.15,0.15 \
--quantile 10 \
--perm 10000 \
--out /file_path/lupus_prs_to_mdd

## PRS Psoriasis 
Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/metaanalysis_inversevariance_7datasets1_RS_ID.tbl \
--A1 Allele1 \
--A2 Allele2 \
--pvalue P.value \
--snp SNP \
--stat Effect \
--beta \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/depression_gender \
--pheno-col Broad_all,Narrow_all,Broad_female,Narrow_female,Broad_male,Narrow_male \
--binary-target T,T,T,T,T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.15,0.15,0.15,0.15,0.15,0.15 \
--quantile 10 \
--perm 10000 \
--out /file_path/psoriasis_prs_to_mdd


## PRS Psoriatic Arthritis
Rscript /file_path/PRSice_v2.3.1.e/PRSice.R \
--dir . \
--prsice /file_path/PRSice_v2.3.1.e/PRSice_linux \
--base /file_path/psa_sumstats.txt \
--target /file_path/binary_pre_qc \
--keep /file_path/post_qc_id_list.fam \
--extract /file_path/post_qc_snp_list.txt \
--pheno-file /file_path/depression_gender \
--pheno-col Broad_all,Narrow_all,Broad_female,Narrow_female,Broad_male,Narrow_male \
--binary-target T,T,T,T,T,T \
--fastscore \
--model add \
--score std \
--cov-file /file_path/covariates.txt \
--cov-col @PC[1-6],batch,assessment_centre \
--cov-factor batch,assessment_centre \
--x-range 6:28866528-33775446 \
--prevalence 0.15,0.15,0.15,0.15,0.15,0.15 \
--quantile 10 \
--perm 10000 \
--out /file_path/psa_prs_to_mdd

```

## PRS for MDD predicting autoimmune diseases

</br>

```{r download_results_mdd_autoimmune, include=FALSE}

# Download results for PRS MDD predicting autoimmune case/control

# Read in data
first <- as.data.frame(fread("/file_path/mdd_prs_first.summary", header = T, sep = "\t"))
second <- as.data.frame(fread("/file_path/mdd_prs_second.summary", header = T, sep = "\t"))
all <- rbind(first, second)
all$auto_pheno <- all$Phenotype
all$auto_disease <- all$Phenotype
all$Set <- rep("MDD")
all$auto_pheno <- gsub(".*_poss", "Possible (combined)", all$auto_pheno)
all$auto_pheno <- gsub(".*_prob", "Probable (combined)", all$auto_pheno)

female <- as.data.frame(fread("/file_path/female_mdd_prs.summary", header = T, sep = "\t"))
female$auto_pheno <- female$Phenotype
female$auto_disease <- female$Phenotype
female$Set <- rep("MDD")
female$auto_pheno <- gsub(".*_poss", "Possible (female)", female$auto_pheno)
female$auto_pheno <- gsub(".*_prob", "Probable (female)", female$auto_pheno)

male <- as.data.frame(fread("/file_path/male_mdd_prs.summary", header = T, sep = "\t"))
male$auto_pheno <- male$Phenotype
male$auto_disease <- male$Phenotype
male$Set <- rep("MDD")
male$auto_pheno <- gsub(".*_poss", "Possible (male)", male$auto_pheno)
male$auto_pheno <- gsub(".*_prob", "Probable (male)", male$auto_pheno)

mdd_to_autoimmune <- rbind(all, female, male)
mdd_to_autoimmune <- mdd_to_autoimmune[,c(14:15,1:13)]

mdd_to_autoimmune$auto_disease <- gsub("pernecious.*", "Pernicious\nAnemia", mdd_to_autoimmune$auto_disease)
mdd_to_autoimmune$auto_disease <- gsub("thyroid.*", "Autoimmune\nThyroid Disease", mdd_to_autoimmune$auto_disease)
mdd_to_autoimmune$auto_disease <- gsub("t1d.*", "Type 1\nDiabetes", mdd_to_autoimmune$auto_disease)
mdd_to_autoimmune$auto_disease <- gsub("ms.*", "Multiple\nSclerosis", mdd_to_autoimmune$auto_disease)
mdd_to_autoimmune$auto_disease <- gsub("myasthenia.*", "Myasthenia\nGravis", mdd_to_autoimmune$auto_disease)
mdd_to_autoimmune$auto_disease <- gsub("coeliac.*", "Coeliac", mdd_to_autoimmune$auto_disease)
mdd_to_autoimmune$auto_disease <- gsub("ibd.*", "Inflammatory\nBowel Disease", mdd_to_autoimmune$auto_disease)
mdd_to_autoimmune$auto_disease <- gsub("psoriasis.*", "Psoriasis", mdd_to_autoimmune$auto_disease)
mdd_to_autoimmune$auto_disease <- gsub("ankylosing.*", "Ankylosing\nSpondylitis", mdd_to_autoimmune$auto_disease)
mdd_to_autoimmune$auto_disease <- gsub("polymyalgia.*", "Polymyalgia\nRheumatica/GCA", mdd_to_autoimmune$auto_disease)
mdd_to_autoimmune$auto_disease <- gsub("psa.*", "Psoriatic\nArthritis", mdd_to_autoimmune$auto_disease)
mdd_to_autoimmune$auto_disease <- gsub("arthritis.*", "Rheumatoid\nArthritis", mdd_to_autoimmune$auto_disease)
mdd_to_autoimmune$auto_disease <- gsub("sjogrens.*", "Sjögren\nSyndrome", mdd_to_autoimmune$auto_disease)
mdd_to_autoimmune$auto_disease <- gsub("sle.*", "Systemic Lupus\nErythematosus", mdd_to_autoimmune$auto_disease)

mdd_to_autoimmune <- mdd_to_autoimmune[- grep("hidradentis", mdd_to_autoimmune$auto_disease),]
mdd_to_autoimmune <- mdd_to_autoimmune[- grep("pemphigoid", mdd_to_autoimmune$auto_disease),]

# Find p-values < 7.102273e-05
mdd_to_autoimmune %>% filter(P < 0.000071)

# Add column for significant findings 
mdd_to_autoimmune$sig_obs <- ifelse(mdd_to_autoimmune$P<0.000071, mdd_to_autoimmune$P, " ")
mdd_to_autoimmune$sig_obs <- as.numeric(mdd_to_autoimmune$sig_obs)
mdd_to_autoimmune$sig_obs <- formatC(mdd_to_autoimmune$sig_obs,format="e",digits = 0)
mdd_to_autoimmune$print.p <- sub("e", "*x*10^", mdd_to_autoimmune$sig_obs)
mdd_to_autoimmune$asterix <- ifelse(mdd_to_autoimmune$P<0.000071, "*", " ")

mdd_to_autoimmune <- mdd_to_autoimmune[order(mdd_to_autoimmune$auto_disease, mdd_to_autoimmune$auto_pheno),] 

mdd_to_autoimmune1 <- mdd_to_autoimmune[1:42,]
mdd_to_autoimmune2 <- mdd_to_autoimmune[43:84,]

```

```{r plot_mdd_to_autoimmune1, echo=FALSE, results = 'asis', fig.align='center', fig.height=4, fig.width=14}

## Plot results for first set of autoimmune traits

Figure1 <- ggplot(data = mdd_to_autoimmune1, aes(x = as.factor(auto_disease), y = PRS.R2.adj, fill = as.factor(auto_pheno))) +
  geom_bar(stat = "identity", position = position_dodge(), width = 0.8) +
  scale_fill_manual(values = c("dodgerblue4", "cadetblue3", "cadetblue1", "red4", "indianred3", "indianred1"), name = "Autoimmune\nPhenotype\n(target trait)") +
  geom_text(aes(label=paste(asterix)), position=position_dodge(width=0.8), 
                             vjust=0.3, hjust=0.5, size = 10, angle = 0) +
  ylab(expression(paste("R"^"2"," ","liability"))) +
  xlab("\nAutoimmune Disease") +
  theme(axis.text=element_text(size=14),
        axis.title=element_text(size=16)) +
  theme(panel.grid.minor.x=element_blank(), 
                         panel.grid.major.x=element_blank(), 
                         panel.grid.minor.y=element_blank(), 
                         panel.grid.major.y=element_blank(),
                         panel.background = element_rect(fill = "white"),
                         axis.line.y = element_line(color="grey"),
                         axis.line.x = element_line(color="grey"),
                        legend.title = element_text(size = 14),
                         legend.text = element_text(size = 12),
                         plot.title = element_text(hjust = 0.5, margin = margin(b=25), size = 18)) + 
  scale_y_continuous(limits=c(0,0.01), expand=c(0,0))

```

```{r print_plot_mdd_to_autoimmune1, echo=FALSE, results = 'asis', fig.align='center', fig.height=4, fig.width=14}

Figure1

```

**Figure:**  Polygenic scores for MDD predicting case/control status for autoimmune diseases (x-axis). Asterisks denote p-value < 7.1x10-5 (bonferroni corrected significance threshold). 

</br>

```{r plot_mdd_to_autoimmune2, echo=FALSE, results = 'asis', fig.align='center', fig.height=4, fig.width=14}

## Plot results for second set of autoimmune traits
Figure2 <- ggplot(data = mdd_to_autoimmune2, aes(x = as.factor(auto_disease), y = PRS.R2.adj, fill = as.factor(auto_pheno))) +
  geom_bar(stat = "identity", position = position_dodge(), width = 0.8) +
  scale_fill_manual(values = c("dodgerblue4", "cadetblue3", "cadetblue1", "red4", "indianred3", "indianred1"), name = "Autoimmune\nPhenotype\n(target trait)") +
  geom_text(aes(label=paste(asterix)), position=position_dodge(width=0.8), 
                             vjust=0.3, hjust=0.5, size = 10, angle = 0) +
  ylab(expression(paste("R"^"2"," ","liability"))) +
  xlab("\nAutoimmune Disease") +
  theme(axis.text=element_text(size=14),
        axis.title=element_text(size=16)) +
  theme(panel.grid.minor.x=element_blank(), 
                         panel.grid.major.x=element_blank(), 
                         panel.grid.minor.y=element_blank(), 
                         panel.grid.major.y=element_blank(),
                         panel.background = element_rect(fill = "white"),
                         axis.line.y = element_line(color="grey"),
                         axis.line.x = element_line(color="grey"),
                        legend.title = element_text(size = 14),
                         legend.text = element_text(size = 12),
                         plot.title = element_text(hjust = 0.5, margin = margin(b=25), size = 18)) + 
  scale_y_continuous(limits=c(0,0.01), expand=c(0,0))

```

```{r print_plot_mdd_to_autoimmune2, echo=FALSE, results = 'asis', fig.align='center', fig.height=4, fig.width=14}

Figure2

```

**Figure:**  Polygenic scores for MDD predicting case/control status for autoimmune diseases (x-axis). Asterisks denote p-value < 7.1x10-5 (bonferroni corrected significance threshold). 

</br>

**Table:** MDD PRS predicting autoimmune case/control status.

```{r print_prs_table_mdd_to_autoimmune, echo=FALSE, results = 'asis'}

# Make table with full set of results for MDD PRS predicting autoimmune case/control

mdd_to_autoimmune$Base <- "Major Depressive Disorder"
mdd_to_autoimmune$Target <- paste0(mdd_to_autoimmune$auto_disease, ", ", mdd_to_autoimmune$auto_pheno) 
mdd_to_autoimmune$Target <- gsub("\n", " ", mdd_to_autoimmune$Target)
rownames(mdd_to_autoimmune) <- NULL

mdd_to_autoimmune_table <- mdd_to_autoimmune[, c(19,20,5,14,10,6,7,11:13)]

mdd_to_autoimmune_table$Prevalence <- percent(mdd_to_autoimmune_table$Prevalence, digits = 2)
mdd_to_autoimmune_table$PRS.R2.adj <- percent(mdd_to_autoimmune_table$PRS.R2.adj, digits = 3)
mdd_to_autoimmune_table$PRS.R2 <- percent(mdd_to_autoimmune_table$PRS.R2, digits = 3)
mdd_to_autoimmune_table$Coefficient <- round(mdd_to_autoimmune_table$Coefficient, digits = 3)
mdd_to_autoimmune_table$Standard.Error <- round(mdd_to_autoimmune_table$Standard.Error, digits = 3)
mdd_to_autoimmune_table$P <- formatC(mdd_to_autoimmune_table$P, format = "e", digits = 1)

# write tables
write.table(mdd_to_autoimmune_table, "/file_path/mdd_to_autoimmune_table", row.names=F, col.name=T, quote=F, sep = "\t")

kable(mdd_to_autoimmune_table, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666", align = "centre")

```

</br>

### PRS at all thresholds (for significant associations)

</br

```{r prepare_plot_all_thresholds_mdd_to_auto, echo=FALSE, results = 'asis', fig.align='center', fig.height=6, fig.width=18}

# Create plots with PRS at all p-value thresholds for significant results in MDD PRS predicting autoimmune case/control

# Read and format data
all1 <- read.table("/file_path/mdd_prs_first.prsice", header = T)
all2 <- read.table("/file_path/mdd_prs_second.prsice", header = T) 
combined <- rbind(all1, all2)
combined$Group <- "(combined)"

female <- read.table("/file_path/female_mdd_prs.prsice", header = T) 
female$Group <- "(female)"

male <- read.table("/file_path/male_mdd_prs.prsice", header = T) 
male$Group <- "(male)"

all <- rbind(combined, female, male)
all$Base <- "MDD"

all$Type <-  gsub(".*_poss", "Possible", all$Pheno)
all$Type <-  gsub(".*_prob", "Probable", all$Type)

all$Disease <- all$Pheno
all$Disease <- gsub("pernecious.*", "Pernicious Anemia", all$Disease)
all$Disease <- gsub("thyroid.*", "Autoimmune Thyroid Disease", all$Disease)
all$Disease <- gsub("t1d.*", "Type 1 Diabetes", all$Disease)
all$Disease <- gsub("ms.*", "Multiple Sclerosis", all$Disease)
all$Disease <- gsub("myasthenia.*", "Myasthenia Gravis", all$Disease)
all$Disease <- gsub("coeliac.*", "Coeliac", all$Disease)
all$Disease <- gsub("ibd.*", "Inflammatory Bowel Disease", all$Disease)
all$Disease <- gsub("psoriasis.*", "Psoriasis", all$Disease)
all$Disease <- gsub("ankylosing.*", "Ankylosing Spondylitis", all$Disease)
all$Disease <- gsub("polymyalgia.*", "Polymyalgia Rheumatica/GCA", all$Disease)
all$Disease <- gsub("psa.*", "Psoriatic Arthritis", all$Disease)
all$Disease <- gsub("arthritis.*", "Rheumatoid Arthritis", all$Disease)
all$Disease <- gsub("sjogrens.*", "Sjögren Syndrome", all$Disease)
all$Disease <- gsub("sle.*", "Systemic Lupus Erythematosus", all$Disease)

all$Title <- paste("PRS for", all$Base, "predicting", all$Type, all$Disease, all$Group)

all <- all[ ,c(14,1:13)]
all$print.p <- formatC(all$P, format="e",digits = 1)
all$print.p <- sub("e", "*x*10^", all$print.p)

one <- all %>%
  filter(Title == "PRS for MDD predicting Possible Coeliac (combined)" | Title == "PRS for MDD predicting Possible Coeliac (female)") 

two <- all %>%
  filter(Title == "PRS for MDD predicting Possible Inflammatory Bowel Disease (combined)" | Title == "PRS for MDD predicting Possible Inflammatory Bowel Disease (female)") 

three <- all %>%
  filter(Title == "PRS for MDD predicting Possible Psoriasis (combined)" | Title == "PRS for MDD predicting Possible Psoriasis (male)") 

four <- all %>%
  filter(Title == "PRS for MDD predicting Probable Psoriasis (combined)" | Title == "PRS for MDD predicting Probable Psoriatic Arthritis (combined)") 
four <- four[order(four$Title), ]

five <- all %>%
  filter(Title == "PRS for MDD predicting Possible Rheumatoid Arthritis (combined)" | Title == "PRS for MDD predicting Possible Rheumatoid Arthritis (female)") 

six <- all %>%
  filter(Title == "PRS for MDD predicting Possible Rheumatoid Arthritis (male)" | Title == "PRS for MDD predicting Probable Rheumatoid Arthritis (combined)") 
six <- six[order(six$Title), ]

seven <- all %>%
  filter(Title == "PRS for MDD predicting Probable Rheumatoid Arthritis (female)" | Title == "PRS for MDD predicting Possible Type 1 Diabetes (combined)") 
#seven <- seven[order(seven$Pheno), ]
seven$Title <- factor(seven$Title, levels = c("PRS for MDD predicting Probable Rheumatoid Arthritis (female)","PRS for MDD predicting Possible Type 1 Diabetes (combined)"))

eight <- all %>%
  filter(Title == "PRS for MDD predicting Possible Type 1 Diabetes (female)" | Title == "PRS for MDD predicting Probable Type 1 Diabetes (combined)") 
eight <- eight[order(eight$Title), ]

# Make list of dfs
thresholds <- list(one, two, three, four, five, six, seven, eight)

# Create plot function
threshold_plot <- function(x) {
  ggplot(x, aes(x = as.factor(Threshold), y = R2, fill = -log10(P))) +
  geom_bar(stat = "identity", position = position_dodge()) +
  scale_fill_gradient2(low = "dodgerblue", high = "firebrick", mid = "dodgerblue", midpoint=1e-4,
                       name = bquote(atop(-log[10] ~ model, italic(P) - value),)) +
  ylab(expression(paste("PRS model fit:  ", R ^ 2, sep = " "))) +
  xlab(expression(italic(P) - value ~ threshold ~ (italic(P)[T]))) +
  theme(axis.text.x = element_text(angle=45, hjust=1, size = 12),
        axis.text.y = element_text(size = 12),
        axis.title = element_text(size = 14),
        plot.title = element_text(hjust = 0.5, size = 18),
        strip.text.x = element_text(size = 12, face = "bold"),
        panel.grid = element_blank()) +
  scale_y_continuous(expand = expand_scale(mult = c(0, .3))) +
  geom_text(aes(label = paste(print.p)), vjust = -1.5, hjust = 0, angle = 45, cex = 3.5, parse = T) +
  facet_grid(~ Title)
}  

one_plot <- threshold_plot(one)
two_plot <- threshold_plot(two)
three_plot <- threshold_plot(three)
four_plot <- threshold_plot(four)
five_plot <- threshold_plot(five)
six_plot <- threshold_plot(six)
seven_plot <- threshold_plot(seven)
eight_plot <- threshold_plot(eight)

```

</br>

```{r print_thresholds_mdd_to_auto1, echo=FALSE, results = 'asis', fig.align='center', fig.height=20, fig.width=15}

grid.arrange(one_plot, two_plot, three_plot, four_plot, ncol = 1)

```

```{r print_thresholds_mdd_to_auto2, echo=FALSE, results = 'asis', fig.align='center', fig.height=20, fig.width=15}

grid.arrange(five_plot, six_plot, seven_plot, eight_plot, ncol = 1)

```

</br>

### Sex interaction analyses

</br>

```{r sex_interactions, echo=FALSE, eval=FALSE}

# For significant sex-stratified associations, test for an interaction between sex and PRS

# read covariates and autoimmune phenotypes and join
covary <- read.table("/file_path/covariates.txt", header = T)
auto_pheno <- read.table("/file_path/new_autoimmune_all", header = T)
auto_covary <- left_join(auto_pheno, covary)

# read PRS at best p-value threshold from sex-stratified analysis. Join with pheno/covariates file. 
coeliac_possible_female <- read.table("/file_path/female_mdd_prs.coeliac_poss.best", header = T)
coeliac_possible_female <- left_join(auto_covary, coeliac_possible_female)

ibd_possible_female <- read.table("/file_path/female_mdd_prs.ibd_poss.best", header = T)
ibd_possible_female <- left_join(auto_covary, ibd_possible_female)

psoriasis_possible_male <- read.table("/file_path/male_mdd_prs.psoriasis_poss.best", header = T)
psoriasis_possible_male <- left_join(auto_covary, psoriasis_possible_male)

ra_possible_female <- read.table("/file_path/female_mdd_prs.arthritis_poss.best", header = T)
ra_possible_female <- left_join(auto_covary, ra_possible_female)

ra_possible_male <- read.table("/file_path/male_mdd_prs.arthritis_poss.best", header = T)
ra_possible_male <- left_join(auto_covary, ra_possible_male)

ra_probable_female <- read.table("/file_path/female_mdd_prs.arthritis_prob.best", header = T)
ra_probable_female <- left_join(auto_covary, ra_probable_female)

t1d_possible_female <- read.table("/file_path/female_mdd_prs.t1d_poss.best", header = T)
t1d_possible_female <- left_join(auto_covary, t1d_possible_female)

# Test for sex interaction
coeliac_poss_fem <- with(coeliac_possible_female, glm(coeliac_poss ~ sex + PRS + sex * PRS + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + as.factor(batch) + as.factor(assessment_centre), family = binomial))

ibd_poss_fem <- with(ibd_possible_female, glm(ibd_poss ~ sex + PRS + sex * PRS + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + as.factor(batch) + as.factor(assessment_centre), family = binomial))

psoriasis_poss_male <- with(psoriasis_possible_male, glm(psoriasis_poss ~ sex + PRS + sex * PRS + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + as.factor(batch) + as.factor(assessment_centre), family = binomial))

ra_poss_fem <- with(ra_possible_female, glm(arthritis_poss ~ sex + PRS + sex * PRS + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + as.factor(batch) + as.factor(assessment_centre), family = binomial))

ra_poss_male <- with(ra_possible_male, glm(arthritis_poss ~ sex + PRS + sex * PRS + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + as.factor(batch) + as.factor(assessment_centre), family = binomial))

ra_prob_fem <- with(ra_probable_female, glm(arthritis_prob ~ sex + PRS + sex * PRS + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + as.factor(batch) + as.factor(assessment_centre), family = binomial))

t1d_poss_fem <- with(t1d_possible_female, glm(t1d_poss ~ sex + PRS + sex * PRS + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + as.factor(batch) + as.factor(assessment_centre), family = binomial))

# Create table with results
results <- function(x){ 
  interaction <- data.frame(
            Sex = coef(summary(x))[2,4], 
            PRS = coef(summary(x))[3,4], 
            Sex_by_PRS = coef(summary(x))[136,4])
}

# Read table with mdd to autoimmune PRS results and join wiht interaction results. Note % female for each disorder. 

mdd_to_autoimmune_table <- as.data.frame(fread("/file_path/mdd_to_autoimmune_table", header = T, sep = "\t"))

coeliac_possfem <- results(coeliac_poss_fem)
coeliac_possfem <- cbind(mdd_to_autoimmune_table[mdd_to_autoimmune_table$Target=="Coeliac, Possible (female)",c(1:3,10)], coeliac_possfem)
coeliac_possfem$`Female (%)` <- "67%"

ibd_possfem <- results(ibd_poss_fem)
ibd_possfem <- cbind(mdd_to_autoimmune_table[mdd_to_autoimmune_table$Target=="Inflammatory Bowel Disease, Possible (female)",c(1:3,10)], ibd_possfem)
ibd_possfem$`Female (%)` <- "51%"

psoriasis_possmale <- results(psoriasis_poss_male)
psoriasis_possmale <- cbind(mdd_to_autoimmune_table[mdd_to_autoimmune_table$Target=="Psoriasis, Possible (male)",c(1:3,10)], psoriasis_possmale)
psoriasis_possmale$`Female (%)` <- "46%"

ra_possfem <- results(ra_poss_fem)
ra_possfem <- cbind(mdd_to_autoimmune_table[mdd_to_autoimmune_table$Target=="Rheumatoid Arthritis, Possible (female)",c(1:3,10)], ra_possfem)
ra_possfem$`Female (%)` <- "67%"

ra_possmale <- results(ra_poss_male)
ra_possmale <- cbind(mdd_to_autoimmune_table[mdd_to_autoimmune_table$Target=="Rheumatoid Arthritis, Possible (male)",c(1:3,10)], ra_possmale)
ra_possmale$`Female (%)` <- "67%"

ra_probfem <- results(ra_prob_fem)
ra_probfem <- cbind(mdd_to_autoimmune_table[mdd_to_autoimmune_table$Target=="Rheumatoid Arthritis, Probable (female)",c(1:3,10)], ra_probfem)
ra_probfem$`Female (%)` <- "70%"

t1d_possfem <- results(t1d_poss_fem)
t1d_possfem <- cbind(mdd_to_autoimmune_table[mdd_to_autoimmune_table$Target=="Type 1 Diabetes, Possible (female)",c(1:3,10)], t1d_possfem)
t1d_possfem$`Female (%)` <- "42%"

# Make one table for all interaction results
interaction <- rbind(coeliac_possfem,ibd_possfem,psoriasis_possmale,ra_possfem,ra_possmale,ra_probfem,t1d_possfem)
rownames(interaction) <- NULL
interaction$Sex <- formatC(interaction$Sex, format = "e", digits = 1)
interaction$PRS <- formatC(interaction$PRS, format = "e", digits = 1)
interaction$Sex_by_PRS <- round(interaction$Sex_by_PRS, digits = 2)
interaction$Combined_pT <- c("0.001","0.3","0.5","0.5","0.5","0.1","0.4")
interaction$P <- as.numeric(interaction$P)

# write tables
write.table(interaction, "/file_path/interaction", row.names=F, col.name=T, quote=F, sep = "\t")

```

**Table:** Sex-interaction results. Columns 1:4 = data from PRS *sex-stratified analyses*. Columns 5:7 = p-values from *sex interaction tests*; regressed PRS at the best threshold from sex-stratified analyses (column 4) on *combined* sample (men and women). Columns 8:9 = % female for each disorder and best threshold from combined analysis. 

</br> 

```{r print_sex_interaction, eval=TRUE, echo=FALSE, message=FALSE}

# read table with mdd to autoimmune PRS and interaction results
interaction <- as.data.frame(fread("/file_path/interaction", header = T, sep = "\t"))
interaction$Sex <- formatC(interaction$Sex, format = "e", digits = 1)

kable(interaction, booktabs = T) %>%
kable_styling() %>%
add_header_above(c("Significant results from original sex-stratified analyses" = 4, "P-values for sex, PRS and sex*PRS in combined (men & women) analyses" = 3, "General Info" = 2))

```

## PRS for autoimmune diseases predicting depression

</br>

```{r download_results_autoimmune_to_mdd, include=FALSE}

# Download results from autoimmune PRS predicting depression case/control status

# Read in data
a <- as.data.frame(fread("/file_path/psa_prs_to_mdd.summary", header = T, sep = "\t"))
a$Set <- "Psoriatic\nArthritis"

b <- as.data.frame(fread("/file_path/coeliac_prs_to_mdd.summary", header = T, sep = "\t"))
b$Set <- "Coeliac"

c <- as.data.frame(fread("/file_path/ibd_prs_to_mdd.summary", header = T, sep = "\t"))
c$Set <- "Inflammatory\nBowel Disease"

d <- as.data.frame(fread("/file_path/lupus_prs_to_mdd.summary", header = T, sep = "\t"))
d$Set <- "Systemic Lupus\nErythematosus"

e <- as.data.frame(fread("/file_path/ms_prs_to_mdd.summary", header = T, sep = "\t"))
e$Set <- "Multiple\nSclerosis"

f <- as.data.frame(fread("/file_path/myasthenia_prs_to_mdd.summary", header = T, sep = "\t"))
f$Set <- "Myasthenia\nGravis"

g <- as.data.frame(fread("/file_path/psoriasis_prs_to_mdd.summary", header = T, sep = "\t"))
g$Set <- "Psoriasis"

h <- as.data.frame(fread("/file_path/ra_prs_to_mdd.summary", header = T, sep = "\t"))
h$Set <- "Rheumatoid\nArthritis"

# combine results
autoimmune_to_mdd <- rbind(a,b,c,d,e,f,g,h)
autoimmune_to_mdd$Phenotype <- gsub("Broad_all", "Any (combined)",autoimmune_to_mdd$Phenotype)
autoimmune_to_mdd$Phenotype <- gsub("Narrow_all", "Stringent (combined)",autoimmune_to_mdd$Phenotype)
autoimmune_to_mdd$Phenotype <- gsub("Broad_female", "Any (female)",autoimmune_to_mdd$Phenotype)
autoimmune_to_mdd$Phenotype <- gsub("Narrow_female", "Stringent (female)",autoimmune_to_mdd$Phenotype)
autoimmune_to_mdd$Phenotype <- gsub("Broad_male", "Any (male)",autoimmune_to_mdd$Phenotype)
autoimmune_to_mdd$Phenotype <- gsub("Narrow_male", "Stringent (male)",autoimmune_to_mdd$Phenotype)

# Find p-values < 7.102273e-05 
autoimmune_to_mdd %>% filter(P < 0.000071)

# Add column for significant findings 
autoimmune_to_mdd$sig_obs <- ifelse(autoimmune_to_mdd$P<0.000071, autoimmune_to_mdd$P, " ")
autoimmune_to_mdd$sig_obs <- as.numeric(autoimmune_to_mdd$sig_obs)
autoimmune_to_mdd$sig_obs <- formatC(autoimmune_to_mdd$sig_obs,format="e",digits = 0)
autoimmune_to_mdd$print.p <- sub("e", "*x*10^", autoimmune_to_mdd$sig_obs)
autoimmune_to_mdd$asterix <- ifelse(autoimmune_to_mdd$P<0.000071, "*", " ")

```

```{r plot_autoimmune_to_mdd, echo=FALSE, results = 'asis', fig.align='center', fig.height=4, fig.width=14}

## Plot results of autoimmune PRS predicting depression

Figure <- ggplot(data = autoimmune_to_mdd, aes(x = as.factor(Set), y = PRS.R2.adj, fill = as.factor(Phenotype))) +
  geom_bar(stat = "identity", position = position_dodge(), width = 0.8) +
  scale_fill_manual(values = c("darkorchid4", "orchid3", "orchid1", "seagreen4", "darkseagreen3", "darkseagreen1"), name = "Depression\nPhenotype\n(target trait)") +
  geom_text(aes(label=paste(asterix)), position=position_dodge(width=0.8), 
                             vjust=0.3, hjust=0.5, size = 10, angle = 0) +
 ylab(expression(paste("R"^"2"," ","liability"))) +
  xlab("\nBase Summary Statistics") +
  theme(axis.text=element_text(size=14),
        axis.title=element_text(size=16)) +
  theme(panel.grid.minor.x=element_blank(), 
                         panel.grid.major.x=element_blank(), 
                         panel.grid.minor.y=element_blank(), 
                         panel.grid.major.y=element_blank(),
                         panel.background = element_rect(fill = "white"),
                         axis.line.y = element_line(color="grey"),
                         axis.line.x = element_line(color="grey"),
                        legend.title = element_text(size = 14),
                         legend.text = element_text(size = 12),
                         plot.title = element_text(hjust = 0.5, margin = margin(b=25), size = 18)) + 
  scale_y_continuous(limits=c(0,0.001), expand=c(0,0)) +
  labs(fill = "Depression Phenotype") 

```

```{r print_plot_autoimmune_to_mdd, echo=FALSE, results = 'asis', fig.align='center', fig.height=4, fig.width=14}

Figure

```

**Figure:**  Polygenic scores for autoimmune diseases (x-axis) predicting depression case/control status. Asterisks denote p-value < 7.1x10-5 (bonferroni corrected significance threshold). 

</br>

**Table:** Autoimmune PRS predicting depression case/control status.

```{r print_prs_table_ad_to_mdd, echo=FALSE, results = 'asis'}

# Create table of results for autoimmune PRS predicting depression

autoimmune_to_mdd$Base <- gsub("\n", " ", autoimmune_to_mdd$Set)
autoimmune_to_mdd$P <- formatC(autoimmune_to_mdd$P, format = "e", digits = 1)

table_autoimmune_to_mdd <- autoimmune_to_mdd[,c(17,1,3,12,8,4:5,9:11)]
table_autoimmune_to_mdd$Prevalence <- percent(table_autoimmune_to_mdd$Prevalence, digits = 0)
table_autoimmune_to_mdd$PRS.R2.adj <- percent(table_autoimmune_to_mdd$PRS.R2.adj, digits = 4)
table_autoimmune_to_mdd$PRS.R2 <- percent(table_autoimmune_to_mdd$PRS.R2, digits = 4)
table_autoimmune_to_mdd$Coefficient <- round(table_autoimmune_to_mdd$Coefficient, digits = 3)
table_autoimmune_to_mdd$Standard.Error <- round(table_autoimmune_to_mdd$Standard.Error, digits = 3)

table_autoimmune_to_mdd <- table_autoimmune_to_mdd[order(table_autoimmune_to_mdd$Base, table_autoimmune_to_mdd$Phenotype),] 
rownames(table_autoimmune_to_mdd) <- NULL

write.table(table_autoimmune_to_mdd, "/file_path/table_autoimmune_to_mdd", row.names=F, col.name=T, quote=F, sep = "\t")

kable(table_autoimmune_to_mdd, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%
  row_spec(0, bold = TRUE, color = "white", background = "#666", align = "centre")

```

</br>

### PRS at all thresholds (for significant associations)

</br>

```{r prepare_plot_all_thresholds_auto_to_mdd, echo=FALSE, results = 'asis', fig.align='center', fig.height=6, fig.width=18}

# Prepare plots for PRS at all p-value thresholds for significant results

# Read and format data
myasthenia <- read.table("/file_path/myasthenia_prs_to_mdd.prsice", header = T) 
myasthenia <- myasthenia[grep(c("all"), myasthenia$Pheno),]
myasthenia$Base <- "PRS for Myasthenia Gravis"
myasthenia$Target <- myasthenia$Pheno
myasthenia$Target <- gsub("Broad_all", "Any Depression (combined)", myasthenia$Target)
myasthenia$Target <- gsub("Narrow_all", "Stringent Depression (combined)", myasthenia$Target)
myasthenia$Title <-  paste(myasthenia$Base, "predicting", myasthenia$Target)
myasthenia$print.p <- formatC(myasthenia$P,format="e",digits = 0)
myasthenia$print.p <- sub("e", "*x*10^", myasthenia$print.p)

threshold_plot_myasthenia <- ggplot(myasthenia, aes(x = as.factor(Threshold), y = R2, fill = -log10(P))) +
  geom_bar(stat = "identity", position = position_dodge()) +
  scale_fill_gradient2(low = "dodgerblue", high = "firebrick", mid = "dodgerblue", midpoint=1e-4,
                       name = bquote(atop(-log[10] ~ model, italic(P) - value),)) +
  ylab(expression(paste("PRS model fit:  ", R ^ 2, sep = " "))) +
  xlab(expression(italic(P) - value ~ threshold ~ (italic(P)[T]))) +
  theme(axis.text.x = element_text(angle=45, hjust=1, size = 12),
        axis.text.y = element_text(size = 12),
        axis.title = element_text(size = 14),
        plot.title = element_text(hjust = 0.5, size = 18),
        strip.text.x = element_text(size = 12, face = "bold"),
        panel.grid = element_blank()) +
  scale_y_continuous(expand = expand_scale(mult = c(0, .3))) +
  geom_text(aes(label = paste(print.p)), vjust = -1.5, hjust = 0, angle = 45, cex = 3.5, parse = T) +
  facet_grid(~ Title)

psoriasis <- read.table("/file_path/psoriasis_prs_to_mdd.prsice", header = T) 
psoriasis <- psoriasis[grep(c("Broad_all"), psoriasis$Pheno),]
psoriasis$Base <- "PRS for Psoriasis"
psoriasis$Target <- psoriasis$Pheno
psoriasis$Target <- gsub("Broad_all", "Any Depression (combined)", psoriasis$Target)
psoriasis$Title <-  paste(psoriasis$Base, "predicting", psoriasis$Target)
psoriasis$print.p <- formatC(psoriasis$P,format="e",digits = 0)
psoriasis$print.p <- sub("e", "*x*10^", psoriasis$print.p)

threshold_plot_psoriasis <- ggplot(psoriasis, aes(x = as.factor(Threshold), y = R2, fill = -log10(P))) +
  geom_bar(stat = "identity", position = position_dodge()) +
  scale_fill_gradient2(low = "dodgerblue", high = "firebrick", mid = "dodgerblue", midpoint=1e-4,
                       name = bquote(atop(-log[10] ~ model, italic(P) - value),)) +
  ylab(expression(paste("PRS model fit:  ", R ^ 2, sep = " "))) +
  xlab(expression(italic(P) - value ~ threshold ~ (italic(P)[T]))) +
  theme(axis.text.x = element_text(angle=45, hjust=1, size = 12),
        axis.text.y = element_text(size = 12),
        axis.title = element_text(size = 14),
        plot.title = element_text(hjust = 0.5, size = 18),
        strip.text.x = element_text(size = 12, face = "bold"),
        panel.grid = element_blank()) +
  scale_y_continuous(expand = expand_scale(mult = c(0, .3))) +
  geom_text(aes(label = paste(print.p)), vjust = -1.5, hjust = 0, angle = 45, cex = 3.5, parse = T) +
  facet_grid(~ Title)

```

```{r print_thresholds_myasthenia, echo=FALSE, results = 'asis', fig.align='center', fig.height=5, fig.width=15}

threshold_plot_myasthenia

```

```{r print_thresholds_psoriasis, echo=FALSE, results = 'asis', fig.align='center', fig.height=5, fig.width=9}

threshold_plot_psoriasis

```

</br>

# Power Calculation

</br>

## SNP-based heritabilities

</br>

```{r calculate_heritability, echo=FALSE, eval=FALSE}

# Calculate SNP heritability (MHC removed) for traits with available summary statistics - this is required to power calculation in AVENGEME

# depression cases: 116,404	controls: 314,990. prevalence 116404/(116404+314990) = 0.2698322
./ldsc.py \
--h2 /file_path/DEPR06_noMHC.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.27 \
--pop-prev 0.15 \
--out /file_path/DEPR06_noMHC

# coeliac cases: 4,533,	controls: 10,750. prevalence 4533/(4533+10750) = 0.2966041
./ldsc.py \
--h2 /file_path/CELI01_noMHC.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.3 \
--pop-prev 0.01 \
--out /file_path/CELI01_noMHC

# ibd cases: 12,882, controls: 21,770. prevalence 12882/(12882+21770) = 0.3717534.
./ldsc.py \
--h2 /file_path/INFB01_noMHC.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.37 \
--pop-prev 0.005 \
--out /file_path/INFB01_noMHC

# ms cases: 9,772, controls: 17,376. prevalence 9772/(9772+17376) = 0.3599529.
./ldsc.py \
--h2 /file_path/SCLE02_noMHC.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.36 \
--pop-prev 0.001 \
--out /file_path/SCLE02_noMHC

# RA cases: 14,361, controls: 43,923. prevalence 14361/(14361+43923) = 0.246397.
./ldsc.py \
--h2 /file_path/RHEU02_noMHC.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.25 \
--pop-prev 0.01 \
--out /file_path/RHEU02_noMHC

# sle cases: 7,219, controls: 15,991. prevalence 7219/(7219+15991) = 0.3110297.
./ldsc.py \
--h2 /file_path/LUPU01_noMHC.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.31 \
--pop-prev 0.001 \
--out /file_path/LUPU01_noMHC

# Remove MHC where necessary and munge

# Myasthenia
# Read data
myasthenia <- read.table("/file_path/myasthenia_gravis_sum_stats_New_AllResults.2_NoSex.txt", header = TRUE)

# Remove MHC
myasthenia_nomhc <- myasthenia[!(myasthenia$CHR==6 & myasthenia$BP > 28866527 & myasthenia$BP < 33775447),]

# Save summary stats without MHC
write.table(myasthenia_nomhc, "/file_path/MYASTHENIA_noMHC", row.names = F, col.names = T, quote = F)

# Munge
./munge_sumstats.py \
--sumstats /file_path/MYASTHENIA_noMHC \
--snp MARKER \
--a1 ALLELE1 \
--a2 ALLELE2 \
--N-cas-col NCASES \
--N-con-col NCONTROLS \
--p PVALUE \
--signed-sumstats OR,1 \
--out /file_path/MYASTHENIA_noMHC_munged

# Heritability
# MG cases: 1,032, controls: 1,998. prevalence 1032/(1032+1998) = 0.3405941.
./ldsc.py \
--h2 /file_path/MYASTHENIA_noMHC_munged.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.34 \
--pop-prev 0.0002 \
--out /file_path/MYASTHENIA_noMHC

# Psoriasis
# Read data
psoriasis <- read.table("/file_path/metaanalysis_inversevariance_7datasets1_RS_ID.tbl", header = TRUE)

# Remove MHC
psoriasis1 <- psoriasis %>% 
   separate(MarkerName,c("CHR","BP"),sep=":")

psoriasis_nomhc <- psoriasis1[!(psoriasis1$CHR=="chr6" & psoriasis1$BP > 28866527 & psoriasis1$BP < 33775447),]

# change p value column heading
colnames(psoriasis_nomhc)[8] <- "P"

# Save summary stats without MHC
write.table(psoriasis_nomhc, "/file_path/PSORIASIS_noMHC", row.names = F, col.names = T, quote = F)

# Munge CASES 19,032, CONTROLS	286,769
./munge_sumstats.py \
--sumstats /file_path/PSORIASIS_noMHC \
--snp SNP \
--a1 Allele1 \
--a2 Allele2 \
--N-cas 19032 \
--N-con 286769 \
--p P \
--signed-sumstats Effect,0 \
--out /file_path/PSORIASIS_noMHC_munged

# Heritability
# Psoriasis cases: 19,032, controls: 286,769. prevalence 19032/(19032+286769) = 0.06223655	
./ldsc.py \
--h2 /file_path/PSORIASIS_noMHC_munged.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.0622	 \
--pop-prev 0.02 \
--out /file_path/PSORIASIS_noMHC

# PsA
# Read data
psa <- read.table("/file_path/psa_sumstats.txt", header = TRUE)

# Remove MHC
psa_nomhc <- psa[!(psa$CHR==6 & psa$POS > 28866527 & psa$POS < 33775447),]

# Save summary stats without MHC
write.table(psa_nomhc, "/file_path/PSA_noMHC", row.names = F, col.names = T, quote = F)

# Munge cases 1,430, controls	1,417
./munge_sumstats.py \
--sumstats /file_path/PSA_noMHC \
--snp SNP \
--a1 A1 \
--a2 A2 \
--N-cas 1430 \
--N-con 1417 \
--p P \
--signed-sumstats OR,1 \
--out /file_path/PSA_noMHC_munged

# Heritability
# PsA cases: 1,430, controls: 1,417. prevalence 1430/(1430+1417) = 0.5022831	
./ldsc.py \
--h2 /file_path/PSA_noMHC_munged.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.5023	 \
--pop-prev 0.005 \
--out /file_path/PSA_noMHC

```

```{r power_mdd_prs_to_auto, echo=FALSE, eval=TRUE, message=FALSE}

# Calculate power for MDD PRS predicting autoimmune case/control using number of SNPs at best p-value threshold.

library(avengeme)

# MDD h2 = Total Liability scale h2: 0.0838 (0.0036)

### MDD PRS predicting Coeliac
# Coeliac h2 = Total Liability scale h2: 0.1621 (0.0303)
base = "MDD"
target = "Coeliac"
base_h2_se = 0.0036
base_h2_intercept = 1.0169
target_h2_se = 0.0303
target_h2_intercept = 1.0681
target_case = 2364
  
## Calculate covariance
n=c(116404+314990,2364+324074) # training (MDD) 116404+314990=431394, target (Coeliac possible) 2364+324074=326438
vg1=0.0838 # h2 snp base MDD 
vg2=0.1621 # h2 snp target Coeliac
cor12=c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5) # rG
cov12=cor12*sqrt(vg1*vg2) # covariance at different rG. 

# Calculate power
nsnp=1123 # independent SNPs at best pT. 
pi0=0.9 # 90% SNPs with no effect in training trait - DOESN'T CHANGE POWER ESTIMATE. 
binary=c(T,T) # training and target binary
prevalence=c(0.15,0.01) # population prevalence in training and target 
sampling=c(116404/(116404+314990),2364/(2364+324074)) # gwas case/control, target case/control. 
pupper=c(0,0.001) # best pT

mdd_to_coeliac <- polygenescore(nsnp=nsnp,n=n,vg1=vg1,cov12=cov12,pi0=pi0,binary=binary,prevalence=prevalence,sampling=sampling,pupper=pupper)

df_mdd_to_coeliac <- data.frame(
      base = rep(base,14),
      base_n = rep(n[1],14),
      base_h2 = vg1,
      base_h2_se = base_h2_se,
      base_h2_intercept = base_h2_intercept,
      base_pop_prev = prevalence[1],
      base_sample_prev = sampling[1],
      target = rep(target,14),
      target_n = rep(n[2],14),
      target_case = target_case,
      target_h2 = vg2,
      target_h2_se = target_h2_se,
      target_h2_intercept = target_h2_intercept,
      target_pop_prev = prevalence[2],
      target_sample_prev = sampling[2],
      best_pT = pupper[2],
      nsnp = nsnp,
      rg = c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5),
      power = mdd_to_coeliac$power,
      r2_at_rg = mdd_to_coeliac$R2,
      p_at_rg = mdd_to_coeliac$p
)

### MDD PRS predicting IBD  
# IBD h2 = Total Liability scale h2: 0.162 (0.0157)
base = "MDD"
target = "Inflammatory Bowel Disease"
base_h2_se = 0.0036
base_h2_intercept = 1.0169
target_h2_se = 0.0157
target_h2_intercept = 1.0608
target_case = 5105

## Calculate covariance
n=c(116404+314990,5105+324074) # training (MDD) 116404+314990=431394, target (IBD possible) 5105+324074=329179
vg1=0.0838 # h2 snp base MDD 
vg2=0.162 # h2 snp target IBD
cor12=c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5) # rG
cov12=cor12*sqrt(vg1*vg2) 

# Calculate power
nsnp=48852 # independent SNPs at best pT. 
pi0=0.9 # 90% SNPs with no effect in training trait - DOESN'T CHANGE POWER ESTIMATE. 
binary=c(T,T) # training and target binary
prevalence=c(0.15,0.005) # population prevalence in training and target 
sampling=c(116404/(116404+314990),5105/(5105+324074)) # gwas case/control, target case/control. 
pupper=c(0,0.300)   

mdd_to_ibd <- polygenescore(nsnp=nsnp,n=n,vg1=vg1,cov12=cov12,pi0=pi0,binary=binary,prevalence=prevalence,sampling=sampling,pupper=pupper)

df_mdd_to_ibd <- data.frame(
      base = rep(base,14),
      base_n = rep(n[1],14),
      base_h2 = vg1,
      base_h2_se = base_h2_se,
      base_h2_intercept = base_h2_intercept,
      base_pop_prev = prevalence[1],
      base_sample_prev = sampling[1],
      target = rep(target,14),
      target_n = rep(n[2],14),
      target_case = target_case,
      target_h2 = vg2,
      target_h2_se = target_h2_se,
      target_h2_intercept = target_h2_intercept,
      target_pop_prev = prevalence[2],
      target_sample_prev = sampling[2],
      best_pT = pupper[2],
      nsnp = nsnp,
      rg = c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5),
      power = mdd_to_ibd$power,
      r2_at_rg = mdd_to_ibd$R2,
      p_at_rg = mdd_to_ibd$p
)

### MDD PRS predicting MS  
# MS h2 = Total Liability scale h2: 0.0185 (0.0101)
base = "MDD"
target = "Multiple Sclerosis"
base_h2_se = 0.0036
base_h2_intercept = 1.0169
target_h2_se = 0.0101
target_h2_intercept = 1.0601
target_case = 1683

## Calculate covariance
n=c(116404+314990,1683+324074) # training (MDD) 116404+314990=431394, target (MS possible) 1683+324074=325757
vg1=0.0838 # h2 snp base MDD 
vg2=0.0185 # h2 snp target MS
cor12=c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5) # rG
cov12=cor12*sqrt(vg1*vg2) 

# Calculate power
nsnp=67956 # independent SNPs at best pT. 
pi0=0.9 # 90% SNPs with no effect in training trait - DOESN'T CHANGE POWER ESTIMATE. 
binary=c(T,T) # training and target binary
prevalence=c(0.15,0.001) # population prevalence in training and target 
sampling=c(116404/(116404+314990),1683/(1683+324074)) # gwas case/control, target case/control. 
pupper=c(0,0.500)  

mdd_to_ms <- polygenescore(nsnp=nsnp,n=n,vg1=vg1,cov12=cov12,pi0=pi0,binary=binary,prevalence=prevalence,sampling=sampling,pupper=pupper)

df_mdd_to_ms <- data.frame(
      base = rep(base,14),
      base_n = rep(n[1],14),
      base_h2 = vg1,
      base_h2_se = base_h2_se,
      base_h2_intercept = base_h2_intercept,
      base_pop_prev = prevalence[1],
      base_sample_prev = sampling[1],
      target = rep(target,14),
      target_n = rep(n[2],14),
      target_case = target_case,
      target_h2 = vg2,
      target_h2_se = target_h2_se,
      target_h2_intercept = target_h2_intercept,
      target_pop_prev = prevalence[2],
      target_sample_prev = sampling[2],
      best_pT = pupper[2],
      nsnp = nsnp,
      rg = c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5),
      power = mdd_to_ms$power,
      r2_at_rg = mdd_to_ms$R2,
      p_at_rg = mdd_to_ms$p
)

### MDD PRS predicting Myasthenia  
# Myasthenia h2 = Total Liability scale h2: 0.1396 (0.0494)
base = "MDD"
target = "Myasthenia Gravis"
base_h2_se = 0.0036
base_h2_intercept = 1.0169
target_h2_se = 0.0494
target_h2_intercept = 1.0189
target_case = 234

## Calculate covariance
n=c(116404+314990,234+324074) # training (MDD) 116404+314990=431394, target (myasthenia possible) 234+324074=324308
vg1=0.0838 # h2 snp base MDD 
vg2=0.1396 # h2 snp target myasthenia
cor12=c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5) # rG
cov12=cor12*sqrt(vg1*vg2) 

# Calculate power
nsnp=99216 # independent SNPs at best pT. 
pi0=0.9 # 90% SNPs with no effect in training trait - DOESN'T CHANGE POWER ESTIMATE. 
binary=c(T,T) # training and target binary
prevalence=c(0.15,0.0002) # population prevalence training and target 
sampling=c(116404/(116404+314990),234/(234+324074)) # gwas case/control, target case/control. 
pupper=c(0,1) 

mdd_to_mg <- polygenescore(nsnp=nsnp,n=n,vg1=vg1,cov12=cov12,pi0=pi0,binary=binary,prevalence=prevalence,sampling=sampling,pupper=pupper)

df_mdd_to_mg <- data.frame(
      base = rep(base,14),
      base_n = rep(n[1],14),
      base_h2 = vg1,
      base_h2_se = base_h2_se,
      base_h2_intercept = base_h2_intercept,
      base_pop_prev = prevalence[1],
      base_sample_prev = sampling[1],
      target = rep(target,14),
      target_n = rep(n[2],14),
      target_case = target_case,
      target_h2 = vg2,
      target_h2_se = target_h2_se,
      target_h2_intercept = target_h2_intercept,
      target_pop_prev = prevalence[2],
      target_sample_prev = sampling[2],
      best_pT = pupper[2],
      nsnp = nsnp,
      rg = c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5),
      power = mdd_to_mg$power,
      r2_at_rg = mdd_to_mg$R2,
      p_at_rg = mdd_to_mg$p
)

### MDD PRS predicting psoriasis  
# Psoriasis h2 = Total Liability scale h2: 0.0838 (0.0086)
base = "MDD"
target = "Psoriasis"
base_h2_se = 0.0036
base_h2_intercept = 1.0169
target_h2_se = 0.0086
target_h2_intercept = 0.9816
target_case = 5459

## Calculate covariance
n=c(116404+314990,5459+324074) # training (MDD) 116404+314990=431394, target (psoriasis) 5459+324074=329533
vg1=0.0838 # h2 snp base MDD 
vg2=0.0838 # h2 snp target psoriasis
cor12=c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5) # rG
cov12=cor12*sqrt(vg1*vg2) 

# Calculate power
nsnp=67956 # independent SNPs at best pT. 
pi0=0.9 # 90% SNPs with no effect in training trait - DOESN'T CHANGE POWER ESTIMATE. 
binary=c(T,T) # training and target binary
prevalence=c(0.15,0.02) # population prevalence training and target 
sampling=c(116404/(116404+314990),5459/(5459+324074)) # gwas case/control, target case/control. 
pupper=c(0,0.5) 

mdd_to_psoriasis <- polygenescore(nsnp=nsnp,n=n,vg1=vg1,cov12=cov12,pi0=pi0,binary=binary,prevalence=prevalence,sampling=sampling,pupper=pupper)

df_mdd_to_psoriasis <- data.frame(
      base = rep(base,14),
      base_n = rep(n[1],14),
      base_h2 = vg1,
      base_h2_se = base_h2_se,
      base_h2_intercept = base_h2_intercept,
      base_pop_prev = prevalence[1],
      base_sample_prev = sampling[1],
      target = rep(target,14),
      target_n = rep(n[2],14),
      target_case = target_case,
      target_h2 = vg2,
      target_h2_se = target_h2_se,
      target_h2_intercept = target_h2_intercept,
      target_pop_prev = prevalence[2],
      target_sample_prev = sampling[2],
      best_pT = pupper[2],
      nsnp = nsnp,
      rg = c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5),
      power = mdd_to_psoriasis$power,
      r2_at_rg = mdd_to_psoriasis$R2,
      p_at_rg = mdd_to_psoriasis$p
)

### MDD PRS predicting PsA  
# PsA h2 = Total Liability scale h2: 0.4544 (0.1065)
base = "MDD"
target = "Psoriatic Arthritis"
base_h2_se = 0.0036
base_h2_intercept = 1.0169
target_h2_se = 0.1065
target_h2_intercept = 1.1299
target_case = 1107

## Calculate covariance
n=c(116404+314990,1107+324074) # training (MDD) 116404+314990=431394, target (psa) 1107+324074=325181
vg1=0.0838 # h2 snp base MDD 
vg2=0.4544 # h2 snp target psa
cor12=c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5) # rG
cov12=cor12*sqrt(vg1*vg2) 

# Calculate power
nsnp=99216 # independent SNPs at best pT. 
pi0=0.9 # 90% SNPs with no effect in training trait - DOESN'T CHANGE POWER ESTIMATE. 
binary=c(T,T) # training and target binary
prevalence=c(0.15,0.005) # population prevalence training and target 
sampling=c(116404/(116404+314990),1107/(1107+324074)) # gwas case/control, target case/control. 
pupper=c(0,1)  

mdd_to_psa <- polygenescore(nsnp=nsnp,n=n,vg1=vg1,cov12=cov12,pi0=pi0,binary=binary,prevalence=prevalence,sampling=sampling,pupper=pupper)

df_mdd_to_psa <- data.frame(
      base = rep(base,14),
      base_n = rep(n[1],14),
      base_h2 = vg1,
      base_h2_se = base_h2_se,
      base_h2_intercept = base_h2_intercept,
      base_pop_prev = prevalence[1],
      base_sample_prev = sampling[1],
      target = rep(target,14),
      target_n = rep(n[2],14),
      target_case = target_case,
      target_h2 = vg2,
      target_h2_se = target_h2_se,
      target_h2_intercept = target_h2_intercept,
      target_pop_prev = prevalence[2],
      target_sample_prev = sampling[2],
      best_pT = pupper[2],
      nsnp = nsnp,
      rg = c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5),
      power = mdd_to_psa$power,
      r2_at_rg = mdd_to_psa$R2,
      p_at_rg = mdd_to_psa$p
)

### MDD PRS predicting RA 
# RA h2 = Total Liability scale h2: 0.1035 (0.0129)
base = "MDD"
target = "Rheumatoid Arthritis"
base_h2_se = 0.0036
base_h2_intercept = 1.0169
target_h2_se = 0.0129
target_h2_intercept = 0.9511
target_case = 6360

## Calculate covariance
n=c(116404+314990,6360+324074) # training (MDD) 116404+314990=431394, target (arthritis) 6360+324074=330434
vg1=0.0838 # h2 snp base MDD 
vg2=0.1035 # h2 snp target RA
cor12=c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5) # rG
cov12=cor12*sqrt(vg1*vg2) 

# Calculate power
nsnp=67956 # independent SNPs at best pT. 
pi0=0.9 # 90% SNPs with no effect in training trait - DOESN'T CHANGE POWER ESTIMATE. 
binary=c(T,T) # training and target binary
prevalence=c(0.15,0.01) # population prevalence training and target 
sampling=c(116404/(116404+314990),6360/(6360+324074)) # gwas case/control, target case/control. 
pupper=c(0,0.5) 

mdd_to_ra <- polygenescore(nsnp=nsnp,n=n,vg1=vg1,cov12=cov12,pi0=pi0,binary=binary,prevalence=prevalence,sampling=sampling,pupper=pupper)

df_mdd_to_ra <- data.frame(
      base = rep(base,14),
      base_n = rep(n[1],14),
      base_h2 = vg1,
      base_h2_se = base_h2_se,
      base_h2_intercept = base_h2_intercept,
      base_pop_prev = prevalence[1],
      base_sample_prev = sampling[1],
      target = rep(target,14),
      target_n = rep(n[2],14),
      target_case = target_case,
      target_h2 = vg2,
      target_h2_se = target_h2_se,
      target_h2_intercept = target_h2_intercept,
      target_pop_prev = prevalence[2],
      target_sample_prev = sampling[2],
      best_pT = pupper[2],
      nsnp = nsnp,
      rg = c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5),
      power = mdd_to_ra$power,
      r2_at_rg = mdd_to_ra$R2,
      p_at_rg = mdd_to_ra$p
)

### MDD PRS predicting SLE 
# SLE he = Total Liability scale h2: 0.103 (0.018)
base = "MDD"
target = "Systemic Lupus Erythematosus"
base_h2_se = 0.0036
base_h2_intercept = 1.0169
target_h2_se = 0.018
target_h2_intercept = 1.1001
target_case = 624

## Calculate covariance
n=c(116404+314990,624+324074) # training (MDD) 116404+314990=431394, target (sle) 624+324074=324698
vg1=0.0838 # h2 snp base MDD 
vg2=0.103 # h2 snp target SLE
cor12=c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5) # rG
cov12=cor12*sqrt(vg1*vg2) 

# Calculate power
nsnp=14137 # independent SNPs at best pT. 
pi0=0.9 # 90% SNPs with no effect in training trait - DOESN'T CHANGE POWER ESTIMATE. 
binary=c(T,T) # training and target binary
prevalence=c(0.15,0.001) # population prevalence training and target 
sampling=c(116404/(116404+314990),624/(624+324074)) # gwas case/control, target case/control. 
pupper=c(0,0.05)  

mdd_to_sle <- polygenescore(nsnp=nsnp,n=n,vg1=vg1,cov12=cov12,pi0=pi0,binary=binary,prevalence=prevalence,sampling=sampling,pupper=pupper)

df_mdd_to_sle <- data.frame(
      base = rep(base,14),
      base_n = rep(n[1],14),
      base_h2 = vg1,
      base_h2_se = base_h2_se,
      base_h2_intercept = base_h2_intercept,
      base_pop_prev = prevalence[1],
      base_sample_prev = sampling[1],
      target = rep(target,14),
      target_n = rep(n[2],14),
      target_case = target_case,
      target_h2 = vg2,
      target_h2_se = target_h2_se,
      target_h2_intercept = target_h2_intercept,
      target_pop_prev = prevalence[2],
      target_sample_prev = sampling[2],
      best_pT = pupper[2],
      nsnp = nsnp,
      rg = c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5),
      power = mdd_to_sle$power,
      r2_at_rg = mdd_to_sle$R2,
      p_at_rg = mdd_to_sle$p
)

# combine power estimates
power <- rbind(df_mdd_to_coeliac,df_mdd_to_ibd,df_mdd_to_ms,df_mdd_to_mg,df_mdd_to_psoriasis,df_mdd_to_psa,df_mdd_to_ra,df_mdd_to_sle)
  
```

```{r plot_h2, echo=FALSE, eval=TRUE, message=FALSE, results = 'asis', fig.align='center', fig.height=5, fig.width=10}

## Plot SNP heritabilities

power$base_h2_lowci <- power$base_h2-1.96*power$base_h2_se
power$base_h2_highci <- power$base_h2+1.96*power$base_h2_se
power$target_h2_lowci <- power$target_h2-1.96*power$target_h2_se
power$target_h2_highci <- power$target_h2+1.96*power$target_h2_se

h2_MDD <- power %>%
  filter(rg==0.1) %>%
  select(base, base_h2, base_h2_lowci, base_h2_highci, base_h2_intercept) %>%
  rename(trait = base, h2 = base_h2, lowci = base_h2_lowci, highci = base_h2_highci, intercept = base_h2_intercept) %>%
  unique() %>%
  mutate(trait = gsub("MDD", "Major Depressive Disorder", trait))

h2_auto <- power %>%
  filter(rg==0.1) %>%
  select(target, target_h2, target_h2_lowci, target_h2_highci, target_h2_intercept) %>%
  rename(trait = target, h2 = target_h2, lowci = target_h2_lowci, highci = target_h2_highci, intercept = target_h2_intercept) 

h2 <- rbind(h2_MDD,h2_auto)

h2$legend <- h2$trait
h2$legend <- gsub("Inflammatory Bowel Disease", "Inflammatory\nBowel Disease", h2$legend)
h2$legend <- gsub("Multiple Sclerosis", "Multiple\nSclerosis", h2$legend)
h2$legend <- gsub("Myasthenia Gravis", "Myasthenia\nGravis", h2$legend)
h2$legend <- gsub("Psoriatic Arthritis", "Psoriatic\nArthritis", h2$legend)
h2$legend <- gsub("Rheumatoid Arthritis", "Rheumatoid\nArthritis", h2$legend)
h2$legend <- gsub("Systemic Lupus Erythematosus", "Systemic Lupus\nErythematosus", h2$legend)
h2$legend <- gsub("Major Depressive Disorder", "Major Depressive\nDisorder", h2$legend)

h2_plot <- ggplot(data = h2, aes(x = as.factor(legend), y = h2)) +
  geom_bar(stat = "identity", position = position_dodge(), width = 0.8, fill = "lightblue") +
  ylab(expression(paste("h"^"2"," ","SNP"," ","liability"))) +
  xlab("Trait") +
  #labs(title="SNP-based heritability excluding MHC", subtitle = "From publicly-available summary statistics") +
  theme(axis.text=element_text(size=11),
        axis.title=element_text(size=12)) +
  theme(panel.grid.minor.x=element_blank(), 
                         panel.grid.major.x=element_blank(), 
                         panel.grid.minor.y=element_blank(), 
                         panel.grid.major.y=element_blank(),
                         panel.background = element_rect(fill = "white"),
                         axis.line.y = element_line(color="grey"),
                         axis.line.x = element_line(color="grey"),
                          axis.text.x=element_text(angle = 30, hjust = 1),
                         plot.title = element_text(hjust = 0.5, size = 14, margin = margin(b=5)),
                        plot.subtitle = element_text(hjust = 0.5, size = 12, margin = margin(b=10))) +
  ylim(-0.05,0.8) +
  geom_errorbar(aes(ymin=lowci, ymax=highci), position = position_dodge(0.8), width = 0.2) +
  annotate("text", x = 1:9, y = Inf,  vjust=3, fontface=1, colour = "grey10",
           size=3, label = rep("Intercept:", 9)) +
  annotate("text", x = 1:9, y = Inf,  vjust=5, fontface=1, colour = "grey10",
           size=3, label = format(round(h2$intercept, digits = 3), nsmall = 3)) 
  
h2_plot

```

**Bar Plot:** SNP heritability of traits using publicly available summary statistics, with MHC removed. Error bars: 95% CIs. Intercepts shown atop bars. Transformed to liability scale using population prevalence estimates: coeliac=1%, IBD=0.5%, MDD=15%, MS=0.1%, MG=0.02%, psoriasis=2%, PsA=0.5%, RA=1%, SLE=0.1%.

</br>

## MDD PRS predicting possible autoimmune diseases in the UKB

</br>

**Table:** Parameter inputs for power calculation in AVENGEME - MDD PRS predicting possible autoimmune case-control status in the UKB.

```{r parameter_table_mdd_prs_to_auto, echo = FALSE, results = 'asis'}

# Create table with parameter inputs for power calculation - MDD PRS predicing autoimmune

power_table <- power %>%
  filter(rg==0.1) %>%
  select(base, base_n, base_h2, base_pop_prev, base_sample_prev, best_pT, nsnp, target, target_n, target_h2, target_pop_prev, target_sample_prev)

power_table$base_n <- formatC(power_table$base_n, big.mark = ",", format = "d")
power_table$base_h2 <- percent(power_table$base_h2, digits = 2)
power_table$base_pop_prev <- percent(power_table$base_pop_prev, digits = 0)
power_table$base_sample_prev <- percent(power_table$base_sample_prev, digits = 0)
power_table$nsnp <- formatC(power_table$nsnp, big.mark = ",", format = "d")
power_table$target_n <- formatC(power_table$target_n, big.mark = ",", format = "d")
power_table$target_h2 <- percent(power_table$target_h2, digits = 2)
power_table$target_pop_prev <- percent(power_table$target_pop_prev, digits = 2)
power_table$target_sample_prev <- percent(power_table$target_sample_prev, digits = 2)

kable(power_table, escape = F, align = 'r') %>%
  kable_styling(c("striped", "condensed"), full_width = T, font_size = 13) %>%
  add_header_above(c("Base GWAS" = 7, "Target Phenotype, UKB" = 5))

```

</br>

```{r plot_power_mdd_prs_to_auto, echo=FALSE, eval=TRUE, message=FALSE, results = 'asis', fig.align='center', fig.height=5, fig.width=10}

# Plot power - MDD PRS predicting autoimmune case/control

power$legend <- paste0(power$target," (",formatC(power$target_case, big.mark = ","),")")

power_plot <- ggplot(data=power, mapping=aes(x=rg, y=power)) +
  geom_line(mapping=aes(colour=legend), size=0.6, show.legend=TRUE) +
  geom_hline(mapping=aes(yintercept=0.8, fill="Power"), size=0.6, color="red") +
  scale_color_hue("Autoimmune Disease (n possible cases in UKB)", guide=guide_legend(order=1)) +
  scale_fill_manual("Power", values = 1, guide=guide_legend(
    override.aes=list(colour=c("red")),order = 2),labels=c("80% power")) +
  labs(title="Power by rG assumptions", subtitle = "MDD PRS predicting possible autoimmune diseases in the UKB", x = "\nrG between MDD and Autoimmune Diseases", y = "Power\n") +
  theme(plot.title = element_text(hjust = 0.5, size = 14, margin = margin(b=5)),
        plot.subtitle = element_text(hjust = 0.5, size = 12, margin = margin(b=10)),
        axis.title = element_text(size = 12),
        axis.text = element_text(size = 10),
        legend.title = element_text(size = 12),
        legend.text = element_text(size = 10))

power_plot

```

**Line Plot:** Power to detect cross-trait associations at different levels of rG. Power calculated for MDD PRS predicing possible case status for eight autoimmune disorders in the UKB. 

</br>

## Autoimmune PRS predicting 'any' depression in the UKB

</br>

```{r power_autoimmune_prs_to_mdd, echo=FALSE, eval=TRUE, message=FALSE}

# Calculate power for autoimmune PRS predicting depression case/control using number of SNPs at best p-value threshold.

library(avengeme)

# Autoimmune PRS predicting any depression

# MDD h2 = Total Liability scale h2: 0.0838 (0.0036)

### MDD Coeliac predicting any depression
# Coeliac h2 = Total Liability scale h2: 0.1621 (0.0303)
base = "Coeliac"
base_case = 4533
base_control = 10750
target = "Any Depression"
target_case = 65075
  
## Calculate covariance
n=c(4533+10750,65075+232552) # training (coeliac) 4533+10750=15283, target (any depression) 65075+232552=297627
vg1=0.1621 # h2 snp base Coeliac 
vg2=0.0838 # h2 snp target MDD
cor12=c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5) # rG
cov12=cor12*sqrt(vg1*vg2) # covariance at different rG

# Calculate power
nsnp=12430 # independent SNPs at best pT. 
pi0=0.9 # 90% SNPs with no effect in training trait - DOESN'T CHANGE POWER ESTIMATE. 
binary=c(T,T) # training and target binary
prevalence=c(0.01,0.15) # population prevalence in training and target 
sampling=c(4533/(4533+10750),65075/(65075+232552)) # gwas case/control, target case/control. 
pupper=c(0,0.200) # best p-value threshold  

coeliac <- polygenescore(nsnp=nsnp,n=n,vg1=vg1,cov12=cov12,pi0=pi0,binary=binary,prevalence=prevalence,sampling=sampling,pupper=pupper)

df_coeliac <- data.frame(
      base = rep(base,14),
      base_n = rep(n[1],14),
      base_case = base_case,
      base_control = base_control,
      base_h2 = vg1,
      base_pop_prev = prevalence[1],
      base_sample_prev = sampling[1],
      target = rep(target,14),
      target_n = rep(n[2],14),
      target_case = target_case,
      target_h2 = vg2,
      target_pop_prev = prevalence[2],
      target_sample_prev = sampling[2],
      best_pT = pupper[2],
      nsnp = nsnp,
      rg = c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5),
      power = coeliac$power,
      r2_at_rg = coeliac$R2,
      p_at_rg = coeliac$p
)

### IBD PRS predicting any depression
# IBD h2 = Total Liability scale h2: 0.162 (0.0157)
base = "Inflammatory Bowel Disease"
base_case = 12882
base_control = 21770
target = "Any Depression"
target_case = 65075

## Calculate covariance
n=c(12882+21770,65075+232552) # training (ibd) 12882+21770=34652, target (any depression) 65075+232552=297627
vg1=0.162 # h2 snp base ibd 
vg2=0.0838 # h2 snp target MDD
cor12=c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5) # rG
cov12=cor12*sqrt(vg1*vg2) 

# Calculate power
nsnp=80201 # independent SNPs at best pT. 
pi0=0.9 # 90% SNPs with no effect in training trait - DOESN'T CHANGE POWER ESTIMATE. 
binary=c(T,T) # training and target binary
prevalence=c(0.005,0.15) # population prevalence in training and target 
sampling=c(12882/(12882+21770),65075/(65075+232552)) # gwas case/control, target case/control. 
pupper=c(0,1)   

ibd <- polygenescore(nsnp=nsnp,n=n,vg1=vg1,cov12=cov12,pi0=pi0,binary=binary,prevalence=prevalence,sampling=sampling,pupper=pupper)

df_ibd <- data.frame(
      base = rep(base,14),
      base_n = rep(n[1],14),
      base_case = base_case,
      base_control = base_control,
      base_h2 = vg1,
      base_pop_prev = prevalence[1],
      base_sample_prev = sampling[1],
      target = rep(target,14),
      target_n = rep(n[2],14),
      target_case = target_case,
      target_h2 = vg2,
      target_pop_prev = prevalence[2],
      target_sample_prev = sampling[2],
      best_pT = pupper[2],
      nsnp = nsnp,
      rg = c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5),
      power = ibd$power,
      r2_at_rg = ibd$R2,
      p_at_rg = ibd$p
)

### MS PRS predicting any depression  
# MS h2 = Total Liability scale h2: 0.0185 (0.0101)
base = "Multiple Sclerosis"
base_case = 9772
base_control = 17376
target = "Any Depression"
target_case = 65075

## Calculate covariance
n=c(9772+17376,65075+232552) # training (ms) 9772+17376=27148, target (any depression) 65075+232552=297627
vg1=0.0185 # h2 snp base MS 
vg2=0.0838 # h2 snp target MDD
cor12=c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5) # rG
cov12=cor12*sqrt(vg1*vg2) 

# Calculate power
nsnp=3240 # independent SNPs at best pT. 
pi0=0.9 # 90% SNPs with no effect in training trait - DOESN'T CHANGE POWER ESTIMATE. 
binary=c(T,T) # training and target binary
prevalence=c(0.001,0.15) # population prevalence in training and target 
sampling=c(9772/(9772+17376),65075/(65075+232552)) # gwas case/control, target case/control. 
pupper=c(0,0.050)  

ms <- polygenescore(nsnp=nsnp,n=n,vg1=vg1,cov12=cov12,pi0=pi0,binary=binary,prevalence=prevalence,sampling=sampling,pupper=pupper)

df_ms <- data.frame(
      base = rep(base,14),
      base_n = rep(n[1],14),
      base_case = base_case,
      base_control = base_control,
      base_h2 = vg1,
      base_pop_prev = prevalence[1],
      base_sample_prev = sampling[1],
      target = rep(target,14),
      target_n = rep(n[2],14),
      target_case = target_case,
      target_h2 = vg2,
      target_pop_prev = prevalence[2],
      target_sample_prev = sampling[2],
      best_pT = pupper[2],
      nsnp = nsnp,
      rg = c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5),
      power = ms$power,
      r2_at_rg = ms$R2,
      p_at_rg = ms$p
)

### Myasthenia PRS predicting any depression  
# Myasthenia h2 = Total Liability scale h2: 0.1396 (0.0494)
base = "Myasthenia Gravis"
base_case = 1032
base_control = 1998
target = "Any Depression"
target_case = 65075

## Calculate covariance
n=c(1032+1998,65075+232552) # training (myasthenia) 1032+1998=3030, target (any depression) 65075+232552=297627
vg1=0.1396 # h2 snp base myasthenia 
vg2=0.0838 # h2 snp target MDD
cor12=c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5) # rG
cov12=cor12*sqrt(vg1*vg2) 

# Calculate power
nsnp=56894 # independent SNPs at best pT. 
pi0=0.9 # 90% SNPs with no effect in training trait - DOESN'T CHANGE POWER ESTIMATE. 
binary=c(T,T) # training and target binary
prevalence=c(0.0002,0.15) # population prevalence training and target 
sampling=c(1032/(1032+1998),65075/(65075+232552)) # gwas case/control, target case/control. 
pupper=c(0,0.3)  

mg <- polygenescore(nsnp=nsnp,n=n,vg1=vg1,cov12=cov12,pi0=pi0,binary=binary,prevalence=prevalence,sampling=sampling,pupper=pupper)

df_mg <- data.frame(
      base = rep(base,14),
      base_n = rep(n[1],14),
      base_case = base_case,
      base_control = base_control,
      base_h2 = vg1,
      base_pop_prev = prevalence[1],
      base_sample_prev = sampling[1],
      target = rep(target,14),
      target_n = rep(n[2],14),
      target_case = target_case,
      target_h2 = vg2,
      target_pop_prev = prevalence[2],
      target_sample_prev = sampling[2],
      best_pT = pupper[2],
      nsnp = nsnp,
      rg = c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5),
      power = mg$power,
      r2_at_rg = mg$R2,
      p_at_rg = mg$p
)

### MDD PRS predicting psoriasis  
# Psoriasis h2 = Total Liability scale h2: 0.0838 (0.0086)
base = "Psoriasis"
base_case = 19032
base_control = 286769
target = "Any Depression"
target_case = 65075

## Calculate covariance
n=c(19032+286769,65075+232552) # training (psoriasis) 19032+286769=305801, target (any depression) 65075+232552=297627
vg1=0.0838 # h2 snp base psoriasis 
vg2=0.0838 # h2 snp target MDD
cor12=c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5) # rG
cov12=cor12*sqrt(vg1*vg2) 

# Calculate power
nsnp=136084 # independent SNPs at best pT. 
pi0=0.9 # 90% SNPs with no effect in training trait - DOESN'T CHANGE POWER ESTIMATE. 
binary=c(T,T) # training and target binary
prevalence=c(0.02,0.15) # population prevalence training and target 
sampling=c(19032/(19032+286769),65075/(65075+232552)) # gwas case/control, target case/control. 
pupper=c(0,1)   

psoriasis <- polygenescore(nsnp=nsnp,n=n,vg1=vg1,cov12=cov12,pi0=pi0,binary=binary,prevalence=prevalence,sampling=sampling,pupper=pupper)

df_psoriasis <- data.frame(
      base = rep(base,14),
      base_n = rep(n[1],14),
      base_case = base_case,
      base_control = base_control,
      base_h2 = vg1,
      base_pop_prev = prevalence[1],
      base_sample_prev = sampling[1],
      target = rep(target,14),
      target_n = rep(n[2],14),
      target_case = target_case,
      target_h2 = vg2,
      target_pop_prev = prevalence[2],
      target_sample_prev = sampling[2],
      best_pT = pupper[2],
      nsnp = nsnp,
      rg = c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5),
      power = psoriasis$power,
      r2_at_rg = psoriasis$R2,
      p_at_rg = psoriasis$p
)

### PsA PRS predicting any depression 
# PsA h2 = Total Liability scale h2: 0.4544 (0.1065)
base = "Psoriatic Arthritis"
base_case = 1430
base_control = 1417
target = "Any Depression"
target_case = 65075

## Calculate covariance
n=c(1430+1417,65075+232552) # training (psa) 1430+1417=2847, target (any depression) 65075+232552=297627
vg1=0.4544 # h2 snp base psa 
vg2=0.0838 # h2 snp target MDD
cor12=c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5) # rG
cov12=cor12*sqrt(vg1*vg2) 

# Calculate power
nsnp=4666 # independent SNPs at best pT. 
pi0=0.9 # 90% SNPs with no effect in training trait - DOESN'T CHANGE POWER ESTIMATE. 
binary=c(T,T) # training and target binary
prevalence=c(0.005,0.15) # population prevalence training and target 
sampling=c(1430/(1430+1417),65075/(65075+232552)) # gwas case/control, target case/control. 
pupper=c(0,0.05) 

psa <- polygenescore(nsnp=nsnp,n=n,vg1=vg1,cov12=cov12,pi0=pi0,binary=binary,prevalence=prevalence,sampling=sampling,pupper=pupper)

df_psa <- data.frame(
      base = rep(base,14),
      base_n = rep(n[1],14),
      base_case = base_case,
      base_control = base_control,
      base_h2 = vg1,
      base_pop_prev = prevalence[1],
      base_sample_prev = sampling[1],
      target = rep(target,14),
      target_n = rep(n[2],14),
      target_case = target_case,
      target_h2 = vg2,
      target_pop_prev = prevalence[2],
      target_sample_prev = sampling[2],
      best_pT = pupper[2],
      nsnp = nsnp,
      rg = c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5),
      power = psa$power,
      r2_at_rg = psa$R2,
      p_at_rg = psa$p
)

### RA PRS predicting any depression 
# RA h2 = Total Liability scale h2: 0.1035 (0.0129)
base = "Rheumatoid Arthritis"
base_case = 14361
base_control = 43923
target = "Any Depression"
target_case = 65075

## Calculate covariance
n=c(14361+43923,65075+232552) # training (ra) 14361+43923=58284, target (any depression) 65075+232552=297627
vg1=0.1035 # h2 snp base RA 
vg2=0.0838 # h2 snp target MDD
cor12=c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5) # rG
cov12=cor12*sqrt(vg1*vg2) 

# Calculate power
nsnp=118649 # independent SNPs at best pT. 
pi0=0.9 # 90% SNPs with no effect in training trait - DOESN'T CHANGE POWER ESTIMATE. 
binary=c(T,T) # training and target binary
prevalence=c(0.01,0.15) # population prevalence training and target 
sampling=c(14361/(14361+43923),65075/(65075+232552)) # gwas case/control, target case/control. 
pupper=c(0,1)   

ra <- polygenescore(nsnp=nsnp,n=n,vg1=vg1,cov12=cov12,pi0=pi0,binary=binary,prevalence=prevalence,sampling=sampling,pupper=pupper)

df_ra <- data.frame(
      base = rep(base,14),
      base_n = rep(n[1],14),
      base_case = base_case,
      base_control = base_control,
      base_h2 = vg1,
      base_pop_prev = prevalence[1],
      base_sample_prev = sampling[1],
      target = rep(target,14),
      target_n = rep(n[2],14),
      target_case = target_case,
      target_h2 = vg2,
      target_pop_prev = prevalence[2],
      target_sample_prev = sampling[2],
      best_pT = pupper[2],
      nsnp = nsnp,
      rg = c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5),
      power = ra$power,
      r2_at_rg = ra$R2,
      p_at_rg = ra$p
)

### SLE PRS predicting any depression 
# SLE he = Total Liability scale h2: 0.103 (0.018)
base = "Systemic Lupus Erythematosus"
base_case = 7219
base_control = 15991
target = "Any Depression"
target_case = 65075

## Calculate covariance
n=c(7219+15991,65075+232552) # training (SLE) 7219+15991=23210, target (any depression) 65075+232552=297627
vg1=0.103 # h2 snp base sle
vg2=0.0838 # h2 snp target MDD
cor12=c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5) # rG
cov12=cor12*sqrt(vg1*vg2) 

# Calculate power
nsnp=836 # independent SNPs at best pT. 
pi0=0.9 # 90% SNPs with no effect in training trait - DOESN'T CHANGE POWER ESTIMATE. 
binary=c(T,T) # training and target binary
prevalence=c(0.001,0.15) # population prevalence training and target 
sampling=c(7219/(7219+15991),65075/(65075+232552)) # gwas case/control, target case/control. 
pupper=c(0,0.001)  

sle <- polygenescore(nsnp=nsnp,n=n,vg1=vg1,cov12=cov12,pi0=pi0,binary=binary,prevalence=prevalence,sampling=sampling,pupper=pupper)

df_sle <- data.frame(
      base = rep(base,14),
      base_n = rep(n[1],14),
      base_case = base_case,
      base_control = base_control,
      base_h2 = vg1,
      base_pop_prev = prevalence[1],
      base_sample_prev = sampling[1],
      target = rep(target,14),
      target_n = rep(n[2],14),
      target_case = target_case,
      target_h2 = vg2,
      target_pop_prev = prevalence[2],
      target_sample_prev = sampling[2],
      best_pT = pupper[2],
      nsnp = nsnp,
      rg = c(0.01,0.02,0.03,0.04,0.05,0.06,0.07,0.08,0.09,0.1,0.2,0.3,0.4,0.5),
      power = sle$power,
      r2_at_rg = sle$R2,
      p_at_rg = sle$p
)

# combine data
power3 <- rbind(df_coeliac,df_ibd,df_ms,df_mg,df_psoriasis,df_psa,df_ra,df_sle)
  
```

**Table:** Parameter inputs for power calculation in AVENGEME - Autoimmune PRS predicting any depression case-control status in the UKB.

```{r parameter_table_autoimmune_prs_to_mdd, echo = FALSE, results = 'asis'}

# Create table with parameter inputs for power calculation - autoimmune PRS predicing depression

power_table3 <- power3 %>%
  filter(rg==0.1) %>%
  select(base, base_n, base_h2, base_pop_prev, base_sample_prev, best_pT, nsnp, target, target_n, target_h2, target_pop_prev, target_sample_prev)

power_table3$base_n <- formatC(power_table3$base_n, big.mark = ",", format = "d")
power_table3$base_h2 <- percent(power_table3$base_h2, digits = 2)
power_table3$base_pop_prev <- percent(power_table3$base_pop_prev, digits = 2)
power_table3$base_sample_prev <- percent(power_table3$base_sample_prev, digits = 0)
power_table3$nsnp <- formatC(power_table3$nsnp, big.mark = ",", format = "d")
power_table3$target_n <- formatC(power_table3$target_n, big.mark = ",", format = "d")
power_table3$target_h2 <- percent(power_table3$target_h2, digits = 2)
power_table3$target_pop_prev <- percent(power_table3$target_pop_prev, digits = 0)
power_table3$target_sample_prev <- percent(power_table3$target_sample_prev, digits = 0)

kable(power_table3, escape = F, align = 'r') %>%
  kable_styling(c("striped", "condensed"), full_width = T, font_size = 13) %>%
  add_header_above(c("Base GWAS" = 7, "Target Phenotype, UKB" = 5))

```

</br>

```{r plot_power_autoimmune_prs_to_mdd, echo=FALSE, eval=TRUE, message=FALSE, results = 'asis', fig.align='center', fig.height=5, fig.width=10}

# Plot power - autoimmune PRS predicting depression case/control

power3$legend <- paste0(
  power3$base,
  " (",
  formatC(power3$base_case, format = "d", big.mark = ","),
  ", ",
  formatC(power3$base_control, format = "d", big.mark = ","),
  ")")

power3_plot <- ggplot(data=power3, mapping=aes(x=rg, y=power)) +
  geom_line(mapping=aes(colour=legend), size=0.6, show.legend=TRUE) +
  geom_hline(mapping=aes(yintercept=0.8, fill="Power"), size=0.6, color="red") +
  scale_color_hue("Autoimmune GWAS (n cases, n controls)", guide=guide_legend(order=1)) +
  scale_fill_manual("Power", values = 1, guide=guide_legend(
    override.aes=list(colour=c("red")),order = 2),labels=c("80% power")) +
  labs(title="Power by rG assumptions", subtitle = "Autoimmune PRS predicting Any Depression in the UKB", x = "\nrG between Autoimmune Diseases and MDD", y = "Power\n") +
  theme(plot.title = element_text(hjust = 0.5, size = 14, margin = margin(b=5)),
        plot.subtitle = element_text(hjust = 0.5, size = 12, margin = margin(b=10)),
        axis.title = element_text(size = 12),
        axis.text = element_text(size = 10),
        legend.title = element_text(size = 12),
        legend.text = element_text(size = 10))

power3_plot

```

**Line Plot:** Power to detect cross-trait associations at different levels of rG. Power calculated for Autoimmune PRS predicing any depression case status in the UKB.

</br>

# Genetic Correlations

</br>

```{r bgenie_gwas_prep_pheno, echo=FALSE, eval=FALSE}

# GWAS of any and stringent depression required for genetic correlation analyses with autoimmune traits

# Prepare any and stringent phenotypes for GWAS

# Residualise 'any' depression (required for linear regression in BGENIE)

Cov <- fread("/file_path/covariates.txt", data.table=F)
Covariates <- Cov[,c(2,5:12)]

dep <- fread("/file_path/depression_gender", data.table=F)
any_dep <- dep[,2:3] %>%
  rename(any_dep = Broad_all)

any_dep_cov <- na.omit(merge(any_dep, Covariates))
rownames(any_dep_cov) <- NULL

# residualise
any_dep_residuals <- data.frame(any_dep_cov$IID)
names(any_dep_residuals) <- "IID"
any_dep_residuals$any_dep_resid <- with(any_dep_cov, residuals(glm(any_dep ~ PC1 + PC2 + PC3 + PC4+ PC5 + PC6 +as.factor(batch) + as.factor(assessment_centre), family="binomial"), type="pearson"))

any_dep_residuals <- any_dep_residuals[,c(1,1,2)]
names(any_dep_residuals) <- c("FID","IID","any_dep_residuals")

write.table(any_dep_residuals, "/file_path/any_dep_residuals", col.names=T, row.names=F, quote=F)

# Residualise 'stringent' depression

stringent_dep <- dep[,c(2,4)] %>%
  rename(stringent_dep = Narrow_all)

stringent_dep_cov <- na.omit(merge(stringent_dep, Covariates))
rownames(stringent_dep_cov) <- NULL

# residualise
stringent_dep_residuals <- data.frame(stringent_dep_cov$IID)
names(stringent_dep_residuals) <- "IID"
stringent_dep_residuals$stringent_dep_resid <- with(stringent_dep_cov, residuals(glm(stringent_dep ~ PC1 + PC2 + PC3 + PC4+ PC5 + PC6 +as.factor(batch) + as.factor(assessment_centre), family="binomial"), type="pearson"))

stringent_dep_residuals <- stringent_dep_residuals[,c(1,1,2)]
names(stringent_dep_residuals) <- c("FID","IID","stringent_dep_residuals")

write.table(stringent_dep_residuals, "/file_path/stringent_dep_residuals", col.names=T, row.names=F, quote=F)

# Prepare phenotypes in format for BGENIE

dat1 <- read.table("/file_path/any_dep_residuals", head=T)
dat1 <- dat1[,2:3]
colnames(dat1)[1] <- "ID"

dat2 <- read.table("/file_path/stringent_dep_residuals", head=T)
dat2 <- dat2[,2:3]
colnames(dat2)[1] <- "ID"

dat3 <- Reduce(function(x, y) merge(x, y, all=TRUE), list(dat1, dat2))

bgen_id <- read.table("/file_path/bgen_ordered_ids.txt", head=F)
names(bgen_id)[1] <- "ID"

dat <- merge(bgen_id, dat3, by = "ID", all.x=TRUE)

# order correctly for BGENIE analysis
dat <- dat[order(match(dat$ID,bgen_id$ID)),] 

dat <- dat[,2:ncol(dat)]

write.table(dat, "/file_path/bgenie_any_stringent_dep", col.names=T, row.names=F, quote=F, sep=" ")

```

```{bash bgenie_gwas, echo=FALSE, eval=FALSE}

# BGENIE script for any and stringent depression GWAS, filtering on HM3 SNPs

for i in {1..22}; do sbatch /file_path/bgenie_any_stringent_hm3.sh $i; done

/file_path/bgenie_v1.2_static1 \
--bgen /file_path/imp_chr${1}_MAF1_INFO4_v1.bgen \
--pheno /file_path/bgenie_any_stringent_dep \
--miss NA \
--include_rsids /file_path/hm3_ids \
--scale_phenotypes \
--scale_genotypes \
--pvals \
--out /file_path/bgenie_any_stringent_hm3_chr${1} \
--thread 8

# Create separate summary statistics for any and stringent depression

for chr in {1..22}; do
  for j in {0..1}; do 
    a=$(( 8 + (4 * $j)))
    b=$(( 9 + (4 * $j)))
    c=$(( 10 + (4 * $j)))
    d=$(( 11 + (4 * $j)))
      zcat /file_path/bgenie_any_stringent_hm3_chr${chr}.gz |\
       awk '{print $1, $2, $3, $4, $5, $6, $7, $'$a', $'$b', $'$c', $'$d', 10^($'$d' * -1)}'\
       > /file_path/bgenie_any_stringent_hm3_chr${chr}_pheno_${j}
  done
done

# Join chromosomes for each phenotype

head -n1 bgenie_any_stringent_hm3_chr1_pheno_0 > bgenie_sumstat_any_dep
for fname in bgenie_any_stringent_hm3_chr*_pheno_0
do
    tail -n+2 $fname >> bgenie_sumstat_any_dep
done

head -n1 bgenie_any_stringent_hm3_chr1_pheno_1 > bgenie_sumstat_stringent_dep
for fname in bgenie_any_stringent_hm3_chr*_pheno_1
do
    tail -n+2 $fname >> bgenie_sumstat_stringent_dep
done

# format column headings
# NB from bgenie wiki: In the regression model we code the first and second alleles as 0 and 1 respectively, so the beta coefficient refers to the effect of having an extra copy of the second allele.

any <- fread("/file_path/bgenie_sumstat_any_dep", head=T)
stringent <- fread("/file_path/bgenie_sumstat_stringent_dep", head=T)

colnames(any) <- c("CHR","SNP","BP","A2","A1","AF","INFO","BETA","SE","T","-log10p","P")
colnames(stringent) <- c("CHR","SNP","BP","A2","A1","AF","INFO","BETA","SE","T","-log10p","P")

write.table(any, gzfile("/file_path/bgenie_dep_any_sumstat.gz"), col.names=T, row.names=F, quote=F, sep=" ")

write.table(stringent, gzfile("/file_path/bgenie_dep_stringent_sumstat.gz"), col.names=T, row.names=F, quote=F, sep=" ")

```

```{r remove_mhc_from_any_and_stringent_mdd, echo=FALSE, eval=FALSE}

# Remove MHC from summary stats

# Any depression

# Read data
any_dep <- read.table(gzfile("/file_path/bgenie_dep_any_sumstat.gz"), head=T) 

# Remove MHC
any_dep_nomhc <- any_dep[!(any_dep$CHR==6 & any_dep$BP > 28866527 & any_dep$BP < 33775447),]

# Save summary stats without MHC
write.table(any_dep_nomhc, "/file_path/ANY_DEP_noMHC", row.names = F, col.names = T, quote = F)

# Stringent depression

# Read data
stringent_dep <- read.table(gzfile("/file_path/bgenie_dep_stringent_sumstat.gz"), head=T) 

# Remove MHC
stringent_dep_nomhc <- stringent_dep[!(stringent_dep$CHR==6 & stringent_dep$BP > 28866527 & stringent_dep$BP < 33775447),]

# Save summary stats without MHC
write.table(stringent_dep_nomhc, "/file_path/STRINGENT_DEP_noMHC", row.names = F, col.names = T, quote = F)

```

```{bash munge_ukb_mdd, echo=FALSE, eval=FALSE}

# Munge depression summary statistics

# Munge any depression: case = 65,075, control = 232,552
/file_path/munge_sumstats.py \
--sumstats /file_path/ANY_DEP_noMHC \
--snp SNP \
--a1 A1 \
--a2 A2 \
--N-cas 65075 \
--N-con 232552 \
--p P \
--signed-sumstats BETA,0 \
--merge-alleles /file_path/w_hm3.snplist \
--out /file_path/ANY_DEP_noMHC_hm3_munged

# Munge stringent depression: case = 14,625, control = 232,552
/file_path/munge_sumstats.py \
--sumstats /file_path/STRINGENT_DEP_noMHC \
--snp SNP \
--a1 A1 \
--a2 A2 \
--N-cas 14625 \
--N-con 232552 \
--p P \
--signed-sumstats BETA,0 \
--merge-alleles /file_path/w_hm3.snplist \
--out /file_path/STRINGENT_DEP_noMHC_hm3_munged

```

```{bash rg_md_autoimmune_gwas_above_5k, echo=FALSE, eval=FALSE}

# Calculate genetic correlations between UKB depression phenotypes and autoimmune diseases with summary statistics available. Exclude autoimmune GWAS with <5k as lack power for LDSC (Myasthenia Gravis and Psoriatic Arthritis)

# rG between ANY depression and six autoimmune disorders

/file_path/ldsc.py \
--rg /file_path/ANY_DEP_noMHC_hm3_munged.sumstats.gz,/file_path/CELI01_noMHC.sumstats.gz,/file_path/INFB01_noMHC.sumstats.gz,/file_path/SCLE02_noMHC.sumstats.gz,/file_path/PSORIASIS_noMHC_hm3_munged.sumstats.gz,/file_path/RHEU02_noMHC.sumstats.gz,/file_path/LUPU01_noMHC.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.22,0.3,0.37,0.36,0.06,0.25,0.31 \
--pop-prev 0.15,0.01,0.005,0.001,0.02,0.01,0.001 \
--out /file_path/rG_ANY_dep_autoimmune

# rG between STRINGENT depression and six autoimmune disorders - using 15% pop prevalence for depression

/file_path/ldsc.py \
--rg /file_path/STRINGENT_DEP_noMHC_hm3_munged.sumstats.gz,/file_path/CELI01_noMHC.sumstats.gz,/file_path/INFB01_noMHC.sumstats.gz,/file_path/SCLE02_noMHC.sumstats.gz,/file_path/PSORIASIS_noMHC_hm3_munged.sumstats.gz,/file_path/RHEU02_noMHC.sumstats.gz,/file_path/LUPU01_noMHC.sumstats.gz \
--ref-ld-chr /file_path/eur_w_ld_chr/ \
--w-ld-chr /file_path/eur_w_ld_chr/ \
--samp-prev 0.06,0.3,0.37,0.36,0.06,0.25,0.31 \
--pop-prev 0.15,0.01,0.005,0.001,0.02,0.01,0.001 \
--out /file_path/rG_STRINGENT_dep_autoimmune

```

```{r make_rg_table, echo=FALSE}

# Make table of rG results

rg_any_dep.log <- read.table("/file_path/rG_ANY_dep_autoimmune.log", sep = ",", stringsAsFactors = FALSE)
rg_stringent_dep.log <- read.table("/file_path/rG_STRINGENT_dep_autoimmune.log", sep = ",", stringsAsFactors = FALSE)

rg_any_dep <- data.frame(x = rg_any_dep.log[193:198,], stringsAsFactors = F)
rg_stringent_dep <- data.frame(x = rg_stringent_dep.log[193:198,], stringsAsFactors = F)

head <- data.frame(x = rg_any_dep.log[192,], stringsAsFactors = F)

rg <- data.frame(rbind(
  strsplit(rg_any_dep$x, " +")[[1]],
  strsplit(rg_any_dep$x, " +")[[2]],
  strsplit(rg_any_dep$x, " +")[[3]],
  strsplit(rg_any_dep$x, " +")[[4]],
  strsplit(rg_any_dep$x, " +")[[5]],
  strsplit(rg_any_dep$x, " +")[[6]],
  strsplit(rg_stringent_dep$x, " +")[[1]],
  strsplit(rg_stringent_dep$x, " +")[[2]],
  strsplit(rg_stringent_dep$x, " +")[[3]],
  strsplit(rg_stringent_dep$x, " +")[[4]],
  strsplit(rg_stringent_dep$x, " +")[[5]],
  strsplit(rg_stringent_dep$x, " +")[[6]]  
  ))
colnames(rg) <- strsplit(head$x, " +")[[1]]

rg[] <- lapply(rg, function(x) gsub("/file_path/ANY_DEP_noMHC_hm3_munged.sumstats.gz", "Any Depression", x))
rg[] <- lapply(rg, function(x) gsub("/file_path/STRINGENT_DEP_noMHC_hm3_munged.sumstats.gz", "Stringent Depression", x))
rg[] <- lapply(rg, function(x) gsub("/file_path/CELI01_noMHC.sumstats.gz", "Coeliac Disease", x))
rg[] <- lapply(rg, function(x) gsub("/file_path/INFB01_noMHC.sumstats.gz", "Inflammatory Bowel Disease", x))
rg[] <- lapply(rg, function(x) gsub("/file_path/SCLE02_noMHC.sumstats.gz", "Multiple Sclerosis", x))
rg[] <- lapply(rg, function(x) gsub("/file_path/PSORIASIS_noMHC_hm3_munged.sumstats.gz", "Psoriasis", x))
rg[] <- lapply(rg, function(x) gsub("/file_path/RHEU02_noMHC.sumstats.gz", "Rheumatoid Arthritis", x))
rg[] <- lapply(rg, function(x) gsub("/file_path/LUPU01_noMHC.sumstats.gz", "Systemic Lupus Erythematosus", x))

# Make 95% CIs
rg$lowci <- format(round(as.numeric(rg$rg)-1.96*as.numeric(rg$se), digits = 2), nsmall = 2)
rg$highci <- format(round(as.numeric(rg$rg)+1.96*as.numeric(rg$se), digits = 2), nsmall = 2)
rg$ci_95 <- paste(rg$lowci,"-",rg$highci)

rg_summary <- rg[,c(2:4,16,7)]

```

Genetic correlation table: Any and Stringent depression with six autoimmune phenotypes

```{r print_rg_table_summary, echo=FALSE, results = 'asis'}

kable(rg_summary, escape = FALSE, format.args = list(big.mark = ',')) %>%
  kable_styling(c("striped","condensed")) %>%   
  row_spec(0, bold = TRUE, color = "white", background = "#666")

```
